{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Colorectal Carcinoma RNAseq - 2\n",
    "```\n",
    "pi:ababaian\n",
    "files: ~/Crown/data/CRC/\n",
    "start: 2017 03 31\n",
    "complete : 2017 04 01\n",
    "```\n",
    "## Introduction\n",
    "\n",
    "[Continuation of CRC_1](./20170325_CRC_RNAseq_PRIVATE.ipynb)\n",
    "\n",
    "If variant allele frequency changes within a patient between cancer and normal, deviations in allele frequency, specifically deviations between the mean RAF of the cancer and normal samples is indicitive of directional selection. This can be used to identify oncogenic/tumour suppressor variations in rRNA if they exist.\n",
    "\n",
    "## Hypothesis\n",
    "\n",
    "1) Null: An allele is not under directional selection if the mean reference allele frequency (or variant allele frequency) is the same between the cancer and normal cell populations.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Materials and Methods\n",
    "\n",
    "1) [Data generated previously](./20170325_CRC_RNAseq_PRIVATE.ipynb) for all of 18S and 28S rRNA in each of the 138 cancer/normal RNAseq libraries.\n",
    "\n",
    "- `18S_crc.gvcf` - DP:AD for chr13:1003660-1005529 across 138 samples (69 patient-tumour matched)\n",
    "- `28S_crc.gvcf` - DP:AD for chr13:1007948-1013560 across 138 samples (69 patient-tumour matched)\n",
    "\n",
    "NOTE: 28S is incorrectly calculated to the end of 45S not the end of 28S, as such for this analysis the extra bases are trimmed in the code manually at 1013018.\n",
    "\n",
    "2) Ran crcAnalysis.r script\n",
    "\n",
    "3) Molecule-wide significance\n",
    "\n",
    "3) Measuring 18E pre-rRNA content\n",
    "\n",
    "- Ran adCalc.sh on gsc on `REGION='chr13:1005530-1005724'` and `OUTPUT='18E_crc.gvcf'`\n",
    "- Ran adCalc.sh on gsc on `REGION='chr13:1006622-1006779'` and `OUTPUT='5.8S_crc.gvcf'`\n",
    "\n",
    "Note: I'm not predicting any change in 5.8S, I just will look but don't consider it part of the screen. The 18E is the ~200 bases downstream of 18S which can be used to measure the relative proportion of 18S-e (pre-rRNA accumulation) which can also explain a decrease in the amount of modified base being read out.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### crcAnalysis.r\n",
    "\n",
    "```\n",
    "# crcAnalysis.R\n",
    "#\n",
    "# Analysis of adCalc.sh\n",
    "# output gvcf files\n",
    "#\n",
    "library(ggplot2)\n",
    "\n",
    "# Import\n",
    "GVCF = read.table('18S_crc.gvcf')\n",
    "  GVCF = data.frame(t(GVCF))\n",
    "  colnames(GVCF) = seq(1, length(GVCF[1,]))\n",
    "\n",
    "# Cut 28S to just 28S (5071 bases)\n",
    "# chr13    1007948    1013018    28S\n",
    "  # trim = which( (apply(GVCF[2,], 1, as.numeric) < 1007949) | \n",
    "  #                 (apply(GVCF[2,], 1 , as.numeric) > 1013018) )\n",
    "  # GVCF = GVCF[, -trim]  \n",
    "\n",
    "  \n",
    "# Cut 18S to just 18S (1869 bases)\n",
    "# chr13    1003661    1005529    18S\n",
    "  trim = which( (apply(GVCF[2,], 1, as.numeric) < 1003661) | \n",
    "                (apply(GVCF[2,], 1 , as.numeric) > 1005529) )\n",
    "  GVCF = GVCF[, -trim]  \n",
    "  \n",
    "\n",
    "refAllele = GVCF[4,]\n",
    "altAllele = GVCF[5,]\n",
    "genCoord  = GVCF[2,]\n",
    "rnaCoord  = seq(1, length(GVCF[2,]))\n",
    "\n",
    "sampleN = length(GVCF[,1]) - 9 # remove 9 header vcf rows\n",
    "bpN     = length(genCoord) # 1869 for 18S; 5071 for 28S\n",
    "\n",
    "# Functions =========================================================\n",
    "\n",
    "# Convert DP:AD string to numeric DP (Total Depth)\n",
    "dpCalc = function(inSTR){\n",
    "# inSTR is from vcf\n",
    "# in format DP:AD\n",
    "# 2000:1500,400,50,50\n",
    "# extract 2000\n",
    "inSTR = as.character(inSTR)\n",
    "as.numeric(unlist(strsplit(inSTR,split=':'))[1])\n",
    "\n",
    "}\n",
    "\n",
    "# Convert DP:AD string to numeric RD for the REFERENCE ALLELE DEPTH\n",
    "# Thus Alternative_Allele_Depth = Total_Depth - Reference_Allele_Depth\n",
    "# for all alternative alleles.\n",
    "rdCalc = function(inSTR){\n",
    "  # inSTR is from vcf\n",
    "  # in format DP:AD\n",
    "  # 2000:1500,400,50,50\n",
    "  # extract 1500\n",
    "  inSTR = as.character(inSTR)\n",
    "  as.numeric(unlist(strsplit(unlist(strsplit(inSTR,split=\":\"))[2], split = \",\"))[1])\n",
    "  \n",
    "}\n",
    "\n",
    "\n",
    "# Calculations ======================================================\n",
    "# Calculate Depth of Coverage (baq > 30)\n",
    "# for all positions\n",
    "\n",
    "#Initialize DP vector\n",
    "DP = vapply( GVCF[-c(1:9),1], dpCalc, 1)\n",
    "\n",
    "#Extend the DP vector for all positions\n",
    "for (i in 2:bpN){\n",
    "DP = cbind(DP,\n",
    "           vapply( GVCF[-c(1:9),i], dpCalc, 1) )\n",
    "}\n",
    "\n",
    "# Calculate Reference Depth of Coverage (baq > 30)\n",
    "# for all positions\n",
    "#\n",
    "#Initialize\n",
    "RD = vapply( GVCF[-c(1:9),1], rdCalc, 1)\n",
    "\n",
    "#The rest\n",
    "for (i in 2:bpN){\n",
    "  RD = cbind(RD,\n",
    "             vapply( GVCF[-c(1:9),i], rdCalc, 1) )\n",
    "}\n",
    "\n",
    "\n",
    "# Reference Allele Frequency\n",
    "# Intra-Library\n",
    "# RD / DP\n",
    "RAF = RD / DP\n",
    "\n",
    "# NOTE: division by zero is possible here and will introduce NAs\n",
    "\n",
    "\n",
    "# Deconvolute Cancer samples from normal samples\n",
    "# Odd Rows = Cancer Sample\n",
    "# Even Rows = Normal Sample\n",
    "# Paired for CRC\n",
    "canRAF = RAF[seq(1,sampleN,2),]\n",
    "normRAF = RAF[seq(2,sampleN,2),]\n",
    "              \n",
    "# Change in Reference Allele Frequency\n",
    "# of Cancer from Normal\n",
    "dRAF = canRAF - normRAF\n",
    "\n",
    "\n",
    "# Calculate some descriptive statistics\n",
    "# about the change in Reference Allele Frequency\n",
    "# Remove NA from calculations (no sequencing depth in a library)\n",
    "mean_dRAF = apply(dRAF,2,mean, na.rm = TRUE)\n",
    "sd_dRAF   = apply(dRAF,2,sd, na.rm = TRUE)\n",
    "var_dRAF  = apply(dRAF,2,var, na.rm = TRUE)\n",
    "mean_DP = apply(DP,2,mean, na.rm = TRUE)\n",
    "\n",
    "# Remove poorly 'covered' positions (i.e. less then 1000x coverage on average)\n",
    "# the magnitude of bias is simply too high at such regions\n",
    "dropPOS = (mean_DP < 1000)\n",
    "\n",
    "canRAF[,dropPOS]  = 0\n",
    "normRAF[,dropPOS] = 0\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### screenPlot_crc.r\n",
    "Script to process crcAnalysis.r data files and plot molecule-wide changes in RAF and significance.\n",
    "\n",
    "```\n",
    "# screenPlot_crc.r\n",
    "#\n",
    "# Screen for changes between allele frequency\n",
    "# between two biological samples\n",
    "# using adCalc.sh / crcAnalysis.r\n",
    "#\n",
    "\n",
    "library(ggplot2)\n",
    "library(rgl)\n",
    "library(grid)\n",
    "\n",
    "# Calculate P-value for difference of means in RAF\n",
    "# between cancer and normal\n",
    "# use as a 'score' \n",
    "\n",
    "# t.test based\n",
    "Pval = t.test(canRAF[,1], normRAF[,1], paired = TRUE)$p.value\n",
    "\n",
    "for (i in 2:bpN){\n",
    "  Pval = cbind(Pval,\n",
    "               t.test(canRAF[,i], normRAF[,i], paired = TRUE)$p.value)\n",
    "}\n",
    "\n",
    "# Score\n",
    "Pscore = -log(Pval)\n",
    "Pscore[is.na(Pscore)] = 0\n",
    "Pscore = as.numeric(Pscore)\n",
    "\n",
    "\n",
    "# Bonferonni correction\n",
    "# p < alpha / m\n",
    "# p < signifiance_cutoff / numberTests\n",
    "# p * numberTests < significance_cutoff\n",
    "\n",
    "Pscore_bon = -log(Pval*bpN)\n",
    "Pscore_bon[is.na(Pscore_bon)] = 0\n",
    "Pscore_bon[Pscore_bon < 0] = 0\n",
    "Pscore_bon = as.numeric(Pscore_bon)\n",
    "\n",
    "# NOTE: this should be re-calculated as a Manhatten plot\n",
    "# for the publication. T-test is a bit basic.\n",
    "\n",
    "\n",
    "# PLOT ==================================================\n",
    "\n",
    "DATA = data.frame(1:bpN) # for 18S\n",
    "  colnames(DATA) = c(\"RNA\")\n",
    "\n",
    "DATA$mean_dRAF = mean_dRAF    \n",
    "DATA$var_dRAF = var_dRAF\n",
    "DATA$mean_DP = mean_DP\n",
    "DATA$Pscore = Pscore\n",
    "DATA$Pscore_bon = Pscore_bon\n",
    "\n",
    "# plot(mean_dRAF)\n",
    "#  plot(var_dRAF)\n",
    "#  plot(mean_DP)\n",
    "#  plot(log(mean_DP))\n",
    "#  plot(Pscore)\n",
    "#  plot(Pscore_bon)\n",
    "# plot3d(mean_DP, Pscore)\n",
    " \n",
    "# PLOTS for 18S\n",
    "PLOT1 = ggplot(DATA, aes(RNA, mean_dRAF)) +\n",
    "  geom_point() + theme_minimal() +\n",
    "  ylim(c(-0.11,0.11))\n",
    "#PLOT1\n",
    "\n",
    "PLOT2 = ggplot(DATA, aes(RNA, var_dRAF)) +\n",
    "  geom_point() + theme_minimal()\n",
    "#PLOT2\n",
    "\n",
    "PLOT3 = ggplot(DATA, aes(RNA, mean_DP)) +\n",
    "  geom_point() + theme_minimal() +\n",
    "  scale_y_log10()\n",
    "#PLOT3\n",
    "\n",
    "PLOT4 = ggplot(DATA, aes(RNA, Pscore)) +\n",
    "  geom_point() + theme_minimal() +\n",
    "  geom_hline(yintercept = -log(0.001/bpN))\n",
    "#PLOT4\n",
    "\n",
    "multiplot <- function(..., plotlist=NULL, file, cols=1, layout=NULL) {\n",
    "\n",
    "  # Make a list from the ... arguments and plotlist\n",
    "  plots <- c(list(...), plotlist)\n",
    "  \n",
    "  numPlots = length(plots)\n",
    "  \n",
    "  # If layout is NULL, then use 'cols' to determine layout\n",
    "  if (is.null(layout)) {\n",
    "    # Make the panel\n",
    "    # ncol: Number of columns of plots\n",
    "    # nrow: Number of rows needed, calculated from # of cols\n",
    "    layout <- matrix(seq(1, cols * ceiling(numPlots/cols)),\n",
    "                     ncol = cols, nrow = ceiling(numPlots/cols))\n",
    "  }\n",
    "  \n",
    "  if (numPlots==1) {\n",
    "    print(plots[[1]])\n",
    "    \n",
    "  } else {\n",
    "    # Set up the page\n",
    "    grid.newpage()\n",
    "    pushViewport(viewport(layout = grid.layout(nrow(layout), ncol(layout))))\n",
    "    \n",
    "    # Make each plot, in the correct location\n",
    "    for (i in 1:numPlots) {\n",
    "      # Get the i,j matrix positions of the regions that contain this subplot\n",
    "      matchidx <- as.data.frame(which(layout == i, arr.ind = TRUE))\n",
    "      \n",
    "      print(plots[[i]], vp = viewport(layout.pos.row = matchidx$row,\n",
    "                                      layout.pos.col = matchidx$col))\n",
    "    }\n",
    "  }\n",
    "}\n",
    "\n",
    "multiplot(PLOT1, PLOT2, PLOT3, PLOT4, cols=1)\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Results\n",
    "\n",
    "### 18S rRNA Change in Reference Allele Frequency\n",
    "\n",
    "![dRAF 18S](../../data/CRC/plot/18S_dRAF.png)\n",
    "\n",
    "A single position 18S.1248U reaches molecule-wide signifiance by t.testing for difference of means at (0.001) significance.\n",
    "\n",
    "This is the hyper-modified base macp-Psi, the increase in reference allele frequency is consistent with a decrease in modification of this base.\n",
    "\n",
    "### 28S rRNA Change in Reference Allele Frequency\n",
    "\n",
    "![dRAF 28S](../../data/CRC/plot/28S_dRAF.png)\n",
    "\n",
    "In 28S another single position also reaches molecule wide signifiance 28S.470A (hgr1.1008418) Note that in 28S there is a region between 3000 and 3600 which is poorly covered and mean coverage drops below 1000 and thus is excluded as a bias-region. \n",
    "\n",
    "\n",
    "In this way I fail to reject the null hypothesis and two positions reach molecule-wide significance, 18S.1248U and 28S.470A for changes in reference allele frequency.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 18S.1248U\n",
    "`basePlot_crc.r` analysis\n",
    "\n",
    "![18S.1248U RAF](../../data/CRC/plot/18S.1248U_RAF.png)\n",
    "\n",
    "```\n",
    "> t.test(POS$cancer_RAF, POS$normal_RAF, paired = TRUE)\n",
    "\n",
    "\tPaired t-test\n",
    "\n",
    "data:  POS$cancer_RAF and POS$normal_RAF\n",
    "t = 8.331, df = 68, p-value = 5.491e-12\n",
    "alternative hypothesis: true difference in means is not equal to 0\n",
    "95 percent confidence interval:\n",
    " 0.08220607 0.13398997\n",
    "sample estimates:\n",
    "mean of the differences \n",
    "               0.108098 \n",
    "\n",
    "> var.test(POS$cancer_RAF, POS$normal_RAF)\n",
    "\n",
    "\tF test to compare two variances\n",
    "\n",
    "data:  POS$cancer_RAF and POS$normal_RAF\n",
    "F = 11.003, num df = 68, denom df = 68, p-value < 2.2e-16\n",
    "alternative hypothesis: true ratio of variances is not equal to 1\n",
    "95 percent confidence interval:\n",
    "  6.812664 17.770178\n",
    "sample estimates:\n",
    "ratio of variances \n",
    "          11.00283 \n",
    "```\n",
    "\n",
    "Which is 1.0256 x 10^-8 molecule-wide significance and ~3.8437 x 10^-8 experiment-wide signifiance which is less then alpha of 0.001. This position is different between cancer and normal by a wide margin and the change is directional towards the reference allele (U). This is likely a marker of strong selection at this position but it's slightly more complicated then that since this is a hyper-modified base...\n",
    "\n",
    "![18S_1248U dRAF](../../data/CRC/plot/18S.1248U_dRAF.png)\n",
    "\n",
    "\n",
    "```\n",
    "> t.test(POS$dRAF, POS$sim_dRAF, paired = FALSE)\n",
    "\n",
    "\tWelch Two Sample t-test\n",
    "\n",
    "data:  POS$dRAF and POS$sim_dRAF\n",
    "t = 5.4939, df = 135.51, p-value = 1.881e-07\n",
    "alternative hypothesis: true difference in means is not equal to 0\n",
    "95 percent confidence interval:\n",
    " 0.06654615 0.14139895\n",
    "sample estimates:\n",
    "  mean of x   mean of y \n",
    "0.108098019 0.004125469 \n",
    "\n",
    "> var.test(POS$dRAF, POS$sim_dRAF)\n",
    "\n",
    "\tF test to compare two variances\n",
    "\n",
    "data:  POS$dRAF and POS$sim_dRAF\n",
    "F = 0.88707, num df = 68, denom df = 68, p-value = 0.6226\n",
    "alternative hypothesis: true ratio of variances is not equal to 1\n",
    "95 percent confidence interval:\n",
    " 0.549253 1.432674\n",
    "sample estimates:\n",
    "ratio of variances \n",
    "         0.8870739 \n",
    "```\n",
    "\n",
    "\n",
    "### 28S.470A\n",
    "`basePlot_crc.R` analysis\n",
    "\n",
    "![28S.470A RAF](../../data/CRC/plot/28S.470A_RAF.png)\n",
    "\n",
    "```\n",
    "> t.test(POS$cancer_RAF, POS$normal_RAF, paired = TRUE)\n",
    "\n",
    "\tPaired t-test\n",
    "\n",
    "data:  POS$cancer_RAF and POS$normal_RAF\n",
    "t = -5.8538, df = 68, p-value = 1.528e-07\n",
    "alternative hypothesis: true difference in means is not equal to 0\n",
    "95 percent confidence interval:\n",
    " -0.020305158 -0.009981117\n",
    "sample estimates:\n",
    "mean of the differences \n",
    "            -0.01514314 \n",
    "\n",
    "> var.test(POS$cancer_RAF, POS$normal_RAF)\n",
    "\n",
    "\tF test to compare two variances\n",
    "\n",
    "data:  POS$cancer_RAF and POS$normal_RAF\n",
    "F = 2.0824, num df = 68, denom df = 68, p-value = 0.002877\n",
    "alternative hypothesis: true ratio of variances is not equal to 1\n",
    "95 percent confidence interval:\n",
    " 1.289365 3.363186\n",
    "sample estimates:\n",
    "ratio of variances \n",
    "          2.082397 \n",
    "```\n",
    "\n",
    "Bonferroni correcting for multiple testing the P-value is 7.886 x 10^-4 across 28S and 1.074 x 10^-3 for the entire experiment. Which by the book rules this position out as being significantly different. I think that's a super conservative intepretation though and something probably is involved with 470A.\n",
    "\n",
    "![28S.470A dRAF](../../data/CRC/plot/28S.470A_dRAF.png)\n",
    "\n",
    "```\n",
    "> t.test(POS$dRAF, POS$sim_dRAF, paired = FALSE)\n",
    "\n",
    "\tWelch Two Sample t-test\n",
    "\n",
    "data:  POS$dRAF and POS$sim_dRAF\n",
    "t = -4.8017, df = 135.81, p-value = 4.098e-06\n",
    "alternative hypothesis: true difference in means is not equal to 0\n",
    "95 percent confidence interval:\n",
    " -0.02434797 -0.01014286\n",
    "sample estimates:\n",
    "   mean of x    mean of y \n",
    "-0.015143137  0.002102281 \n",
    "\n",
    "> var.test(POS$dRAF, POS$sim_dRAF)\n",
    "\n",
    "\tF test to compare two variances\n",
    "\n",
    "data:  POS$dRAF and POS$sim_dRAF\n",
    "F = 1.0781, num df = 68, denom df = 68, p-value = 0.7574\n",
    "alternative hypothesis: true ratio of variances is not equal to 1\n",
    "95 percent confidence interval:\n",
    " 0.6675255 1.7411761\n",
    "sample estimates:\n",
    "ratio of variances \n",
    "          1.078091\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Conclusion\n",
    "\n",
    "I failed to reject the null hypothesis. There is a significant difference between 18S.1248U in cancer and normal. This is an incredibly important position in the ribosome, it's at the core of the P-site and a hyper-modified base to macp-Psi which convolutes the analysis but I don't believe this is simply chance or noise, this is the biochemical signiture of a difference between a sub-population of rRNA in cancer cells and normal cells.\n",
    "\n",
    "[Continued in CRC 3](./20170401_CRC_RNAseq_3_PRIVATE.ipynb)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 5.8S dRAF\n",
    "\n",
    "Just because I can't not look at this point.\n",
    "\n",
    "![5.8S dRAF](../../data/CRC/plot/5.8S_dRAF.png)\n",
    "\n",
    "No signifiance points : )\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Addendum -- 170423\n",
    "\n",
    "In another parallel analysis of T-ALL, I identified two more modifications in detectable by RNA-seq which are likely RNA nucleotide differences [See Database](https://people.biochem.umass.edu/fournierlab/3dmodmap/hum28sseq.php).\n",
    "\n",
    "I hypothesized that if there is a pan-ribobiogenesis phenotype in the CRC samples then other modifications should also be 'hypo-modified'.\n",
    "\n",
    "### 18S.1851A\n",
    "Modification to di-me6_A\n",
    "\n",
    "`basePlot_crc.R` analysis\n",
    "\n",
    "![18S.1851A RAF](../../data/CRC/plot/18S.1851A_RAF.png)\n",
    "\n",
    "```\n",
    "> t.test(POS$cancer_RAF, POS$normal_RAF, paired = TRUE)\n",
    "\n",
    "\tPaired t-test\n",
    "\n",
    "data:  POS$cancer_RAF and POS$normal_RAF\n",
    "t = -1.5446, df = 68, p-value = 0.1271\n",
    "alternative hypothesis: true difference in means is not equal to 0\n",
    "95 percent confidence interval:\n",
    " -0.010416566  0.001326732\n",
    "sample estimates:\n",
    "mean of the differences \n",
    "           -0.004544917 \n",
    "\n",
    "> var.test(POS$cancer_RAF, POS$normal_RAF)\n",
    "\n",
    "\tF test to compare two variances\n",
    "\n",
    "data:  POS$cancer_RAF and POS$normal_RAF\n",
    "F = 0.51365, num df = 68, denom df = 68, p-value = 0.006684\n",
    "alternative hypothesis: true ratio of variances is not equal to 1\n",
    "95 percent confidence interval:\n",
    " 0.3180367 0.8295680\n",
    "sample estimates:\n",
    "ratio of variances \n",
    "         0.5136468 \n",
    "```\n",
    "\n",
    "There is no evidence that this position is 'hypo-modified', if anything it's reading out more (although non-significant) error reads.\n",
    "\n",
    "### 28S.1321A\n",
    "Modification to me1_A.\n",
    "\n",
    "`basePlot_crc.R` analysis\n",
    "\n",
    "![28S.1321A RAF](../../data/CRC/plot/28S.1321A_RAF.png)\n",
    "\n",
    "```\n",
    "> t.test(POS$cancer_RAF, POS$normal_RAF, paired = TRUE)\n",
    "\n",
    "\tPaired t-test\n",
    "\n",
    "data:  POS$cancer_RAF and POS$normal_RAF\n",
    "t = -2.4384, df = 68, p-value = 0.01737\n",
    "alternative hypothesis: true difference in means is not equal to 0\n",
    "95 percent confidence interval:\n",
    " -0.0083813933 -0.0008371938\n",
    "sample estimates:\n",
    "mean of the differences \n",
    "           -0.004609294 \n",
    "\n",
    "> var.test(POS$cancer_RAF, POS$normal_RAF)\n",
    "\n",
    "\tF test to compare two variances\n",
    "\n",
    "data:  POS$cancer_RAF and POS$normal_RAF\n",
    "F = 1.2623, num df = 68, denom df = 68, p-value = 0.3392\n",
    "alternative hypothesis: true ratio of variances is not equal to 1\n",
    "95 percent confidence interval:\n",
    " 0.7815606 2.0386256\n",
    "sample estimates:\n",
    "ratio of variances \n",
    "          1.262264 \n",
    "```\n",
    "\n",
    "This position is not molecule-wide or experiment-wide significant but by itself there is a difference. It's key to note that the order of this change is much less then that of the 1248U difference.\n",
    "\n",
    "### 28S.4532U\n",
    "Modification to me3_U\n",
    "\n",
    "\n",
    "`basePlot_crc.R` analysis\n",
    "\n",
    "![28S.4532U RAF](../../data/CRC/plot/28S.4532U_RAF.png)\n",
    "\n",
    "```\n",
    "> t.test(POS$cancer_RAF, POS$normal_RAF, paired = TRUE)\n",
    "\n",
    "\tPaired t-test\n",
    "\n",
    "data:  POS$cancer_RAF and POS$normal_RAF\n",
    "t = -0.19977, df = 68, p-value = 0.8423\n",
    "alternative hypothesis: true difference in means is not equal to 0\n",
    "95 percent confidence interval:\n",
    " -0.002957874  0.002419535\n",
    "sample estimates:\n",
    "mean of the differences \n",
    "          -0.0002691691 \n",
    "\n",
    "> var.test(POS$cancer_RAF, POS$normal_RAF)\n",
    "\n",
    "\tF test to compare two variances\n",
    "\n",
    "data:  POS$cancer_RAF and POS$normal_RAF\n",
    "F = 0.7182, num df = 68, denom df = 68, p-value = 0.1749\n",
    "alternative hypothesis: true ratio of variances is not equal to 1\n",
    "95 percent confidence interval:\n",
    " 0.4446933 1.1599398\n",
    "sample estimates:\n",
    "ratio of variances \n",
    "         0.7182043 \n",
    "```\n",
    "\n",
    "And this position is not different between cancer and normal.\n",
    "\n",
    "\n",
    "Altogether this supports the notion that there isn't an over-whemling ribosome biogenesis phenotype which can explain the 1248U reference allele frequency.\n",
    "\n",
    "\n",
    "### Controls --- Bona-fide unmodified bases\n",
    "\n",
    "One could argue that this is 'read error'. That's in a different order of magnitude. Positions 4030, 4899 (high GC), and 4900 (high conservation) are shown below.\n",
    "\n",
    "\n",
    "![Control position 4030](../../data/CRC/plot/18S.4030_ctrl.png)\n",
    "\n",
    "![High GC Control Position 4899](../../data/CRC/plot/18S.4899_hiGC_ctrl.png)\n",
    "\n",
    "![Control position 4900](../../data/CRC/plot/18S.4900_ctrl.png)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
