{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Colorectal Carcinoma RNA-seq - 1\n",
    "```\n",
    "pi:ababaian\n",
    "files: ~/Crown/data/CRC/\n",
    "start: 2017 03 25\n",
    "complete : 2017 03 31\n",
    "```\n",
    "## Introduction\n",
    "\n",
    "I'm fairly certain that 28S.59G/A is conserved across individuals from the DNAseq/RNAseq. I emailed the manuscript to Jon Dinmen, and he had the same question as everyone else. What's the biological function?\n",
    "\n",
    "Casually observing the AzaSen experiment, there wasn't any all-or-none effects but there is a pretty consistent change at 28S.59 between Replicating and Non-Replicating Cells and between primary fibroblasts (p53 -/+) and immortalized fibroblasts (p53 -/-)\n",
    "\n",
    "![28S 59 G / A Alleles](../../data/CRC/plot/azaSen_28S.png)\n",
    "\n",
    "I think this position may be a pretty decent 'assay', while I don't know what the consequences of either of these alleles are, and they may simple be in linkage with the relevant variants but since they are so easy to study (moderate GC content at this area) then it's worth trying.\n",
    "\n",
    "The trend is: A% increases between primary vs. transformed and decreases with cell senescence. Thus...\n",
    "\n",
    "### Hypothesis\n",
    "\n",
    "1) In primary colorectal carcinoma (CRC) RNA-seq, the relative abundance of 28S.A will increase in the cancer sample relative to adjacent normal tissues.\n",
    "\n",
    "2) CRC samples which are p53 mutated (I should be able to look this up) will have an increased 28S.A allele frequency."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Materials and Methods\n",
    "This analysis will be done on the GSC; data is already there just need to pipe it in.\n",
    "\n",
    "`crc_align_hgr1.fa`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "# crc_align_hgr1.fa\n",
    "# rDNA alignment pipeline\n",
    "# for CRC data on GSC to hgr1\n",
    "# 170325 -- 1750 build\n",
    "# xhost10 \n",
    "\n",
    "# Control Panel -------------------------------\n",
    "\n",
    "# Project Dir\n",
    "  BASE='/home/ababaian/projects/rDNA/CRC'\n",
    "  cd $BASE\n",
    "\n",
    "# Sequencing Data\n",
    "  CRC_DIR='/projects/magerlab/mkarimi/Colon/human/Colon'\n",
    "  LIB_LIST='crc_data.txt' # list of crc data fastq files\n",
    "  \n",
    "# CPU\n",
    "  THREADS='4'\n",
    "  \n",
    "# Initialize start-up sequence ----------------\n",
    "# Make working directory\n",
    "  mkdir -p align\n",
    "  mkdir -p flagstat\n",
    "  \n",
    "#Resources\n",
    "  aws s3 cp s3://crownproject/resources/hgr1.fa ./\n",
    "  samtools faidx hgr1.fa\n",
    "  bowtie2-build hgr1.fa hgr1\n",
    "\n",
    "# GATK variant calling resources\n",
    "  aws s3 cp s3://crownproject/resources/hgr1.gatk.fa ./\n",
    "  aws s3 cp s3://crownproject/resources/hgr1.gatk.fa.fai ./\n",
    "  aws s3 cp s3://crownproject/resources/hgr1.gatk.dict ./\n",
    "\n",
    "\n",
    "# ---------------------------------------------\n",
    "# SCRIPT LOOP ---------------------------------\n",
    "# ---------------------------------------------\n",
    "# For each line in input LIB_LIST; run the pipeline\n",
    "\n",
    "cat $LIB_LIST | while read LINE\n",
    "do\n",
    "    #Initialize Runls\n",
    "    echo \"Start Iteration:\"\n",
    "    echo \"  $LINE\"\n",
    "    echo ''\n",
    "    \n",
    "    LIBRARY=$(echo $LINE | cut -f1 -d' ' -) # Library Name\n",
    "    RGSM=$(echo $LINE | cut -f2 -d' ' -)    # Sample / Patient Identifer\n",
    "    RGID=$(echo $LINE | cut -f3 -d' ' -)    # Read Group ID\n",
    "    RGLB=$(echo $LINE | cut -f3 -d' ' -)    # Library Name. Accession Number\n",
    "    RGPL='ILLUMINA'                   # Sequencing Platform.\n",
    "    RGPO=$(echo $LINE | cut -f4 -d' ' -)    # Patient Population\n",
    "\n",
    "    FASTQ1=$(echo $LINE | cut -f5 -d' ' -)  # Filename Read 1\n",
    "    FASTQ2=$(echo $LINE | cut -f6 -d' ' -)  # Filename Read 2\n",
    "    \n",
    "    FQ1=\"$CRC_DIR/$FASTQ1\"            # Fastq1 Filepath\n",
    "    FQ2=\"$CRC_DIR/$FASTQ2\"            # Fastq2 Filepath\n",
    "    \n",
    "    # Extract Sequencing Run Info\n",
    "    RGPU=$(gzip -dc $FQ1 | head -n1 - | cut -f1 -d':' | cut -f2 -d' ')\n",
    "    \n",
    "    echo \" Library: $LIBRARY\"\n",
    "    echo \" SAMPLE: $RGSM\"\n",
    "    echo \" ID: $RGID\"\n",
    "    echo \" RGLIB: $RGLB\"\n",
    "    echo \" Platform: $RGPL\"\n",
    "    echo \" Population: $RGPO\"\n",
    "    echo \" RunName: $RGPU\"\n",
    "    echo \" Fastq1: $FQ1\"\n",
    "    echo \" Fastq2: $FQ2\"\n",
    "    \n",
    "    # Bowtie2: align to genome\n",
    "    bowtie2 --very-sensitive-local -p $THREADS --rg-id $RGID \\\n",
    "      --rg LB:$RGLB --rg SM:$RGSM \\\n",
    "      --rg PL:$RGPL --rg PU:$RGPU \\\n",
    "      -x hgr1 -1 $FQ1 -2 $FQ2 | \\\n",
    "      samtools view -bS - > aligned_unsorted.bam\n",
    "      \n",
    "    # Calcualte library flagstats\n",
    "    samtools flagstat aligned_unsorted.bam > aligned_unsorted.flagstat\n",
    "\n",
    "    # Read Subset ------------------------------\n",
    "    # Extract mapped reads, and their unmapped pairs\n",
    "\n",
    "      # Extract Header\n",
    "      samtools view -H aligned_unsorted.bam > align.header.tmp\n",
    "\n",
    "      # Unmapped reads with mapped pairs\n",
    "      # Extract Mapped Reads\n",
    "      # and their unmapped pairs\n",
    "      samtools view -b -F 4 aligned_unsorted.bam > align.F4.bam #mapped\n",
    "      samtools view -b -f 4 -F 8 aligned_unsorted.bam > align.f4F8.bam #unmapped pairs\n",
    "\n",
    "      # Extract just the 45S unit\n",
    "      #aws s3 cp s3://crownproject/resources/rDNA_45s.bed ./\n",
    "      #samtools view -b -L rDNA_45s.bed align.F4.bam > align.F4.45s.bam\n",
    "\n",
    "      # What are the mapped readnames\n",
    "      samtools view align.F4.bam | cut -f1 - > read.names.tmp\n",
    "\n",
    "      # Extract mapped reads\n",
    "      samtools view align.F4.bam | grep -Ff read.names.tmp - > align.F4.tmp.sam\n",
    "\n",
    "\n",
    "      # Extract cases of read pairs mapped on edge of region of interest\n",
    "      # -------|======= R O I ======| ----------\n",
    "      # read:                  ====---====\n",
    "      samtools view align.F4.bam | grep -Ff read.names.tmp - > align.F4.tmp.sam\n",
    "\n",
    "      # Complete mapped reads list\n",
    "      #cut -f1 align.F4.tmp.sam > read.names.45s.long.tmp\n",
    "\n",
    "      # Extract unmapped reads with a mapped pair\n",
    "      samtools view align.f4F8.bam | grep -Ff read.names.tmp - > align.f4F8.tmp.sam\n",
    "\n",
    "      # Re-compile bam file\n",
    "      cat align.header.tmp align.F4.tmp.sam align.f4F8.tmp.sam | samtools view -bS - > align.hgr1.tmp.bam\n",
    "        samtools sort align.hgr1.tmp.bam align.hgr1\n",
    "        samtools index align.hgr1.bam\n",
    "        samtools flagstat align.hgr1.bam > align.hgr1.flagstat\n",
    "\n",
    "      # Clean up \n",
    "      rm *tmp* align.F4.bam align.f4F8.bam\n",
    "\n",
    "    # Rename/remove the total Bam Files\n",
    "      rm aligned_unsorted.bam\n",
    "      mv aligned_unsorted.flagstat flagstat/$LIBRARY.flagstat\n",
    "\n",
    "    # Rename the hgr Bam files\n",
    "      mv align.hgr1.bam align/$LIBRARY.hgr1.bam\n",
    "      mv align.hgr1.bam.bai align/$LIBRARY.hgr1.bam.bai\n",
    "      mv align.hgr1.flagstat flagstat/$LIBRARY.hgr1.flagstat\n",
    "\n",
    "done\n",
    "\n",
    "# Primary VCF ----------------------------\n",
    "# N/A\n",
    "\n",
    "# Script complete"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### crc_data.xls / crc_data.txt\n",
    "```\n",
    "ID\tIndividual\tAccession\tPopulation\tRead_1\tRead_2\n",
    "220_c\t587220\t587220\tCRC\t587220_1_1.fastq.gz\t587220_1_2.fastq.gz\n",
    "220_n\t587220\t587221\tCRC\t587221_1_1.fastq.gz\t587221_1_2.fastq.gz\n",
    "222_c\t587222\t587222\tCRC\t587222_1_1.fastq.gz\t587222_1_2.fastq.gz\n",
    "222_n\t587222\t587223\tCRC\t587223_1_1.fastq.gz\t587223_1_2.fastq.gz\n",
    "...\n",
    "```\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Results\n",
    "\n",
    "### Sample 220 - Pilot\n",
    "\n",
    "In the first sample ran, manually inspected position 1008007 (28S.59). There is a 5% relative increase in 28S.59A expression.\n",
    "\n",
    "Each library is taking ~20 minutes to analyze;  est. 69 will be done in a ~23 hours\n",
    "\n",
    "#### Normal\n",
    "```\n",
    "chr13:1,008,007\n",
    "Total count: 63472\n",
    "A : 28190 (44%, 26995+, 1195- )\n",
    "C : 30 (0%, 30+, 0- )\n",
    "G : 35239 (56%, 33890+, 1349- )\n",
    "T : 11 (0%, 10+, 1- )\n",
    "N : 2 (0%, 2+, 0- )\n",
    "---------------\n",
    "DEL: 22\n",
    "INS: 3\n",
    "```\n",
    "\n",
    "#### Cancer\n",
    "```\n",
    "chr13:1,008,007\n",
    "Total count: 21730\n",
    "A : 10703 (49%, 10019+, 684- )\n",
    "C : 16 (0%, 14+, 2- )\n",
    "G : 10996 (51%, 10398+, 598- )\n",
    "T : 14 (0%, 14+, 0- )\n",
    "N : 1 (0%, 1+, 0- )\n",
    "---------------\n",
    "DEL: 4\n",
    "INS: 1\n",
    "```\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Methods - Automating Allele Counts\n",
    "I can't (read: don't want to) go into each of the 138 libraries and manaully extract counts for 1,008,007 (28S.59). That's a computers job.\n",
    "\n",
    "28s58.bed: `chr13 1008006 1008007`\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bcftools 1.4\r\n",
      "Using htslib 1.4\r\n",
      "Copyright (C) 2016 Genome Research Ltd.\r\n",
      "License Expat: The MIT/Expat license\r\n",
      "This is free software: you are free to change and redistribute it.\r\n",
      "There is NO WARRANTY, to the extent permitted by law.\r\n"
     ]
    }
   ],
   "source": [
    "# Updated bctools to version 1.4\n",
    "bcftools --version"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[mpileup] 1 samples in 1 input files\r\n",
      "chr13\t1008007\t.\tG\tA,T,C\t0\t.\t.\tDP:AD\t20257:10361,9868,14,13\r\n"
     ]
    }
   ],
   "source": [
    "cd ~/Crown/data/CRC/\n",
    "\n",
    "#samtools mpileup -vu -f hgr1.fa \\\n",
    "#  --max-depth 1000000 --min-BQ 30 \\\n",
    "#  -t DP,AD \\\n",
    "#  -l 28s58.bed 220_n.hgr1.bam | \n",
    "# tail -n1 - | cut -f 4,5,9,10 -\n",
    "\n",
    "BAM='220_c.hgr1.bam'\n",
    "# If using multiple samples per run\n",
    "# add: --ignore-RG\n",
    "\n",
    "bcftools mpileup -f hgr1.fa \\\n",
    "  --max-depth 1000000 --min-BQ 30 \\\n",
    "  -a FORMAT/DP,AD \\\n",
    "  -r chr13:1008007 \\\n",
    "  $BAM |\n",
    "  bcftools annotate -x INFO,FORMAT/PL - |\n",
    "  bcftools view -O v -H -\n",
    "  "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### adCalc.sh\n",
    "\n",
    "This method slowly processes through each library indepenently, this can be greatly sped-up if all the libraries are read in one pass. Thus the adCalc.sh script was written.\n",
    "\n",
    "I'm showing an example run locally here but the same thing can be done to a region as shown below.\n",
    "\n",
    "Initially just the 59G position was analyzed but on the GSC I also ran 18S, 28S on each of the 138 libraries. (After initial analysis of 59G shown below). Samples took ~20 minutes for alignment per library and the gvcf file generation was ~10 minutes each. Only base calls of quality BAQ > 30 were used for the analysis.\n",
    "\n",
    "#### OUTPUT FILES\n",
    "\n",
    "- `18S_crc.gvcf` - DP:AD for chr13:1003660-1005529 across 138 samples (69 patient-tumour matched)\n",
    "- `28S_crc.gvcf` - DP:AD for chr13:1007948-1013560 across 138 samples (69 patient-tumour matched)\n",
    "\n",
    "The data is alpha-sorted the same as `crc_data.txt`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[mpileup] 2 samples in 2 input files\r\n",
      "chr13\t1008007\t.\tG\tA,C,T\t0\t.\t.\tDP:AD\t20257:10361,9868,13,14\t60045:33639,26366,30,10\r\n"
     ]
    }
   ],
   "source": [
    "#!/bin/bash\n",
    "# ADcalc.sh\n",
    "# Allelic Depth Calculator\n",
    "# for a position (28S.59G>A)\n",
    "\n",
    "# Controls -----------------\n",
    "REGION='chr13:1008007'\n",
    "OUTPUT='28S_59G.vcf'\n",
    "DEPTH='100000'\n",
    "BAMLIST='bam.list'\n",
    "\n",
    "ls *.bam > bam.list\n",
    "\n",
    "# Iterate through every bam file in directory\n",
    "# look-up position and return VCF\n",
    "\n",
    "    bcftools mpileup -f hgr1.fa \\\n",
    "  --max-depth $DEPTH --min-BQ 30 \\\n",
    "  -a FORMAT/DP,AD \\\n",
    "  -r $REGION \\\n",
    "  --ignore-RG \\\n",
    "  -b $BAMLIST |\n",
    "  bcftools annotate -x INFO,FORMAT/PL - |\n",
    "  bcftools view -O v -H -\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "# ADcalc.sh\n",
    "# Allelic Depth Calculator\n",
    "# for a range of coordinates\n",
    "\n",
    "# Controls -----------------\n",
    "REGION='chr13:1007948-10130018'\n",
    "OUTPUT='28S_crc.gvcf'\n",
    "DEPTH='100000'\n",
    "BAMLIST='bam.list'\n",
    "\n",
    "# Iterate through every bam file in directory\n",
    "# look-up position and return VCF\n",
    "\n",
    "    bcftools mpileup -f hgr1.fa \\\n",
    "  --max-depth $DEPTH --min-BQ 30 \\\n",
    "  -a FORMAT/DP,AD \\\n",
    "  -r $REGION \\\n",
    "  --ignore-RG \\\n",
    "  -b $BAMLIST |\n",
    "  bcftools annotate -x INFO,FORMAT/PL - |\n",
    "  bcftools view -O v -H - >> $OUTPUT\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## R Analysis of Change in Variant Allele Frequency\n",
    "\n",
    "The processed data was downloaded from the GSC and analyzed in R.\n",
    "\n",
    "`~/Crown/data/CRC/crcAnalysis.r`\n",
    "\n",
    "### crcAnalysis Script Core\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "# crcAnalysis.R\n",
    "#\n",
    "# Analysis of adCalc.sh\n",
    "# output gvcf files\n",
    "#\n",
    "library(ggplot2)\n",
    "\n",
    "# Import\n",
    "GVCF = read.table('28S_crc.gvcf')\n",
    "GVCF = data.frame(t(GVCF))\n",
    "\n",
    "colnames(GVCF) = seq(1, length(GVCF[1,]))\n",
    "\n",
    "refAllele = GVCF[4,]\n",
    "altAllele = GVCF[5,]\n",
    "genCoord  = GVCF[2,]\n",
    "rnaCoord  = seq(-1, (length(GVCF[1,]) - 2))\n",
    "\n",
    "sampleN = length(GVCF[,1]) - 9 # remove 9 header vcf rows\n",
    "bpN     = length(genCoord)\n",
    "\n",
    "# Functions =========================================================\n",
    "\n",
    "# Convert DP:AD string to numeric DP (Total Depth)\n",
    "dpCalc = function(inSTR){\n",
    "# inSTR is from vcf\n",
    "# in format DP:AD\n",
    "# 2000:1500,400,50,50\n",
    "# extract 2000\n",
    "inSTR = as.character(inSTR)\n",
    "as.numeric(unlist(strsplit(inSTR,split=':'))[1])\n",
    "\n",
    "}\n",
    "\n",
    "# Convert DP:AD string to numeric RD for the REFERENCE ALLELE DEPTH\n",
    "# Thus Alternative_Allele_Depth = Total_Depth - Reference_Allele_Depth\n",
    "# for all alternative alleles.\n",
    "rdCalc = function(inSTR){\n",
    "  # inSTR is from vcf\n",
    "  # in format DP:AD\n",
    "  # 2000:1500,400,50,50\n",
    "  # extract 1500\n",
    "  inSTR = as.character(inSTR)\n",
    "  as.numeric(unlist(strsplit(unlist(strsplit(inSTR,split=\":\"))[2], split = \",\"))[1])\n",
    "  \n",
    "}\n",
    "\n",
    "\n",
    "# Calculations ======================================================\n",
    "# Calculate Depth of Coverage (baq > 30)\n",
    "# for all positions\n",
    "\n",
    "\n",
    "#Initialize DP vector\n",
    "DP = vapply( GVCF[-c(1:9),1], dpCalc, 1)\n",
    "\n",
    "#Extend the DP vector for all positions\n",
    "for (i in 2:bpN){\n",
    "DP = cbind(DP,\n",
    "           vapply( GVCF[-c(1:9),i], dpCalc, 1) )\n",
    "}\n",
    "\n",
    "# Calculate Reference Depth of Coverage (baq > 30)\n",
    "# for all positions\n",
    "#\n",
    "#Initialize\n",
    "RD = vapply( GVCF[-c(1:9),1], rdCalc, 1)\n",
    "\n",
    "#The rest\n",
    "for (i in 2:bpN){\n",
    "  RD = cbind(RD,\n",
    "             vapply( GVCF[-c(1:9),i], rdCalc, 1) )\n",
    "}\n",
    "\n",
    "\n",
    "# Reference Allele Frequency\n",
    "# Intra-Library\n",
    "# RD / DP\n",
    "RAF = RD / DP\n",
    "\n",
    "# NOTE: division by zero is possible here and will introduce NAs\n",
    "\n",
    "\n",
    "# Deconvolute Cancer samples from normal samples\n",
    "# Odd Rows = Cancer Sample\n",
    "# Even Rows = Normal Sample\n",
    "# Paired for CRC\n",
    "canRAF = RAF[seq(1,sampleN,2),]\n",
    "normRAF = RAF[seq(2,sampleN,2),]\n",
    "              \n",
    "# Change in Reference Allele Frequency\n",
    "# of Cancer from Normal\n",
    "dRAF = canRAF - normRAF\n",
    "\n",
    "\n",
    "# Calculate some descriptive statistics\n",
    "# about the change in Reference Allele Frequency\n",
    "# Remove NA from calculations (no sequencing depth in a library)\n",
    "mean_dRAF = apply(dRAF,2,mean, na.rm = TRUE)\n",
    "sd_dRAF   = apply(dRAF,2,sd, na.rm = TRUE)\n",
    "var_dRAF  = apply(dRAF,2,var, na.rm = TRUE)\n",
    "mean_DP = apply(DP,2,mean, na.rm = TRUE)\n",
    "\n",
    "# Remove poorly 'covered' positions (i.e. less then 1000x coverage on average)\n",
    "# the magnitude of bias is simply too high at such regions\n",
    "dropPOS = (mean_DP < 10000)\n",
    "\n",
    "canRAF[,dropPOS]  = 0\n",
    "normRAF[,dropPOS] = 0\n",
    "\n",
    "\n",
    "# Calculate P-value for difference of means in RAF\n",
    "# between cancer and normal\n",
    "# use as a 'score' \n",
    "\n",
    "# t.test based\n",
    "Pval = t.test(canRAF[,1], normRAF[,1], paired = FALSE)$p.value\n",
    "\n",
    "for (i in 2:bpN){\n",
    "  Pval = cbind(Pval,\n",
    "               t.test(canRAF[,i], normRAF[,i], paired = FALSE)$p.value)\n",
    "}\n",
    "\n",
    "# Score\n",
    "Pscore = -log(Pval)\n",
    "Pscore[is.na(Pscore)] = 0\n",
    "Pscore = as.numeric(Pscore)\n",
    "\n",
    "\n",
    "# Bonferonni correction\n",
    "  # p < alpha / m\n",
    "  # p < signifiance_cutoff / numberTests\n",
    "  # p * numberTests < significance_cutoff\n",
    "\n",
    "Pscore_bon = -log(Pval*bpN)\n",
    "Pscore_bon[is.na(Pscore_bon)] = 0\n",
    "Pscore_bon[Pscore_bon < 0] = 0\n",
    "Pscore_bon = as.numeric(Pscore_bon)\n",
    "\n",
    "# NOTE: this should be re-calculated as a Manhatten plot\n",
    "# for the publication. T-test is a bit basic.\n",
    "\n",
    " plot(mean_dRAF)\n",
    " plot(var_dRAF)\n",
    " plot(mean_DP)\n",
    " plot(log(mean_DP))\n",
    " plot(Pscore)\n",
    " plot(Pscore_bon)\n",
    "```\n",
    "\n",
    "Plot a single position Script\n",
    "\n",
    "```\n",
    "# basePlot_crc.r\n",
    "#\n",
    "# Analysis and plot the data\n",
    "# for a single base (after running crcAnalysis.r)\n",
    "#\n",
    "\n",
    "library(ggplot2)\n",
    "library(reshape2)\n",
    "\n",
    "# Position of interest\n",
    "# in hgr1 / chr13 coordinates \n",
    "\n",
    "pos ='1008007' # G58A / hgr1:1008007\n",
    "  pos = which(genCoord == pos)\n",
    "\n",
    "  \n",
    "# Change in allele frequency between cancer-normal\n",
    "POS = data.frame(dRAF[,pos])\n",
    "  colnames(POS) = \"dRAF\"\n",
    "\n",
    "# Raw reference allele frequency\n",
    "POS$cancer_RAF = canRAF[,pos]\n",
    "POS$normal_RAF = normRAF[,pos]\n",
    "\n",
    "# Depth of coverage at position of interest\n",
    "POS$cancer_DP = DP[seq(1, sampleN, 2), pos]\n",
    "POS$normal_DP = DP[seq(2, sampleN, 2), pos]\n",
    "\n",
    "\n",
    "# Plot canRAF vs. normRAF\n",
    "POSDATA = melt(POS[,2:3])\n",
    "\n",
    "PLOT = ggplot(POSDATA, aes(variable, value)) +\n",
    "  geom_boxplot(stat = 'boxplot') + \n",
    "  geom_jitter( width = 0.2)\n",
    "PLOT\n",
    "\n",
    "# Test for signifiance\n",
    "t.test(POS$cancer_RAF, POS$normal_RAF, paired = TRUE)\n",
    "var.test(POS$cancer_RAF, POS$normal_RAF)\n",
    "\n",
    "\n",
    "# Plot change in reference allele frequency\n",
    "# comapred to a normal distribution with the same\n",
    "# standard deviation (null hypothesis)\n",
    "POS$sim_dRAF = rnorm(sampleN/2, mean = 0,\n",
    "                     sd = sd(POS$dRAF))\n",
    "\n",
    "POSDATA = melt(POS[,c(1,6)])\n",
    "\n",
    "# Plot dRAF alone\n",
    "# PLOT = ggplot(POS, aes('delta 28S.r.59G', dRAF)) +\n",
    "#   geom_boxplot(stat = 'boxplot') +\n",
    "#   geom_jitter(width = 0.2)\n",
    "# PLOT\n",
    "\n",
    "# Plot dRAF vs. Normal Distribution\n",
    "PLOT = ggplot(POSDATA, aes(variable, value)) +\n",
    "  geom_boxplot(stat = 'boxplot') + \n",
    "  geom_jitter( width = 0.2)\n",
    "PLOT\n",
    "\n",
    "t.test(POS$dRAF, POS$sim_dRAF, paired = FALSE)\n",
    "var.test(POS$dRAF, POS$sim_dRAF)\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 28S.r.G59A\n",
    "\n",
    "Testing the hypothesis; there should be an increase in A59 in the colorectal carcinoma samples relative to the their normal controls.\n",
    "\n",
    "#### Cancer vs. Normal Reference Allele Frequency (intra-sample)\n",
    "![28S_59G_RAF Boxplot](../../data/CRC/plot/28S_59G_RAF.png)\n",
    "\n",
    "```\n",
    "> mean(POS$cancer_RAF)\n",
    "[1] 0.6393323\n",
    "> mean(POS$normal_RAF)\n",
    "[1] 0.6445118\n",
    "```\n",
    "\n",
    "```\n",
    "> t.test(POS$cancer_RAF, POS$normal_RAF, paired = TRUE)\n",
    "\n",
    "\tPaired t-test\n",
    "\n",
    "data:  POS$cancer_RAF and POS$normal_RAF\n",
    "t = -0.43754, df = 68, p-value = 0.6631\n",
    "alternative hypothesis: true difference in means is not equal to 0\n",
    "95 percent confidence interval:\n",
    " -0.02880163  0.01844265\n",
    "sample estimates:\n",
    "mean of the differences \n",
    "           -0.005179492 \n",
    "\n",
    "> var.test(POS$cancer_RAF, POS$normal_RAF)\n",
    "\n",
    "\tF test to compare two variances\n",
    "\n",
    "data:  POS$cancer_RAF and POS$normal_RAF\n",
    "F = 1.2305, num df = 68, denom df = 68, p-value = 0.3946\n",
    "alternative hypothesis: true ratio of variances is not equal to 1\n",
    "95 percent confidence interval:\n",
    " 0.7618793 1.9872890\n",
    "sample estimates:\n",
    "ratio of variances \n",
    "          1.230477 \n",
    "```\n",
    "\n",
    "\n",
    "There is no difference between the mean or variance of 28S.59A frequency between Cancer and Normal as a population (and with paired testing).\n",
    "\n",
    "\n",
    "#### Cancer-Normal matched change in Reference Allele Frequency (inter-patient)\n",
    "The difference between (Cancer - Normal) RAF compared to a normal distribution (mean = 0, sd = 0.0983329)\n",
    "\n",
    "\n",
    "![28S 59G dRAF](../../data/CRC/plot/28S_59G_dRAF.png)\n",
    "\n",
    "```\n",
    "> t.test(POS$dRAF, POS$sim_dRAF, paired = FALSE)\n",
    "\n",
    "\tWelch Two Sample t-test\n",
    "\n",
    "data:  POS$dRAF and POS$sim_dRAF\n",
    "t = 0.94326, df = 135.85, p-value = 0.3472\n",
    "alternative hypothesis: true difference in means is not equal to 0\n",
    "95 percent confidence interval:\n",
    " -0.01703680  0.04811067\n",
    "sample estimates:\n",
    "   mean of x    mean of y \n",
    "-0.005179492 -0.020716424 \n",
    "\n",
    "> var.test(POS$dRAF, POS$sim_dRAF)\n",
    "\n",
    "\tF test to compare two variances\n",
    "\n",
    "data:  POS$dRAF and POS$sim_dRAF\n",
    "F = 1.0683, num df = 68, denom df = 68, p-value = 0.7861\n",
    "alternative hypothesis: true ratio of variances is not equal to 1\n",
    "95 percent confidence interval:\n",
    " 0.6614696 1.7253798\n",
    "sample estimates:\n",
    "ratio of variances \n",
    "           1.06831 \n",
    "```\n",
    "\n",
    "#### Discussion 28S.59G\n",
    "\n",
    "The 28S.59G has a global-normal average allele frequency of 64.45% which is pretty consistent with the previous iVAF measurements ~34% for this position globally. Most interestingly is that there can be such substantial change in allele frequency in the matched cancers (upto 30%) which given the ~10+ years to develop + cytogenetic abnormalaties is not completely unexpected.\n",
    "\n",
    "This experiment fails to reject the null hypothesis, there is no difference in RAF between matched cancer and normal samples. If anything this position changes perfectly neutrall.\n",
    "\n",
    "This is pretty exciting because it means that this method can measure changes in variant/reference allele frequency  quite well and if there's neutral change, that also means deviations from neutrallity will be a good indicator for selection. The cancer-normal pairs can be analyzed as evolutionary trajectories for rDNA flucuations.\n",
    "\n",
    "Notably; none of the samples reached fixation in either direction, it's unclear at the moment if this is because my sample size is too small or if this is because both alleles of 28S.59G|A are neccesary. Worth following up on.\n",
    "\n",
    "```\n",
    "> range(POS$normal_RAF)\n",
    "[1] 0.1482212 0.9874421\n",
    "> range(POS$cancer_RAF)\n",
    "[1] 0.1427057 0.9807644\n",
    "```\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "Follow-up Experiment: [ Selection of rRNA variant alleles in CRC ](./20170331_CRC_RNAseq_2_PRIVATE.ipynb)\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
