{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Molecular 1248U MACP Assay -- 1\n",
    "```\n",
    "PI: ababaian\n",
    "Start Date: 2017 04 03\n",
    "End Date: 2017 05 06\n",
    "```\n",
    "\n",
    "## Introduction\n",
    "\n",
    "### Objective\n",
    "- Establish a molecular screen to test for the 18S.1248 U -> C change from RT.\n",
    "- Establish protocols for the extraction and manipulation of rRNA\n",
    "- Screen Colorectal Carcinoma / Hodgkins Lymphoma / T-ALL / Control cDNA for MACP incorporation.\n",
    "\n",
    "## Materials and Methods\n",
    "\n",
    "### Cell Culture\n",
    "#### 170403 -- Growing HT115 Cells\n",
    "\n",
    "[HT115](https://www.phe-culturecollections.org.uk/products/celllines/generalcell/detail.jsp?refId=85061104&collection=ecacc_gc) cells (cat# ECACC 85061104) were taken from FLs box; 293I (oct 3rd). Unknown passage, unknown starting density.\n",
    "\n",
    "Cells initially are clumpy/round (first thaw) but flatten out and look like typical fibroblasts. Stated as growing in 'Islands' but I'm not sure I see this, they are perhaps a bit patchy. ~5% of the cells round up under these growth conditions and shed into suspension, possibly worth changing media between passages depending on how they look.\n",
    "\n",
    "##### Growth Medium\n",
    "In hood mixed up some growth medium.\n",
    "- DMEM 500 mL (Stemcell)\n",
    "- 75 mL FBS (15%)\n",
    "- 5.75 200 mM L-glutamine (2 mM) (Stemcell)\n",
    "\n",
    "##### Thawing Cells\n",
    "1 mL of cells in the freezer media were directly plated into 5mL of *4'C cold* media (5 mL in T25).\n",
    "\n",
    "Next AM media exchanged after 1x PBS wash. Arbitrarily designated cells passage 10.\n",
    "\n",
    "##### Growing Cells\n",
    "Passage cells at between 70-80% confluency\n",
    "\n",
    "Cells should be seeded at 2-4 e4 cells/ml^2.\n",
    "\tT25: 5e5-1e6\n",
    "\tT75: 2e6-4e6\n",
    "\n",
    "#### 170405\n",
    "Cells are at ~80% confluency. Cells passaged from T25 flask and seeded into a T75 flask.\n",
    "\n",
    "##### Passaging Cells\n",
    "- Old media is removed, cells are washed in 5 mL warmed PBS\n",
    "- 1.5 mL (T25), 2 mL (T75) of Trypsin-EDTA is added to the cells, incubated at 37'C for 1-3 minutes untill cells dislodge with tapping.\n",
    "- 3.5 mL (T25) or 8 mL (T75) growth media added to inactive Trypsin.\n",
    "- Cells spun at 300xg for 5 minutes to pellet\n",
    "- Pellet resuspended in 5 mL (T25) or 10 mL (T75) PBS for wash and spun again\n",
    "- When counting cells, 0.5 mL of cell suspsenion aliquoted here for further dilution/counting.\n",
    "- Cells resuspended in 5 mL (T25) or 10 mL (T75) media and seeded at the appropriate density.\n",
    "\n",
    "##### Cell Counting\n",
    "Unless otherwise noted this is the protocol for counting cells; I often simply report the two cell count numbers and the final concentration.\n",
    "\n",
    "- Aliquot of cells is resuspended / mixed to homogenize. Ensure a fine single-cell suspension here (no clumps)\n",
    "- 10 uL of cell suspension is mixed with 90 uL of Trypan-Blue for a 10x dilution\n",
    "- 10 uL of the diluted cells are plated into a Hemocytometer, both sides independently.\n",
    "- Count the 5 squares, corners and middle. Top and Right boundary is considered 'in'.\n",
    "- 59 cells in first count; 43 cells in second count.\n",
    "- Sum both counts for a count in 10 squares total. (102 in this case)\n",
    "- 102 cells in 1 mm^3 or in 1 uL\n",
    "\n",
    "- <Cells> * <Dilution Factor> * 1000 (for mL)\n",
    "- 102 * 10 * 1000 = 1.02 x 10^6 cells/mL\n",
    "- 1.02e6 cells/mL with a total of 5 mL suspension\n",
    "\n",
    "#### 170405 - 170422\n",
    "I was a bit bad with notes on cell culture here but took care of the cells pretty much as above. Nothing interesting to note\n",
    "\n",
    "#### HCT115 p13 - Trizol Extraction\n",
    "A T75 flask of HT115 were grown to ~80% confluency. Cells were trypsinized by SOP and washed 1x in 10 mL PBS and counted.\n",
    "\n",
    "- Cells were resuspended in PBS again and 3e6 cells in suspension were aliquoted into 1.5 mL tubes in triplicate\n",
    "- Cells were pelleted at 500g 5m\n",
    "- 0.5 mL trizol reagent was added to the 3e6 cell pellets\n",
    "- Pellet was pipetted and vortexed to homogenize.\n",
    "- Tubes incubated for 5 minutes at RT and then immediatly placed into -80'C freezer\n",
    "\n",
    "* Frozen samples kept in -80'C DM Freezer #7, Rack 8, Box 1 'Artem HL Box'\n",
    "\n",
    "#### RNA/DNA Extraction\n",
    "Followed [Trizol](https://tools.thermofisher.com/content/sfs/manuals/trizol_reagent.pdf) SOP for RNA/DNA extraction with the following modifications:\n",
    "\n",
    "- Two HCT115 Trizol samples thawed on ice, incubated 5 minutes at RT\n",
    "- 0.5 mL of reagent added to 3e6 washed cell pellet. Volumes adjusted accordingly throughout.\n",
    "\n",
    "* DNA WASH -- Sodium Citrate Solution (0.1 M)\n",
    "- 50 mL ddH2O\n",
    "- 1.4705 g Dihydrate sodium Cirate (294.1 g /mol)\n",
    "\n",
    "* Sodium Hydroxide (1M)\n",
    "- 100 mL ddH2O\n",
    "- 3.9997 g NaOH\n",
    "- warmed up at 37'C bath to dissolve\n",
    "\n",
    "* DNA SUSPEND -- Sodium Hydroxide (8 mM)\n",
    "- 49.6 mL ddH2O\n",
    "- 0.4  mL 1M NaOH\n",
    "\n",
    "- DNA resuspended in 100 uL 8mM NaOH ddH2O, stored in 4'C\n",
    "- RNA resuspended in 30 uL ddH2O, stored in 4'C\n",
    "\n",
    "RNA-yield measured on [NanoDrop](./plot/170421.txt)\n",
    "| Sample      | ng/uL  | A260/280 | A260/230 |\n",
    "|-------------|--------|----------|----------|\n",
    "|ht115 gDNA 1 | 3.37   | 0.23 | 0.03 |\n",
    "|ht115 gDNA 2 | 19.85  | 0.65 | 0.09 |\n",
    "|ht115 RNA 1  | 754.89 | 1.95 | 1.82 |\n",
    "|ht115 RNA 2  | 752.56 | 1.92 | 2.07 |\n",
    "\n",
    "RNA looks pretty good, gDNA extraction is quite poor, I notably lost the pellet in the first sample but continued anyways. May be able to PCR something still. I tested the pH of the Trizol and it's sub pH 4.5 which means it's not gone basic on me (i.e. DNA goes into aquous phase). Try to resuspend in lower volume next time.\n",
    "\n",
    "\n",
    "#### DNAase Treament\n",
    "Followed [Turbo DNA-free](https://tools.thermofisher.com/content/sfs/manuals/cms_055740.pdf) kit protocols. (page 4,5)\n",
    "- Incubated for 60m, all DNase added at start\n",
    "- Samples processed are diluted 2x fold during processing\n",
    "\n",
    "#### Reverse Transcription\n",
    "Followed [Superscript III](https://tools.thermofisher.com/content/sfs/manuals/superscriptIII_man.pdf) protocol. A modified Moloney Murine Leukemia Virus RT.\n",
    "\n",
    "* RT Reaction\n",
    "- 1 ul Random Primer (200 ng/ uL)\n",
    "- 1 ul dNTP          (10 mM)\n",
    "- 5 uL total RNA + 6 uL ddH2O; or\n",
    "- 11 ul RNA DNase Tx\n",
    "\n",
    "(Can make this as 'mastermix', just w/out RT)\n",
    "- 4 uL 5x 1-strand Buffer\n",
    "- 1 uL DTT\n",
    "- 1 uL RNaseOUT\n",
    "- 1 uL Superscript\n",
    "\n",
    "| Sample #    | RNA         |RT |\n",
    "|---|-----------------------|---|\n",
    "| 1 | ht115 DNase Treated 1 | + |\n",
    "| 2 | ht115 DNase Treated 2 | + |\n",
    "| 3 | ht115 1               | + |\n",
    "| 4 | ht115 2               | + |\n",
    "| 5 | ht115 DNase Treated 1 | - |\n",
    "| 6 | ht115 DNase Treated 2 | - |\n",
    "| 7 | ddH2O                 | + |\n",
    "| 8 | ddH2O                 | + |\n",
    "\n",
    "* RT Cycle\n",
    "- 25'C for 5 minutes\n",
    "- 55'C (hot) for 60 minutes\n",
    "- 70'C for 15 minutes to deactivate\n",
    "\n",
    "* 1:10 Final / Working Dilution\n",
    "- 18 uL of cDNA solution\n",
    "- 162 uL ddH2O\n",
    "\n",
    "![HCT115 cDNA](../../data2/macp/plot/170426_nd.jpg)\n",
    "\n",
    "Yield is reasonable for dilute cDNA. The quality is not ideal for 'clean' DNA solutions.\n",
    "\n",
    "#### PCR-assay\n",
    "* Primers\n",
    "```\n",
    ">Frag6_f1_EcoRI\n",
    "TAGATGAATTCCGTTCGAAGACGATCAGATACCG\n",
    ">Frag8_r1_XbaI\n",
    "AAGTTGTCTAGATCGTTCGTTATCGGAATTAACCAG\n",
    "```\n",
    "\n",
    "Note: Frag6 primer has an extra 'C' in the EcoRI cut site. GAATTCC instead of GAATTC. Need to re-order.\n",
    "\n",
    "* 24x Master Mix\n",
    "- 60 uL 10x PCR Buffer\n",
    "- 24 uL 50mM MgCl2\n",
    "- 24 uL 10mM dNTP\n",
    "- 12 uL 100uM Primer 1\n",
    "- 12 uL 100uM Primer 2\n",
    "- 6  uL Taq (Platinum)\n",
    "- 442uL ddH2O\n",
    "\n",
    "- Aliquot 29 uL x samples\n",
    "- Add 1 uL cDNA solution\n",
    "\n",
    "| Sample #    | Sample      |\n",
    "|---|-----------------------|\n",
    "| 1 | ht115 DNase Treated 1 |\n",
    "| 2 | ht115 DNase Treated 2 |\n",
    "| 3 | ht115 1               |\n",
    "| 4 | ht115 2               |\n",
    "| 5 | ht115 noRT            |\n",
    "| 6 | ht115 noRT            |\n",
    "| 7 | ddH2O + RT            |\n",
    "| 8 | ddH2O + RT            |\n",
    "| 9 | ht115 gDNA 1          |\n",
    "| 10| ht115 gDNA 2          |\n",
    "| 11| L428 cDNA             |\n",
    "| 12| HEK293T cDNA          |\n",
    "| 13| K562 cDNA             |\n",
    "| 14| HepG2 gDNA            |\n",
    "| 15| H2O                   |\n",
    "| 16| H2O                   |\n",
    "\n",
    "These sample-wells were taken from frozen stocks available in the lab. Not PCR'd\n",
    "| 17| ht115 total RNA       |\n",
    "| 18| ht115 total RNA       |\n",
    "\n",
    "* PCR-Program\n",
    "- 94'C 5 min\n",
    "\n",
    "25xCycles\n",
    "- 94'C 30s\n",
    "- 58'C 30s\n",
    "- 72'C 30s\n",
    "\n",
    "- 72'C 7m\n",
    "\n",
    "* Gel\n",
    "- 2 g Agarose (2%)\n",
    "- 100 mL TBE buffer\n",
    "- 10 uL Safe-View added to gel\n",
    "- Use large (0.5L) flask and heat slowly to reduce bubbles\n",
    "\n",
    "- Ran: 60V for 40 min; 80V for 40 min.\n",
    "\n",
    "![17426_rtPCR.JPG](../../data2/macp/plot/rtpcr_17426.JPG)\n",
    "\n",
    "No amplicons; most likely the PCR is non-optimal. The lab thermocyler is 'decalibrated' by temperature so maybe that was the problem. Re-try with another thermocycler and gradiant-PCR to optimize.\n",
    "\n",
    "The crappy debris I believe is from the cheap tape I used to seal the gel when casting, going to use the fancy label tape next time which shouldn't leave residue like this.\n",
    "\n",
    "#### 170427 - PCR optimization\n",
    "Since there is no amplicon yesterday run PCR optimization of 18S.1248U analysis.\n",
    "\n",
    "\n",
    "Gradiant PCR / Optimization\n",
    "\tTemperature Range: 50'C - 60'C\n",
    "\tCycles: 35 cycles (overkill)\n",
    "\tAdditives: +- 3% DMSO\n",
    "\n",
    "Master Mix (20x rxns)\n",
    "- 60 ul 10x PCR Buffer\n",
    "- 24 ul 50 mM MgCl2\n",
    "- 24 ul 10 mM dNTPs\n",
    "- 12 ul Primer 1 (f1)\n",
    "- 12 ul Primer 2 (r1)\n",
    "- 6  ul Platinum Taq\n",
    "- 424 ddH2O\n",
    "\n",
    "--> Split 281 uL of master mix into two tubes\n",
    "- Add 9 uL of H2O (-) or 9 uL of DMSO (+)\n",
    "\n",
    "Aliquot 29 ul/PCR tube; add 1 uL of cDNA. All samples loaded with HEK 293T (1:20) cDNA from freezer.\n",
    "\n",
    "| Sample #    | Note        |\n",
    "|---|-----------------------|\n",
    "| 1 | 60.0 'C |\n",
    "| 2 | 59.2 'C |\n",
    "| 3 | 58.0 'C |\n",
    "| 4 | 56.1 'C |\n",
    "| 5 | 53.8 'C |\n",
    "| 6 | 51.9 'C |\n",
    "| 7 | 50.7 'C |\n",
    "| 8 | 50.7 'C |\n",
    "| 9 | 60.0 'C |\n",
    "| 10 | 59.2 'C DMSO|\n",
    "| 11 | 58.0 'C DMSO|\n",
    "| 12 | 56.1 'C DMSO|\n",
    "| 13 | 53.8 'C DMSO|\n",
    "| 14 | 51.9 'C DMSO|\n",
    "| 15 | 50.7 'C DMSO|\n",
    "| 16 | 50.7 'C DMSO|\n",
    "| 17 | 56.1 'C |\n",
    "| 18 | 56.1 'C DMSO|\n",
    "\n",
    "PCR Program 'AB001'\n",
    "(Ran on BioRad T100 TFL-#1 Thermocyler)\n",
    "- 95'C 10 min (longer denaturation)\n",
    "35x\n",
    "- 95'C 30s\n",
    "- 50~60'C 30s\n",
    "- 72'C 30s\n",
    "\n",
    "-72'C 7 min extension\n",
    "\n",
    "Amplicons stored at 4'C \n",
    "\n",
    "Gel/Run\n",
    "10 uL Amplicon + 2 uL Load Dye (6x).\n",
    "1.8% Agar gel cast + 10uL SafeView.\n",
    "Ran 10m 60V, 40m 100V.\n",
    "\n",
    "Well for Sample 1 mis-loaded (4uL lost). The loadign wasn't the best.\n",
    "\n",
    "![rtpcr 170427](../../data2/macp/plot/rtpcr_170427_1.JPG)\n",
    "\n",
    "No obvious bands. Note the DNA load dye (Bromophenol blue?) runs at ~300-400 bp and might be casting a 'shadow'.\n",
    "\n",
    "The 2% DMSO does amplify something but it's too high to be the 18S fragment I'm trying for. Needs to be further optimized.\n",
    "\n",
    "#### 170429 - PCR optimization II\n",
    "\n",
    "Since it's not a temperature-dependent optmization it's most likely some sort of secondary structure in the amplicon which is inhibiting amplification. I'm going to go ahead with a narrower range of amplification (52'C - 58'C) and try and change the reaction conditions\n",
    "\n",
    "- Use gDNA as the starting material, incase 1848U issues with the M-MLV Reverse Transcriptase used (Superscript III). HepG2 gDNA used.\n",
    "- Increase primer concentrations to 1 uM (max suggested)\n",
    "- Vary MgCl2; reducing should de-stabilize DNA hairpin formation. 2mM --> 1 mM, 1.5 mM\n",
    "- Vary DMSO % from 0,2,4,6,8 %\n",
    "\n",
    "\n",
    "| |Tube  A|   B|    C|    D|    E|    F|\n",
    "|--|-----|-----|-----|-----|-----|-----|\n",
    "|PCR Buffer [10x]|    15|    15|    15|    15|    15|    15|\n",
    "|MgCl_2 [50 mM]|    6|    6|    6|    6|    3|    4.5|\n",
    "|dNTPs [10 mM]|    6|    6|    6|    6|    6|    6|\n",
    "|Primer 1 [10 uM]|    6|    6|    6|    6|    6|    6|\n",
    "|Primer 2 [10 uM]|    6|    6|    6|    6|    6|    6|\n",
    "|DMSO|    0|    2.4|    4.8|    6.2|    0|    0|\n",
    "|ddH2O|    104|    101.6|    99.2|    97.8|    107|    105.5|\n",
    "|Taq [5 U /uL]|    2|    2|    2|    2|    2|    2|\n",
    "|DNA [ x ]|    5|    5|    5|    5|    5|    5|\n",
    "|    |    |    |    |    |    |\n",
    "|Total Volume|    150|    150|    150|    150|    150|\t150|\n",
    "\n",
    "- Each mastermix was aliquoted 4x30uL into PCR tubes\n",
    "- Ran on AB001 PCR program using 52-58'C gradiant\n",
    "- Left-over mastermix 30ul were aliquoted into 'Tube #5s', 1.5 uL HT115 cDNA added and ran at only 54'C\n",
    "\n",
    "1.6 % agarose gel cast, 10uL (1:10000) Safeview added to gel\n",
    "\n",
    "Gel Ran 60V 20m, 100V 30m\n",
    "\n",
    "\n",
    "#### DNA loading Dye - Oj\n",
    "I made my own DNA-loading dye containing only Orange G dye which runs at ~50 bp. This should prevent the 'UV shadow' over my expected bands with bromophenol blue. In the future once gel/run conditions are optimized switching to only Xylene Cyan (sp?) which runs at 1 kB is probably the best option as that will be above all the bands of interest.\n",
    "\n",
    "6x Oj DNA loading Dye\n",
    "* 40 mL ddH2O\n",
    "* 9 mL Glycerol\n",
    "* 0.0604 g Orange G\n",
    "* Fill with ddH2O to 50 mL\n",
    "* Vortex thoroughly\n",
    "\n",
    "I made 12x 0.5 mL aliquots, 10x of them put into -20'C freezer with enzymes. Two tubes kept at 4'C for immediate use. The remaining ~42 mL also frozen as a 'master stock mix'.\n",
    "\n",
    "#### 170501 - PCR optimization III\n",
    "\n",
    "New primers arrived. Stock and working solutions made up by SOP.\n",
    "\n",
    ">Frag6_fwd1_EcoRI\n",
    "TAGAT GAATTC GTTCGAAGACGATCAGATACCG\n",
    ">Frag6_fwd2_EcoRI\n",
    "TAGAT GAATTC TCAGATACCGTCGTAGTTCCG\n",
    "\n",
    ">Frag6_f1\n",
    "GTTCGAAGACGATCAGATACCG\n",
    ">Frag8_r1\n",
    "TCGTTCGTTATCGGAATTAACCAG\n",
    "\n",
    ">18S_F3\n",
    "ACTTTCGATGGTAGTCGCCGTG\n",
    ">18S_R3\n",
    "ATTGGAGCTGGAATTACCGCGG\n",
    "\n",
    "\n",
    "The B-actin primers (albeit they were old and frozen strangly, sublimation?) didn't work. Could be cDNA failure, re-do PCR from square one, don't skip out on controls : )\n",
    "\n",
    "[rt PCR Set-up 170501](./files/170501_rtPCR_optimization_III.ods)\n",
    "\n",
    "- cDNA: HT115 cDNA #2\n",
    "- AB001 PCR program ran, samples loadeded 52'C - 58 'C like previously\n",
    "- 1.5 % gel cast, ran 60V for 65 minutes.\n",
    "\n",
    "Lanes Top:\n",
    "-, Ladder , A5, B5, C5, D5, E5, Primer 34nt, -, E1, E2, E3, E4\n",
    "\n",
    "Lanes Bot:\n",
    "Ladder, A1, A2, A3, A4, B1, B2, B3/4 mix, B3, B4, C1, C2, C3, C4, D1, D2, D3, D4\n",
    "\n",
    "There appears to be something in B3/4 which means maybe there is product but at sub-detection limits (i.e. poor PCR efficiency). The primers at the bottom seem to be non-dimers, they run at ~35 nt along with the free primer mix used. Notably the B-Actin is missing free primers which probably means that batch was no good/degraded.\n",
    "\n",
    "![rtpcr 170504](../../data2/macp/plot/rtpcr_170504_2_CE.JPG)\n",
    "\n",
    "Contrast enhanced image via GIMP levels function.\n",
    "\n",
    "#### 170505 - PCR Optimization IV\n",
    "\n",
    "Taking this back to square one I'm repeating the RT-PCR and getting something! Running 3x reactions on 4x samples + H2O control.\n",
    "\n",
    "- Samples: A) HT115 cDNA 1, B) HT115 cDNA 2, C) K562 cDNA, D) L428 cDNA, E) H2O\n",
    "- PCR-35x PCR program: on the thermocycler in DM lab. Created an 'art' user on the machine to store all my programs.\n",
    "- 1.5% gel, ran 65 m\n",
    "\n",
    "Primers from Mark\n",
    "```\n",
    ">B_Actin_F1\n",
    "AAGGCCAACCGTGAAAGAT\n",
    ">B_Actin_R1\n",
    "GTGGTACGACCAGAGGCATAC\n",
    "```\n",
    "\n",
    "It worked! This primer pair works under 'easy' conditions. Only 3% DMSO is non-standard. I used new reagents for this batch which might have been the issue so throw out the ones which didn't work incase that was the issue.\n",
    "\n",
    "HT115 reverse transcription didn't work, but the L428 and K562 cDNA amplified beautifully. I know the HT115 RNA is intact (test again?) so I can do the RT one more time, talk to Kat about it.\n",
    "\n",
    "B-actin primers work well as a control for cDNA conversion, B2M are more likely to form primer-dimers under these conditions so don't use that.\n",
    "\n",
    "Note the primers DO form primer-dimers which means I have to be a little bit careful with conditions, may require RE+ and RE- loading to show it's not primer-dimers when doing the full HifI assay.\n",
    "\n",
    "I need to plan the next few experiments carefully and do this right / think about controls. For instance this also has to work on Genomic DNA which may be trickier as a 'fully unmodified' control, and get HEK/HUVEC cDNA that's high quality as a 'fully modified' (assumed) control.\n",
    "\n",
    "##### PCR-35x (Basic PCR)\n",
    "\n",
    "- 94'C 5m\n",
    "\n",
    "35x Cycles\n",
    "- 94'C 30s\n",
    "- 55'C 30s\n",
    "- 72'C 30s\n",
    "\n",
    "- 72'C 7m\n",
    "- 4'C INF\n",
    "\n",
    "\n",
    "#### 170505 - HinFI Digestion Test\n",
    "\n",
    "I have two amplicons, K562 and L428. From RNA-seq I'm expecting that K562 is 100% modified, unsure about L428.\n",
    "\n",
    "*General Restriction Enyzme Set-up (25 uL)*\n",
    "- ~0.5 ug DNA\n",
    "- 2.5 uL 10x Buffer (CutSmart)\n",
    "- 5 U/rxn RE (stock is 10,000 U / mL)\n",
    "- Incubation 37'C 1h; 80'C 20m (`hinfi` PCR program set-up)\n",
    "\n",
    "For each A) K562 and B) L428; test differing DNA/Amplicon input. Do everything on ice.\n",
    "\n",
    "|       |   2 |   5 |  10 |\n",
    "|-------|-----|-----|-----|\n",
    "| ddH2O | 20  | 17  |  12 |\n",
    "|10x Buf| 2.5 | 2.5 | 2.5 |\n",
    "| Ampl. | 2   | 5   | 10  |\n",
    "| HinFI | 0.5 | 0.5 | 0.5 |\n",
    "\n",
    "Note: In reaction A) K562, I accidently switched up the Amplicon input of 2 ul and 10 ul samples.\n",
    "\n",
    "#### 170506 - HinFI Gel\n",
    "\n",
    "- Cast 2.5% agarose gel; using mini-gel set-up.\n",
    "- 8x lane thin-well (5 x 5 x 1 mm well should be 25 uL...)\n",
    "- Ran 60V for 80 minutes\n",
    "\n",
    "Gel is not the prettiest but HinFI still works (after 19 years at -20C)\n",
    "\n",
    "![MACP Assay HT115 Setup](../../data2/macp/plot/macp_170506_2.JPG)\n",
    "\n",
    "Lanes:\n",
    "- 1: Ladder\n",
    "- 2: K562 - 10 ul\n",
    "- 3: K562 - 5 ul\n",
    "- 4: K562 - 2 ul\n",
    "- 5: L428 - 2 ul\n",
    "- 6: L428 - 5 ul\n",
    "- 7: L428 - 10 ul\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
