{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Colorectal Carcinoma Cell Lines\n",
    "```\n",
    "pi:ababaian\n",
    "files: ~/Crown/data/crc_lines\n",
    "start: 2017 05 12\n",
    "complete : 2017 05 19\n",
    "```\n",
    "## Introduction\n",
    "\n",
    "In the lab we have CRC cell lines. I have the notion that 1248U may be differentially modified in CRC, so to screen some publically available RNA-seq for variants in rRNA which can be rapidly confirmed with cell lines in the freezer.\n",
    "\n",
    "[Fasterius](http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0171435) published HCT116, RKO and HKE3 RNA-seq [+GEO](https://www.ncbi.nlm.nih.gov/sra/SRX1745244[accn]) and has references for a few more libraries.\n",
    "\n",
    "\n",
    "There is an even more comprehensive Cell Line sequencing project [here](www.nature.com/nbt/journal/v33/n3/full/nbt.3080.html) but the data is locked up at EBI.\n",
    "### Hypothesis / Objective\n",
    "- Screen for rare expressed variants / hypo-mods in rRNA of CRC cell lines\n",
    "\n",
    "Break up analysis into two arms; HCT116, RKO and HKE3 and the remaining lines.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Material and Methods\n",
    "\n",
    "Initialize Workspace"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": []
    }
   ],
   "source": [
    "CROWN='/home/artem/Crown'\n",
    "DATA_HOME='/home/artem/Crown/data/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": []
    }
   ],
   "source": [
    "cd $DATA_HOME\n",
    "mkdir -p crc_lines\n",
    "cd crc_lines"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "2017-05-13T16:58:07 prefetch.2.8.2: 1) Downloading 'SRR3479764'...\r\n",
      "2017-05-13T16:58:07 prefetch.2.8.2:  Downloading via http...\r\n",
      "2017-05-13T17:22:18 prefetch.2.8.2 sys: transfer interrupted while reading file within network system module - mbedtls_ssl_read returned -26880 ( SSL - Connection requires a read call )\r\n",
      "2017-05-13T17:22:18 prefetch.2.8.2 int: transfer interrupted while reading file within network system module - x\r\n",
      "2017-05-13T17:22:18 prefetch.2.8.2: 1) failed to download SRR3479764\r\n"
     ]
    }
   ],
   "source": [
    "# Acquire Data from SRA\n",
    "# HCT116\n",
    "prefetch SRR3479755\n",
    "prefetch SRR3479757\n",
    "\n",
    "# RKO\n",
    "prefetch SRR3479758\n",
    "prefetch SRR3479760\n",
    "\n",
    "# HKE3\n",
    "prefetch SRR3479763\n",
    "# prefetch SRR3479764"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "prefetch SRR3479764 \n",
    "# ran in a seperate script window\n",
    "# failed 4x, continue without for now\n",
    "# download another replicate\n",
    "# also tried prefetch SRR3479762"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2017-05-13T19:18:24 fastq-dump.2.8.2 err: process canceled while executing process - failed SRR3479757\r\n"
     ]
    }
   ],
   "source": [
    "# Dump to FASTQ format in crc_lines folder\n",
    "\n",
    "# HCT116\n",
    "fastq-dump --gzip --split-files SRR3479755\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Read 37049468 spots for SRR3479757\r\n",
      "Written 37049468 spots for SRR3479757\r\n",
      "Read 35763277 spots for SRR3479758\r\n",
      "Written 35763277 spots for SRR3479758\r\n",
      "Read 33883402 spots for SRR3479760\r\n",
      "Written 33883402 spots for SRR3479760\r\n"
     ]
    }
   ],
   "source": [
    "fastq-dump --gzip --split-files SRR3479757\n",
    "\n",
    "#RKO\n",
    "fastq-dump --gzip --split-files SRR3479758\n",
    "fastq-dump --gzip --split-files SRR3479760"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Read 67249613 spots for SRR3479763\r\n",
      "Written 67249613 spots for SRR3479763\r\n"
     ]
    }
   ],
   "source": [
    "#HKE3\n",
    "fastq-dump --gzip --split-files SRR3479763\n",
    "#fastq-dump --gzip --split-files SRR3479764"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "hct116_1\tHCT116\tSRX1745244\tinVitro\tSRR3479755_1.fastq.gz\tSRR3479755_2.fastq.gz\r\n",
      "hct116_2\tHCT116\tSRX1745244\tinVitro\tSRR3479757_1.fastq.gz\tSRR3479757_2.fastq.gz\r\n",
      "rko_1\tRKO\tSRX1745244\tinVitro\tSRR3479758_1.fastq.gz\tSRR3479758_2.fastq.gz\r\n",
      "rko_2\tRKO\tSRX1745244\tinVitro\tSRR3479760_1.fastq.gz\tSRR3479760_2.fastq.gz\r\n",
      "hke3_1\tHKE3\tSRX1745244\tinVitro\tSRR3479763_1.fastq.gz\tSRR3479763_2.fastq.gz\r\n"
     ]
    }
   ],
   "source": [
    "cat crc_lines_data.txt\n",
    "# Note: SRR3479764 was removed\n",
    "# from initial analysis. Repeated download\n",
    "# errors."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Settings:\r\n",
      "  Output files: \"hgr1.*.bt2\"\r\n",
      "  Line rate: 6 (line is 64 bytes)\r\n",
      "  Lines per side: 1 (side is 64 bytes)\r\n",
      "  Offset rate: 4 (one in 16)\r\n",
      "  FTable chars: 10\r\n",
      "  Strings: unpacked\r\n",
      "  Max bucket size: default\r\n",
      "  Max bucket size, sqrt multiplier: default\r\n",
      "  Max bucket size, len divisor: 4\r\n",
      "  Difference-cover sample period: 1024\r\n",
      "  Endianness: little\r\n",
      "  Actual local endianness: little\r\n",
      "  Sanity checking: disabled\r\n",
      "  Assertions: disabled\r\n",
      "  Random seed: 0\r\n",
      "  Sizeofs: void*:8, int:4, long:8, size_t:8\r\n",
      "Input files DNA, FASTA:\r\n",
      "  hgr1.fa\r\n",
      "Building a SMALL index\r\n",
      "Reading reference sizes\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "  Time reading reference sizes: 00:00:00\r\n",
      "Calculating joined length\r\n",
      "Writing header\r\n",
      "Reserving space for joined string\r\n",
      "Joining reference sequences\r\n",
      "  Time to join reference sequences: 00:00:00\r\n",
      "bmax according to bmaxDivN setting: 4072\r\n",
      "Using parameters --bmax 3054 --dcv 1024\r\n",
      "  Doing ahead-of-time memory usage test\r\n",
      "  Passed!  Constructing with these parameters: --bmax 3054 --dcv 1024\r\n",
      "Constructing suffix-array element generator\r\n",
      "Building DifferenceCoverSample\r\n",
      "  Building sPrime\r\n",
      "  Building sPrimeOrder\r\n",
      "  V-Sorting samples\r\n",
      "  V-Sorting samples time: 00:00:00\r\n",
      "  Allocating rank array\r\n",
      "  Ranking v-sort output\r\n",
      "  Ranking v-sort output time: 00:00:00\r\n",
      "  Invoking Larsson-Sadakane on ranks\r\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:00\r\n",
      "  Sanity-checking and returning\r\n",
      "Building samples\r\n",
      "Reserving space for 12 sample suffixes\r\n",
      "Generating random suffixes\r\n",
      "QSorting 12 sample offsets, eliminating duplicates\r\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\r\n",
      "Multikey QSorting 12 samples\r\n",
      "  (Using difference cover)\r\n",
      "  Multikey QSorting samples time: 00:00:00\r\n",
      "Calculating bucket sizes\r\n",
      "  Binary sorting into buckets\r\n",
      "  10%\r\n",
      "  20%\r\n",
      "  30%\r\n",
      "  40%\r\n",
      "  50%\r\n",
      "  60%\r\n",
      "  70%\r\n",
      "  80%\r\n",
      "  90%\r\n",
      "  100%\r\n",
      "  Binary sorting into buckets time: 00:00:00\r\n",
      "Splitting and merging\r\n",
      "  Splitting and merging time: 00:00:00\r\n",
      "Split 1, merged 6; iterating...\r\n",
      "  Binary sorting into buckets\r\n",
      "  10%\r\n",
      "  20%\r\n",
      "  30%\r\n",
      "  40%\r\n",
      "  50%\r\n",
      "  60%\r\n",
      "  70%\r\n",
      "  80%\r\n",
      "  90%\r\n",
      "  100%\r\n",
      "  Binary sorting into buckets time: 00:00:00\r\n",
      "Splitting and merging\r\n",
      "  Splitting and merging time: 00:00:00\r\n",
      "Avg bucket size: 2035.12 (target: 3053)\r\n",
      "Converting suffix-array elements to index image\r\n",
      "Allocating ftab, absorbFtab\r\n",
      "Entering Ebwt loop\r\n",
      "Getting block 1 of 8\r\n",
      "  Reserving size (3054) for bucket\r\n",
      "  Calculating Z arrays\r\n",
      "  Calculating Z arrays time: 00:00:00\r\n",
      "  Entering block accumulator loop:\r\n",
      "  10%\r\n",
      "  20%\r\n",
      "  30%\r\n",
      "  40%\r\n",
      "  50%\r\n",
      "  60%\r\n",
      "  70%\r\n",
      "  80%\r\n",
      "  90%\r\n",
      "  100%\r\n",
      "  Block accumulator loop time: 00:00:00\r\n",
      "  Sorting block of length 1732\r\n",
      "  (Using difference cover)\r\n",
      "  Sorting block time: 00:00:00\r\n",
      "Returning block of 1733\r\n",
      "Getting block 2 of 8\r\n",
      "  Reserving size (3054) for bucket\r\n",
      "  Calculating Z arrays\r\n",
      "  Calculating Z arrays time: 00:00:00\r\n",
      "  Entering block accumulator loop:\r\n",
      "  10%\r\n",
      "  20%\r\n",
      "  30%\r\n",
      "  40%\r\n",
      "  50%\r\n",
      "  60%\r\n",
      "  70%\r\n",
      "  80%\r\n",
      "  90%\r\n",
      "  100%\r\n",
      "  Block accumulator loop time: 00:00:00\r\n",
      "  Sorting block of length 2288\r\n",
      "  (Using difference cover)\r\n",
      "  Sorting block time: 00:00:00\r\n",
      "Returning block of 2289\r\n",
      "Getting block 3 of 8\r\n",
      "  Reserving size (3054) for bucket\r\n",
      "  Calculating Z arrays\r\n",
      "  Calculating Z arrays time: 00:00:00\r\n",
      "  Entering block accumulator loop:\r\n",
      "  10%\r\n",
      "  20%\r\n",
      "  30%\r\n",
      "  40%\r\n",
      "  50%\r\n",
      "  60%\r\n",
      "  70%\r\n",
      "  80%\r\n",
      "  90%\r\n",
      "  100%\r\n",
      "  Block accumulator loop time: 00:00:00\r\n",
      "  Sorting block of length 1839\r\n",
      "  (Using difference cover)\r\n",
      "  Sorting block time: 00:00:00\r\n",
      "Returning block of 1840\r\n",
      "Getting block 4 of 8\r\n",
      "  Reserving size (3054) for bucket\r\n",
      "  Calculating Z arrays\r\n",
      "  Calculating Z arrays time: 00:00:00\r\n",
      "  Entering block accumulator loop:\r\n",
      "  10%\r\n",
      "  20%\r\n",
      "  30%\r\n",
      "  40%\r\n",
      "  50%\r\n",
      "  60%\r\n",
      "  70%\r\n",
      "  80%\r\n",
      "  90%\r\n",
      "  100%\r\n",
      "  Block accumulator loop time: 00:00:00\r\n",
      "  Sorting block of length 2985\r\n",
      "  (Using difference cover)\r\n",
      "  Sorting block time: 00:00:00\r\n",
      "Returning block of 2986\r\n",
      "Getting block 5 of 8\r\n",
      "  Reserving size (3054) for bucket\r\n",
      "  Calculating Z arrays\r\n",
      "  Calculating Z arrays time: 00:00:00\r\n",
      "  Entering block accumulator loop:\r\n",
      "  10%\r\n",
      "  20%\r\n",
      "  30%\r\n",
      "  40%\r\n",
      "  50%\r\n",
      "  60%\r\n",
      "  70%\r\n",
      "  80%\r\n",
      "  90%\r\n",
      "  100%\r\n",
      "  Block accumulator loop time: 00:00:00\r\n",
      "  Sorting block of length 2663\r\n",
      "  (Using difference cover)\r\n",
      "  Sorting block time: 00:00:00\r\n",
      "Returning block of 2664\r\n",
      "Getting block 6 of 8\r\n",
      "  Reserving size (3054) for bucket\r\n",
      "  Calculating Z arrays\r\n",
      "  Calculating Z arrays time: 00:00:00\r\n",
      "  Entering block accumulator loop:\r\n",
      "  10%\r\n",
      "  20%\r\n",
      "  30%\r\n",
      "  40%\r\n",
      "  50%\r\n",
      "  60%\r\n",
      "  70%\r\n",
      "  80%\r\n",
      "  90%\r\n",
      "  100%\r\n",
      "  Block accumulator loop time: 00:00:00\r\n",
      "  Sorting block of length 597\r\n",
      "  (Using difference cover)\r\n",
      "  Sorting block time: 00:00:00\r\n",
      "Returning block of 598\r\n",
      "Getting block 7 of 8\r\n",
      "  Reserving size (3054) for bucket\r\n",
      "  Calculating Z arrays\r\n",
      "  Calculating Z arrays time: 00:00:00\r\n",
      "  Entering block accumulator loop:\r\n",
      "  10%\r\n",
      "  20%\r\n",
      "  30%\r\n",
      "  40%\r\n",
      "  50%\r\n",
      "  60%\r\n",
      "  70%\r\n",
      "  80%\r\n",
      "  90%\r\n",
      "  100%\r\n",
      "  Block accumulator loop time: 00:00:00\r\n",
      "  Sorting block of length 2605\r\n",
      "  (Using difference cover)\r\n",
      "  Sorting block time: 00:00:00\r\n",
      "Returning block of 2606\r\n",
      "Getting block 8 of 8\r\n",
      "  Reserving size (3054) for bucket\r\n",
      "  Calculating Z arrays\r\n",
      "  Calculating Z arrays time: 00:00:00\r\n",
      "  Entering block accumulator loop:\r\n",
      "  10%\r\n",
      "  20%\r\n",
      "  30%\r\n",
      "  40%\r\n",
      "  50%\r\n",
      "  60%\r\n",
      "  70%\r\n",
      "  80%\r\n",
      "  90%\r\n",
      "  100%\r\n",
      "  Block accumulator loop time: 00:00:00\r\n",
      "  Sorting block of length 1572\r\n",
      "  (Using difference cover)\r\n",
      "  Sorting block time: 00:00:00\r\n",
      "Returning block of 1573\r\n",
      "Exited Ebwt loop\r\n",
      "fchr[A]: 0\r\n",
      "fchr[C]: 1929\r\n",
      "fchr[G]: 7764\r\n",
      "fchr[T]: 13716\r\n",
      "fchr[$]: 16288\r\n",
      "Exiting Ebwt::buildToDisk()\r\n",
      "Returning from initFromVector\r\n",
      "Wrote 4199959 bytes to primary EBWT file: hgr1.1.bt2\r\n",
      "Wrote 4080 bytes to secondary EBWT file: hgr1.2.bt2\r\n",
      "Re-opening _in1 and _in2 as input streams\r\n",
      "Returning from Ebwt constructor\r\n",
      "Headers:\r\n",
      "    len: 16288\r\n",
      "    bwtLen: 16289\r\n",
      "    sz: 4072\r\n",
      "    bwtSz: 4073\r\n",
      "    lineRate: 6\r\n",
      "    offRate: 4\r\n",
      "    offMask: 0xfffffff0\r\n",
      "    ftabChars: 10\r\n",
      "    eftabLen: 20\r\n",
      "    eftabSz: 80\r\n",
      "    ftabLen: 1048577\r\n",
      "    ftabSz: 4194308\r\n",
      "    offsLen: 1019\r\n",
      "    offsSz: 4076\r\n",
      "    lineSz: 64\r\n",
      "    sideSz: 64\r\n",
      "    sideBwtSz: 48\r\n",
      "    sideBwtLen: 192\r\n",
      "    numSides: 85\r\n",
      "    numLines: 85\r\n",
      "    ebwtTotLen: 5440\r\n",
      "    ebwtTotSz: 5440\r\n",
      "    color: 0\r\n",
      "    reverse: 0\r\n",
      "Total time for call to driver() for forward index: 00:00:00\r\n",
      "Reading reference sizes\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "Warning: Encountered empty reference sequence\r\n",
      "  Time reading reference sizes: 00:00:00\r\n",
      "Calculating joined length\r\n",
      "Writing header\r\n",
      "Reserving space for joined string\r\n",
      "Joining reference sequences\r\n",
      "  Time to join reference sequences: 00:00:00\r\n",
      "  Time to reverse reference sequence: 00:00:00\r\n",
      "bmax according to bmaxDivN setting: 4072\r\n",
      "Using parameters --bmax 3054 --dcv 1024\r\n",
      "  Doing ahead-of-time memory usage test\r\n",
      "  Passed!  Constructing with these parameters: --bmax 3054 --dcv 1024\r\n",
      "Constructing suffix-array element generator\r\n",
      "Building DifferenceCoverSample\r\n",
      "  Building sPrime\r\n",
      "  Building sPrimeOrder\r\n",
      "  V-Sorting samples\r\n",
      "  V-Sorting samples time: 00:00:00\r\n",
      "  Allocating rank array\r\n",
      "  Ranking v-sort output\r\n",
      "  Ranking v-sort output time: 00:00:00\r\n",
      "  Invoking Larsson-Sadakane on ranks\r\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:00\r\n",
      "  Sanity-checking and returning\r\n",
      "Building samples\r\n",
      "Reserving space for 12 sample suffixes\r\n",
      "Generating random suffixes\r\n",
      "QSorting 12 sample offsets, eliminating duplicates\r\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\r\n",
      "Multikey QSorting 12 samples\r\n",
      "  (Using difference cover)\r\n",
      "  Multikey QSorting samples time: 00:00:00\r\n",
      "Calculating bucket sizes\r\n",
      "  Binary sorting into buckets\r\n",
      "  10%\r\n",
      "  20%\r\n",
      "  30%\r\n",
      "  40%\r\n",
      "  50%\r\n",
      "  60%\r\n",
      "  70%\r\n",
      "  80%\r\n",
      "  90%\r\n",
      "  100%\r\n",
      "  Binary sorting into buckets time: 00:00:00\r\n",
      "Splitting and merging\r\n",
      "  Splitting and merging time: 00:00:00\r\n",
      "Split 1, merged 6; iterating...\r\n",
      "  Binary sorting into buckets\r\n",
      "  10%\r\n",
      "  20%\r\n",
      "  30%\r\n",
      "  40%\r\n",
      "  50%\r\n",
      "  60%\r\n",
      "  70%\r\n",
      "  80%\r\n",
      "  90%\r\n",
      "  100%\r\n",
      "  Binary sorting into buckets time: 00:00:00\r\n",
      "Splitting and merging\r\n",
      "  Splitting and merging time: 00:00:00\r\n",
      "Avg bucket size: 2035.12 (target: 3053)\r\n",
      "Converting suffix-array elements to index image\r\n",
      "Allocating ftab, absorbFtab\r\n",
      "Entering Ebwt loop\r\n",
      "Getting block 1 of 8\r\n",
      "  Reserving size (3054) for bucket\r\n",
      "  Calculating Z arrays\r\n",
      "  Calculating Z arrays time: 00:00:00\r\n",
      "  Entering block accumulator loop:\r\n",
      "  10%\r\n",
      "  20%\r\n",
      "  30%\r\n",
      "  40%\r\n",
      "  50%\r\n",
      "  60%\r\n",
      "  70%\r\n",
      "  80%\r\n",
      "  90%\r\n",
      "  100%\r\n",
      "  Block accumulator loop time: 00:00:00\r\n",
      "  Sorting block of length 2911\r\n",
      "  (Using difference cover)\r\n",
      "  Sorting block time: 00:00:00\r\n",
      "Returning block of 2912\r\n",
      "Getting block 2 of 8\r\n",
      "  Reserving size (3054) for bucket\r\n",
      "  Calculating Z arrays\r\n",
      "  Calculating Z arrays time: 00:00:00\r\n",
      "  Entering block accumulator loop:\r\n",
      "  10%\r\n",
      "  20%\r\n",
      "  30%\r\n",
      "  40%\r\n",
      "  50%\r\n",
      "  60%\r\n",
      "  70%\r\n",
      "  80%\r\n",
      "  90%\r\n",
      "  100%\r\n",
      "  Block accumulator loop time: 00:00:00\r\n",
      "  Sorting block of length 2208\r\n",
      "  (Using difference cover)\r\n",
      "  Sorting block time: 00:00:00\r\n",
      "Returning block of 2209\r\n",
      "Getting block 3 of 8\r\n",
      "  Reserving size (3054) for bucket\r\n",
      "  Calculating Z arrays\r\n",
      "  Calculating Z arrays time: 00:00:00\r\n",
      "  Entering block accumulator loop:\r\n",
      "  10%\r\n",
      "  20%\r\n",
      "  30%\r\n",
      "  40%\r\n",
      "  50%\r\n",
      "  60%\r\n",
      "  70%\r\n",
      "  80%\r\n",
      "  90%\r\n",
      "  100%\r\n",
      "  Block accumulator loop time: 00:00:00\r\n",
      "  Sorting block of length 2514\r\n",
      "  (Using difference cover)\r\n",
      "  Sorting block time: 00:00:00\r\n",
      "Returning block of 2515\r\n",
      "Getting block 4 of 8\r\n",
      "  Reserving size (3054) for bucket\r\n",
      "  Calculating Z arrays\r\n",
      "  Calculating Z arrays time: 00:00:00\r\n",
      "  Entering block accumulator loop:\r\n",
      "  10%\r\n",
      "  20%\r\n",
      "  30%\r\n",
      "  40%\r\n",
      "  50%\r\n",
      "  60%\r\n",
      "  70%\r\n",
      "  80%\r\n",
      "  90%\r\n",
      "  100%\r\n",
      "  Block accumulator loop time: 00:00:00\r\n",
      "  Sorting block of length 637\r\n",
      "  (Using difference cover)\r\n",
      "  Sorting block time: 00:00:00\r\n",
      "Returning block of 638\r\n",
      "Getting block 5 of 8\r\n",
      "  Reserving size (3054) for bucket\r\n",
      "  Calculating Z arrays\r\n",
      "  Calculating Z arrays time: 00:00:00\r\n",
      "  Entering block accumulator loop:\r\n",
      "  10%\r\n",
      "  20%\r\n",
      "  30%\r\n",
      "  40%\r\n",
      "  50%\r\n",
      "  60%\r\n",
      "  70%\r\n",
      "  80%\r\n",
      "  90%\r\n",
      "  100%\r\n",
      "  Block accumulator loop time: 00:00:00\r\n",
      "  Sorting block of length 2648\r\n",
      "  (Using difference cover)\r\n",
      "  Sorting block time: 00:00:00\r\n",
      "Returning block of 2649\r\n",
      "Getting block 6 of 8\r\n",
      "  Reserving size (3054) for bucket\r\n",
      "  Calculating Z arrays\r\n",
      "  Calculating Z arrays time: 00:00:00\r\n",
      "  Entering block accumulator loop:\r\n",
      "  10%\r\n",
      "  20%\r\n",
      "  30%\r\n",
      "  40%\r\n",
      "  50%\r\n",
      "  60%\r\n",
      "  70%\r\n",
      "  80%\r\n",
      "  90%\r\n",
      "  100%\r\n",
      "  Block accumulator loop time: 00:00:00\r\n",
      "  Sorting block of length 2008\r\n",
      "  (Using difference cover)\r\n",
      "  Sorting block time: 00:00:00\r\n",
      "Returning block of 2009\r\n",
      "Getting block 7 of 8\r\n",
      "  Reserving size (3054) for bucket\r\n",
      "  Calculating Z arrays\r\n",
      "  Calculating Z arrays time: 00:00:00\r\n",
      "  Entering block accumulator loop:\r\n",
      "  10%\r\n",
      "  20%\r\n",
      "  30%\r\n",
      "  40%\r\n",
      "  50%\r\n",
      "  60%\r\n",
      "  70%\r\n",
      "  80%\r\n",
      "  90%\r\n",
      "  100%\r\n",
      "  Block accumulator loop time: 00:00:00\r\n",
      "  Sorting block of length 2048\r\n",
      "  (Using difference cover)\r\n",
      "  Sorting block time: 00:00:00\r\n",
      "Returning block of 2049\r\n",
      "Getting block 8 of 8\r\n",
      "  Reserving size (3054) for bucket\r\n",
      "  Calculating Z arrays\r\n",
      "  Calculating Z arrays time: 00:00:00\r\n",
      "  Entering block accumulator loop:\r\n",
      "  10%\r\n",
      "  20%\r\n",
      "  30%\r\n",
      "  40%\r\n",
      "  50%\r\n",
      "  60%\r\n",
      "  70%\r\n",
      "  80%\r\n",
      "  90%\r\n",
      "  100%\r\n",
      "  Block accumulator loop time: 00:00:00\r\n",
      "  Sorting block of length 1307\r\n",
      "  (Using difference cover)\r\n",
      "  Sorting block time: 00:00:00\r\n",
      "Returning block of 1308\r\n",
      "Exited Ebwt loop\r\n",
      "fchr[A]: 0\r\n",
      "fchr[C]: 1929\r\n",
      "fchr[G]: 7764\r\n",
      "fchr[T]: 13716\r\n",
      "fchr[$]: 16288\r\n",
      "Exiting Ebwt::buildToDisk()\r\n",
      "Returning from initFromVector\r\n",
      "Wrote 4199959 bytes to primary EBWT file: hgr1.rev.1.bt2\r\n",
      "Wrote 4080 bytes to secondary EBWT file: hgr1.rev.2.bt2\r\n",
      "Re-opening _in1 and _in2 as input streams\r\n",
      "Returning from Ebwt constructor\r\n",
      "Headers:\r\n",
      "    len: 16288\r\n",
      "    bwtLen: 16289\r\n",
      "    sz: 4072\r\n",
      "    bwtSz: 4073\r\n",
      "    lineRate: 6\r\n",
      "    offRate: 4\r\n",
      "    offMask: 0xfffffff0\r\n",
      "    ftabChars: 10\r\n",
      "    eftabLen: 20\r\n",
      "    eftabSz: 80\r\n",
      "    ftabLen: 1048577\r\n",
      "    ftabSz: 4194308\r\n",
      "    offsLen: 1019\r\n",
      "    offsSz: 4076\r\n",
      "    lineSz: 64\r\n",
      "    sideSz: 64\r\n",
      "    sideBwtSz: 48\r\n",
      "    sideBwtLen: 192\r\n",
      "    numSides: 85\r\n",
      "    numLines: 85\r\n",
      "    ebwtTotLen: 5440\r\n",
      "    ebwtTotSz: 5440\r\n",
      "    color: 0\r\n",
      "    reverse: 1\r\n",
      "Total time for backward call to driver() for mirror index: 00:00:00\r\n"
     ]
    }
   ],
   "source": [
    "#!/bin/bash\n",
    "# crc_lines_align_hgr1.fa\n",
    "# rDNA alignment pipeline\n",
    "# for crc cell line data on local machine\n",
    "\n",
    "# Control Panel -------------------------------\n",
    "\n",
    "# Project Dir\n",
    "  BASE='/home/artem/Crown/data/crc_lines/'\n",
    "  cd $BASE\n",
    "\n",
    "# Sequencing Data\n",
    "  CRC_DIR='/home/artem/Crown/data/crc_lines/'\n",
    "  LIB_LIST='crc_lines_data.txt' # list of fastq files in hgr1 format\n",
    "  \n",
    "# CPU\n",
    "  THREADS='1'\n",
    "  \n",
    "# Initialize start-up sequence ----------------\n",
    "# Make working directory\n",
    "  mkdir -p align\n",
    "  mkdir -p flagstat\n",
    "\n",
    "#Resources\n",
    "  cp $CROWN/resources/hgr1/hgr1.fa ./\n",
    "  samtools faidx hgr1.fa\n",
    "  bowtie2-build hgr1.fa hgr1\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": []
    }
   ],
   "source": [
    "# script below originally interrupted\n",
    "# completed 3/5 alignments. restarting at 4/5\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Start Iteration:\r\n",
      "  rko_2\tRKO\tSRX1745244\tinVitro\tSRR3479760_1.fastq.gz\tSRR3479760_2.fastq.gz\r\n",
      "\r\n",
      "Lib: rko_2\r\n",
      "RGSM: RKO\r\n",
      "RGID: SRX1745244\r\n",
      "RGLB: SRX1745244\r\n",
      "RGPL: ILLUMINA\r\n",
      "RGPO: inVitro\r\n",
      "FQ: /home/artem/Crown/data/crc_lines//SRR3479760_1.fastq.gz /home/artem/Crown/data/crc_lines//SRR3479760_2.fastq.gz\r\n",
      "\r\n",
      "\r\n",
      "\r\n",
      "gzip: stdout: Broken pipe\r\n",
      "33883402 reads; of these:\r\n",
      "  33883402 (100.00%) were paired; of these:\r\n",
      "    33610018 (99.19%) aligned concordantly 0 times\r\n",
      "    273310 (0.81%) aligned concordantly exactly 1 time\r\n",
      "    74 (0.00%) aligned concordantly >1 times\r\n",
      "    ----\r\n",
      "    33610018 pairs aligned concordantly 0 times; of these:\r\n",
      "      13862 (0.04%) aligned discordantly 1 time\r\n",
      "    ----\r\n",
      "    33596156 pairs aligned 0 times concordantly or discordantly; of these:\r\n",
      "      67192312 mates make up the pairs; of these:\r\n",
      "        67180873 (99.98%) aligned 0 times\r\n",
      "        11408 (0.02%) aligned exactly 1 time\r\n",
      "        31 (0.00%) aligned >1 times\r\n",
      "0.86% overall alignment rate\r\n",
      "Start Iteration:\r\n",
      "  hke3_1\tHKE3\tSRX1745244\tinVitro\tSRR3479763_1.fastq.gz\tSRR3479763_2.fastq.gz\r\n",
      "\r\n",
      "Lib: hke3_1\r\n",
      "RGSM: HKE3\r\n",
      "RGID: SRX1745244\r\n",
      "RGLB: SRX1745244\r\n",
      "RGPL: ILLUMINA\r\n",
      "RGPO: inVitro\r\n",
      "FQ: /home/artem/Crown/data/crc_lines//SRR3479763_1.fastq.gz /home/artem/Crown/data/crc_lines//SRR3479763_2.fastq.gz\r\n",
      "\r\n",
      "\r\n",
      "\r\n",
      "gzip: stdout: Broken pipe\r\n",
      "67249613 reads; of these:\r\n",
      "  67249613 (100.00%) were paired; of these:\r\n",
      "    66802479 (99.34%) aligned concordantly 0 times\r\n",
      "    446941 (0.66%) aligned concordantly exactly 1 time\r\n",
      "    193 (0.00%) aligned concordantly >1 times\r\n",
      "    ----\r\n",
      "    66802479 pairs aligned concordantly 0 times; of these:\r\n",
      "      68840 (0.10%) aligned discordantly 1 time\r\n",
      "    ----\r\n",
      "    66733639 pairs aligned 0 times concordantly or discordantly; of these:\r\n",
      "      133467278 mates make up the pairs; of these:\r\n",
      "        133392604 (99.94%) aligned 0 times\r\n",
      "        74617 (0.06%) aligned exactly 1 time\r\n",
      "        57 (0.00%) aligned >1 times\r\n",
      "0.82% overall alignment rate\r\n"
     ]
    }
   ],
   "source": [
    "# ---------------------------------------------\n",
    "# SCRIPT LOOP ---------------------------------\n",
    "# ---------------------------------------------\n",
    "# For each line in input LIB_LIST; run the pipeline\n",
    "\n",
    "\n",
    "cat $LIB_LIST | while read LINE\n",
    "do\n",
    "    #Initialize Run\n",
    "    echo \"Start Iteration:\"\n",
    "    echo \"  $LINE\"\n",
    "    echo ''\n",
    "    \n",
    "    LIBRARY=$(echo $LINE | cut -f1 -d' ' -) # Library Name\n",
    "    RGSM=$(echo $LINE | cut -f2 -d' ' -)    # Sample / Patient Identifer\n",
    "    RGID=$(echo $LINE | cut -f3 -d' ' -)    # Read Group ID\n",
    "    RGLB=$(echo $LINE | cut -f3 -d' ' -)    # Library Name. Accession Number\n",
    "    RGPL='ILLUMINA'                   # Sequencing Platform.\n",
    "    RGPO=$(echo $LINE | cut -f4 -d' ' -)    # Patient Population\n",
    "\n",
    "    FASTQ1=$(echo $LINE | cut -f5 -d' ' -)  # Filename Read 1\n",
    "    FASTQ2=$(echo $LINE | cut -f6 -d' ' -)  # Filename Read 2  \n",
    "\n",
    "    FQ1=\"$CRC_DIR/$FASTQ1\"            # Fastq1 Filepath\n",
    "    FQ2=\"$CRC_DIR/$FASTQ2\"            # Fastq1 Filepath\n",
    "\n",
    "    echo \"Lib: $LIBRARY\"\n",
    "    echo \"RGSM: $RGSM\"\n",
    "    echo \"RGID: $RGID\"\n",
    "    echo \"RGLB: $RGLB\"\n",
    "    echo \"RGPL: $RGPL\"\n",
    "    echo \"RGPO: $RGPO\"\n",
    "    echo \"FQ: $FQ1 $FQ2\"\n",
    "    echo ''\n",
    "    echo ''  \n",
    "\n",
    "    # Extract Sequencing Run Info\n",
    "    RGPU=$(gzip -dc $FQ1 | head -n1 - | cut -f1 -d'.' | cut -f2 -d'@')\n",
    "\n",
    "    # Bowtie2: align to genome\n",
    "    bowtie2 --very-sensitive-local -p $THREADS --rg-id $RGID \\\n",
    "      --rg LB:$RGLB --rg SM:$RGSM \\\n",
    "      --rg PL:$RGPL --rg PU:$RGPU \\\n",
    "      -x hgr1 -1 $FQ1 -2 $FQ2 |\\\n",
    "      samtools view -bS - > aligned_unsorted.bam\n",
    "\n",
    "    # Calcualte library flagstats\n",
    "    samtools flagstat aligned_unsorted.bam > aligned_unsorted.flagstat\n",
    "\n",
    "\n",
    "\n",
    "    # Read Subset ------------------------------\n",
    "    # Extract mapped reads, and their unmapped pairs\n",
    "\n",
    "      # Extract Header\n",
    "      samtools view -H aligned_unsorted.bam > align.header.tmp\n",
    "\n",
    "      # Unmapped reads with mapped pairs\n",
    "      # Extract Mapped Reads\n",
    "      # and their unmapped pairs\n",
    "      samtools view -b -F 4 aligned_unsorted.bam > align.F4.bam #mapped\n",
    "      samtools view -b -f 4 -F 8 aligned_unsorted.bam > align.f4F8.bam #unmapped pairs\n",
    "\n",
    "      # What are the mapped readnames\n",
    "      samtools view align.F4.bam | cut -f1 - > read.names.tmp\n",
    "\n",
    "      # Extract mapped reads\n",
    "      samtools view align.F4.bam | grep -Ff read.names.tmp - > align.F4.tmp.sam\n",
    "\n",
    "\n",
    "      # Extract cases of read pairs mapped on edge of region of interest\n",
    "      # -------|======= R O I ======| ----------\n",
    "      # read:                  ====---====\n",
    "      samtools view align.F4.bam | grep -Ff read.names.tmp - > align.F4.tmp.sam\n",
    "\n",
    "      # Complete mapped reads list\n",
    "      #cut -f1 align.F4.tmp.sam > read.names.45s.long.tmp\n",
    "\n",
    "      # Extract unmapped reads with a mapped pair\n",
    "      samtools view align.f4F8.bam | grep -Ff read.names.tmp - > align.f4F8.tmp.sam\n",
    "\n",
    "      # Re-compile bam file\n",
    "      cat align.header.tmp align.F4.tmp.sam align.f4F8.tmp.sam | samtools view -bS - > align.hgr1.tmp.bam\n",
    "        samtools sort align.hgr1.tmp.bam -o align.hgr1.bam\n",
    "\n",
    "\n",
    "    # mv align.hgr1 align.hgr1.bam\n",
    "        samtools index align.hgr1.bam\n",
    "        samtools flagstat align.hgr1.bam > align.hgr1.flagstat\n",
    "\n",
    "      # Clean up \n",
    "      rm *tmp* align.F4.bam align.f4F8.bam\n",
    "\n",
    "    # Rename the hgr Bam files\n",
    "      mv align.hgr1.bam align/$LIBRARY.hgr1.bam\n",
    "      mv align.hgr1.bam.bai align/$LIBRARY.hgr1.bam.bai\n",
    "      mv align.hgr1.flagstat flagstat/$LIBRARY.hgr1.flagstat\n",
    "\n",
    "    # Rename the total Bam Files\n",
    "      rm aligned_unsorted.bam\n",
    "      mv aligned_unsorted.flagstat flagstat/$LIBRARY.flagstat\n",
    "\n",
    "done\n",
    "\n",
    "# Script complete"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Inspection of reads in IGV for macp\n",
    "\n",
    "#### HCT116\n",
    "\n",
    "```\n",
    "HCT116 - 1\n",
    "chr13:1,004,908\n",
    "<hr>Total count: 2596\n",
    "A      : 100  (4%,     53+,   47- )\n",
    "C      : 1185  (46%,     572+,   613- )\n",
    "G      : 110  (4%,     50+,   60- )\n",
    "T      : 1201  (46%,     516+,   685- )\n",
    "N      : 0\n",
    "---------------\n",
    "DEL: 101\n",
    "INS: 0\n",
    "\n",
    "HCT116 - 2\n",
    "chr13:1,004,908\n",
    "<hr>Total count: 4716\n",
    "A      : 145  (3%,     69+,   76- )\n",
    "C      : 1885  (40%,     862+,   1023- )\n",
    "G      : 191  (4%,     112+,   79- )\n",
    "T      : 2495  (53%,     1164+,   1331- )\n",
    "N      : 0\n",
    "---------------\n",
    "DEL: 214\n",
    "INS: 0\n",
    "```\n",
    "\n",
    "#### HKE\n",
    "\n",
    "```\n",
    "HKE - 1\n",
    "chr13:1,004,908\n",
    "<hr>Total count: 4312\n",
    "A      : 163  (4%,     99+,   64- )\n",
    "C      : 1696  (39%,     1038+,   658- )\n",
    "G      : 132  (3%,     86+,   46- )\n",
    "T      : 2321  (54%,     1467+,   854- )\n",
    "N      : 0\n",
    "---------------\n",
    "DEL: 222\n",
    "INS: 0\n",
    "```\n",
    "\n",
    "#### RKO\n",
    "\n",
    "```\n",
    "RKO - 1\n",
    "chr13:1,004,908\n",
    "<hr>Total count: 675\n",
    "A      : 21  (3%,     11+,   10- )\n",
    "C      : 206  (31%,     117+,   89- )\n",
    "G      : 18  (3%,     14+,   4- )\n",
    "T      : 430  (64%,     229+,   201- )\n",
    "N      : 0\n",
    "---------------\n",
    "DEL: 25\n",
    "INS: 0\n",
    "\n",
    "RKO - 2\n",
    "chr13:1,004,908\n",
    "<hr>Total count: 1537\n",
    "A      : 29  (2%,     12+,   17- )\n",
    "C      : 501  (33%,     250+,   251- )\n",
    "G      : 33  (2%,     15+,   18- )\n",
    "T      : 974  (63%,     435+,   539- )\n",
    "N      : 0\n",
    "---------------\n",
    "DEL: 63\n",
    "INS: 0\n",
    "```\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "2017-05-17T02:33:58 prefetch.2.8.2: 1) Downloading 'SRR3479756'...\r\n",
      "2017-05-17T02:33:58 prefetch.2.8.2:  Downloading via http...\r\n",
      "2017-05-17T04:26:09 prefetch.2.8.2: 1) 'SRR3479756' was downloaded successfully\r\n",
      "\r\n",
      "2017-05-17T04:26:26 prefetch.2.8.2: 1) Downloading 'SRR3479759'...\r\n",
      "2017-05-17T04:26:26 prefetch.2.8.2:  Downloading via http...\r\n",
      "2017-05-17T06:45:23 prefetch.2.8.2: 1) 'SRR3479759' was downloaded successfully\r\n",
      "Read 29388187 spots for SRR3479756\r\n",
      "Written 29388187 spots for SRR3479756\r\n",
      "Read 33832789 spots for SRR3479759\r\n",
      "Written 33832789 spots for SRR3479759\r\n"
     ]
    }
   ],
   "source": [
    "# Acquire Data set II\n",
    "# HCT116\n",
    "prefetch SRR3479756\n",
    "\n",
    "# RKO\n",
    "prefetch SRR3479759\n",
    "\n",
    "fastq-dump --gzip --split-files SRR3479756\n",
    "fastq-dump --gzip --split-files SRR3479759"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "hct116_3\tHCT116\tSRX1745244\tinVitro\tSRR3479756_1.fastq.gz\tSRR3479756_2.fastq.gz\r\n",
      "rko_3\tRKO\tSRX1745244\tinVitro\tSRR3479759_1.fastq.gz\tSRR3479759_2.fastq.gz\r\n",
      "Start Iteration:\r\n",
      "  rko_3\tRKO\tSRX1745244\tinVitro\tSRR3479759_1.fastq.gz\tSRR3479759_2.fastq.gz\r\n",
      "\r\n",
      "Lib: rko_3\r\n",
      "RGSM: RKO\r\n",
      "RGID: SRX1745244\r\n",
      "RGLB: SRX1745244\r\n",
      "RGPL: ILLUMINA\r\n",
      "RGPO: inVitro\r\n",
      "FQ: /home/artem/Crown/data/crc_lines//SRR3479759_1.fastq.gz /home/artem/Crown/data/crc_lines//SRR3479759_2.fastq.gz\r\n",
      "\r\n",
      "\r\n",
      "\r\n",
      "gzip: stdout: Broken pipe\r\n",
      "33832789 reads; of these:\r\n",
      "  33832789 (100.00%) were paired; of these:\r\n",
      "    33557460 (99.19%) aligned concordantly 0 times\r\n",
      "    275273 (0.81%) aligned concordantly exactly 1 time\r\n",
      "    56 (0.00%) aligned concordantly >1 times\r\n",
      "    ----\r\n",
      "    33557460 pairs aligned concordantly 0 times; of these:\r\n",
      "      15811 (0.05%) aligned discordantly 1 time\r\n",
      "    ----\r\n",
      "    33541649 pairs aligned 0 times concordantly or discordantly; of these:\r\n",
      "      67083298 mates make up the pairs; of these:\r\n",
      "        67072724 (99.98%) aligned 0 times\r\n",
      "        10555 (0.02%) aligned exactly 1 time\r\n",
      "        19 (0.00%) aligned >1 times\r\n",
      "0.88% overall alignment rate\r\n"
     ]
    }
   ],
   "source": [
    "# Project Dir\n",
    "  BASE='/home/artem/Crown/data/crc_lines/'\n",
    "  cd $BASE\n",
    "\n",
    "# Sequencing Data\n",
    "  CRC_DIR='/home/artem/Crown/data/crc_lines/'\n",
    "  LIB_LIST='crc_lines_data_2.txt' # list of fastq files in hgr1 format\n",
    "  cat $LIB_LIST\n",
    "  \n",
    "  # HCT116 - 3 ran and made, jupyter crashed afterwards\n",
    "  # use a bash termianl in teh future, jupyter is not as stable\n",
    "\n",
    "  LIB_LIST='crc_lines_data_3.txt' # list of fastq files in hgr1 format\n",
    "\n",
    "# CPU\n",
    "  THREADS='1'\n",
    "\n",
    "cat $LIB_LIST\n",
    "\n",
    "# ---------------------------------------------\n",
    "# SCRIPT LOOP ---------------------------------\n",
    "# ---------------------------------------------\n",
    "# For each line in input LIB_LIST; run the pipeline\n",
    "\n",
    "\n",
    "cat $LIB_LIST | while read LINE\n",
    "do\n",
    "    #Initialize Run\n",
    "    echo \"Start Iteration:\"\n",
    "    echo \"  $LINE\"\n",
    "    echo ''\n",
    "    \n",
    "    LIBRARY=$(echo $LINE | cut -f1 -d' ' -) # Library Name\n",
    "    RGSM=$(echo $LINE | cut -f2 -d' ' -)    # Sample / Patient Identifer\n",
    "    RGID=$(echo $LINE | cut -f3 -d' ' -)    # Read Group ID\n",
    "    RGLB=$(echo $LINE | cut -f3 -d' ' -)    # Library Name. Accession Number\n",
    "    RGPL='ILLUMINA'                   # Sequencing Platform.\n",
    "    RGPO=$(echo $LINE | cut -f4 -d' ' -)    # Patient Population\n",
    "\n",
    "    FASTQ1=$(echo $LINE | cut -f5 -d' ' -)  # Filename Read 1\n",
    "    FASTQ2=$(echo $LINE | cut -f6 -d' ' -)  # Filename Read 2  \n",
    "\n",
    "    FQ1=\"$CRC_DIR/$FASTQ1\"            # Fastq1 Filepath\n",
    "    FQ2=\"$CRC_DIR/$FASTQ2\"            # Fastq1 Filepath\n",
    "\n",
    "    echo \"Lib: $LIBRARY\"\n",
    "    echo \"RGSM: $RGSM\"\n",
    "    echo \"RGID: $RGID\"\n",
    "    echo \"RGLB: $RGLB\"\n",
    "    echo \"RGPL: $RGPL\"\n",
    "    echo \"RGPO: $RGPO\"\n",
    "    echo \"FQ: $FQ1 $FQ2\"\n",
    "    echo ''\n",
    "    echo ''  \n",
    "\n",
    "    # Extract Sequencing Run Info\n",
    "    RGPU=$(gzip -dc $FQ1 | head -n1 - | cut -f1 -d'.' | cut -f2 -d'@')\n",
    "\n",
    "    # Bowtie2: align to genome\n",
    "    bowtie2 --very-sensitive-local -p $THREADS --rg-id $RGID \\\n",
    "      --rg LB:$RGLB --rg SM:$RGSM \\\n",
    "      --rg PL:$RGPL --rg PU:$RGPU \\\n",
    "      -x hgr1 -1 $FQ1 -2 $FQ2 |\\\n",
    "      samtools view -bS - > aligned_unsorted.bam\n",
    "\n",
    "    # Calcualte library flagstats\n",
    "    samtools flagstat aligned_unsorted.bam > aligned_unsorted.flagstat\n",
    "\n",
    "\n",
    "\n",
    "    # Read Subset ------------------------------\n",
    "    # Extract mapped reads, and their unmapped pairs\n",
    "\n",
    "      # Extract Header\n",
    "      samtools view -H aligned_unsorted.bam > align.header.tmp\n",
    "\n",
    "      # Unmapped reads with mapped pairs\n",
    "      # Extract Mapped Reads\n",
    "      # and their unmapped pairs\n",
    "      samtools view -b -F 4 aligned_unsorted.bam > align.F4.bam #mapped\n",
    "      samtools view -b -f 4 -F 8 aligned_unsorted.bam > align.f4F8.bam #unmapped pairs\n",
    "\n",
    "      # What are the mapped readnames\n",
    "      samtools view align.F4.bam | cut -f1 - > read.names.tmp\n",
    "\n",
    "      # Extract mapped reads\n",
    "      samtools view align.F4.bam | grep -Ff read.names.tmp - > align.F4.tmp.sam\n",
    "\n",
    "\n",
    "      # Extract cases of read pairs mapped on edge of region of interest\n",
    "      # -------|======= R O I ======| ----------\n",
    "      # read:                  ====---====\n",
    "      samtools view align.F4.bam | grep -Ff read.names.tmp - > align.F4.tmp.sam\n",
    "\n",
    "      # Complete mapped reads list\n",
    "      #cut -f1 align.F4.tmp.sam > read.names.45s.long.tmp\n",
    "\n",
    "      # Extract unmapped reads with a mapped pair\n",
    "      samtools view align.f4F8.bam | grep -Ff read.names.tmp - > align.f4F8.tmp.sam\n",
    "\n",
    "      # Re-compile bam file\n",
    "      cat align.header.tmp align.F4.tmp.sam align.f4F8.tmp.sam | samtools view -bS - > align.hgr1.tmp.bam\n",
    "        samtools sort align.hgr1.tmp.bam -o align.hgr1.bam\n",
    "\n",
    "\n",
    "    # mv align.hgr1 align.hgr1.bam\n",
    "        samtools index align.hgr1.bam\n",
    "        samtools flagstat align.hgr1.bam > align.hgr1.flagstat\n",
    "\n",
    "      # Clean up \n",
    "      rm *tmp* align.F4.bam align.f4F8.bam\n",
    "\n",
    "    # Rename the hgr Bam files\n",
    "      mv align.hgr1.bam align/$LIBRARY.hgr1.bam\n",
    "      mv align.hgr1.bam.bai align/$LIBRARY.hgr1.bam.bai\n",
    "      mv align.hgr1.flagstat flagstat/$LIBRARY.hgr1.flagstat\n",
    "\n",
    "    # Rename the total Bam Files\n",
    "      rm aligned_unsorted.bam\n",
    "      mv aligned_unsorted.flagstat flagstat/$LIBRARY.flagstat\n",
    "\n",
    "done\n",
    "\n",
    "# Script complete"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": []
    }
   ],
   "source": [
    "# Clean-up directory\n",
    "\n",
    "mkdir fq\n",
    "\n",
    "mv *.fastq.gz fq/\n",
    "mv *.txt fq/\n",
    "\n",
    "rm hgr1*"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Results / Discussion\n",
    "\n",
    "### RAF at 1248U\n",
    "\n",
    "\n",
    "| Cell Line | rep 1 | rep 2 | rep 3 |\n",
    "|-----------|-------|-------|-------|\n",
    "| HCT116    |   46  |  53   |   51  |\n",
    "| RKO       |   64  |  63   |   56  |\n",
    "| HKE3      |   54  |       |       |\n",
    "\n",
    "\n",
    "RKO is a good candidate for having a hypo-modification phenotype (Student's T-test between HCT116 and RKO, P = 0.028. All samples are essentially negative for 18S-E precursor so it's not an apparent biogenesis phenotype. Confirm in the lab is the next move\n",
    "\n",
    "\n",
    "### 18S.990A > G variant\n",
    "\n",
    "While inspecting the RNA-seq libraries in IGV there was a notable difference between HCT116 and RKO libraries at `chr13:1004650`. A ~10% expressed variant at a conserved position (and protein contact) 18S.990A>G in HCT116.\n",
    "\n",
    "![IGV view of 990A](../../data/crc_lines/plot/170519_18S.A990G_hct116_variant.png)\n",
    "\n",
    "This position is highly conserved (although not perfectly) in eukaryotes but not prokaryotes/archae. Sequence context is:\n",
    "\n",
    "`gacggacc A gagcgaaag`\n",
    "\n",
    "![Conservation](../../data/crc_lines/plot/170519_18S.A990G.eukAlgn.png)\n",
    "\n",
    "It's involved in secondary structure (helix 21a) with base-pairing to A1001 within 18S.\n",
    "\n",
    "![18S helix 21a](../../data/crc_lines/plot/170519_18S.A990_2ndStruc.png)\n",
    "\n",
    "From the 4UG0 structure, it looks like this a change to G at this position would disrupt the base-pairing interaction although this would require simulation/structural analysis to see if there's another local minima by shifting things around.\n",
    "\n",
    "![helix 21 structure](../../data/crc_lines/plot/170519_990A.hct116.png)\n",
    "\n",
    "---\n",
    "\n",
    "This variant is rare in the DNA-seq dataset (107 genomes) being called in only 2/107 genomes. So this is clearly a variant haplotype and likely has biochemical consequences. The protein chain in the above image is S3a, which wouldn't really be disrupted in the interaction but I'm thinking that entire helix 21 may fold differently.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Homo_sapiens_18S.990A\r\n",
      ">Mus_Musculus_18S.991A\r\n",
      ">C_elegans_18S.907A\r\n",
      ">D_melanogaster_18S.U1019\r\n",
      ">A_thaliana_18S.937A\r\n",
      ">S_cerevisae_18S.933A\r\n",
      ">E_coli_18S.722G\r\n"
     ]
    }
   ],
   "source": [
    "# For future reference the homologous positions of A990\n",
    "cat plot/170519_18S.990A.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "QED"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
