{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# CRC Differential Expression - Modification\n",
    "```\n",
    "pi:ababaian\n",
    "files: ~/Crown/data2/crcmod_dge\n",
    "start: 2017 08 23\n",
    "complete : YYYY MM DD\n",
    "```\n",
    "## Introduction\n",
    "\n",
    "20170825 Writeup\n",
    "\n",
    "Quite a simple experiment, in the CRC data there are two classes of cancers, those with 1248U hypo-modification and  \"normal\" levels of modification.\n",
    "\n",
    "The main motivation is trying to understand what the underlying reason or mechanism for this difference is. An obvious candidate would be a recurrent deficiency in nucleotide synthesis, methionine (SAM) metabolism, pseudouradine synthesases or something along those lines.\n",
    "\n",
    "Since the hypo-modification is continuous, the data was seperated into two groups based on ranking each CRC library. From the 69 CRC sequencing libraries, the top 15 hypomodified and bottom 15 controls were selected and their respective 'normal' tisue controls for a 2-way analysis (CRC_Hypo vs. CRC_Control & CRC vs. Normal).\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "## Objective\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Materials and Methods\n",
    "\n",
    "#### CRC Libraries\n",
    "The alignments were taken from the previous CRC hg19 alignments from the LIONS project (by MMK). These are from the genentech publication.\n",
    "```\n",
    "@PG\tID:TopHat\tVN:2.0.8\tCL:/home/ababaian/software/tophat-2.0.8/tophat -p 6 -r 200 --report-secondary-alignments -o /scratch/epigenomics/mkarimi/human/libs/587221/ /home/ababaian/resources/hg19r/hg19r /scratch/epigenomics/mkarimi/human/Colon/587221_1_1.fastq.gz /scratch/epigenomics/mkarimi/human/Colon/587221_1_2.fastq.gz\n",
    "```\n",
    "\n",
    "ERRORS\n",
    "```\n",
    "1 587271 SEGMENTATION FAULT\n",
    "2 587290 ALIGNMENT FILE MISSING -- Hypo\n",
    "3 587288 ALIGNMENT FILE CORRUPTED -- Hypo\n",
    "4 587328 ALIGNMENT FILE CORRUPTED -- Ctrl\n",
    "```\n",
    "\n",
    "#### Ranking CRC Libraries\n",
    "In `crcmod30_dge.xlsx` all the libraries were ranked based on their 1248U Reference Allele Frequency. Two crc15 libraries (*) had errors in the bam files and were omitted, so the next two on the ranked list were taken instead.\n",
    "\n",
    "The respective normal libraries are one number greater then the ones listed below.\n",
    "\n",
    "*crc15_hypo*\n",
    "```\n",
    "587264\n",
    "587302\n",
    "587384\n",
    "587290*\n",
    "587270\n",
    "587288*\n",
    "587344\n",
    "587278\n",
    "587254\n",
    "587260\n",
    "587380\n",
    "587346\n",
    "587292\n",
    "587350\n",
    "587386\n",
    "587268\n",
    "587322\n",
    "```\n",
    "*crc15_ctrl*\n",
    "```\n",
    "587300\n",
    "587348\n",
    "587340\n",
    "587360\n",
    "587358\n",
    "587392\n",
    "587338\n",
    "587236\n",
    "587342\n",
    "587294\n",
    "587372\n",
    "587400\n",
    "587230\n",
    "587398\n",
    "587316\n",
    "```\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "`runDiff_crc30.sh`\n",
    "\n",
    "```\n",
    "#!/bin/bash\n",
    "\n",
    "cd /home/ababaian/projects/Colon_DGE/\n",
    "\n",
    "touch RUNNING\n",
    "\n",
    "# CuffDiff on all Cancer Samples\n",
    "\n",
    "\n",
    "cuffdiff -p 8 -o CRCMOD30/ -L norm15_ctrl,norm15_hypo,crc15_ctrl,crc15_hypo \\\n",
    "gencode.v26lift37.basic.annotation.gtf \\\n",
    "$(cat norm15_ctrl.bams) \\\n",
    "$(cat norm15_hypo.bams) \\\n",
    "$(cat crc15_ctrl.bams)  \\\n",
    "$(cat crc15_hypo.bams)\n",
    "```\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "`crc30_analysis.r`\n",
    "\n",
    "```\n",
    "# crc30_analysis.r\n",
    "\n",
    "library(cummeRbund)\n",
    "\n",
    "# CRC Analysis ==============================================================\n",
    "cuff = readCufflinks(dir='./cufflinks/', genome = 'hg38' , gtfFile = \"gencode.v26lift37.basic.annotation.gtf\")\n",
    "cuff\n",
    "\n",
    "#Below command is equivalent to looking at the gene_exp.diff file that we spent a lot of time parsing yesterday\n",
    "  gene_diff_data <- diffData(genes(cuff))   \n",
    "\n",
    "#Do gene_diff_data followed by tab to see all the variables in this data object\n",
    "  sig_gene_data  <- subset(gene_diff_data,\n",
    "                           #(sample_1 == 'crc15_ctrl'),\n",
    "                           #(sample_2 == 'crc15_hypo'),\n",
    "                           (significant ==  'yes'),\n",
    "                           level='genes')\n",
    "  \n",
    "  # CRC ctrl vs. CRC hypo comparison\n",
    "  crcmod_gene_data = gene_diff_data[(gene_diff_data$sample_1 == 'crc15_ctrl' & gene_diff_data$sample_2 == 'crc15_hypo'),]\n",
    "  \n",
    "  crcmod_sig_data =  subset(crcmod_gene_data,\n",
    "                            (significant ==  'yes'),\n",
    "                            level='genes')\n",
    "\n",
    "  write.table(crcmod_sig_data, file = 'crc30_mod.csv', quote = F,sep = '\\t')\n",
    "  \n",
    "  \n",
    "#Retrive significant gene IDs (XLOC) with a pre-specified alpha\n",
    "#Use returned identifiers to create a CuffGeneSet object with all relevant info for given genes\n",
    "  diffGeneIDs <- getSig(cuff, level=\"genes\", alpha=0.05)\n",
    "  diffGenes <-   getGenes(cuff, diffGeneIDs)\n",
    "\n",
    "# Add Feature Data\n",
    "  CRC30 = cbind(featureNames(diffGenes)[,2], sig_gene_data)\n",
    "  rm(sig_gene_data, gene_diff_data, diffGenes, diffGeneIDs, cuff)\n",
    "\n",
    "  write.table( sig_gene_data, file = 'crc30_diffExp.csv', append = F, quote = F, sep ='\\t', row.names = F)\n",
    "  \n",
    "\n",
    "# Analysis  ==============================================================\n",
    "\n",
    "#Below command is equivalent to looking at the gene_exp.diff file that we spent a lot of time parsing yesterday\n",
    "gene_diff_data <- diffData(genes(cuff))   \n",
    "\n",
    "#Do gene_diff_data followed by tab to see all the variables in this data object\n",
    "sig_gene_data  <- subset(gene_diff_data, (significant ==  'yes'))\n",
    "\n",
    "#Retrive significant gene IDs (XLOC) with a pre-specified alpha\n",
    "#Use returned identifiers to create a CuffGeneSet object with all relevant info for given genes\n",
    "diffGeneIDs <- getSig(cuff,level=\"genes\", alpha=0.05)\n",
    "diffGenes <- getGenes(cuff,diffGeneIDs)\n",
    "\n",
    "# Plot all pair-wise comparisons    \n",
    "mySigMat <- sigMatrix(cuff, level=\"genes\", alpha=0.05) # constructs the plot\n",
    "mySigMat\n",
    "\n",
    "\n",
    "myGene <- getGene(cuff, \"RPPH1\")\n",
    "#  expressionPlot(myGene) # Will collapse replicates and only show gene level FPKM\n",
    "#  expressionPlot(myGene, replicates=T) # Will show replicate FPKMs\n",
    "  expressionPlot(myGene, replicates=T, logMode = T ) # Show as bar plot instead\n",
    "  \n",
    "#myGene <- getGene(cuff, \"SI\")\n",
    "#  expressionPlot(myGene, replicates=T,logMode = T ) # Show as bar plot instead\n",
    "# \n",
    "```\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Results\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### lncRNA genes\n",
    "```\n",
    "gene_id\tGene name\tGene description\tsample_1\tsample_2\tstatus\tvalue_1\tvalue_2\tlog2_fold_change\ttest_stat\tp_value\tq_value\tsignificant\n",
    "ENSG00000269900\tRMRP\tRNA component of mitochondrial RNA processing endoribonuclease [Source:HGNC Symbol;Acc:HGNC:10031]\tcrc15_ctrl\tcrc15_hypo\tOK\t681.025\t41.9002\t-4.02268\t-8.96698\t5.00E-05\t0.000968818\tyes\n",
    "ENSG00000259001\tAL355075.4\tRNase P -- RPPH1(ENSG00000252678.1)\tcrc15_ctrl\tcrc15_hypo\tOK\t187.579\t6.17257\t-4.92549\t-2.37484\t0.00015\t0.00249348\tyes\n",
    "RNU4-2\tRNA, U4 small nuclear 2 [Source:HGNC Symbol;Acc:HGNC:10193]\tcrc15_ctrl\tcrc15_hypo\tOK\t743.103\t20.466\t-5.18226\t-69.1121\t9.00E-04\t0.0107869\tyes\n",
    "SNORD96A\tsmall nucleolar RNA, C/D box 96A [Source:HGNC Symbol;Acc:HGNC:32758]\tcrc15_ctrl\tcrc15_hypo\tOK\t1126.22\t1081.91\t-0.0578974\t-5.56E-05\t0.001\t0.0117372\tyes\n",
    "MT-RNR1\tmitochondrially encoded 12S RNA [Source:HGNC Symbol;Acc:HGNC:7470]\tcrc15_ctrl\tcrc15_hypo\tOK\t680.171\t281.846\t-1.27099\t-1.13E+00\t0.00135\t0.0148756\tyes\n",
    "```\n",
    "\n",
    "One discriminating gene *RPPH1* is upregulated quite highly in CRC-control libraries. This is an RNAzyme, component of RNAseP and may play a role in rRNA processing and definetly in tRNA processing.\n",
    "\n",
    "![RPHH1 CRC30](../../data2/crcmod_dge/plot/RPPH1_crc30.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### DAVID\n",
    "\n",
    "All the output genes either Upreglated or Downregulated were put into DAVID 6.8 (https://david.ncifcrf.gov/summary.jsp).\n",
    "\n",
    "Relating to metabolism as defined by the original purpose there is downregulation of 'Triglyceride catabolism' in 1248U hypo-modified libraries. Notably defined by the downregulation of FABP1,2,3. \n",
    "```\n",
    "Category\tTerm\tCount\t%\tPValue\tGenes\tList Total\tPop Hits\tPop Total\tFold Enrichment\tBonferroni\tBenjamini\tFDR\n",
    "GOTERM_BP_DIRECT\tGO:0019433~triglyceride catabolic process\t5\t3.448275862\t3.18E-05\tAPOB, FABP3, FABP1, FABP2, CPS1\t125\t25\t16792\t26.8672\t0.020225308\t0.020225308\t0.047365501\n",
    "```\n",
    "There is also non-significant, although enriched Retinol metabolism downregulation.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Cox survival curve on RPPH1, using top 33 and bottom 33 percentiles on oncolnc.org\n",
    "http://www.oncolnc.org/kaplan/?lower=33&upper=33&cancer=COAD&gene_id=85495&raw=RPPH1&species=mRNA\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Discussion\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
