{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# hgr1 TCGA Pilot Experiment\n",
    "```\n",
    "pi:ababaian\n",
    "files: ~/Crown/data2/tcga_0\n",
    "start: 2018 08 14\n",
    "complete : 2018 08 16\n",
    "```\n",
    "## Introduction\n",
    "\n",
    "Pilot run script of hgr1-alignment of TCGA RNA-seq data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Materials and Methods\n",
    "\n",
    "### DRAFT Scripts"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### queenB.sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "code_folding": [
     0
    ]
   },
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "# queenB.sh\n",
    "# 20180814 build\n",
    "# EC2 Launch / Control Script\n",
    "#\n",
    "\n",
    "# 1. queenB script is initialized locally and input files\n",
    "#    are parsed ready for cluster analaysis\n",
    "# 2. queenB launches instances, logs in to it and runs the\n",
    "#    droneB.sh script remotely.\n",
    "# 3. The droneB script is executed on the instance and it\n",
    "#    launches a `screen` on the instance and loads and \n",
    "#    starts to perform the $TASK (gather.sh) script.\n",
    "# 4. TASK script should include a instance shut-down\n",
    "#    command to close instance upon completion.\n",
    "#\n",
    "\n",
    "# Control Panel =========================\n",
    "# EC2 TASK Script - script for droneB to execute\n",
    "TASK=\"s3://crownproject/scripts/hgr1_align_v2.tcga.0.sh\"\n",
    "\n",
    "# Parameter file:\n",
    "# Each line of PARAMETERS will be input to STDIN of\n",
    "# the droneB script which can then be used to run the\n",
    "# TASK script.\n",
    "# i.e. bash droneB.sh <line_N_of_PARAMETERS>\n",
    "# PARAMETERS=\"tcga0_input.txt\"\n",
    "PARAMETERS=$1\n",
    "\n",
    "# EC2 Set-up\n",
    "instanceTYPE='c4.xlarge'\n",
    "imageID='ami-0031fd61f932bdef9' #AMI TCGA\n",
    "\n",
    "devNAME='/dev/sda1' # /dev/sda1 for Crown-AMI\n",
    "volSIZE='200' # in Gb\n",
    "\n",
    "# Number of instances to launch\n",
    "#COUNT=2 # predetermined number\n",
    "COUNT=$(wc -l $PARAMETERS | cut -f 1 -d' ' ) # for each input argument\n",
    "\n",
    "# Security\n",
    "keyNAME='CrownKey'\n",
    "keyPATH=\"/home/artem/.ssh/CrownKey.pem\"\n",
    "secGROUP='crown-group'\n",
    "\n",
    "# =======================================\n",
    "\n",
    "for ITER in $(seq 1 $COUNT)\n",
    "do\n",
    "\n",
    "  # Extract Parameters/Arguments ----------\n",
    "\n",
    "  ARGS=$(sed -n \"$ITER\"p $PARAMETERS | sed 's/\\t/ /g' - )\n",
    "\n",
    "  echo \"Launch instance # $ITER\"\n",
    "  echo \"Instance Type: $instanceTYPE\"\n",
    "  echo \"AMI Image: $imageID\"\n",
    "  echo \"Run Script: $TASK\"\n",
    "  echo \"Parameters: $ARGS\"\n",
    "\n",
    "  # Launch an instance --------------------\n",
    "  # NOTE: each iteration of the for loop launches one instance\n",
    "  # therefore each loop launches only one instance\n",
    "  aws ec2 run-instances --image-id $imageID --count 1 \\\n",
    "   --instance-type $instanceTYPE --key-name $keyNAME \\\n",
    "   --block-device-mappings DeviceName=$devNAME,Ebs={VolumeSize=$volSIZE} \\\n",
    "   --security-groups $secGROUP > launch.tmp\n",
    "\n",
    "  # Another alternative is to use --user-data droneB.sh \n",
    "  # which will run at instance boot-up\n",
    "  # passing arguments to it may be challenging\n",
    "\n",
    "  # Retrieve instance ID\n",
    "  instanceID=$(cat launch.tmp | \\\n",
    "    egrep -o -e 'InstanceId[\":/A-Za-z0-9_ \\\\-]*' - |\\\n",
    "    cut -f2 -d' ' - | xargs)\n",
    "\n",
    "  echo \"Instance ID: $instanceID\"\n",
    "\n",
    "\n",
    "  # Add a few minute wait here to allow for Public DNS to be assigned\n",
    "  # otherwise ssh doesn't work\n",
    "  sleep 180s\n",
    "\n",
    "  # Retrieve public DNS\n",
    "  aws ec2 describe-instances --instance-ids $instanceID > launch2.tmp\n",
    "\n",
    "  pubDNS=$(cat launch2.tmp | \\\n",
    "    egrep -o -m 1 -e 'PublicDnsName[.\":/A-Za-z0-9_ \\\\-]*' - |\\\n",
    "    cut -f2 -d' ' - | xargs)\n",
    "\n",
    "  echo \"Public DNS: $pubDNS\"\n",
    "\n",
    "  # Access the instance -------------------\n",
    "\n",
    "  LOGIN=\"ubuntu@$pubDNS\" \n",
    "\n",
    "  ssh -i $keyPATH \\\n",
    "    -o StrictHostKeyChecking=no \\\n",
    "    $LOGIN 'bash -s' < droneB.sh $TASK $(echo $ARGS)\n",
    "\n",
    "  # Cleanup\n",
    "  rm *.tmp\n",
    "\n",
    "  echo ''\n",
    "  echo ''\n",
    "\n",
    "done\n",
    "\n",
    "# end of script"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### droneB.sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "code_folding": [
     0
    ]
   },
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "# droneB.sh\n",
    "#\n",
    "\n",
    "# This script-layer is neccesary to launch a screen session\n",
    "# on each ec2-machine. The pipeline is run within that session\n",
    "# and the output is logged. This allows 'looking in' on sessions\n",
    "# as they are running.\n",
    "\n",
    "# Commands to run on server-side\n",
    "# ===============================================================\n",
    "\n",
    "SCRIPTPATH=$1\n",
    "\n",
    "SCRIPT=$(basename $1)\n",
    "\n",
    "shift # drop first (TASK or SCRIPT variable)\n",
    "\n",
    "# Download pipeline / droneB's function\n",
    "  aws s3 cp $SCRIPTPATH ./\n",
    "\n",
    "  chmod 777 *.sh\n",
    "\n",
    "# open screen; run gather.sh function. -L logged\n",
    "  screen -Ldmt sh ~/$SCRIPT $@\n",
    "\n",
    "\n",
    "# ===============================================================\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### hgr1_align_v2.tcga.0.sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "code_folding": [
     0
    ]
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "^C\r\n"
     ]
    }
   ],
   "source": [
    "#!/bin/bash\n",
    "# 1kg_align_v2.tcga.0.sh\n",
    "# rDNA alignment pipeline\n",
    "# 180813 build -- TCGA\n",
    "# AMI: crown-180813 - ami-0031fd61f932bdef9\n",
    "# EC2: c4.2xlarge (8cpu / 15 gb)\n",
    "# EC2: c4.xlarge  (4cpu / 8  gb)\n",
    "# Storage: 300 Gb\n",
    "#\n",
    "\n",
    "# Input Requirements --------------------------\n",
    "\n",
    "# $1 : Library name and file-output name\n",
    "# $2 : Library population/analysis set\n",
    "# $3 : Library UUID\n",
    "\n",
    "# Control Panel -------------------------------\n",
    "# CPU\n",
    "\tTHREADS='3'\n",
    "\n",
    "# Sequencing Data\n",
    "\tLIBRARY=$1 # Library/ File name\n",
    "\n",
    "# TCGA FILE UUID\n",
    "  UUID=$3\n",
    "\n",
    " # FastQ File-names\n",
    "    FQ0=\"$LIBRARY.tmp.sort.0.fq\"\n",
    "    FQ1=\"$LIBRARY.tmp.sort.1.fq\"\n",
    "    FQ2=\"$LIBRARY.tmp.sort.2.fq\"\n",
    "    \n",
    "# Read Group Data\n",
    "# Extract from downloaded BAM file / input\n",
    "\tRGPO=$2 # Patient Population\n",
    "\n",
    "\t#RGSM= # Sample. Patient Identifer\n",
    "\t#RGID= # Read Group ID. Accession Number\n",
    "    \n",
    "\tRGLB=$LIBRARY # Library Name. Accession Number\n",
    "\tRGPL='ILLUMINA'  # Sequencing Platform.\n",
    "    \n",
    "\t# Extract Sequencing Run Info\n",
    "\t#  RGPU=$(gzip -dc $FQ1 | head -n1 - | cut -f1 -d':' | cut -f2 -d' ')\n",
    "\n",
    "# Initialize wordir ---------------------------\n",
    "\n",
    "# Make working directory\n",
    "  mkdir -p align; cd align\n",
    "\n",
    "# Copy hgrX genome and create bowtie2 index\n",
    "  cp ~/resources/hgr1/* ./\n",
    "  \n",
    "# Download RNAseq BAM file\n",
    "# with a UUID as input\n",
    "  gdc-client download -t ~/resources/gdc.token -d ./ \\\n",
    "  -n $THREADS\\\n",
    "  $FILE_UUID\n",
    "  \n",
    "# Link the RNA-seq bamfile which is called by its UID to workdir\n",
    "  ln -s */*.bam input.bam\n",
    "  \n",
    "# Extract ReadGroup Sample Name (SM)\n",
    "  RGSM=$(samtools view -H input.bam | grep '^@RG' | sed \"s/.*SM:\\([^\\t]*\\).*/\\1/g\" | uniq )\n",
    "\n",
    "# Extract ReadGroup identifier (ID)\n",
    "  RGID=$(samtools view -H input.bam | grep '^@RG' | sed \"s/.*ID:\\([^\\t]*\\).*/\\1/g\" | uniq )\n",
    "\n",
    "# Convert input bam file to fastq files for re-alignment\n",
    "samtools sort -@ $THREADS-n input.bam | \\\n",
    "    samtools fastq -@ $THREADS -O \\\n",
    "    -0 $FQ0 \\\n",
    "    -1 $FQ1 \\\n",
    "    -2 $FQ2 -\n",
    "\n",
    "# SINGLE END READS ====================================================\n",
    "\n",
    "if [ -s $FQ0 ]\n",
    "then\n",
    "    # Single-End Extracted Reads Alignment\n",
    "\n",
    "    # Extract Sequencing Run Info\n",
    "    #RGPU=$(gzip -dc $FQ0| head -n1 - | cut -f1 -d':' | cut -f2 -d' ')\n",
    "    RGPU=$(head -n1 $FQ0 | cut -f1 -d':' | cut -f2 -d' ')\n",
    "\n",
    "    # Bowtie2: align to genome\n",
    "    bowtie2 --very-sensitive-local -p $THREADS \\\n",
    "      --rg-id $RGID --rg LB:$RGLB --rg SM:$RGSM \\\n",
    "      --rg PL:$RGPL --rg PU:$RGPU \\\n",
    "      -x hgr1 -U $FQ0 | \\\n",
    "      samtools view -bS - > aligned_unsorted.bam\n",
    "     \n",
    "    # Calculate library flagstats\n",
    "      samtools flagstat aligned_unsorted.bam > aligned_unsorted.flagstat\n",
    "      rm $FQ0 # Remove fastq files to save space\n",
    "\n",
    "    # Read Subset ------------------------------\n",
    "    # Extract mapped reads, and their unmapped pairs\n",
    "\n",
    "      # Extract Header\n",
    "      samtools view -H aligned_unsorted.bam > align.header.tmp\n",
    "\n",
    "      # Extract Mapped Reads\n",
    "      samtools view -b -F 4 aligned_unsorted.bam | \\\n",
    "      samtools sort > align.hgr1.bam #mapped\n",
    "      \n",
    "    # Calcualte library flagstats\n",
    "    samtools index align.hgr1.bam\n",
    "    samtools flagstat align.hgr1.bam > align.hgr1.flagstat\n",
    "\n",
    "    # Rename the total Bam Files\n",
    "      #mv aligned_unsorted.bam $LIBRARY.se.bam\n",
    "      #mv aligned_unsorted.bam.bai $LIBRARY.se.bam.bai\n",
    "      mv aligned_unsorted.flagstat $LIBRARY.se.flagstat\n",
    "\n",
    "    # Rename the hgr-aligned Bam files\n",
    "      mv align.hgr1.bam $LIBRARY.hgr1.se.bam\n",
    "      mv align.hgr1.bam.bai $LIBRARY.hgr1.se.bam.bai\n",
    "      mv align.hgr1.flagstat $LIBRARY.hgr1.se.flagstat\n",
    "      \n",
    "    # Alignments (Full)\n",
    "    aws s3 cp $LIBRARY.se.flagstat s3://crownproject/tcga/\n",
    "\n",
    "    # Alignments (Aligned)\n",
    "    aws s3 cp $LIBRARY.hgr1.se.bam s3://crownproject/tcga/\n",
    "    aws s3 cp $LIBRARY.hgr1.se.bam.bai s3://crownproject/tcga/\n",
    "    aws s3 cp $LIBRARY.hgr1.se.flagstat s3://crownproject/tcga/\n",
    "\n",
    "fi\n",
    "\n",
    "# PAIRED END READS ====================================================\n",
    "\n",
    "\n",
    "if [ -s $FQ1 ]\n",
    "then\n",
    "    # Paired-End Extracted Reads Alignment\n",
    "\n",
    "    # Extract Sequencing Run Info\n",
    "    RGPU=$(gzip -dc $FQ1| head -n1 - | cut -f1 -d':' | cut -f2 -d' ')\n",
    "    \n",
    "    # Bowtie2: align to genome\n",
    "    bowtie2 --very-sensitive-local -p $THREADS \\\n",
    "      --rg-id $RGID --rg LB:$RGLB --rg SM:$RGSM \\\n",
    "      --rg PL:$RGPL --rg PU:$RGPU \\\n",
    "      -x hgr1 -1 $FQ1 -2 $FQ2 | \\\n",
    "      samtools view -bS - > aligned_unsorted.bam\n",
    "      \n",
    "    # Calcualte library flagstats\n",
    "      samtools flagstat aligned_unsorted.bam > aligned_unsorted.flagstat\n",
    "      \n",
    "      rm $FQ1 $FQ2 # Remove fastq files to save space\n",
    "\n",
    "      \n",
    "    # Read Subset ------------------------------\n",
    "    # Extract mapped reads, and their unmapped pairs\n",
    "\n",
    "      # Extract Header\n",
    "      samtools view -H aligned_unsorted.bam > align.header.tmp\n",
    "\n",
    "      # Unmapped reads with mapped pairs\n",
    "      # Extract Mapped Reads\n",
    "      # and their unmapped pairs\n",
    "      samtools view -b -F 4 aligned_unsorted.bam > align.F4.bam #mapped\n",
    "      samtools view -b -f 4 -F 8 aligned_unsorted.bam > align.f4F8.bam #unmapped pairs\n",
    "\n",
    "      # Extract just the 45S unit\n",
    "      #aws s3 cp s3://crownproject/resources/rDNA_45s.bed ./\n",
    "      #samtools view -b -L rDNA_45s.bed align.F4.bam > align.F4.45s.bam\n",
    "\n",
    "      # What are the mapped readnames\n",
    "      samtools view align.F4.bam | cut -f1 - > read.names.tmp\n",
    "\n",
    "      # Extract mapped reads\n",
    "      samtools view align.F4.bam | grep -Ff read.names.tmp - > align.F4.tmp.sam\n",
    "\n",
    "\n",
    "      # Extract cases of read pairs mapped on edge of region of interest\n",
    "      # -------|======= R O I ======| ----------\n",
    "      # read:                  ====---====\n",
    "      samtools view align.F4.bam | grep -Ff read.names.tmp - > align.F4.tmp.sam\n",
    "\n",
    "      # Complete mapped reads list\n",
    "      #cut -f1 align.F4.tmp.sam > read.names.45s.long.tmp\n",
    "\n",
    "      # Extract unmapped reads with a mapped pair\n",
    "      samtools view align.f4F8.bam | grep -Ff read.names.tmp - > align.f4F8.tmp.sam\n",
    "\n",
    "      # Re-compile bam file\n",
    "      cat align.header.tmp align.F4.tmp.sam align.f4F8.tmp.sam | samtools view -bS - > align.hgr1.tmp.bam\n",
    "        samtools sort align.hgr1.tmp.bam align.hgr1\n",
    "        samtools index align.hgr1.bam\n",
    "        samtools flagstat align.hgr1.bam > align.hgr1.flagstat\n",
    "\n",
    "      # Clean up \n",
    "      rm *tmp* align.F4.bam align.f4F8.bam\n",
    "\n",
    "    # Rename the total Bam Files\n",
    "      mv aligned_unsorted.bam $LIBRARY.bam\n",
    "      mv aligned_unsorted.bam.bai $LIBRARY.bam.bai\n",
    "      mv aligned_unsorted.flagstat $LIBRARY.flagstat\n",
    "\n",
    "    # Rename the hgr-aligned Bam files\n",
    "      mv align.hgr1.bam $LIBRARY.hgr1.bam\n",
    "      mv align.hgr1.bam.bai $LIBRARY.hgr1.bam.bai\n",
    "      mv align.hgr1.flagstat $LIBRARY.hgr1.flagstat\n",
    "    \n",
    "fi\n",
    "\n",
    "rm $FQ0 $FQ1 $FQ2 # Remove fastq files to save space  \n",
    "  \n",
    "# Alignments (Full)\n",
    " aws s3 cp $LIBRARY.se.flagstat s3://crownproject/tcga/\n",
    "\n",
    "# Alignments (Aligned)\n",
    "  aws s3 cp $LIBRARY.hgr1.bam s3://crownproject/tcga/\n",
    "  aws s3 cp $LIBRARY.hgr1.bam.bai s3://crownproject/tcga/\n",
    "  aws s3 cp $LIBRARY.hgr1.flagstat s3://crownproject/tcga/\n",
    "\n",
    "# VCF\n",
    " aws s3 cp $LIBRARY.hgr1.vcf s3://crownproject/tcga/\n",
    " aws s3 cp $LIBRARY.hgr1.vcf.idx s3://crownproject/tcga/\n",
    " \n",
    "# Shutdown and Terminate instance\n",
    "EC2ID=$(ec2metadata --instance-id)\n",
    "aws ec2 terminate-instances --instance-ids $EC2ID\n",
    "\n",
    "# Script complete\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### tcga0_input.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "TCGA-F4-6461-01A\tTCGA-COAD\t01581d3a-8427-41db-82c2-a7dd880a3937\n",
    "TCGA-G4-6315-01A\tTCGA-COAD\t017e4572-bbde-4e44-9de2-22e7d2573603\n",
    "TCGA-AA-3529-01A\tTCGA-COAD\t01871314-b195-4142-8149-1cff7ea8c3b4\n",
    "TCGA-AZ-6599-01A\tTCGA-COAD\t0188bc58-2deb-4978-98c3-3028d6fa61f8\n",
    "TCGA-QG-A5Z1-01A\tTCGA-COAD\t019ae823-acfc-49a6-a144-3b363b15a0dc"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Pilot Run 1 - 180814 1448\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Completed 7.0 KiB/7.0 KiB with 1 file(s) remaining\r",
      "upload: ./hgr1_align_v2.tcga.0.sh to s3://crownproject/scripts/hgr1_align_v2.tcga.0.sh\r\n"
     ]
    }
   ],
   "source": [
    "# Copy tcga script to S3\n",
    "cd /home/artem/Desktop/Crown/data2/tcga_0\n",
    "aws s3 cp hgr1_align_v2.tcga.0.sh s3://crownproject/scripts/hgr1_align_v2.tcga.0.sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Launch instance # 1\r\n",
      "Instance Type: c4.xlarge\r\n",
      "AMI Image: ami-0031fd61f932bdef9\r\n",
      "Run Script: s3://crownproject/scripts/hgr1_align_v2.tcga.0.sh\r\n",
      "Parameters: TCGA-F4-6461-01A TCGA-COAD 01581d3a-8427-41db-82c2-a7dd880a3937\r\n",
      "Instance ID: i-0559a94aff30c5a00\r\n",
      "Public DNS: ec2-54-212-59-228.us-west-2.compute.amazonaws.com\r\n",
      "Warning: Permanently added 'ec2-54-212-59-228.us-west-2.compute.amazonaws.com,54.212.59.228' (ECDSA) to the list of known hosts.\r",
      "\r\n",
      "download: s3://crownproject/scripts/hgr1_align_v2.tcga.0.sh to ./hgr1_align_v2.tcga.0.sh\r\n",
      "\r\n",
      "\r\n",
      "Launch instance # 2\r\n",
      "Instance Type: c4.xlarge\r\n",
      "AMI Image: ami-0031fd61f932bdef9\r\n",
      "Run Script: s3://crownproject/scripts/hgr1_align_v2.tcga.0.sh\r\n",
      "Parameters: TCGA-G4-6315-01A TCGA-COAD 017e4572-bbde-4e44-9de2-22e7d2573603\r\n",
      "Instance ID: i-0668ef286d8c2af74\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "bash queenB.sh tcga0_input.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Completed 6.8 KiB/6.8 KiB with 1 file(s) remaining\r",
      "upload: ./hgr1_align_v2.tcga.0.sh to s3://crownproject/scripts/hgr1_align_v2.tcga.0.sh\r\n"
     ]
    }
   ],
   "source": [
    "# Stop with interrupt\n",
    "# Didn't work; the instances shut down prematurely. I'll re-test by turning off self-close of instances\n",
    "# (Line 223/224 of align script commented out)\n",
    "aws s3 cp hgr1_align_v2.tcga.0.sh s3://crownproject/scripts/hgr1_align_v2.tcga.0.sh"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "`bash queenB.sh tcga0_input0.tsv # two files only`\n",
    " \n",
    "Note: ran in seperate window.\n",
    "\n",
    "--- Bugs ---\n",
    "* Only first line of input ran, second parameter didn't run. -- EOL char --> changed to awk command DONE\n",
    "* Found bug on line 55-57 of tcga align script. gdc-client uses FILE_UUID, parameter is UUID -- DONE\n",
    "* Input file must be SPACE delimited, not TAB delimited (changed) -- DONE\n",
    "* screenlog `/home/ubuntu/hgr1_align_v2.tcga.0.sh: line 55: gdc-client: command not found` change command to an explicit call `~/bin/gdc-client`. The samtools version being called is also the older one which makes me think that the most recent updates to AMI are not functioning. -- DONE (but sloppy)\n",
    "\n",
    "--- TODO ---\n",
    "* Add support to copy the screenlog.0 file which is generated and save it to the output tcga folder such that each run instance can be debugged and go back to it to confirm all commands run successfully -- DONE\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bash: cd: /home/artem/Crown/data2/tcga0/: No such file or directory\r\n",
      "Completed 7.0 KiB/7.0 KiB with 1 file(s) remaining\r",
      "upload: ./hgr1_align_v2.tcga.0.sh to s3://crownproject/scripts/hgr1_align_v2.tcga.0.sh\r\n",
      "Launch instance # 1\r\n",
      "Instance Type: c4.xlarge\r\n",
      "AMI Image: ami-0031fd61f932bdef9\r\n",
      "Run Script: s3://crownproject/scripts/hgr1_align_v2.tcga.0.sh\r\n",
      "Parameters: TCGA-F4-6461-01A TCGA-COAD 01581d3a-8427-41db-82c2-a7dd880a3937\r\n",
      "Instance ID: i-09dbea3ca5362b27c\r\n",
      "Public DNS: ec2-34-219-137-196.us-west-2.compute.amazonaws.com\r\n",
      "Warning: Permanently added 'ec2-34-219-137-196.us-west-2.compute.amazonaws.com,34.219.137.196' (ECDSA) to the list of known hosts.\r",
      "\r\n",
      "download: s3://crownproject/scripts/hgr1_align_v2.tcga.0.sh to ./hgr1_align_v2.tcga.0.sh\r\n",
      "\r\n",
      "\r\n",
      "Launch instance # 2\r\n",
      "Instance Type: c4.xlarge\r\n",
      "AMI Image: ami-0031fd61f932bdef9\r\n",
      "Run Script: s3://crownproject/scripts/hgr1_align_v2.tcga.0.sh\r\n",
      "Parameters: TCGA-G4-6315-01A TCGA-COAD 017e4572-bbde-4e44-9de2-22e7d2573603\r\n",
      "Instance ID: i-0255b652d1082e594\r\n",
      "Public DNS: ec2-34-209-170-244.us-west-2.compute.amazonaws.com\r\n",
      "Warning: Permanently added 'ec2-34-209-170-244.us-west-2.compute.amazonaws.com,34.209.170.244' (ECDSA) to the list of known hosts.\r",
      "\r\n",
      "download: s3://crownproject/scripts/hgr1_align_v2.tcga.0.sh to ./hgr1_align_v2.tcga.0.sh\r\n",
      "\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "# Pilot 2 with above bugfixes\n",
    "cd ~/Crown/data2/tcga0/\n",
    "aws s3 cp hgr1_align_v2.tcga.0.sh s3://crownproject/scripts/hgr1_align_v2.tcga.0.sh\n",
    "bash queenB.sh tcga0_input0.tsv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "From TCGA.screenlog the problem is with a `samtools sort` command:\n",
    "\n",
    "```\n",
    "{running correctly}...\n",
    "...\n",
    "1.04% overall alignment rate\n",
    "[bam_sort] Use -T PREFIX / -o FILE to specify temporary and final output files\n",
    "Usage: samtools sort [options...] [in.bam]\n",
    "...\n",
    "{errors}\n",
    "```\n",
    "\n",
    "Lines 101-103 of `hgr1_align_v2.tcga.0.sh` script updated to use samtools 1.9 command syntax\n",
    "```\n",
    "      # Extract Mapped Reads\n",
    "      ~/bin/samtools view -b -F 4 aligned_unsorted.bam | \\\n",
    "      ~/bin/samtools sort -@ $THREADS -O BAM - > align.hgr1.bam #mapped\n",
    "```\n",
    "\n",
    "Running only a single bam file for testing now since iteration is working and re-run with updated script."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bash: cd: /home/artem/Crown/data2/tcga0/: No such file or directory\r\n",
      "\r\n",
      "TCGA-F4-6461-01A TCGA-COAD 01581d3a-8427-41db-82c2-a7dd880a3937\r\n",
      "\r\n",
      "Completed 7.1 KiB/7.1 KiB with 1 file(s) remaining\r",
      "upload: ./hgr1_align_v2.tcga.0.sh to s3://crownproject/scripts/hgr1_align_v2.tcga.0.sh\r\n",
      "Launch instance # 1\r\n",
      "Instance Type: c4.xlarge\r\n",
      "AMI Image: ami-0031fd61f932bdef9\r\n",
      "Run Script: s3://crownproject/scripts/hgr1_align_v2.tcga.0.sh\r\n",
      "Parameters: TCGA-F4-6461-01A TCGA-COAD 01581d3a-8427-41db-82c2-a7dd880a3937\r\n",
      "Instance ID: i-0c7296d75f8dc68eb\r\n",
      "Public DNS: ec2-34-221-200-179.us-west-2.compute.amazonaws.com\r\n",
      "Warning: Permanently added 'ec2-34-221-200-179.us-west-2.compute.amazonaws.com,34.221.200.179' (ECDSA) to the list of known hosts.\r",
      "\r\n",
      "download: s3://crownproject/scripts/hgr1_align_v2.tcga.0.sh to ./hgr1_align_v2.tcga.0.sh\r\n",
      "\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "# Pilot 2 with above bugfixes\n",
    "cd ~/Crown/data2/tcga0/\n",
    "echo ''; cat tcga0_input0.tsv; echo ''\n",
    "aws s3 cp hgr1_align_v2.tcga.0.sh s3://crownproject/scripts/hgr1_align_v2.tcga.0.sh\n",
    "bash queenB.sh tcga0_input0.tsv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This version now worked and there are bam files on S3 : )\n",
    "\n",
    "Trying with remaining 4 libraries.\n",
    "\n",
    "Errors in screenlog\n",
    "```\n",
    "mv: cannot stat 'aligned_unsorted.bam.bai': No such file or directory\n",
    "\n",
    "The user-provided path TCGA-F4-6461-01A.se.flagstat does not exist.\n",
    "```\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bash: cd: /home/artem/Crown/data2/tcga0/: No such file or directory\r\n",
      " \r\n",
      "TCGA-G4-6315-01A\tTCGA-COAD\t017e4572-bbde-4e44-9de2-22e7d2573603\r\n",
      "TCGA-AA-3529-01A\tTCGA-COAD\t01871314-b195-4142-8149-1cff7ea8c3b4\r\n",
      "TCGA-AZ-6599-01A\tTCGA-COAD\t0188bc58-2deb-4978-98c3-3028d6fa61f8\r\n",
      "TCGA-QG-A5Z1-01A\tTCGA-COAD\t019ae823-acfc-49a6-a144-3b363b15a0dc \r\n",
      "Completed 7.1 KiB/7.1 KiB with 1 file(s) remaining\r",
      "upload: ./hgr1_align_v2.tcga.0.sh to s3://crownproject/scripts/hgr1_align_v2.tcga.0.sh\r\n",
      "Launch instance # 1\r\n",
      "Instance Type: c4.xlarge\r\n",
      "AMI Image: ami-0031fd61f932bdef9\r\n",
      "Run Script: s3://crownproject/scripts/hgr1_align_v2.tcga.0.sh\r\n",
      "Parameters: TCGA-G4-6315-01A TCGA-COAD 017e4572-bbde-4e44-9de2-22e7d2573603\r\n",
      "Instance ID: i-0e27041d09664f9af\r\n",
      "Public DNS: ec2-34-219-143-181.us-west-2.compute.amazonaws.com\r\n",
      "Warning: Permanently added 'ec2-34-219-143-181.us-west-2.compute.amazonaws.com,34.219.143.181' (ECDSA) to the list of known hosts.\r",
      "\r\n",
      "download: s3://crownproject/scripts/hgr1_align_v2.tcga.0.sh to ./hgr1_align_v2.tcga.0.sh\r\n",
      "\r\n",
      "\r\n",
      "Launch instance # 2\r\n",
      "Instance Type: c4.xlarge\r\n",
      "AMI Image: ami-0031fd61f932bdef9\r\n",
      "Run Script: s3://crownproject/scripts/hgr1_align_v2.tcga.0.sh\r\n",
      "Parameters: TCGA-AA-3529-01A TCGA-COAD 01871314-b195-4142-8149-1cff7ea8c3b4\r\n",
      "Instance ID: i-0f016b52877670e1f\r\n",
      "Public DNS: ec2-52-39-24-49.us-west-2.compute.amazonaws.com\r\n",
      "Warning: Permanently added 'ec2-52-39-24-49.us-west-2.compute.amazonaws.com,52.39.24.49' (ECDSA) to the list of known hosts.\r",
      "\r\n",
      "download: s3://crownproject/scripts/hgr1_align_v2.tcga.0.sh to ./hgr1_align_v2.tcga.0.sh\r\n",
      "\r\n",
      "\r\n",
      "Launch instance # 3\r\n",
      "Instance Type: c4.xlarge\r\n",
      "AMI Image: ami-0031fd61f932bdef9\r\n",
      "Run Script: s3://crownproject/scripts/hgr1_align_v2.tcga.0.sh\r\n",
      "Parameters: TCGA-AZ-6599-01A TCGA-COAD 0188bc58-2deb-4978-98c3-3028d6fa61f8\r\n",
      "Instance ID: i-077b416ecbf4e6194\r\n",
      "Public DNS: ec2-34-213-153-85.us-west-2.compute.amazonaws.com\r\n",
      "Warning: Permanently added 'ec2-34-213-153-85.us-west-2.compute.amazonaws.com,34.213.153.85' (ECDSA) to the list of known hosts.\r",
      "\r\n",
      "download: s3://crownproject/scripts/hgr1_align_v2.tcga.0.sh to ./hgr1_align_v2.tcga.0.sh\r\n",
      "\r\n",
      "\r\n",
      "Launch instance # 4\r\n",
      "Instance Type: c4.xlarge\r\n",
      "AMI Image: ami-0031fd61f932bdef9\r\n",
      "Run Script: s3://crownproject/scripts/hgr1_align_v2.tcga.0.sh\r\n",
      "Parameters: TCGA-QG-A5Z1-01A TCGA-COAD 019ae823-acfc-49a6-a144-3b363b15a0dc\r\n",
      "Instance ID: i-09a3ec647fcfbf6cc\r\n",
      "Public DNS: ec2-34-218-232-197.us-west-2.compute.amazonaws.com\r\n",
      "Warning: Permanently added 'ec2-34-218-232-197.us-west-2.compute.amazonaws.com,34.218.232.197' (ECDSA) to the list of known hosts.\r",
      "\r\n",
      "download: s3://crownproject/scripts/hgr1_align_v2.tcga.0.sh to ./hgr1_align_v2.tcga.0.sh\r\n",
      "\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "# Pilot 3 with above bugfixes\n",
    "cd ~/Crown/data2/tcga0/\n",
    "echo ' '; cat tcga0_input.tsv; echo ' '\n",
    "\n",
    "aws s3 cp hgr1_align_v2.tcga.0.sh s3://crownproject/scripts/hgr1_align_v2.tcga.0.sh\n",
    "bash queenB.sh tcga0_input.tsv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Results\n",
    "\n",
    "Alignment worked successfully : ) This should be adaquete for piping through larger amounts of RNA-seq data, especially the matched cancer-normal controls.\n",
    "\n",
    "Note: The single-end RNA-seq libraries (se) are quite error prone over the 18S/28S regions since they are shorter and older reads. They should not be used for variant calling of 18S/28S/5.8S.\n",
    "The converse is true about 5S rRNA, since this smaller lower GC content RNA is often lost in the larger fragment size paired-end libraries, it is sequenced very well on both strands in the se libraries. These rRNA should be treated seperately based on which type of library each sample was prepared from.\n",
    "\n",
    "![18S Alignment Output](../../data2/tcga_0/align/tcga0_18S.png)\n",
    "\n",
    "![5S Alignment Output](../../data2/tcga_0/align/tcga0_5S.png)\n",
    "\n",
    "Good to go!\n",
    "\n",
    "\n",
    "Archived these libraries to: s3://crownproject/tcga-coad0/\n",
    "\n",
    "```\n",
    "aws s3 mv --recursive --include \"*hgr1*\" s3://crownproject/tcga/ s3://crownproject/tcga/tcga-coad0/\n",
    "```\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
