{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Initialize Amazon EC2 Instance for TCGA Analysis\n",
    "```\n",
    "pi:ababaian\n",
    "files: ~/Crown/data2/tcga_0/\n",
    "start: 2018 07 30\n",
    "complete : 2018 08 14\n",
    "Addendum: 2018 09 06\n",
    "```\n",
    "## Introduction\n",
    "\n",
    "I've been granted access to TCGA data amd was awarded $8K of Amazon AWS credits. I have 11 months to process the entire cohort of TCGA DNA- and RNA-seq data to get alignments against hgr1 and thus build a data-base of rDNA alignments which to analyze.\n",
    "\n",
    "First step is to set-up a pipeline instance of EC2 which I can then run in parallel and process large amounts of TCGA data.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "## Objective\n",
    "\n",
    "Initialize an EC2 instance such that the entire TCGA analysis pipeline can be run in a single step (use hgr1 align pipeline). \n",
    "\n",
    "This instance will be private/encrypted though and my TCGA/dbGAP credentials can be on it so that each instance can download TCGA data directly.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Materials and Methods\n",
    "\n",
    "### Crown 1800813 TCGA Instance Image (AMI) Set-up\n",
    "\n",
    "Launched an EC2 instance based on Crown 170220 AMI\n",
    "\n",
    "```\n",
    "ssh -i \"~/.ssh/glitch.pem\" ubuntu@ec2-34-216-2-13.us-west-2.compute.amazonaws.com \n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Install hgr1 resource into AMI image\n",
    "\n",
    "# In Resource folder create dir for each genome type\n",
    "cd ~/resources/\n",
    "\n",
    "mkdir hg38\n",
    "mv hg38*.* hg38/\n",
    "\n",
    "mkdir hgr0\n",
    "mv hgr*.* hgr0/\n",
    "\n",
    "# Download hgr1 resource genome\n",
    "mkdir hgr1\n",
    "cd hgr1\n",
    "aws s3 cp s3://crownproject/resources/hgr1.fa ./\n",
    "samtools faidx hgr1.fa\n",
    "bowtie2-build hgr1.fa hgr1\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[sudo] password for artem: \r\n"
     ]
    }
   ],
   "source": [
    "# Install gdc-client into AMI image\n",
    "\n",
    "cd ~/software/\n",
    "\n",
    "# Download TCGA-GDC File Transfer Tool (ubuntu)\n",
    "wget https://gdc.cancer.gov/system/files/authenticated%20user/0/gdc-client_v1.3.0_Ubuntu14.04_x64.zip\n",
    "  sudo apt install unzip\n",
    "  unzip gdc-client_v1.3.0_Ubuntu14.04_x64.zip\n",
    "  mv gdc-client ~/bin/\n",
    "  \n",
    "# Copy over gdc token to resouurces\n",
    "cd ~/resources/\n",
    "vim gdc.token #copied over manually\n",
    "chmod 400 gdc.token\n",
    "\n",
    "# Files can now be downloaded from a manifest file (created on GDC)\n",
    "# gdc-client download -t ~/resources/gdc.token -d ./ -m manifest.test \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Install HISAT2 into AMI image\n",
    "\n",
    "cd ~/software/\n",
    "wget ftp://ftp.ccb.jhu.edu/pub/infphilo/hisat2/downloads/hisat2-2.1.0-Linux_x86_64.zip\n",
    "unzip hisat2*\n",
    "mv hisat2-2.1.0-Linux_x86_64.zip hisat2-2.1.0\n",
    "\n",
    "# add hisat2 dir to PATH\n",
    "export PATH=\"/home/ubuntu/software/hisat2-2.1.0:$PATH\"\n",
    "\n",
    "# clean-up some older software zips to one folder\n",
    "mkdir zips\n",
    "mv cufflinks-2.2.1.Linux_x86_64.tar.gz zips/\n",
    "mv gdc-client_v1.3.0_Ubuntu14.04_x64.zip  zips/\n",
    "\n",
    "# Install Samtools 1.9 to AMI Image\n",
    "\n",
    "cd ~/software/\n",
    "wget https://github.com/samtools/samtools/releases/download/1.9/samtools-1.9.tar.bz2\n",
    "tar -xf samtools-1.9.tar.bz2\n",
    "# Install dependencies\n",
    "sudo apt-get update  # Ensure the package list is up to date\n",
    "sudo apt-get install autoconf automake make gcc perl \\\n",
    "  zlib1g-dev libbz2-dev liblzma-dev libcurl4-gnutls-dev \\\n",
    "  libssl-dev libncurses5-dev\n",
    "\n",
    "# Install samtools\n",
    "cd samtools-1.9/\n",
    "./configure\n",
    "make\n",
    "make DESTDIR=$PWD install\n",
    "\n",
    "# samtools back-version compatibility\n",
    "which samtools\n",
    "\n",
    "mv /home/ubuntu/software/pitchfork/deployment/bin/samtools ~/bin/samtools_0.1.20\n",
    "ln /home/ubuntu/software/samtools-1.9/samtools ~/bin/samtools\n",
    "\n",
    "# Requires re-login to function.\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Running GDC Client Locally to debug"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "id\tfilename\tmd5\tsize\tstate\r\n",
      "9f52e91b-cf6f-4701-a651-096979267ae6\t5192c580-43f3-4f41-81f2-cfd68b7a78fd_gdc_realn_rehead.bam\t03bdc3e675ec69f2a110cedce1ad52a6\t2240824619\tsubmitted\r\n",
      "\r\n",
      "-rwxr-xr-x 1 artem artem  24M Aug 14  2017 gdc-client\r\n",
      "-rw-rw-r-- 1 artem artem  24M Jul 30 14:44 gdc-client_v1.3.0_Ubuntu14.04_x64.zip\r\n",
      "-rw-rw-r-- 1 artem artem  175 Jul 30 15:33 gdc_manifest_20180730_223304.txt\r\n",
      "-r-------- 1 artem artem 1.1K Jul 30 14:49 gdc.token\r\n",
      "\r\n",
      "9f52e91b-cf6f-4701-a651-096979267ae6:\r\n",
      "total 2.1G\r\n",
      "drwxrwxr-x 3 artem artem 4.0K Jul 30 17:09 .\r\n",
      "drwxrwxr-x 3 artem artem 4.0K Jul 30 16:53 ..\r\n",
      "-rw-rw-r-- 1 artem artem 5.4M Jul 30 17:09 5192c580-43f3-4f41-81f2-cfd68b7a78fd_gdc_realn_rehead.bai\r\n",
      "-rw-rw-r-- 1 artem artem 2.1G Jul 30 17:09 5192c580-43f3-4f41-81f2-cfd68b7a78fd_gdc_realn_rehead.bam\r\n",
      "-rw-rw-r-- 1 artem artem  410 Jul 30 17:09 annotations.txt\r\n",
      "drwxrwxr-x 2 artem artem 4.0K Jul 30 17:09 logs\r\n",
      "\r\n",
      "select bam file only\r\n",
      "\r\n",
      "-rw-rw-r-- 1 artem artem 2.1G Jul 30 17:09 9f52e91b-cf6f-4701-a651-096979267ae6/5192c580-43f3-4f41-81f2-cfd68b7a78fd_gdc_realn_rehead.bam\r\n"
     ]
    }
   ],
   "source": [
    "### Running gdc-client locally\n",
    "cd ~/Desktop/gdc\n",
    "\n",
    "# From the GDC website; created a manifest file containing a \n",
    "# single rnaseq bam file for testing.\n",
    "\n",
    "cat gdc_manifest*\n",
    "echo ''\n",
    "echo ''\n",
    "\n",
    "#./gdc-client download -t gdc.token -d ./ -n 2 -m gdc_manifest_20180730_223304.txt\n",
    "# downlaoaded\n",
    "\n",
    "ls -alh *\n",
    "\n",
    "echo ''\n",
    "echo 'select bam file only'\n",
    "echo ''\n",
    "ls -alh */*.bam\n",
    "\n",
    "# can also use command to select bam filename specifically\n",
    "# cut -f2 gdc_manifest* | tail -n-1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Compare HISAT2 vs. bowtie2 alignment for hgr1\n",
    "# to test if I should switch over alignment protocol\n",
    "\n",
    "# Extracted reads from bamfile into fastq_0 format.\n",
    "\n",
    "# HISAT2: align to genome\n",
    "    hisat2 -p 2 \\\n",
    "      -x hgr1 -U tmp.sort.0.fq | \\\n",
    "      samtools view -bS - > aligned_unsorted.bam\n",
    "      \n",
    "    # 30120644 reads; of these:\n",
    "    # 30120644 (100.00%) were unpaired; of these:\n",
    "    # 29522710 (98.01%) aligned 0 times\n",
    "    # 597927 (1.99%) aligned exactly 1 time\n",
    "    # 7 (0.00%) aligned >1 times\n",
    "    # 1.99% overall alignment rate\n",
    "    \n",
    "    # Bowtie2: align to genome\n",
    "    bowtie2 --very-sensitive-local -p 2 \\\n",
    "      -x hgr1 -U tmp.sort.0.fq | \\\n",
    "      samtools view -bS - > aligned_unsorted.bt2.bam\n",
    "      \n",
    "    # 30120644 reads; of these:\n",
    "    # 30120644 (100.00%) were unpaired; of these:\n",
    "    # 29448773 (97.77%) aligned 0 times\n",
    "    # 669940 (2.22%) aligned exactly 1 time\n",
    "    # 1931 (0.01%) aligned >1 times\n",
    "    # 2.23% overall alignment rate\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# samtools tricks for extracting data\n",
    "\n",
    "# Extract ReadGroup Sample Name (SM)\n",
    "samtools view -H */*.bam | grep '^@RG' | sed \"s/.*SM:\\([^\\t]*\\).*/\\1/g\" | uniq\n",
    "\n",
    "# Extract ReadGroup identifier (ID)\n",
    "samtools view -H */*.bam | grep '^@RG' | sed \"s/.*SM:\\([^\\t]*\\).*/\\1/g\" | uniq\n",
    "\n",
    "# Extract ReadGroup platform (PL)\n",
    "samtools view -H */*.bam | grep '^@RG' | sed \"s/.*PL:\\([^\\t]*\\).*/\\1/g\" | uniq\n",
    "\n",
    "\n",
    "# Convert input bam file to fastq files for re-alignment\n",
    "samtools sort -n initial.bam | samtools fastq -O \\\n",
    "    -0 tmp.sort.0.fq \\\n",
    "    -1 tmp.sort.1.fq \\\n",
    "    -2 tmp.sort.2.fq - \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Running test data on EC2\n",
    "\n",
    "Launched crown-180813 AMI on a t2.xlarge instance with 80 Gb storage.\n",
    "\n",
    "Running through test script below manually to debug / test commands"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "# TESTING SCRIPT (NON PRODUCTION)\n",
    "# rDNA alignment pipeline\n",
    "# 180731 build -- TCGA\n",
    "# AMI: crown-XXXXX - ami-XXXXX\n",
    "# EC2: c4.2xlarge (8cpu / 15 gb)\n",
    "# EC2: c4.xlarge  (4cpu / 8  gb)\n",
    "# Storage: 300 Gb\n",
    "#\n",
    "\n",
    "# Control Panel -------------------------------\n",
    "# CPU\n",
    "\tTHREADS='3'\n",
    "\n",
    "# Sequencing Data\n",
    "\tLIBRARY=\"test_align\" # Library/ File name\n",
    "#\tLIBRARY=$1 # Library/ File name\n",
    "\n",
    "# TCGA FILE UUID\n",
    "    UUID='9f52e91b-cf6f-4701-a651-096979267ae'\n",
    "#   UUID=$2\n",
    "\n",
    "    # FastQ File-names\n",
    "    FQ0=\"$LIBRARY.tmp.sort.0.fq\"\n",
    "    FQ1=\"$LIBRARY.tmp.sort.1.fq\"\n",
    "    FQ2=\"$LIBRARY.tmp.sort.2.fq\"\n",
    "    \n",
    "# Read Group Data\n",
    "# Extract from downloaded BAM file / input\n",
    "\tRGPO=\"CRC\" # Patient Population\n",
    "#\tRGPO=$2 # Patient Population\n",
    "\n",
    "\t#RGSM=   # Sample. Patient Identifer\n",
    "\t#RGID= # Read Group ID. Accession Number\n",
    "    \n",
    "\tRGLB=$LIBRARY # Library Name. Accession Number\n",
    "\tRGPL='ILLUMINA'  # Sequencing Platform.\n",
    "    \n",
    "\t# Extract Sequencing Run Info\n",
    "\t#  RGPU=$(gzip -dc $FQ1 | head -n1 - | cut -f1 -d':' | cut -f2 -d' ')\n",
    "\n",
    "# Initialize wordir ---------------------------\n",
    "\n",
    "# Make working directory\n",
    "  mkdir -p align; cd align\n",
    "\n",
    "# Copy hgrX genome and create bowtie2 index\n",
    "  cp ~/resources/hgr1/* ./\n",
    "  \n",
    "# Download RNAseq BAM file\n",
    "# with a UUID as input\n",
    "  gdc-client download -t ~/resources/gdc.token -d ./ \\\n",
    "  -n $THREADS\\\n",
    "  $FILE_UUID\n",
    "  \n",
    "# Link the RNA-seq bamfile which is called by its UID to workdir\n",
    "  ln -s */*.bam input.bam\n",
    "  \n",
    "# Extract ReadGroup Sample Name (SM)\n",
    "  RGSM=$(samtools view -H input.bam | grep '^@RG' | sed \"s/.*SM:\\([^\\t]*\\).*/\\1/g\" | uniq )\n",
    "\n",
    "# Extract ReadGroup identifier (ID)\n",
    "  RGID=$(samtools view -H input.bam | grep '^@RG' | sed \"s/.*ID:\\([^\\t]*\\).*/\\1/g\" | uniq )\n",
    "\n",
    "# Convert input bam file to fastq files for re-alignment\n",
    "samtools sort -@ $THREADS-n input.bam | \\\n",
    "    samtools fastq -@ $THREADS -O \\\n",
    "    -0 $FQ0 \\\n",
    "    -1 $FQ1 \\\n",
    "    -2 $FQ2 -\n",
    "\n",
    "# SINGLE END READS ====================================================\n",
    "\n",
    "if [ -s $FQ0 ]\n",
    "then\n",
    "    # Single-End Extracted Reads Alignment\n",
    "\n",
    "    # Extract Sequencing Run Info\n",
    "    #RGPU=$(gzip -dc $FQ0| head -n1 - | cut -f1 -d':' | cut -f2 -d' ')\n",
    "    RGPU=$(head -n1 $FQ0 | cut -f1 -d':' | cut -f2 -d' ')\n",
    "\n",
    "    # Bowtie2: align to genome\n",
    "    bowtie2 --very-sensitive-local -p $THREADS \\\n",
    "      --rg-id $RGID --rg LB:$RGLB --rg SM:$RGSM \\\n",
    "      --rg PL:$RGPL --rg PU:$RGPU \\\n",
    "      -x hgr1 -U $FQ0 | \\\n",
    "      samtools view -bS - > aligned_unsorted.bam\n",
    "     \n",
    "    # Calculate library flagstats\n",
    "      samtools flagstat aligned_unsorted.bam > aligned_unsorted.flagstat\n",
    "      rm $FQ0 # Remove fastq files to save space\n",
    "\n",
    "    # Read Subset ------------------------------\n",
    "    # Extract mapped reads, and their unmapped pairs\n",
    "\n",
    "      # Extract Header\n",
    "      samtools view -H aligned_unsorted.bam > align.header.tmp\n",
    "\n",
    "      # Extract Mapped Reads\n",
    "      samtools view -b -F 4 aligned_unsorted.bam | \\\n",
    "      samtools sort > align.hgr1.bam #mapped\n",
    "      \n",
    "    # Calcualte library flagstats\n",
    "    samtools index align.hgr1.bam\n",
    "    samtools flagstat align.hgr1.bam > align.hgr1.flagstat\n",
    "\n",
    "    # Rename the total Bam Files\n",
    "      #mv aligned_unsorted.bam $LIBRARY.se.bam\n",
    "      #mv aligned_unsorted.bam.bai $LIBRARY.se.bam.bai\n",
    "      mv aligned_unsorted.flagstat $LIBRARY.se.flagstat\n",
    "\n",
    "    # Rename the hgr-aligned Bam files\n",
    "      mv align.hgr1.bam $LIBRARY.hgr1.se.bam\n",
    "      mv align.hgr1.bam.bai $LIBRARY.hgr1.se.bam.bai\n",
    "      mv align.hgr1.flagstat $LIBRARY.hgr1.se.flagstat\n",
    "      \n",
    "    # Alignments (Full)\n",
    "    aws s3 cp $LIBRARY.se.flagstat s3://crownproject/tcga/\n",
    "\n",
    "    # Alignments (Aligned)\n",
    "    aws s3 cp $LIBRARY.hgr1.se.bam s3://crownproject/tcga/\n",
    "    aws s3 cp $LIBRARY.hgr1.se.bam.bai s3://crownproject/tcga/\n",
    "    aws s3 cp $LIBRARY.hgr1.se.flagstat s3://crownproject/tcga/\n",
    "\n",
    "fi\n",
    "\n",
    "# PAIRED END READS ====================================================\n",
    "\n",
    "\n",
    "if [ -s $FQ1 ]\n",
    "then\n",
    "    # Paired-End Extracted Reads Alignment\n",
    "\n",
    "    # Extract Sequencing Run Info\n",
    "    RGPU=$(gzip -dc $FQ1| head -n1 - | cut -f1 -d':' | cut -f2 -d' ')\n",
    "    \n",
    "    # Bowtie2: align to genome\n",
    "    bowtie2 --very-sensitive-local -p $THREADS \\\n",
    "      --rg-id $RGID --rg LB:$RGLB --rg SM:$RGSM \\\n",
    "      --rg PL:$RGPL --rg PU:$RGPU \\\n",
    "      -x hgr1 -1 $FQ1 -2 $FQ2 | \\\n",
    "      samtools view -bS - > aligned_unsorted.bam\n",
    "      \n",
    "    # Calcualte library flagstats\n",
    "      samtools flagstat aligned_unsorted.bam > aligned_unsorted.flagstat\n",
    "      \n",
    "      rm $FQ1 $FQ2 # Remove fastq files to save space\n",
    "\n",
    "      \n",
    "    # Read Subset ------------------------------\n",
    "    # Extract mapped reads, and their unmapped pairs\n",
    "\n",
    "      # Extract Header\n",
    "      samtools view -H aligned_unsorted.bam > align.header.tmp\n",
    "\n",
    "      # Unmapped reads with mapped pairs\n",
    "      # Extract Mapped Reads\n",
    "      # and their unmapped pairs\n",
    "      samtools view -b -F 4 aligned_unsorted.bam > align.F4.bam #mapped\n",
    "      samtools view -b -f 4 -F 8 aligned_unsorted.bam > align.f4F8.bam #unmapped pairs\n",
    "\n",
    "      # Extract just the 45S unit\n",
    "      #aws s3 cp s3://crownproject/resources/rDNA_45s.bed ./\n",
    "      #samtools view -b -L rDNA_45s.bed align.F4.bam > align.F4.45s.bam\n",
    "\n",
    "      # What are the mapped readnames\n",
    "      samtools view align.F4.bam | cut -f1 - > read.names.tmp\n",
    "\n",
    "      # Extract mapped reads\n",
    "      samtools view align.F4.bam | grep -Ff read.names.tmp - > align.F4.tmp.sam\n",
    "\n",
    "\n",
    "      # Extract cases of read pairs mapped on edge of region of interest\n",
    "      # -------|======= R O I ======| ----------\n",
    "      # read:                  ====---====\n",
    "      samtools view align.F4.bam | grep -Ff read.names.tmp - > align.F4.tmp.sam\n",
    "\n",
    "      # Complete mapped reads list\n",
    "      #cut -f1 align.F4.tmp.sam > read.names.45s.long.tmp\n",
    "\n",
    "      # Extract unmapped reads with a mapped pair\n",
    "      samtools view align.f4F8.bam | grep -Ff read.names.tmp - > align.f4F8.tmp.sam\n",
    "\n",
    "      # Re-compile bam file\n",
    "      cat align.header.tmp align.F4.tmp.sam align.f4F8.tmp.sam | samtools view -bS - > align.hgr1.tmp.bam\n",
    "        samtools sort align.hgr1.tmp.bam align.hgr1\n",
    "        samtools index align.hgr1.bam\n",
    "        samtools flagstat align.hgr1.bam > align.hgr1.flagstat\n",
    "\n",
    "      # Clean up \n",
    "      rm *tmp* align.F4.bam align.f4F8.bam\n",
    "\n",
    "    # Rename the total Bam Files\n",
    "      mv aligned_unsorted.bam $LIBRARY.bam\n",
    "      mv aligned_unsorted.bam.bai $LIBRARY.bam.bai\n",
    "      mv aligned_unsorted.flagstat $LIBRARY.flagstat\n",
    "\n",
    "    # Rename the hgr-aligned Bam files\n",
    "      mv align.hgr1.bam $LIBRARY.hgr1.bam\n",
    "      mv align.hgr1.bam.bai $LIBRARY.hgr1.bam.bai\n",
    "      mv align.hgr1.flagstat $LIBRARY.hgr1.flagstat\n",
    "    \n",
    "fi\n",
    "\n",
    "rm $FQ0 $FQ1 $FQ2 # Remove fastq files to save space  \n",
    "  \n",
    "# Alignments (Full)\n",
    " aws s3 cp $LIBRARY.se.flagstat s3://crownproject/tcga/\n",
    "\n",
    "# Alignments (Aligned)\n",
    "  aws s3 cp $LIBRARY.hgr1.bam s3://crownproject/tcga/\n",
    "  aws s3 cp $LIBRARY.hgr1.bam.bai s3://crownproject/tcga/\n",
    "  aws s3 cp $LIBRARY.hgr1.flagstat s3://crownproject/tcga/\n",
    "\n",
    "# VCF\n",
    " aws s3 cp $LIBRARY.hgr1.vcf s3://crownproject/tcga/\n",
    " aws s3 cp $LIBRARY.hgr1.vcf.idx s3://crownproject/tcga/\n",
    " \n",
    "# Shutdown and Terminate instance\n",
    "EC2ID=$(ec2metadata --instance-id)\n",
    "aws ec2 terminate-instances --instance-ids $EC2ID\n",
    "\n",
    "# Script complete"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### TCGA-COAD RNA-seq selection\n",
    "\n",
    "For the initial trail of the multiple-launch pipeline (using queenB / droneB scripts) I first need a selection of RNA-seq files from TCGA to perform alignment against. Let's start with 5 examples.\n",
    "\n",
    "On the [GDC website](https://portal.gdc.cancer.gov); I used the advanced selection to download a `Manifest File`, `Sample File`, `Clinical File` and `Exposure File` for all files which match the following selection criteria:\n",
    "\n",
    "```\n",
    "cases.primary_site in [\"Colorectal\"] and cases.project.project_id in [\"TCGA-COAD\"] and files.data_category in [\"Raw Sequencing Data\",\"Transcriptome Profiling\"] and files.data_format in [\"BAM\"] and files.experimental_strategy in [\"RNA-Seq\"]\n",
    "```\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "id\tfilename\tmd5\tsize\tstate\r\n",
      "01581d3a-8427-41db-82c2-a7dd880a3937\t9e79af22-063c-4f44-a353-3066a096e9c8_gdc_realn_rehead.bam\ta872da35d9fff4cdd63cadc386045983\t6299788578\tsubmitted\r\n",
      "017e4572-bbde-4e44-9de2-22e7d2573603\t5ef172da-d39f-4f80-89a9-10f656a441ba_gdc_realn_rehead.bam\tee95d6e07a8892e76216e1627c58f236\t6167063349\tsubmitted\r\n",
      "01871314-b195-4142-8149-1cff7ea8c3b4\teb089e55-398c-4f58-84d6-484bc5f5707d_gdc_realn_rehead.bam\t2649ec6ab10affb57bb84e88f34d6420\t2254640489\tsubmitted\r\n",
      "0188bc58-2deb-4978-98c3-3028d6fa61f8\tc513f9b0-764d-4db0-a163-9d2b28111957_gdc_realn_rehead.bam\tdbf7c6bd6bf2b6888e55dd3f3e6d30e0\t5795499393\tsubmitted\r\n",
      "\r\n",
      "case_id\tsubmitter_id\tproject_id\tgender\tyear_of_birth\trace\tethnicity\tyear_of_death\tclassification_of_tumor\tlast_known_disease_status\tprimary_diagnosis\ttumor_stage\tage_at_diagnosis\tvital_status\tmorphology\tdays_to_death\tdays_to_last_known_disease_status\tdays_to_recurrence\ttumor_grade\ttissue_or_organ_of_origin\tdays_to_birth\tprogression_or_recurrence\tprior_malignancy\tsite_of_resection_or_biopsy\tdays_to_last_follow_up\ttherapeutic_agents\ttreatment_intent_type\ttreatment_or_therapy\r\n",
      "e3b6018c-ab2c-47c5-b183-a295d0110835\tTCGA-DM-A1D9\tTCGA-COAD\tfemale\t1933\twhite\tnot hispanic or latino\t--\tnot reported\tnot reported\tC18.0\tstage iia\t24514\talive\t8140/3\t--\t--\t--\tnot reported\tC18.0\t-24514.0\tnot reported\tnot reported\tC18.0\t4270.0\t--\t--\t--\r\n",
      "dc22eba5-6fb7-4875-b30a-99fa00758301\tTCGA-D5-6920\tTCGA-COAD\tfemale\t1934\twhite\tnot hispanic or latino\t--\tnot reported\tnot reported\tC18.7\tstage iia\t28124\talive\t8140/3\t--\t--\t--\tnot reported\tC18.7\t-28124.0\tnot reported\tnot reported\tC18.7\t377.0\t--\t--\t--\r\n",
      "bc42fe2d-1a33-428d-a601-b1871f4c8d9b\tTCGA-AA-3966\tTCGA-COAD\tfemale\t1920\tnot reported\tnot reported\t--\tnot reported\tnot reported\tC18.9\tstage iia\t32537\talive\t8480/3\t--\t--\t--\tnot reported\tC18.9\t-32537.0\tnot reported\tnot reported\tC18.9\t61.0\t--\t--\t--\r\n",
      "44c2854e-3f8c-48e1-b4d5-1b5778d4fce4\tTCGA-AA-3858\tTCGA-COAD\tmale\t1942\tnot reported\tnot reported\t--\tnot reported\tnot reported\tC18.9\tstage i\t24806\talive\t8140/3\t--\t--\t--\tnot reported\tC18.9\t-24806.0\tnot reported\tnot reported\tC18.9\t945.0\t--\t--\t--\r\n",
      "\r\n",
      "File ID\tFile Name\tData Category\tData Type\tProject ID\tCase ID\tSample ID\tSample Type\r\n",
      "76083065-80bd-4549-a157-27eac51b4ca2\t57a094ea-70d5-488a-b48c-053b66387bd8_gdc_realn_rehead.bam\tRaw Sequencing Data\tAligned Reads\tTCGA-COAD\tTCGA-AA-3831\tTCGA-AA-3831-01A\tPrimary Tumor\r\n",
      "0ef08b70-4b13-4a13-9b6f-f96cda68bd01\t0d52bbb6-caec-4d5b-bea6-9b5670c3790f_gdc_realn_rehead.bam\tRaw Sequencing Data\tAligned Reads\tTCGA-COAD\tTCGA-CK-5913\tTCGA-CK-5913-01A\tPrimary Tumor\r\n",
      "958a5e6e-92b2-41a7-8ed4-3c464869120c\t81fac8d9-65ad-4cf8-adef-df09862c729c_gdc_realn_rehead.bam\tRaw Sequencing Data\tAligned Reads\tTCGA-COAD\tTCGA-AA-3930\tTCGA-AA-3930-01A\tPrimary Tumor\r\n",
      "af831284-873e-4530-8bd6-bde434a78a1a\t0a3c82ca-4ca5-4b0f-b8c1-b02ab8e8bd66_gdc_realn_rehead.bam\tRaw Sequencing Data\tAligned Reads\tTCGA-COAD\tTCGA-CM-5861\tTCGA-CM-5861-01A\tPrimary Tumor\r\n"
     ]
    }
   ],
   "source": [
    "cd /home/artem/Desktop/Crown/data2/tcga_0/coad\n",
    "\n",
    "head -n5 COAD_rnaseq_manifest.txt\n",
    "echo ''\n",
    "\n",
    "head -n5 COAD_clinical_table.tsv\n",
    "echo ''\n",
    "\n",
    "head -n5 COAD_sample_table.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TCGA-F4-6461-01A\tTCGA-COAD\t01581d3a-8427-41db-82c2-a7dd880a3937\r\n",
      "TCGA-G4-6315-01A\tTCGA-COAD\t017e4572-bbde-4e44-9de2-22e7d2573603\r\n",
      "TCGA-AA-3529-01A\tTCGA-COAD\t01871314-b195-4142-8149-1cff7ea8c3b4\r\n",
      "TCGA-AZ-6599-01A\tTCGA-COAD\t0188bc58-2deb-4978-98c3-3028d6fa61f8\r\n",
      "TCGA-QG-A5Z1-01A\tTCGA-COAD\t019ae823-acfc-49a6-a144-3b363b15a0dc"
     ]
    }
   ],
   "source": [
    "# Which can be simplied to the input parameters:\n",
    "\n",
    "cat ../tcga0_input.tsv\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Discussion\n",
    "\n",
    "This should be the neccesary pre-requisites to run the analysis and debug in next experiment."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Addendum - 20180906\n",
    "\n",
    "Initial hgr1 alignments of TCGA data is now complete (see TCGA_3_General). Load all this data from S3 to an AMI image containing the data organized for analysis.\n",
    "\n",
    "Run initial MACP GVCF analysis.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Login to EC2 TCGA AMI\n",
    "mkdir tcga; cd tcga\n",
    "\n",
    "# Download TCGA data\n",
    "aws s3 cp --recursive s3://crownproject/tcga/ ./\n",
    "\n",
    "# Move COAD and LUSC data so folder names are matching\n",
    "mv tcga-lusc/ TCGA-LUSC\n",
    "mv tcga-coad-1/ TCGA-COAD\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cd ~/software\n",
    "\n",
    "wget https://github.com/samtools/bcftools/releases/download/1.9/bcftools-1.9.tar.bz2\n",
    "tar -xvf bcftools-1.9.tar.bz2\n",
    "\n",
    "mv bcftools-1.9.tar.bz2 zips/\n",
    "\n",
    "./configure --prefix=$PWD\n",
    "make\n",
    "make install\n",
    "\n",
    "cp bin/* ~/bin/\n",
    "\n",
    "sudo mv /usr/bin/bcftools /usr/bin/bcftools_0.1.19\n",
    "\n",
    "# Move to home directory and run ADcalc.sh script\n",
    "cd ~\n",
    "\n",
    "screen -L\n",
    "\n",
    "bash scripts/ADcalc.sh #see below\n",
    "\n",
    "mkdir tcga/vcf; cd tcga\n",
    "\n",
    "mv *.vcf vcf/\n",
    "mv ~/screenlog.0 18S_1248_vcf.log\n",
    "\n",
    "aws s3 cp --recursive ./ s3://crownproject/tcga/vcf/\n",
    "\n",
    "# Created crown-180906 EC2 AMI: ami-0ed79a4de8e2ca023\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "# ADcalc.sh\n",
    "# Allelic Depth Calculator\n",
    "# for a position\n",
    "\n",
    "cd ~/tcga/\n",
    "\n",
    "# Controls -----------------\n",
    "REGION='chr13:1004900-1004915'\n",
    "OUTPUT='18S_1248.vcf'\n",
    "DEPTH='100000'\n",
    "BAMLIST='bam.list.tmp'\n",
    "\n",
    "# Iterate through every TCGA Cancer Type\n",
    "for TYPE in $(ls)\n",
    "do\n",
    "    echo Analyzing $TYPE...\n",
    "    \n",
    "    cd $TYPE\n",
    "\n",
    "    ls *.bam > bam.list.tmp\n",
    "\n",
    "    # Iterate through every bam file in directory\n",
    "    # look-up position and return VCF\n",
    "        bcftools mpileup -f ~/resources/hgr1/hgr1.fa \\\n",
    "      --max-depth $DEPTH --min-BQ 30 \\\n",
    "      -a FORMAT/DP,AD \\\n",
    "      -r \"$REGION\" \\\n",
    "      --ignore-RG \\\n",
    "      -b $BAMLIST |\n",
    "      bcftools annotate -x INFO,FORMAT/PL - |\n",
    "      bcftools view -O v -H - \\\n",
    "      > ../$TYPE.$OUTPUT\n",
    "      \n",
    "    rm bam.list.tmp\n",
    "    \n",
    "    cd ..\n",
    "done"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
