{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# TCGA RNA-seq hgr1 alignments Analysis I\n",
    "```\n",
    "pi:ababaian\n",
    "files: ~/Crown/data2/tcga_analysis_1\n",
    "start: 2018 09 10\n",
    "complete : 2018 08 13\n",
    "```\n",
    "## Introduction\n",
    "\n",
    "hgr1 Alignment of the TCGA-XXX data. Only libraries for which there is a matched-normal from the same sample.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Materials and Methods\n",
    "\n",
    "#### TCGA Data Input\n",
    "\n",
    "Search Term for limiting files\n",
    "```\n",
    "cases.project.project_id in [\"TCGA-LUSC\"] and files.data_category in [\"Raw Sequencing Data\",\"Transcriptome Profiling\"] and files.data_format in [\"BAM\"] and files.experimental_strategy in [\"RNA-Seq\"]\n",
    "```\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TCGA-BL-A13J-01B.flagstat\tTCGA-BL-A13J-11A.hgr1.bam\r\n",
      "TCGA-BL-A13J-01B.hgr1.bam\tTCGA-BL-A13J-11A.hgr1.bam.bai\r\n",
      "TCGA-BL-A13J-01B.hgr1.bam.bai\tTCGA-BL-A13J-11A.hgr1.flagstat\r\n",
      "TCGA-BL-A13J-01B.hgr1.flagstat\ttmp.bam\r\n",
      "TCGA-BL-A13J-11A.flagstat\r\n"
     ]
    }
   ],
   "source": [
    "WORKDIR='/home/artem/Crown/data2/tcga_analysis'\n",
    "cd $WORKDIR\n",
    "\n",
    "# List downloaded BLCA files\n",
    "ls BLCA*"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "From IGV in TCGA-BL-A13J-11A\n",
    "```\n",
    "chr13:1,004,908\n",
    "<hr>Total count: 3542\n",
    "A      : 151  (4%,     63+,   88- )\n",
    "C      : 1646  (46%,     725+,   921- )\n",
    "G      : 203  (6%,     100+,   103- )\n",
    "T      : 1542  (44%,     649+,   893- )\n",
    "N      : 0\n",
    "---------------\n",
    "DEL: 152\n",
    "INS: 0\n",
    "```\n",
    "\n",
    "From initial ADcalc calulation (BLCA Files)\n",
    "```\n",
    "chr13:1004908\t\t\t\t\t\n",
    "FILENAME   DP  T\tC\tG\tA\n",
    "TCGA-BL-A13J-01B.hgr1.bam\t0\t0\t0\t0\t0\n",
    "TCGA-BL-A13J-11A.hgr1.bam\t0\t0\t0\t0\t0\n",
    "TCGA-BT-A20N-01A.hgr1.bam\t0\t0\t0\t0\t0\n",
    "TCGA-BT-A20N-11A.hgr1.bam\t2672\t1370\t1053\t148\t101\n",
    "TCGA-BT-A20Q-01A.hgr1.bam\t2\t0\t2\t0\t0\n",
    "TCGA-BT-A20Q-11A.hgr1.bam\t2155\t1040\t902\t133\t80\n",
    "TCGA-BT-A20R-01A.hgr1.bam\t0\t0\t0\t0\t0\n",
    "TCGA-BT-A20R-11A.hgr1.bam\t0\t0\t0\t0\t0\n",
    "TCGA-BT-A20U-01A.hgr1.bam\t0\t0\t0\t0\t0\n",
    "TCGA-BT-A20U-11A.hgr1.bam\t2863\t1667\t969\t127\t100\n",
    "TCGA-BT-A20W-01A.hgr1.bam\t0\t0\t0\t0\t0\n",
    "TCGA-BT-A20W-11A.hgr1.bam\t2\t2\t0\t0\t0\n",
    "TCGA-BT-A2LA-01A.hgr1.bam\t7\t5\t2\t0\t0\n",
    "TCGA-BT-A2LA-11A.hgr1.bam\t11044\t5534\t4582\t529\t399\n",
    "TCGA-BT-A2LB-01A.hgr1.bam\t6507\t3001\t2904\t353\t249\n",
    "TCGA-BT-A2LB-11A.hgr1.bam\t1\t0\t1\t0\t0\n",
    "TCGA-CU-A0YN-01A.hgr1.bam\t4701\t3317\t1124\t147\t113\n",
    "TCGA-CU-A0YN-11A.hgr1.bam\t12642\t5685\t5800\t649\t508\n",
    "TCGA-CU-A0YR-01A.hgr1.bam\t5010\t2877\t1788\t210\t135\n",
    "TCGA-CU-A0YR-11A.hgr1.bam\t7513\t3574\t3259\t393\t287\n",
    "TCGA-GC-A3BM-01A.hgr1.bam\t1\t1\t0\t0\t0\n",
    "TCGA-GC-A3BM-11A.hgr1.bam\t43\t10\t26\t5\t2\n",
    "TCGA-GC-A3WC-01A.hgr1.bam\t2013\t1121\t761\t75\t56\n",
    "TCGA-GC-A3WC-11A.hgr1.bam\t2\t1\t1\t0\t0\n",
    "TCGA-GC-A6I3-01A.hgr1.bam\t5103\t2614\t2063\t225\t201\n",
    "TCGA-GC-A6I3-11A.hgr1.bam\t3220\t1596\t1362\t158\t104\n",
    "TCGA-GD-A2C5-01A.hgr1.bam\t0\t0\t0\t0\t0\n",
    "TCGA-GD-A2C5-11A.hgr1.bam\t0\t0\t0\t0\t0\n",
    "TCGA-GD-A3OP-01A.hgr1.bam\t4586\t2551\t1707\t186\t142\n",
    "TCGA-GD-A3OP-11A.hgr1.bam\t7151\t3550\t2967\t382\t252\n",
    "TCGA-GD-A3OQ-01A.hgr1.bam\t6680\t3374\t2716\t312\t278\n",
    "TCGA-GD-A3OQ-11A.hgr1.bam\t5821\t2796\t2480\t315\t230\n",
    "TCGA-K4-A3WV-01A.hgr1.bam\t0\t0\t0\t0\t0\n",
    "TCGA-K4-A3WV-11A.hgr1.bam\t5304\t2557\t2261\t284\t202\n",
    "TCGA-K4-A54R-01A.hgr1.bam\t0\t0\t0\t0\t0\n",
    "TCGA-K4-A54R-11A.hgr1.bam\t1685\t865\t668\t100\t52\n",
    "TCGA-K4-A5RI-01A.hgr1.bam\t1\t0\t1\t0\t0\n",
    "TCGA-K4-A5RI-11A.hgr1.bam\t3284\t1660\t1369\t150\t105\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bash: cd: BLCA: No such file or directory\r\n",
      " Default Run ------------------------------------\r\n",
      "[mpileup] 2 samples in 2 input files\r\n",
      "\r\n",
      "------------------------------------\r\n",
      "[mpileup] 2 samples in 2 input files\r\n",
      "\r\n",
      "------------------------------------\r\n",
      "[mpileup] 2 samples in 2 input files\r\n",
      "chr13\t1004908\t.\tT\tC,G,A\t0\t.\t.\tDP:AD\t0:0,0,0,0\t3519:1528,1641,201,149\r\n",
      "\r\n",
      "------------------------------------\r\n",
      "[mpileup] 2 samples in 2 input files\r\n",
      "chr13\t1004908\t.\tT\tC,G,A\t0\t.\t.\tDP:AD\t0:0,0,0,0\t3463:1499,1619,198,147\r\n",
      "\r\n",
      "UNC13-SN749:141:D0J7WACXX:8:1208:9671:19372\t137\tchr13\t1004621\t44\t48M\t=\t1004621\t0\tGAAATTCTTGGACCGGCGCAAGACGGACCAGAGCGAAAGCATTTGCCA\tCCCFFFFFHHHHHJJJJIIJJJJJJJHIJJIJIJJJJHHHFFFFFFCC\tAS:i:96\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:48\tYT:Z:UP\tRG:Z::120110_UNC13-SN749_0141_AD0J7WACXX_TTAGGC_L008\r\n",
      "UNC13-SN749:141:D0J7WACXX:8:1208:9671:19372\t89\tchr13\t1004893\t44\t48M\t=\t1004893\t0\tGCGGCTTAATTTGACTCAACACGGGAAACCTCACCCGGCCCGGACACG\tFHHIJJJJJJJJJIIJJJJIHJJJJIHHIHGGEJJHHHHHFFFFFBCB\tAS:i:96\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:48\tYT:Z:UP\tRG:Z::120110_UNC13-SN749_0141_AD0J7WACXX_TTAGGC_L008\r\n"
     ]
    }
   ],
   "source": [
    "# Testing BCFtools\n",
    "cd BLCA\n",
    "\n",
    "echo ' Default Run ------------------------------------'\n",
    "bcftools mpileup -f ~/Crown/resources/hgr1/hgr1.fa \\\n",
    "  --max-depth 10000 -a FORMAT/DP,AD -r chr13:1004908 \\\n",
    "  --ignore-RG --min-BQ 30\\\n",
    "  -b tmp.bam |\\\n",
    "  bcftools annotate -x INFO,FORMAT/PL - |\\\n",
    "  bcftools view -O v -H - | tail -n1 -\n",
    "echo ''\n",
    "\n",
    "echo '------------------------------------'\n",
    "# Excllude BAQ 30 cutoff\n",
    "bcftools mpileup -f ~/Crown/resources/hgr1/hgr1.fa \\\n",
    "  --max-depth 10000 -a FORMAT/DP,AD -r chr13:1004908 \\\n",
    "  --ignore-RG \\\n",
    "  -b tmp.bam |\\\n",
    "  bcftools annotate -x INFO,FORMAT/PL - |\\\n",
    "  bcftools view -O v -H - | tail -n1 -\n",
    "echo ''\n",
    "\n",
    "echo '------------------------------------'\n",
    "# Add A: count orphans (do not discard anomolous reads)\n",
    "bcftools mpileup -f ~/Crown/resources/hgr1/hgr1.fa \\\n",
    "  --max-depth 10000 -a FORMAT/DP,AD -r chr13:1004908 \\\n",
    "  --ignore-RG -A \\\n",
    "  -b tmp.bam |\\\n",
    "  bcftools annotate -x INFO,FORMAT/PL - |\\\n",
    "  bcftools view -O v -H - | tail -n1 -\n",
    "echo ''\n",
    "\n",
    "echo '------------------------------------'\n",
    "# Add B: disable per-base alignment quality score cutoffs (no reads gained)\n",
    "# or still use --min-BQ 30 (some reads lost)\n",
    "bcftools mpileup -f ~/Crown/resources/hgr1/hgr1.fa \\\n",
    "  --max-depth 10000 -a FORMAT/DP,AD -r chr13:1004908 \\\n",
    "  --ignore-RG --min-BQ 30 -A  \\\n",
    "  -b tmp.bam |\\\n",
    "  bcftools annotate -x INFO,FORMAT/PL - |\\\n",
    "  bcftools view -O v -H - | tail -n1 -\n",
    "echo ''\n",
    "\n",
    "\n",
    " samtools view TCGA-BL-A13J-11A.hgr1.bam |\\\n",
    "   grep UNC13-SN749:141:D0J7WACXX:8:1208:9671:1937 -       "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Notes:\n",
    "- The issue with the original ADcalc script returning zeros is that the reads over 1248 are flagged 73/89/137 --> common setting of flag:8 Mate unmapped. Even though if you grep for a single read you can see above that both read and it's mate are in fact mapped eyt the 8 flag was still set. This is likely an issue with inserte size from teh aligner but can be corrected by adding the `-A` tag to `bcftools` runs.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "# ADcalc.sh\n",
    "# Allelic Depth Calculator\n",
    "# for a position\n",
    "\n",
    "cd ~/tcga/\n",
    "\n",
    "# Controls -----------------\n",
    "REGION='chr13:1004908'\n",
    "OUTPUT='18S_1248.vcf'\n",
    "DEPTH='100000'\n",
    "BAMLIST='bam.list.tmp'\n",
    "\n",
    "# -A: do not discard anomolous read pairs\n",
    "\n",
    "# Iterate through every TCGA Cancer Type\n",
    "for TYPE in $(ls)\n",
    "do\n",
    "    echo Analyzing $TYPE...\n",
    "    \n",
    "    cd $TYPE\n",
    "\n",
    "    ls *.bam > bam.list.tmp\n",
    "\n",
    "    # Iterate through every bam file in directory\n",
    "    # look-up position and return VCF\n",
    "        bcftools mpileup -f ~/resources/hgr1/hgr1.fa \\\n",
    "      --max-depth $DEPTH --min-BQ 30 -A \\\n",
    "      -a FORMAT/DP,AD \\\n",
    "      -r \"$REGION\" \\\n",
    "      --ignore-RG \\\n",
    "      -b $BAMLIST |\n",
    "      bcftools annotate -x INFO,FORMAT/PL - |\n",
    "      bcftools view -O v -H - |\\\n",
    "      tail -n1 - \\\n",
    "      > ../$TYPE.$OUTPUT\n",
    "      \n",
    "    rm bam.list.tmp\n",
    "    \n",
    "    cd ..\n",
    "done\n",
    "\n",
    "# On AMI image with TCGA data loaded (crown-180906 ami-0ed79a4de8e2ca023)\n",
    "# ran above script to generate:\n",
    "# s3://crownproject/tcga/vcf_1248/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Results -- Initial Run (pre-failed runs)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Completed 5.0 KiB/202.2 KiB with 26 file(s) remaining\r",
      "download: s3://crownproject/tcga/vcf_1248/TCGA-BRCA.18S_1248macp.vcf to vcf/TCGA-BRCA.18S_1248macp.vcf\r\n",
      "Completed 5.0 KiB/202.2 KiB with 25 file(s) remaining\r",
      "Completed 5.9 KiB/202.2 KiB with 25 file(s) remaining\r",
      "download: s3://crownproject/tcga/vcf_1248/TCGA-BLCA.18S_1248macp.vcf to vcf/TCGA-BLCA.18S_1248macp.vcf\r\n",
      "Completed 5.9 KiB/202.2 KiB with 24 file(s) remaining\r",
      "Completed 6.1 KiB/202.2 KiB with 24 file(s) remaining\r",
      "download: s3://crownproject/tcga/vcf_1248/TCGA-CESC.18S_1248macp.vcf to vcf/TCGA-CESC.18S_1248macp.vcf\r\n",
      "Completed 6.1 KiB/202.2 KiB with 23 file(s) remaining\r",
      "Completed 40.2 KiB/202.2 KiB with 23 file(s) remaining\r",
      "download: s3://crownproject/tcga/vcf_1248/18S_1248macp.vcf.bamlist to vcf/18S_1248macp.vcf.bamlist\r\n",
      "Completed 40.2 KiB/202.2 KiB with 22 file(s) remaining\r",
      "Completed 42.1 KiB/202.2 KiB with 22 file(s) remaining\r",
      "Completed 42.4 KiB/202.2 KiB with 22 file(s) remaining\r",
      "Completed 43.5 KiB/202.2 KiB with 22 file(s) remaining\r",
      "Completed 44.9 KiB/202.2 KiB with 22 file(s) remaining\r",
      "Completed 45.3 KiB/202.2 KiB with 22 file(s) remaining\r",
      "download: s3://crownproject/tcga/vcf_1248/TCGA-HNSC.18S_1248macp.vcf to vcf/TCGA-HNSC.18S_1248macp.vcf\r\n",
      "Completed 45.3 KiB/202.2 KiB with 21 file(s) remaining\r",
      "download: s3://crownproject/tcga/vcf_1248/TCGA-KIRP.18S_1248macp.vcf to vcf/TCGA-KIRP.18S_1248macp.vcf\r\n",
      "Completed 45.3 KiB/202.2 KiB with 20 file(s) remaining\r",
      "Completed 47.5 KiB/202.2 KiB with 20 file(s) remaining\r",
      "download: s3://crownproject/tcga/vcf_1248/TCGA-LIHC.18S_1248macp.vcf to vcf/TCGA-LIHC.18S_1248macp.vcf\r\n",
      "Completed 47.5 KiB/202.2 KiB with 19 file(s) remaining\r",
      "download: s3://crownproject/tcga/vcf_1248/TCGA-ESCA.18S_1248macp.vcf to vcf/TCGA-ESCA.18S_1248macp.vcf\r\n",
      "Completed 47.5 KiB/202.2 KiB with 18 file(s) remaining\r",
      "Completed 49.1 KiB/202.2 KiB with 18 file(s) remaining\r",
      "download: s3://crownproject/tcga/vcf_1248/TCGA-COAD.18S_1248macp.vcf to vcf/TCGA-COAD.18S_1248macp.vcf\r\n",
      "Completed 49.1 KiB/202.2 KiB with 17 file(s) remaining\r",
      "download: s3://crownproject/tcga/vcf_1248/TCGA-KICH.18S_1248macp.vcf to vcf/TCGA-KICH.18S_1248macp.vcf\r\n",
      "Completed 49.1 KiB/202.2 KiB with 16 file(s) remaining\r",
      "Completed 51.3 KiB/202.2 KiB with 16 file(s) remaining\r",
      "Completed 53.7 KiB/202.2 KiB with 16 file(s) remaining\r",
      "download: s3://crownproject/tcga/vcf_1248/TCGA-CHOL.18S_1248macp.vcf to vcf/TCGA-CHOL.18S_1248macp.vcf\r\n",
      "Completed 53.7 KiB/202.2 KiB with 15 file(s) remaining\r",
      "download: s3://crownproject/tcga/vcf_1248/TCGA-LUAD.18S_1248macp.vcf to vcf/TCGA-LUAD.18S_1248macp.vcf\r\n",
      "Completed 53.7 KiB/202.2 KiB with 14 file(s) remaining\r",
      "download: s3://crownproject/tcga/vcf_1248/TCGA-LUSC.18S_1248macp.vcf to vcf/TCGA-LUSC.18S_1248macp.vcf\r\n",
      "Completed 53.7 KiB/202.2 KiB with 13 file(s) remaining\r",
      "Completed 53.8 KiB/202.2 KiB with 13 file(s) remaining\r",
      "download: s3://crownproject/tcga/vcf_1248/TCGA-SKCM.18S_1248macp.vcf to vcf/TCGA-SKCM.18S_1248macp.vcf\r\n",
      "Completed 53.8 KiB/202.2 KiB with 12 file(s) remaining\r",
      "Completed 56.1 KiB/202.2 KiB with 12 file(s) remaining\r",
      "Completed 56.3 KiB/202.2 KiB with 12 file(s) remaining\r",
      "Completed 56.4 KiB/202.2 KiB with 12 file(s) remaining\r",
      "download: s3://crownproject/tcga/vcf_1248/TCGA-PRAD.18S_1248macp.vcf to vcf/TCGA-PRAD.18S_1248macp.vcf\r\n",
      "Completed 56.4 KiB/202.2 KiB with 11 file(s) remaining\r",
      "download: s3://crownproject/tcga/vcf_1248/TCGA-PAAD.18S_1248macp.vcf to vcf/TCGA-PAAD.18S_1248macp.vcf\r\n",
      "Completed 56.4 KiB/202.2 KiB with 10 file(s) remaining\r",
      "download: s3://crownproject/tcga/vcf_1248/TCGA-SARC.18S_1248macp.vcf to vcf/TCGA-SARC.18S_1248macp.vcf\r\n",
      "Completed 56.4 KiB/202.2 KiB with 9 file(s) remaining\r",
      "Completed 56.6 KiB/202.2 KiB with 9 file(s) remaining\r",
      "download: s3://crownproject/tcga/vcf_1248/TCGA-PCPG.18S_1248macp.vcf to vcf/TCGA-PCPG.18S_1248macp.vcf\r\n",
      "Completed 56.6 KiB/202.2 KiB with 8 file(s) remaining\r",
      "Completed 86.0 KiB/202.2 KiB with 8 file(s) remaining\r",
      "download: s3://crownproject/tcga/vcf_1248/TCGA-PAN.18S_1248macp.vcf to vcf/TCGA-PAN.18S_1248macp.vcf\r\n",
      "Completed 86.0 KiB/202.2 KiB with 7 file(s) remaining\r",
      "Completed 86.8 KiB/202.2 KiB with 7 file(s) remaining\r",
      "Completed 87.2 KiB/202.2 KiB with 7 file(s) remaining\r",
      "Completed 89.7 KiB/202.2 KiB with 7 file(s) remaining\r",
      "download: s3://crownproject/tcga/vcf_1248/TCGA-READ.18S_1248macp.vcf to vcf/TCGA-READ.18S_1248macp.vcf\r\n",
      "Completed 89.7 KiB/202.2 KiB with 6 file(s) remaining\r",
      "download: s3://crownproject/tcga/vcf_1248/TCGA-THCA.18S_1248macp.vcf to vcf/TCGA-THCA.18S_1248macp.vcf\r\n",
      "Completed 89.7 KiB/202.2 KiB with 5 file(s) remaining\r",
      "download: s3://crownproject/tcga/vcf_1248/TCGA-STAD.18S_1248macp.vcf to vcf/TCGA-STAD.18S_1248macp.vcf\r\n",
      "Completed 89.7 KiB/202.2 KiB with 4 file(s) remaining\r",
      "Completed 89.8 KiB/202.2 KiB with 4 file(s) remaining\r",
      "download: s3://crownproject/tcga/vcf_1248/TCGA-THYM.18S_1248macp.vcf to vcf/TCGA-THYM.18S_1248macp.vcf\r\n",
      "Completed 89.8 KiB/202.2 KiB with 3 file(s) remaining\r",
      "Completed 93.0 KiB/202.2 KiB with 3 file(s) remaining\r",
      "download: s3://crownproject/tcga/vcf_1248/TCGA-KIRC.18S_1248macp.vcf to vcf/TCGA-KIRC.18S_1248macp.vcf\r\n",
      "Completed 93.0 KiB/202.2 KiB with 2 file(s) remaining\r",
      "Completed 96.1 KiB/202.2 KiB with 2 file(s) remaining\r",
      "download: s3://crownproject/tcga/vcf_1248/screen.18S_1248macp.log to vcf/screen.18S_1248macp.log\r\n",
      "Completed 96.1 KiB/202.2 KiB with 1 file(s) remaining\r",
      "Completed 202.2 KiB/202.2 KiB with 1 file(s) remaining\r",
      "download: s3://crownproject/tcga/vcf_1248/bamlist.alh.txt to vcf/bamlist.alh.txt\r\n"
     ]
    }
   ],
   "source": [
    "# TCGA-Analysis Download data files\n",
    "cd $WORKDIR\n",
    "\n",
    "aws s3 cp --recursive s3://crownproject/tcga/vcf_1248/ ./vcf/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rm: cannot remove 'TCGA-PAN.18S_1248macp.vcf': No such file or directory\r\n"
     ]
    }
   ],
   "source": [
    "cd vcf\n",
    "#rm TCGA-PAN.18S_1248macp.vcf\n",
    "\n",
    "for FILE in $(ls *vcf)\n",
    "do\n",
    "    cat $FILE |\\\n",
    "      sed 's/^[A-Za-z0-9\\t .,]*:AD\\t//g' - |\\\n",
    "      sed 's/\\t/\\n/g' - \\\n",
    "      >> TCGA-18S_1248.tsv\n",
    "done\n",
    "\n",
    "cd .."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Results -- 180914 TCGA-paired data checkpoint\n",
    "\n",
    "Initial scripts above used for data exploration. The ~130 failed libraries were re-analyzed in TCGA-3-generic re-run experiment. Will repeat these analyses for that dataset.\n",
    "\n",
    "\n",
    "Note: `tcga_analysis_1` data from previous analysis (used above) will be deleted"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
