{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# TCGA-COAD Batch Effects\n",
    "```\n",
    "pi:ababaian\n",
    "files: ~/Crown/data2/tcga_batch/\n",
    "start: 2019 08 08\n",
    "complete : 2019 08 22\n",
    "```\n",
    "## Introduction\n",
    "\n",
    "Batch effects effect RNA-seq library prep. In the lab we have determined that even lot numbers of RT enzyme (SSIII) can alter the VAF at macp.1248. To test if the 'hypo-modification' is simply a batch effect, the VAF for each RNA-seq library will be taken and broken down into distinct batches by when the samples were prepared/sequenced.\n",
    "\n",
    "The null hypothesis is that there is no hypo-modification and the effect is simply a batch-effect in TCGA-COAD, therefore the per-batch variance of the normal samples will equal the per-batch variance of the CRC samples.\n",
    "\n",
    "\n",
    "### TCGA Data Collection Overview\n",
    "(Taken from https://bioinformatics.mdanderson.org/public-software/tcga-batch-effects/)\n",
    "\n",
    "\n",
    "\"Below is a quick overview of the various steps involved in the TCGA Data Collection Process, each of which could potentially introduce systematic errors. Relevant terms are highlighted in bold and explanatory links have been provided wherever possible.\n",
    "\n",
    "- Biospecimens consisting of Tumor and Normal tissue samples and clinical metadata are collected from patients donors at various Tissue Source Sites (TSS). Each TSS is identified by its unique TSS ID.\n",
    "- These biospecimens are transported to TCGA Biospecimen Core Resources (BCR) laboratories, which ensure these specimens meet the TCGA biospecimen criteria . Specimens of sufficient quality are cataloged, processed and stored for analysis. Any patient identifying information is removed during the process.\n",
    "- Specimens are grouped into batches consisting of a fixed set of patient cases of the same disease study. Each batch is uniquely determined by the first shipment of a group of analytes from the Biospecimen Core Resource.\n",
    "- Analytes are transported to sequencing centers on various ShipDates and processed into usable data on plates each with its own unique PlateID.\n",
    "- This data are then sent to TCGA Genome Characterization Centers and Genome Sequencing Centers (CGCCandGSC) for interpretation.\n",
    "- TCGA Genome Characterization Centers analyze many of the genetic changes involved in cancer including how the genome is rearranged or how gene expression changes in tumors compared to normal cells.\n",
    "- High-throughput TCGA Genome Sequencing Centers identify the changes in DNA sequence associated with specific types of cancer.\n",
    "- The information that is generated by the TCGA Research Network is centrally managed at the TCGA Data Coordinating Center and made available on the TCGA Data Portal by release date.\n",
    "\"\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Materials and Methods"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "TCGA batch information was downloaded from: [BatchEffectsViewer App](https://bioinformatics.mdanderson.org/BatchEffectsViewer)\n",
    "\n",
    "- TCGA-COAD Disease and  RNA-seq Counts workflow was selected\n",
    "- The file `GDC.2018.11.20.1200..current._COAD_RNASeq.counts.zip` was downloaded (press \"Download Archive\")\n",
    "\n",
    "![TCGA-COAD Batch Effects](../../data2/tcga_batch/plot/BatchEffectViewer_COAD.png)\n",
    "\n",
    "\n",
    "- In that zip was the `BatchData.tsv` file listing sample-batch"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Results / Discussion\n",
    "\n",
    "Ran `crc_batch.Rmd` script"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "### VAF by Batch Effect\n",
    "\n",
    "Sepearte VAF values by batch to see differences in mean.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "There are clear batch effects in the data as expected. Overall hypo-macp is observed across each data-set (relative to intra-batch controls) which rules out that this is simply a batch-specific effect. This does raise a cautionary flag that the pan-TCGA data needs to be normalized by batch for reliable measurment, especially in the large cancer cohorts without normal controls (i.e. DLBCL).\n",
    "\n",
    "![VAF by BatchId](../../data2/tcga_batch/plot/macpVAF_batch.png)\n",
    "\n",
    "![VAF by date](../../data2/tcga_batch/plot/macpVAF_date.png)\n",
    "\n",
    "![VAF by plate](../../data2/tcga_batch/plot/macpVAF_plate.png)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### VAF and CMS reanalyzed\n",
    "\n",
    "Breakdown CMS classification per batch, and look at VAF values between CMS groups"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "The batches `116.77.0`, `76.78.0`, `138.78.0`, `89.77.0` have a high number of samples, show consistency and normal controls within batch to test if there is a correlation between CMS group and hypo-macp.\n",
    "\n",
    "Batch `116.77.0` has significant difference between groups by ANNOVA at p = 0.0143 (unadjusted). Tukey follow-up shows no single comparison as significant. Other batches do not replicate this finding although may trend towards decrease in CMS2.\n",
    "\n",
    "![VAF/CMS 1](../../data2/tcga_batch/plot/macpVAF.cms.batch_116.77.0.png) \n",
    "![VAF/CMS 2](../../data2/tcga_batch/plot/macpVAF.cms.batch_76.78.0.png)\n",
    "![VAF/CMS 3](../../data2/tcga_batch/plot/macpVAF.cms.batch_138.78.0.png)\n",
    "![VAF/CMS 4](../../data2/tcga_batch/plot/macpVAF.cms.batch_89.77.0.png)\n",
    "\n",
    "\n",
    "#### All Stat Tests\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[1] \"Batch: 76.78.0\"\n",
    "[1] \"\"\n",
    "            Df Sum Sq Mean Sq F value Pr(>F)\n",
    "cms          4 0.0881 0.02202   0.818  0.526\n",
    "Residuals   24 0.6461 0.02692               \n",
    "  Tukey multiple comparisons of means\n",
    "    95% family-wise confidence level\n",
    "\n",
    "Fit: aov(formula = VAF ~ cms, data = MACP.batch)\n",
    "\n",
    "$cms\n",
    "                   diff        lwr       upr     p adj\n",
    "CMS2-CMS1   0.068785980 -0.2494036 0.3869755 0.9674370\n",
    "CMS3-CMS1  -0.087372505 -0.4820381 0.3072931 0.9645496\n",
    "CMS4-CMS1   0.006826659 -0.3154164 0.3290698 0.9999963\n",
    "NOLBL-CMS1 -0.068913107 -0.4380889 0.3002627 0.9808993\n",
    "CMS3-CMS2  -0.156158485 -0.4743480 0.1620311 0.6055123\n",
    "CMS4-CMS2  -0.061959321 -0.2840500 0.1601314 0.9212515\n",
    "NOLBL-CMS2 -0.137699087 -0.4236615 0.1482633 0.6221757\n",
    "CMS4-CMS3   0.094199164 -0.2280439 0.4164423 0.9081439\n",
    "NOLBL-CMS3  0.018459398 -0.3507164 0.3876352 0.9998858\n",
    "NOLBL-CMS4 -0.075739766 -0.3662058 0.2147262 0.9372712\n",
    "\n",
    "[1] \"Batch: 29.82.0\"\n",
    "[1] \"\"\n",
    "[1] \"Batch: 28.85.0\"\n",
    "[1] \"\"\n",
    "            Df Sum Sq  Mean Sq F value Pr(>F)\n",
    "cms          4 0.0122 0.003061    0.16  0.956\n",
    "Residuals   23 0.4393 0.019100               \n",
    "  Tukey multiple comparisons of means\n",
    "    95% family-wise confidence level\n",
    "\n",
    "Fit: aov(formula = VAF ~ cms, data = MACP.batch)\n",
    "\n",
    "$cms\n",
    "                   diff        lwr       upr     p adj\n",
    "CMS2-CMS1   0.093092963 -0.3288315 0.5150174 0.9644745\n",
    "CMS3-CMS1   0.074944411 -0.3967815 0.5466703 0.9893656\n",
    "CMS4-CMS1   0.099843578 -0.3476749 0.5473620 0.9630428\n",
    "NOLBL-CMS1  0.117432076 -0.3393145 0.5741787 0.9394179\n",
    "CMS3-CMS2  -0.018148552 -0.2765234 0.2402263 0.9995529\n",
    "CMS4-CMS2   0.006750615 -0.2042116 0.2177128 0.9999804\n",
    "NOLBL-CMS2  0.024339113 -0.2055516 0.2542299 0.9977602\n",
    "CMS4-CMS3   0.024899166 -0.2734465 0.3232448 0.9991176\n",
    "NOLBL-CMS3  0.042487665 -0.2695297 0.3545050 0.9940785\n",
    "NOLBL-CMS4  0.017588499 -0.2564595 0.2916365 0.9996872\n",
    "\n",
    "[1] \"Batch: 116.77.0\"\n",
    "[1] \"\"\n",
    "            Df  Sum Sq  Mean Sq F value Pr(>F)  \n",
    "cms          4 0.09716 0.024289   3.522 0.0143 *\n",
    "Residuals   43 0.29659 0.006897                 \n",
    "---\n",
    "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
    "  Tukey multiple comparisons of means\n",
    "    95% family-wise confidence level\n",
    "\n",
    "Fit: aov(formula = VAF ~ cms, data = MACP.batch)\n",
    "\n",
    "$cms\n",
    "                  diff           lwr        upr     p adj\n",
    "CMS2-CMS1  -0.10621445 -0.2273518618 0.01492296 0.1104153\n",
    "CMS3-CMS1  -0.01438454 -0.1419086440 0.11313956 0.9976072\n",
    "CMS4-CMS1  -0.04530513 -0.1697259910 0.07911573 0.8369462\n",
    "NOLBL-CMS1  0.03145403 -0.1412144225 0.20412248 0.9850197\n",
    "CMS3-CMS2   0.09182991 -0.0007761007 0.18443592 0.0529221\n",
    "CMS4-CMS2   0.06090932 -0.0273744662 0.14919311 0.3005356\n",
    "NOLBL-CMS2  0.13766848 -0.0110859164 0.28642288 0.0816604\n",
    "CMS4-CMS3  -0.03092059 -0.1277820977 0.06594092 0.8919668\n",
    "NOLBL-CMS3  0.04583857 -0.1081614148 0.19983856 0.9140904\n",
    "NOLBL-CMS4  0.07675916 -0.0746810917 0.22819941 0.6040186\n",
    "\n",
    "[1] \"Batch: 41.78.0\"\n",
    "[1] \"\"\n",
    "            Df Sum Sq Mean Sq F value Pr(>F)\n",
    "cms          4 0.0775 0.01938   0.941  0.449\n",
    "Residuals   43 0.8854 0.02059               \n",
    "  Tukey multiple comparisons of means\n",
    "    95% family-wise confidence level\n",
    "\n",
    "Fit: aov(formula = VAF ~ cms, data = MACP.batch)\n",
    "\n",
    "$cms\n",
    "                    diff        lwr        upr     p adj\n",
    "CMS2-CMS1   0.0294009311 -0.1266027 0.18540457 0.9829979\n",
    "CMS3-CMS1  -0.0316366816 -0.2953310 0.23205759 0.9969585\n",
    "CMS4-CMS1  -0.0314182987 -0.2178783 0.15504171 0.9888202\n",
    "NOLBL-CMS1 -0.0820907842 -0.2622283 0.09804670 0.6942069\n",
    "CMS3-CMS2  -0.0610376128 -0.3180550 0.19597977 0.9605463\n",
    "CMS4-CMS2  -0.0608192298 -0.2377107 0.11607227 0.8632758\n",
    "NOLBL-CMS2 -0.1114917153 -0.2817056 0.05872221 0.3513788\n",
    "CMS4-CMS3   0.0002183829 -0.2763465 0.27678327 1.0000000\n",
    "NOLBL-CMS3 -0.0504541026 -0.3227964 0.22188817 0.9840482\n",
    "NOLBL-CMS4 -0.0506724855 -0.2491743 0.14782935 0.9491098\n",
    "\n",
    "[1] \"Batch: 123.74.0\"\n",
    "[1] \"\"\n",
    "            Df  Sum Sq  Mean Sq F value Pr(>F)\n",
    "cms          4 0.01309 0.003273   0.418  0.794\n",
    "Residuals   25 0.19563 0.007825               \n",
    "  Tukey multiple comparisons of means\n",
    "    95% family-wise confidence level\n",
    "\n",
    "Fit: aov(formula = VAF ~ cms, data = MACP.batch)\n",
    "\n",
    "$cms\n",
    "                    diff        lwr       upr     p adj\n",
    "CMS2-CMS1  -0.0009391152 -0.1318641 0.1299858 1.0000000\n",
    "CMS3-CMS1   0.0755654040 -0.1327349 0.2838657 0.8221582\n",
    "CMS4-CMS1   0.0189240978 -0.1155332 0.1533814 0.9934816\n",
    "NOLBL-CMS1  0.0360572754 -0.1267786 0.1988931 0.9649794\n",
    "CMS3-CMS2   0.0765045192 -0.1265875 0.2795965 0.8016395\n",
    "CMS4-CMS2   0.0198632129 -0.1063750 0.1461014 0.9900446\n",
    "NOLBL-CMS2  0.0369963905 -0.1191217 0.1931144 0.9554819\n",
    "CMS4-CMS3  -0.0566413062 -0.2620282 0.1487456 0.9251111\n",
    "NOLBL-CMS3 -0.0395081286 -0.2644982 0.1854819 0.9849767\n",
    "NOLBL-CMS4  0.0171331776 -0.1419588 0.1762252 0.9976797\n",
    "\n",
    "[1] \"Batch: 138.78.0\"\n",
    "[1] \"\"\n",
    "            Df Sum Sq Mean Sq F value Pr(>F)\n",
    "cms          4 0.0588 0.01469   1.238   0.31\n",
    "Residuals   41 0.4866 0.01187               \n",
    "  Tukey multiple comparisons of means\n",
    "    95% family-wise confidence level\n",
    "\n",
    "Fit: aov(formula = VAF ~ cms, data = MACP.batch)\n",
    "\n",
    "$cms\n",
    "                   diff         lwr       upr     p adj\n",
    "CMS2-CMS1   0.056247518 -0.10037390 0.2128689 0.8425680\n",
    "CMS3-CMS1   0.031966062 -0.13415604 0.1980882 0.9814615\n",
    "CMS4-CMS1   0.099121436 -0.03931365 0.2375565 0.2645385\n",
    "NOLBL-CMS1  0.051120559 -0.13085709 0.2330982 0.9286782\n",
    "CMS3-CMS2  -0.024281456 -0.18090288 0.1323400 0.9917657\n",
    "CMS4-CMS2   0.042873918 -0.08400393 0.1697518 0.8695604\n",
    "NOLBL-CMS2 -0.005126959 -0.17847510 0.1682212 0.9999880\n",
    "CMS4-CMS3   0.067155373 -0.07127971 0.2055905 0.6411561\n",
    "NOLBL-CMS3  0.019154497 -0.16282315 0.2011321 0.9981548\n",
    "NOLBL-CMS4 -0.048000877 -0.20511098 0.1091092 0.9056272\n",
    "\n",
    "[1] \"Batch: 89.77.0\"\n",
    "[1] \"\"\n",
    "            Df Sum Sq  Mean Sq F value Pr(>F)\n",
    "cms          4 0.0174 0.004341   0.376  0.824\n",
    "Residuals   36 0.4156 0.011546               \n",
    "  Tukey multiple comparisons of means\n",
    "    95% family-wise confidence level\n",
    "\n",
    "Fit: aov(formula = VAF ~ cms, data = MACP.batch)\n",
    "\n",
    "$cms\n",
    "                    diff         lwr       upr     p adj\n",
    "CMS2-CMS1  -0.0257630866 -0.15801913 0.1064930 0.9800560\n",
    "CMS3-CMS1   0.0873980523 -0.23978533 0.4145814 0.9384336\n",
    "CMS4-CMS1   0.0006891637 -0.14563169 0.1470100 1.0000000\n",
    "NOLBL-CMS1  0.0102465139 -0.16560927 0.1861023 0.9998155\n",
    "CMS3-CMS2   0.1131611389 -0.20425337 0.4305756 0.8428648\n",
    "CMS4-CMS2   0.0264522504 -0.09648186 0.1493864 0.9713103\n",
    "NOLBL-CMS2  0.0360096006 -0.12092433 0.1929435 0.9638501\n",
    "CMS4-CMS3  -0.0867088886 -0.41023648 0.2368187 0.9377292\n",
    "NOLBL-CMS3 -0.0771515384 -0.41506508 0.2607620 0.9644876\n",
    "NOLBL-CMS4  0.0095573502 -0.15939942 0.1785141 0.9998359\n",
    "\n",
    "[1] \"Batch: 157.71.0\"\n",
    "[1] \"\"\n",
    "            Df Sum Sq Mean Sq F value Pr(>F)\n",
    "cms          3 0.0327 0.01090   0.989  0.415\n",
    "Residuals   23 0.2536 0.01102               \n",
    "  Tukey multiple comparisons of means\n",
    "    95% family-wise confidence level\n",
    "\n",
    "Fit: aov(formula = VAF ~ cms, data = MACP.batch)\n",
    "\n",
    "$cms\n",
    "                 diff         lwr        upr     p adj\n",
    "CMS2-CMS1 -0.09796660 -0.25961741 0.06368422 0.3579995\n",
    "CMS3-CMS1 -0.02870456 -0.23415916 0.17675004 0.9798770\n",
    "CMS4-CMS1 -0.05486808 -0.20233118 0.09259502 0.7340745\n",
    "CMS3-CMS2  0.06926204 -0.13124112 0.26976520 0.7752485\n",
    "CMS4-CMS2  0.04309852 -0.09738383 0.18358087 0.8305333\n",
    "CMS4-CMS3 -0.02616352 -0.21541441 0.16308736 0.9804780\n",
    "\n",
    "[1] \"Batch: 300.56.0\"\n",
    "[1] \"\"\n",
    "            Df  Sum Sq  Mean Sq F value Pr(>F)\n",
    "cms          2 0.01249 0.006244   0.713  0.516\n",
    "Residuals    9 0.07880 0.008755               \n",
    "  Tukey multiple comparisons of means\n",
    "    95% family-wise confidence level\n",
    "\n",
    "Fit: aov(formula = VAF ~ cms, data = MACP.batch)\n",
    "\n",
    "$cms\n",
    "                 diff        lwr       upr     p adj\n",
    "CMS3-CMS2 0.009787615 -0.1897402 0.2093155 0.9897239\n",
    "CMS4-CMS2 0.069197368 -0.1060499 0.2444447 0.5362928\n",
    "CMS4-CMS3 0.059409753 -0.1313754 0.2501949 0.6717371\n",
    "\n",
    "[1] \"Batch: 36.83.0\"\n",
    "[1] \"\"\n",
    "            Df   Sum Sq   Mean Sq F value Pr(>F)\n",
    "cms          4 0.003489 0.0008722   1.172  0.347\n",
    "Residuals   25 0.018604 0.0007442               \n",
    "  Tukey multiple comparisons of means\n",
    "    95% family-wise confidence level\n",
    "\n",
    "Fit: aov(formula = VAF ~ cms, data = MACP.batch)\n",
    "\n",
    "$cms\n",
    "                   diff         lwr        upr     p adj\n",
    "CMS2-CMS1  -0.021741256 -0.07271185 0.02922934 0.7213337\n",
    "CMS3-CMS1  -0.008680727 -0.07409526 0.05673381 0.9947963\n",
    "CMS4-CMS1  -0.019300196 -0.07595084 0.03735045 0.8526033\n",
    "NOLBL-CMS1  0.007264335 -0.05392536 0.06845403 0.9966126\n",
    "CMS3-CMS2   0.013060529 -0.03791006 0.06403112 0.9416213\n",
    "CMS4-CMS2   0.002441060 -0.03665160 0.04153372 0.9997282\n",
    "NOLBL-CMS2  0.029005591 -0.01641598 0.07442716 0.3557424\n",
    "CMS4-CMS3  -0.010619469 -0.06727012 0.04603118 0.9808642\n",
    "NOLBL-CMS3  0.015945062 -0.04524463 0.07713475 0.9381634\n",
    "NOLBL-CMS4  0.026564531 -0.02515020 0.07827926 0.5666790\n",
    "\n",
    "[1] \"Batch: 45.80.0\"\n",
    "[1] \"\"\n",
    "            Df    Sum Sq   Mean Sq\n",
    "cms          3 0.0005848 0.0001949\n",
    "  Tukey multiple comparisons of means\n",
    "    95% family-wise confidence level\n",
    "\n",
    "Fit: aov(formula = VAF ~ cms, data = MACP.batch)\n",
    "\n",
    "$cms\n",
    "                    diff lwr upr p adj\n",
    "CMS2-CMS1  -0.0049986887 NaN NaN   NaN\n",
    "CMS4-CMS1   0.0238863287 NaN NaN   NaN\n",
    "NOLBL-CMS1 -0.0057144199 NaN NaN   NaN\n",
    "CMS4-CMS2   0.0288850174 NaN NaN   NaN\n",
    "NOLBL-CMS2 -0.0007157312 NaN NaN   NaN\n",
    "NOLBL-CMS4 -0.0296007486 NaN NaN   NaN\n",
    "\n",
    "[1] \"Batch: 33.79.0\"\n",
    "[1] \"\"\n",
    "            Df  Sum Sq  Mean Sq F value Pr(>F)  \n",
    "cms          4 0.05204 0.013009   3.358  0.046 *\n",
    "Residuals   12 0.04649 0.003874                 \n",
    "---\n",
    "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
    "  Tukey multiple comparisons of means\n",
    "    95% family-wise confidence level\n",
    "\n",
    "Fit: aov(formula = VAF ~ cms, data = MACP.batch)\n",
    "\n",
    "$cms\n",
    "                  diff         lwr       upr     p adj\n",
    "CMS2-CMS1   0.08723902 -0.12486141 0.2993394 0.6900263\n",
    "CMS3-CMS1   0.21667519 -0.02631637 0.4596667 0.0892179\n",
    "CMS4-CMS1   0.07679835 -0.14053991 0.2941366 0.7901872\n",
    "NOLBL-CMS1  0.17530668 -0.06768488 0.4182982 0.2103806\n",
    "CMS3-CMS2   0.12943617 -0.02963914 0.2885115 0.1335874\n",
    "CMS4-CMS2  -0.01044067 -0.12661285 0.1057315 0.9983080\n",
    "NOLBL-CMS2  0.08806766 -0.07100766 0.2471430 0.4347741\n",
    "CMS4-CMS3  -0.13987684 -0.30587168 0.0261180 0.1151845\n",
    "NOLBL-CMS3 -0.04136851 -0.23977029 0.1570333 0.9603948\n",
    "NOLBL-CMS4  0.09850833 -0.06748651 0.2645032 0.3714838\n",
    "\n",
    "[1] \"Batch: 30.80.0\"\n",
    "[1] \"\"\n",
    "            Df  Sum Sq  Mean Sq F value Pr(>F)\n",
    "cms          4 0.03411 0.008526   0.393  0.809\n",
    "Residuals   10 0.21685 0.021685               \n",
    "  Tukey multiple comparisons of means\n",
    "    95% family-wise confidence level\n",
    "\n",
    "Fit: aov(formula = VAF ~ cms, data = MACP.batch)\n",
    "\n",
    "$cms\n",
    "                   diff        lwr       upr     p adj\n",
    "CMS2-CMS1  -0.018563874 -0.3120289 0.2749012 0.9995035\n",
    "CMS3-CMS1   0.040846958 -0.3646333 0.4463272 0.9969405\n",
    "CMS4-CMS1   0.172359905 -0.3585383 0.7032581 0.8182348\n",
    "NOLBL-CMS1 -0.011682141 -0.5425804 0.5192161 0.9999925\n",
    "CMS3-CMS2   0.059410832 -0.3362973 0.4551190 0.9861281\n",
    "CMS4-CMS2   0.190923779 -0.3325489 0.7143965 0.7515401\n",
    "NOLBL-CMS2  0.006881733 -0.5165910 0.5303544 0.9999990\n",
    "CMS4-CMS3   0.131512947 -0.4620493 0.7250752 0.9446944\n",
    "NOLBL-CMS3 -0.052529099 -0.6460913 0.5410332 0.9981482\n",
    "NOLBL-CMS4 -0.184042046 -0.8694287 0.5013446 0.8965756\n",
    "\n",
    "[1] \"Batch: 66.81.0\"\n",
    "[1] \"\"\n",
    "            Df  Sum Sq Mean Sq F value Pr(>F)\n",
    "cms          3 0.08189 0.02730   2.277  0.157\n",
    "Residuals    8 0.09590 0.01199               \n",
    "  Tukey multiple comparisons of means\n",
    "    95% family-wise confidence level\n",
    "\n",
    "Fit: aov(formula = VAF ~ cms, data = MACP.batch)\n",
    "\n",
    "$cms\n",
    "                   diff         lwr       upr     p adj\n",
    "CMS2-CMS1   0.036650601 -0.21941045 0.2927117 0.9660708\n",
    "CMS3-CMS1   0.004713347 -0.28157161 0.2909983 0.9999427\n",
    "NOLBL-CMS1  0.310862899 -0.09400517 0.7157310 0.1423333\n",
    "CMS3-CMS2  -0.031937254 -0.28799830 0.2241238 0.9769807\n",
    "NOLBL-CMS2  0.274212298 -0.10987928 0.6583039 0.1804487\n",
    "NOLBL-CMS3  0.306149552 -0.09871852 0.7110176 0.1498735\n",
    "\n",
    "[1] \"Batch: 132.71.0\"\n",
    "[1] \"\"\n",
    "            Df  Sum Sq  Mean Sq F value Pr(>F)\n",
    "cms          3 0.02485 0.008283   1.097  0.405\n",
    "Residuals    8 0.06042 0.007553               \n",
    "  Tukey multiple comparisons of means\n",
    "    95% family-wise confidence level\n",
    "\n",
    "Fit: aov(formula = VAF ~ cms, data = MACP.batch)\n",
    "\n",
    "$cms\n",
    "                  diff        lwr       upr     p adj\n",
    "CMS2-CMS1   0.01194094 -0.2886698 0.3125517 0.9992005\n",
    "CMS3-CMS1   0.05233410 -0.2588276 0.3634958 0.9470580\n",
    "NOLBL-CMS1 -0.12296258 -0.5165545 0.2706293 0.7536010\n",
    "CMS3-CMS2   0.04039316 -0.1392561 0.2200425 0.8864457\n",
    "NOLBL-CMS2 -0.13490352 -0.4355143 0.1657073 0.5130104\n",
    "NOLBL-CMS3 -0.17529668 -0.4864584 0.1358650 0.3379948\n",
    "\n",
    "[1] \"Batch: 172.69.0\"\n",
    "[1] \"\"\n",
    "[1] \"Batch: 154.69.0\"\n",
    "[1] \"\"\n",
    "            Df  Sum Sq  Mean Sq F value Pr(>F)\n",
    "cms          2 0.00623 0.003113   0.292  0.761\n",
    "Residuals    4 0.04263 0.010658               \n",
    "  Tukey multiple comparisons of means\n",
    "    95% family-wise confidence level\n",
    "\n",
    "Fit: aov(formula = VAF ~ cms, data = MACP.batch)\n",
    "\n",
    "$cms\n",
    "                  diff        lwr       upr     p adj\n",
    "CMS3-CMS2  0.002207149 -0.3164429 0.3208572 0.9996641\n",
    "NOLBL-CMS2 0.086654112 -0.3639851 0.5372934 0.7841145\n",
    "NOLBL-CMS3 0.084446963 -0.3269285 0.4958224 0.7596589\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Batch + CMS Normalized Comparison\n",
    "\n",
    "Within a batch and CMS, create a comparison set to look at Hypo-modified CRC vs. Normo-modified CRC samples\n",
    "\n",
    "```\n",
    "# Batch Selection:\n",
    "batch <- '116.77.0'\n",
    "cms   <- 'CMS2'\n",
    "\n",
    "MACP.select <- MACP[ which(MACP$cms == cms & MACP$batch == batch), ]\n",
    "MACP.select <- MACP.select[ order(MACP.select$VAF), ]\n",
    "MACP.select[, c(2,4,10,14,15)]\n",
    "\n",
    "====================================================================\n",
    "\n",
    "             lib.name           pt    batch       VAF hypo.macp\n",
    "2487 TCGA-D5-6532-01A TCGA-D5-6532 116.77.0 0.1143430     .hypo *\n",
    "2256 TCGA-AA-3712-01A TCGA-AA-3712 116.77.0 0.2150752     .hypo *\n",
    "2556 TCGA-G4-6298-01A TCGA-G4-6298 116.77.0 0.2157952     .hypo *\n",
    "2380 TCGA-AU-3779-01A TCGA-AU-3779 116.77.0 0.2779931     .hypo *\n",
    "2568 TCGA-G4-6317-01A TCGA-G4-6317 116.77.0 0.3044773    .normo *\n",
    "2554 TCGA-G4-6295-01A TCGA-G4-6295 116.77.0 0.3149887    .normo\n",
    "2493 TCGA-D5-6538-01A TCGA-D5-6538 116.77.0 0.3346310    .normo\n",
    "2562 TCGA-G4-6307-01A TCGA-G4-6307 116.77.0 0.3408818    .normo\n",
    "2253 TCGA-AA-3697-01A TCGA-AA-3697 116.77.0 0.3434516    .normo\n",
    "2231 TCGA-AA-3660-01A TCGA-AA-3660 116.77.0 0.3670422    .normo\n",
    "2492 TCGA-D5-6537-01A TCGA-D5-6537 116.77.0 0.3807558    .normo\n",
    "2143 TCGA-A6-5667-01A TCGA-A6-5667 116.77.0 0.3960426    .normo *\n",
    "2233 TCGA-AA-3662-01A TCGA-AA-3662 116.77.0 0.4062256    .normo * \n",
    "2552 TCGA-G4-6293-01A TCGA-G4-6293 116.77.0 0.4151551    .normo *\n",
    "2567 TCGA-G4-6315-01A TCGA-G4-6315 116.77.0 0.4326472    .normo *\n",
    "2488 TCGA-D5-6533-01A TCGA-D5-6533 116.77.0 0.4345105    .normo *\n",
    "```\n",
    "\n",
    "\n",
    "\n",
    "```\n",
    "# Batch Selection:\n",
    "\n",
    "batch <- '116.77.0'\n",
    "cms   <- 'CMS4'\n",
    "\n",
    "MACP.select <- MACP[ which(MACP$cms == cms & MACP$batch == batch), ]\n",
    "MACP.select <- MACP.select[ order(MACP.select$VAF), ]\n",
    "\n",
    "MACP.select[, c(2,4,10,14,15)]\n",
    "====================================================================\n",
    "\n",
    "             lib.name           pt    batch       VAF hypo.macp\n",
    "2566 TCGA-G4-6314-01A TCGA-G4-6314 116.77.0 0.2554967     .hypo *\n",
    "2451 TCGA-CM-5344-01A TCGA-CM-5344 116.77.0 0.2678113     .hypo *\n",
    "2538 TCGA-F4-6463-01A TCGA-F4-6463 116.77.0 0.3092360    .normo *\n",
    "2453 TCGA-CM-5349-01A TCGA-CM-5349 116.77.0 0.3221711    .normo *\n",
    "2565 TCGA-G4-6311-01A TCGA-G4-6311 116.77.0 0.3324151    .normo *\n",
    "2085 TCGA-A6-2675-01A TCGA-A6-2675 116.77.0 0.3413247    .normo\n",
    "2555 TCGA-G4-6297-01A TCGA-G4-6297 116.77.0 0.3836968    .normo\n",
    "2452 TCGA-CM-5348-01A TCGA-CM-5348 116.77.0 0.4278800    .normo\n",
    "2491 TCGA-D5-6536-01A TCGA-D5-6536 116.77.0 0.4371242    .normo *\n",
    "2558 TCGA-G4-6302-01A TCGA-G4-6302 116.77.0 0.4777553    .normo *\n",
    "2385 TCGA-AY-6196-01A TCGA-AY-6196 116.77.0 0.5022196    .normo *\n",
    "2496 TCGA-D5-6541-01A TCGA-D5-6541 116.77.0 0.5104515    .normo *\n",
    "2564 TCGA-G4-6310-01A TCGA-G4-6310 116.77.0 0.5256267    .normo *\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
