{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Regenerate GVCF Files\n",
    "```\n",
    "pi:ababaian\n",
    "files: ~/Crown/.. (var0\n",
    "start: 2019 06 19\n",
    "complete : YYYY MM DD\n",
    "```\n",
    "## Introduction\n",
    "\n",
    "I have been using the `ADcalc*.sh` script to create a common set of GVCF files for downstream analysis. The TCGA and CCLE data was proccessed uniformly.\n",
    "\n",
    "I will re-process the CRC and hCAGE data for cell lines and primary cells to make GVCF files for analysis. This will make cross cohort analysis simplified.\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2011_hCAGE_protocol.pdf       ADcalc_cage.sh  hCAGE_gvcf.log  screenlog.0\r\n",
      "2014_MORAI_workflow_CAGE.pdf  cell_lines      primary         TssHmm_Info.pdf\r\n"
     ]
    }
   ],
   "source": [
    "cd ~/Crown/data2/hCAGE/\n",
    "ls"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "## Objective\n",
    "\n",
    "1. Generate `GVCF` files for hCAGE, CRC1 and CRC4 data-sets\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Materials and Methods -- hCAGE\n",
    "\n",
    "Launch an AMI instance and run ADcalc script on data.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Initialization\n",
    "\n",
    "Initialize data on AMI instance, run ADcalc and output data to respective folder"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# DNS:ec2-34-221-202-76.us-west-2.compute.amazonaws.com\n",
    "# AMI: ami-0b375c9c58cb4a7a2 (TCGA aligner)\n",
    "# Instance: m4.4xlarge\n",
    "# Storage: 400 Gb\n",
    "\n",
    "# ON REMOTE:\n",
    "\n",
    "## Copy hCAGE files into it's dir\n",
    "#mkdir -p ~/hCAGE; cd hCAGE;\n",
    "#aws s3 cp --recursive s3://crownproject/hCAGE/ ./\n",
    "\n",
    "## PATHS:\n",
    "## ~/hCAGE/cell_lines/BAM/*\n",
    "## ~/hCAGE/primary/BAM/*\n",
    "\n",
    "# cd ~/hCAGE/cell_lines/BAM/\n",
    "\n",
    "## Run ADcalc script.sh\n",
    "# screen -L\n",
    "# bash ADcalc_tcga.sh\n",
    "\n",
    "# aws s3 cp screenlog.0 s3://crownproject/ccle/logs/ccle.gvcf.log\n",
    "\n",
    "aws s3 cp s3://crownproject/ccle/logs/ccle.gvcf.log ./\n",
    "cat ccle.gvcf.log\n",
    "\n",
    "## DONE"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Scripts\n",
    "`ADcalc_hCAGE.sh` @ s3://crownproject/hCAGE/ADcalc_hCAGE.sh\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Completed 5.7 KiB/8.7 MiB with 23 file(s) remaining\r",
      "upload: ./00_TCGA_Run_template.ipynb to s3://crownproject/hCAGE/gvcf/00_TCGA_Run_template.ipynb\r\n",
      "Completed 5.7 KiB/8.7 MiB with 22 file(s) remaining\r",
      "Completed 11.4 KiB/8.7 MiB with 22 file(s) remaining\r",
      "upload: .ipynb_checkpoints/00_TCGA_Run_template-checkpoint.ipynb to s3://crownproject/hCAGE/gvcf/.ipynb_checkpoints/00_TCGA_Run_template-checkpoint.ipynb\r\n",
      "Completed 11.4 KiB/8.7 MiB with 21 file(s) remaining\r",
      "Completed 17.1 KiB/8.7 MiB with 21 file(s) remaining\r",
      "upload: .ipynb_checkpoints/20190618_RE_GVCF-checkpoint.ipynb to s3://crownproject/hCAGE/gvcf/.ipynb_checkpoints/20190618_RE_GVCF-checkpoint.ipynb\r\n",
      "Completed 17.1 KiB/8.7 MiB with 20 file(s) remaining\r",
      "Completed 57.5 KiB/8.7 MiB with 20 file(s) remaining\r",
      "upload: .ipynb_checkpoints/20190506_TCGA_Run2_Pilot-checkpoint.ipynb to s3://crownproject/hCAGE/gvcf/.ipynb_checkpoints/20190506_TCGA_Run2_Pilot-checkpoint.ipynb\r\n",
      "Completed 57.5 KiB/8.7 MiB with 19 file(s) remaining\r",
      "Completed 61.5 KiB/8.7 MiB with 19 file(s) remaining\r",
      "Completed 62.4 KiB/8.7 MiB with 19 file(s) remaining\r",
      "Completed 65.5 KiB/8.7 MiB with 19 file(s) remaining\r",
      "upload: .ipynb_checkpoints/20190507_Crown_EC2_Update-checkpoint.ipynb to s3://crownproject/hCAGE/gvcf/.ipynb_checkpoints/20190507_Crown_EC2_Update-checkpoint.ipynb\r\n",
      "Completed 65.5 KiB/8.7 MiB with 18 file(s) remaining\r",
      "upload: ./00_template.ipynb to s3://crownproject/hCAGE/gvcf/00_template.ipynb\r\n",
      "Completed 65.5 KiB/8.7 MiB with 17 file(s) remaining\r",
      "upload: .ipynb_checkpoints/Untitled-checkpoint.ipynb to s3://crownproject/hCAGE/gvcf/.ipynb_checkpoints/Untitled-checkpoint.ipynb\r\n",
      "Completed 65.5 KiB/8.7 MiB with 16 file(s) remaining\r",
      "Completed 321.5 KiB/8.7 MiB with 16 file(s) remaining\r",
      "Completed 577.5 KiB/8.7 MiB with 16 file(s) remaining\r",
      "Completed 833.5 KiB/8.7 MiB with 16 file(s) remaining\r",
      "Completed 1.1 MiB/8.7 MiB with 16 file(s) remaining  \r",
      "Completed 1.2 MiB/8.7 MiB with 16 file(s) remaining  \r",
      "upload: .ipynb_checkpoints/20190509_TCGA_gvcf-checkpoint.ipynb to s3://crownproject/hCAGE/gvcf/.ipynb_checkpoints/20190509_TCGA_gvcf-checkpoint.ipynb\r\n",
      "Completed 1.2 MiB/8.7 MiB with 15 file(s) remaining\r",
      "Completed 1.5 MiB/8.7 MiB with 15 file(s) remaining\r",
      "Completed 1.5 MiB/8.7 MiB with 15 file(s) remaining\r",
      "upload: .ipynb_checkpoints/00_template-checkpoint.ipynb to s3://crownproject/hCAGE/gvcf/.ipynb_checkpoints/00_template-checkpoint.ipynb\r\n",
      "Completed 1.5 MiB/8.7 MiB with 14 file(s) remaining\r",
      "Completed 1.5 MiB/8.7 MiB with 14 file(s) remaining\r",
      "upload: ./20190507_Crown_EC2_Update.ipynb to s3://crownproject/hCAGE/gvcf/20190507_Crown_EC2_Update.ipynb\r\n",
      "Completed 1.5 MiB/8.7 MiB with 13 file(s) remaining\r",
      "Completed 1.5 MiB/8.7 MiB with 13 file(s) remaining\r",
      "upload: ./20190506_TCGA_Run2_Pilot.ipynb to s3://crownproject/hCAGE/gvcf/20190506_TCGA_Run2_Pilot.ipynb\r\n",
      "Completed 1.5 MiB/8.7 MiB with 12 file(s) remaining\r",
      "Completed 1.7 MiB/8.7 MiB with 12 file(s) remaining\r",
      "Completed 1.8 MiB/8.7 MiB with 12 file(s) remaining\r",
      "cancelled: ctrl-c received                         \r\n"
     ]
    }
   ],
   "source": [
    "#!/bin/bash\n",
    "# ADcalc_hCAGE.sh\n",
    "# Allelic Depth Calculator\n",
    "# for a position\n",
    "#\n",
    "# s3://crownproject/hCAGE/ADcalc_hcage.sh\n",
    "\n",
    "# Controls -----------------\n",
    "DEPTH='100000' #Max per file DP\n",
    "\n",
    "# Regions in hgr1.fa reference genome\n",
    "REGIONS=('chr13:1003660-1005529' 'chr13:1005529-1005629' \\\n",
    "        'chr13:10219-10340' 'chr13:1006622-1006779' 'chr13:1007948-1013018')\n",
    "\n",
    "# Corresponding region/gene names\n",
    "GENES=('18S' '18SE' '5S' '5.8S' '28S')\n",
    "\n",
    "# 18S  1870\n",
    "# 18SE 101\n",
    "# 5S   122\n",
    "# 5.8S 158\n",
    "# 28S  5071\n",
    "\n",
    "# Terminate instances upon completion (for debugging)\n",
    "TERMINATE='FALSE'\n",
    "\n",
    "# S3 Output directory\n",
    "S3DIR='s3://crownproject/hCAGE/gvcf/'\n",
    "BAMLIST='bam.list.tmp'\n",
    "\n",
    "# Script ------------------ ------------------------------\n",
    "cd ~/hCAGE/cell_lines/\n",
    "mkdir -p GVCF #Output Folder\n",
    "TYPE='hCAGE_cell_lines' # hardcode single ccle run\n",
    "cd BAM\n",
    "\n",
    "#for TYPE in $(echo \"hgr1\")\n",
    "#do\n",
    "    echo Analyzing $TYPE...\n",
    "    #cd $TYPE\n",
    "\n",
    "    ls *.bam > bam.list.tmp\n",
    "    ls *.bam > ../GVCF/$TYPE.bamlist\n",
    "          \n",
    "    for index in ${!GENES[*]}\n",
    "    do\n",
    "      printf \"Started processing %s\\n\" ${GENES[$index]}\n",
    "      OUTPUT=\"../GVCF/$TYPE.${GENES[$index]}.gvcf\"\n",
    "\n",
    "      # Iterate through every bam file in directory\n",
    "      # look-up position and return VCF\n",
    "      bcftools mpileup -f ~/resources/hgr1/hgr1.fa \\\n",
    "        --max-depth $DEPTH -A --min-BQ 30 \\\n",
    "        -a FORMAT/DP,AD \\\n",
    "        -r ${REGIONS[$index]} \\\n",
    "        --ignore-RG \\\n",
    "        -b $BAMLIST | \\\n",
    "        bcftools annotate -x INFO,FORMAT/PL - | \\\n",
    "        bcftools view -O v - \\\n",
    "        >> $OUTPUT\n",
    "\n",
    "      RESULTS+=(\"$OUTPUT\")\n",
    "      printf \"Done with %s \\n\" ${GENES[$index]}\n",
    "      printf \"%s\\n\" ${REGIONS[$index]}\n",
    "\n",
    "    done\n",
    "\n",
    "    rm bam.list.tmp\n",
    "\n",
    "#    cd .. # move to tcga folder to reset\n",
    "#done\n",
    "\n",
    "# Copy GVCF output to AWS S3\n",
    "cd ../GVCF\n",
    "aws s3 cp --recursive ./ $S3DIR\n",
    "\n",
    "# -------------------------------------------\n",
    "cd ~/hCAGE/primary/\n",
    "mkdir -p GVCF #Output Folder\n",
    "TYPE='hCAGE_cell_lines' # hardcode single ccle run\n",
    "cd BAM\n",
    "\n",
    "#for TYPE in $(echo \"hgr1\")\n",
    "#do\n",
    "    echo Analyzing $TYPE...\n",
    "    #cd $TYPE\n",
    "\n",
    "    ls *.bam > bam.list.tmp\n",
    "    ls *.bam > ../GVCF/$TYPE.bamlist\n",
    "          \n",
    "    for index in ${!GENES[*]}\n",
    "    do\n",
    "      printf \"Started processing %s\\n\" ${GENES[$index]}\n",
    "      OUTPUT=\"../GVCF/$TYPE.${GENES[$index]}.gvcf\"\n",
    "\n",
    "      # Iterate through every bam file in directory\n",
    "      # look-up position and return VCF\n",
    "      bcftools mpileup -f ~/resources/hgr1/hgr1.fa \\\n",
    "        --max-depth $DEPTH -A --min-BQ 30 \\\n",
    "        -a FORMAT/DP,AD \\\n",
    "        -r ${REGIONS[$index]} \\\n",
    "        --ignore-RG \\\n",
    "        -b $BAMLIST | \\\n",
    "        bcftools annotate -x INFO,FORMAT/PL - | \\\n",
    "        bcftools view -O v - \\\n",
    "        >> $OUTPUT\n",
    "\n",
    "      RESULTS+=(\"$OUTPUT\")\n",
    "      printf \"Done with %s \\n\" ${GENES[$index]}\n",
    "      printf \"%s\\n\" ${REGIONS[$index]}\n",
    "\n",
    "    done\n",
    "\n",
    "    rm bam.list.tmp\n",
    "\n",
    "#    cd .. # move to tcga folder to reset\n",
    "#done\n",
    "\n",
    "# Copy GVCF output to AWS S3\n",
    "cd ../GVCF\n",
    "aws s3 cp --recursive ./ $S3DIR\n",
    "\n",
    "\n",
    "# Shutdown and Terminate instance\n",
    "EC2ID=$(ec2metadata --instance-id)\n",
    "sleep 20s # to catch errors\n",
    "\n",
    "if [ \"$TERMINATE\" = TRUE ]\n",
    "then\n",
    "  echo \"Run Complete -- Shutting down instance.\"\n",
    "  aws ec2 terminate-instances --instance-ids $EC2ID\n",
    "else\n",
    "  echo \"Run Complete -- Instance is online.\"\n",
    "fi\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Completed 17.3 KiB/17.3 KiB with 1 file(s) remaining\r",
      "download: s3://crownproject/hCAGE/hCAGE_gvcf.log to ./hCAGE_gvcf.log\r\n",
      "\u001b[01;32mubuntu@ip-172-31-17-124\u001b[00m:\u001b[01;34m~/hCAGE\u001b[00m$ ls\r",
      "\r\n",
      "\u001b[0m\u001b[01;32mADcalc_hCAGE.sh\u001b[0m  \u001b[01;34mcell_lines\u001b[0m  \u001b[01;34mgvcf\u001b[0m  \u001b[01;34mprimary\u001b[0m  screenlog.0\r",
      "\r\n",
      "\u001b[01;32mubuntu@ip-172-31-17-124\u001b[00m:\u001b[01;34m~/hCAGE\u001b[00m$ bash ADcalc_hCAGE.sh \r",
      "\r\n",
      "Analyzing hCAGE_cell_lines...\r",
      "\r\n",
      "Started processing 18S\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: acute_lymphoblastic_leukemia__T-ALL__cell_line_aHPB-ALL.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: acute_myeloid_leukemia__FAB_M4__cell_line_aFKH-1.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: acute_myeloid_leukemia__FAB_M7__cell_line_aMKPL-1.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: B_lymphoblastoid_cell_line_a_GM12878_ENCODE_c_biol_rep3.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: colon_carcinoma_cell_line_aCOLO-320.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: glioma_cell_line_aGI-1.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Hep-2_cells_treated_with_Streptococci_strain__c_biol_rep1.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: hepatoblastoma_cell_line_aHuH-6.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: osteosarcoma_cell_line_a143B_fTK_e_-_neo_e_R_.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: papillary_adenocarcinoma_cell_line_a8505C.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: papillotubular_adenocarcinoma_cell_line_aTGBC18TKB.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Wilms__tumor_cell_line_aHFWT.bam.bai\r",
      "\r\n",
      "[mpileup] 266 samples in 266 input files\r",
      "\r\n",
      "Warning: Potential memory hog, up to 26600000M reads in the pileup!\r",
      "\r\n",
      "Done with 18S \r",
      "\r\n",
      "chr13:1003660-1005529\r",
      "\r\n",
      "Started processing 18SE\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: acute_lymphoblastic_leukemia__T-ALL__cell_line_aHPB-ALL.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: acute_myeloid_leukemia__FAB_M4__cell_line_aFKH-1.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: acute_myeloid_leukemia__FAB_M7__cell_line_aMKPL-1.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: B_lymphoblastoid_cell_line_a_GM12878_ENCODE_c_biol_rep3.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: colon_carcinoma_cell_line_aCOLO-320.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: glioma_cell_line_aGI-1.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Hep-2_cells_treated_with_Streptococci_strain__c_biol_rep1.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: hepatoblastoma_cell_line_aHuH-6.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: osteosarcoma_cell_line_a143B_fTK_e_-_neo_e_R_.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: papillary_adenocarcinoma_cell_line_a8505C.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: papillotubular_adenocarcinoma_cell_line_aTGBC18TKB.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Wilms__tumor_cell_line_aHFWT.bam.bai\r",
      "\r\n",
      "[mpileup] 266 samples in 266 input files\r",
      "\r\n",
      "Warning: Potential memory hog, up to 26600000M reads in the pileup!\r",
      "\r\n",
      "Done with 18SE \r",
      "\r\n",
      "chr13:1005529-1005629\r",
      "\r\n",
      "Started processing 5S\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: acute_lymphoblastic_leukemia__T-ALL__cell_line_aHPB-ALL.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: acute_myeloid_leukemia__FAB_M4__cell_line_aFKH-1.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: acute_myeloid_leukemia__FAB_M7__cell_line_aMKPL-1.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: B_lymphoblastoid_cell_line_a_GM12878_ENCODE_c_biol_rep3.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: colon_carcinoma_cell_line_aCOLO-320.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: glioma_cell_line_aGI-1.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Hep-2_cells_treated_with_Streptococci_strain__c_biol_rep1.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: hepatoblastoma_cell_line_aHuH-6.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: osteosarcoma_cell_line_a143B_fTK_e_-_neo_e_R_.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: papillary_adenocarcinoma_cell_line_a8505C.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: papillotubular_adenocarcinoma_cell_line_aTGBC18TKB.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Wilms__tumor_cell_line_aHFWT.bam.bai\r",
      "\r\n",
      "[mpileup] 266 samples in 266 input files\r",
      "\r\n",
      "Warning: Potential memory hog, up to 26600000M reads in the pileup!\r",
      "\r\n",
      "Done with 5S \r",
      "\r\n",
      "chr13:10219-10340\r",
      "\r\n",
      "Started processing 5.8S\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: acute_lymphoblastic_leukemia__T-ALL__cell_line_aHPB-ALL.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: acute_myeloid_leukemia__FAB_M4__cell_line_aFKH-1.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: acute_myeloid_leukemia__FAB_M7__cell_line_aMKPL-1.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: B_lymphoblastoid_cell_line_a_GM12878_ENCODE_c_biol_rep3.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: colon_carcinoma_cell_line_aCOLO-320.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: glioma_cell_line_aGI-1.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Hep-2_cells_treated_with_Streptococci_strain__c_biol_rep1.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: hepatoblastoma_cell_line_aHuH-6.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: osteosarcoma_cell_line_a143B_fTK_e_-_neo_e_R_.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: papillary_adenocarcinoma_cell_line_a8505C.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: papillotubular_adenocarcinoma_cell_line_aTGBC18TKB.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Wilms__tumor_cell_line_aHFWT.bam.bai\r",
      "\r\n",
      "[mpileup] 266 samples in 266 input files\r",
      "\r\n",
      "Warning: Potential memory hog, up to 26600000M reads in the pileup!\r",
      "\r\n",
      "Done with 5.8S \r",
      "\r\n",
      "chr13:1006622-1006779\r",
      "\r\n",
      "Started processing 28S\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: acute_lymphoblastic_leukemia__T-ALL__cell_line_aHPB-ALL.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: acute_myeloid_leukemia__FAB_M4__cell_line_aFKH-1.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: acute_myeloid_leukemia__FAB_M7__cell_line_aMKPL-1.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: B_lymphoblastoid_cell_line_a_GM12878_ENCODE_c_biol_rep3.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: colon_carcinoma_cell_line_aCOLO-320.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: glioma_cell_line_aGI-1.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Hep-2_cells_treated_with_Streptococci_strain__c_biol_rep1.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: hepatoblastoma_cell_line_aHuH-6.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: osteosarcoma_cell_line_a143B_fTK_e_-_neo_e_R_.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: papillary_adenocarcinoma_cell_line_a8505C.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: papillotubular_adenocarcinoma_cell_line_aTGBC18TKB.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Wilms__tumor_cell_line_aHFWT.bam.bai\r",
      "\r\n",
      "[mpileup] 266 samples in 266 input files\r",
      "\r\n",
      "Warning: Potential memory hog, up to 26600000M reads in the pileup!\r",
      "\r\n",
      "Done with 28S \r",
      "\r\n",
      "chr13:1007948-1013018\r",
      "\r\n",
      "upload: ./hCAGE_cell_lines.5S.gvcf to s3://crownproject/hCAGE/gvcf/hCAGE_cell_lines.5S.gvcf\r",
      "\r\n",
      "Completed 1 of 8 part(s) with 5 file(s) remaining\r",
      "upload: ./hCAGE_cell_lines.bamlist to s3://crownproject/hCAGE/gvcf/hCAGE_cell_lines.bamlist\r",
      "\r\n",
      "Completed 2 of 8 part(s) with 4 file(s) remaining\r",
      "upload: ./hCAGE_cell_lines.18SE.gvcf to s3://crownproject/hCAGE/gvcf/hCAGE_cell_lines.18SE.gvcf\r",
      "\r\n",
      "Completed 3 of 8 part(s) with 3 file(s) remaining\r",
      "upload: ./hCAGE_cell_lines.5.8S.gvcf to s3://crownproject/hCAGE/gvcf/hCAGE_cell_lines.5.8S.gvcf\r",
      "\r\n",
      "Completed 4 of 8 part(s) with 2 file(s) remaining\r",
      "upload: ./hCAGE_cell_lines.18S.gvcf to s3://crownproject/hCAGE/gvcf/hCAGE_cell_lines.18S.gvcf\r",
      "\r\n",
      "Completed 5 of 8 part(s) with 1 file(s) remaining\r",
      "Completed 6 of 8 part(s) with 1 file(s) remaining\r",
      "Completed 7 of 8 part(s) with 1 file(s) remaining\r",
      "Completed 8 of 8 part(s) with 1 file(s) remaining\r",
      "upload: ./hCAGE_cell_lines.28S.gvcf to s3://crownproject/hCAGE/gvcf/hCAGE_cell_lines.28S.gvcf\r",
      "\r\n",
      "Analyzing hCAGE_cell_lines...\r",
      "\r\n",
      "Started processing 18S\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Adipocyte_-_breast_c_donor2.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Alveolar_Epithelial_Cells_c_donor1.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Amniotic_Epithelial_Cells_c_donor2.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Astrocyte_-_cerebellum_c_donor1.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Fibroblast_-_Gingival_c_donor4__GFH2_.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Fibroblast_-_Mammary_c_donor1.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Hepatic_Sinusoidal_Endothelial_Cells_c_donor2.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Mast_cell_c_expanded_and_stimulated_c_donor8.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Neurons_c_donor3.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Retinal_Pigment_Epithelial_Cells_c_donor3.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Sebocyte_c_donor1.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Smooth_Muscle_Cells_-_Pulmonary_Artery_c_donor1.bam.bai\r",
      "\r\n",
      "[mpileup] 512 samples in 512 input files\r",
      "\r\n",
      "Warning: Potential memory hog, up to 51200000M reads in the pileup!\r",
      "\r\n",
      "Done with 18S \r",
      "\r\n",
      "chr13:1003660-1005529\r",
      "\r\n",
      "Started processing 18SE\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Adipocyte_-_breast_c_donor2.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Alveolar_Epithelial_Cells_c_donor1.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Amniotic_Epithelial_Cells_c_donor2.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Astrocyte_-_cerebellum_c_donor1.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Fibroblast_-_Gingival_c_donor4__GFH2_.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Fibroblast_-_Mammary_c_donor1.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Hepatic_Sinusoidal_Endothelial_Cells_c_donor2.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Mast_cell_c_expanded_and_stimulated_c_donor8.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Neurons_c_donor3.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Retinal_Pigment_Epithelial_Cells_c_donor3.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Sebocyte_c_donor1.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Smooth_Muscle_Cells_-_Pulmonary_Artery_c_donor1.bam.bai\r",
      "\r\n",
      "[mpileup] 512 samples in 512 input files\r",
      "\r\n",
      "Warning: Potential memory hog, up to 51200000M reads in the pileup!\r",
      "\r\n",
      "Done with 18SE \r",
      "\r\n",
      "chr13:1005529-1005629\r",
      "\r\n",
      "Started processing 5S\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Adipocyte_-_breast_c_donor2.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Alveolar_Epithelial_Cells_c_donor1.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Amniotic_Epithelial_Cells_c_donor2.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Astrocyte_-_cerebellum_c_donor1.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Fibroblast_-_Gingival_c_donor4__GFH2_.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Fibroblast_-_Mammary_c_donor1.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Hepatic_Sinusoidal_Endothelial_Cells_c_donor2.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Mast_cell_c_expanded_and_stimulated_c_donor8.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Neurons_c_donor3.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Retinal_Pigment_Epithelial_Cells_c_donor3.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Sebocyte_c_donor1.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Smooth_Muscle_Cells_-_Pulmonary_Artery_c_donor1.bam.bai\r",
      "\r\n",
      "[mpileup] 512 samples in 512 input files\r",
      "\r\n",
      "Warning: Potential memory hog, up to 51200000M reads in the pileup!\r",
      "\r\n",
      "Done with 5S \r",
      "\r\n",
      "chr13:10219-10340\r",
      "\r\n",
      "Started processing 5.8S\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Adipocyte_-_breast_c_donor2.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Alveolar_Epithelial_Cells_c_donor1.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Amniotic_Epithelial_Cells_c_donor2.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Astrocyte_-_cerebellum_c_donor1.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Fibroblast_-_Gingival_c_donor4__GFH2_.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Fibroblast_-_Mammary_c_donor1.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Hepatic_Sinusoidal_Endothelial_Cells_c_donor2.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Mast_cell_c_expanded_and_stimulated_c_donor8.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Neurons_c_donor3.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Retinal_Pigment_Epithelial_Cells_c_donor3.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Sebocyte_c_donor1.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Smooth_Muscle_Cells_-_Pulmonary_Artery_c_donor1.bam.bai\r",
      "\r\n",
      "[mpileup] 512 samples in 512 input files\r",
      "\r\n",
      "Warning: Potential memory hog, up to 51200000M reads in the pileup!\r",
      "\r\n",
      "Done with 5.8S \r",
      "\r\n",
      "chr13:1006622-1006779\r",
      "\r\n",
      "Started processing 28S\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Adipocyte_-_breast_c_donor2.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Alveolar_Epithelial_Cells_c_donor1.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Amniotic_Epithelial_Cells_c_donor2.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Astrocyte_-_cerebellum_c_donor1.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Fibroblast_-_Gingival_c_donor4__GFH2_.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Fibroblast_-_Mammary_c_donor1.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Hepatic_Sinusoidal_Endothelial_Cells_c_donor2.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Mast_cell_c_expanded_and_stimulated_c_donor8.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Neurons_c_donor3.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Retinal_Pigment_Epithelial_Cells_c_donor3.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Sebocyte_c_donor1.bam.bai\r",
      "\r\n",
      "[W::hts_idx_load2] The index file is older than the data file: Smooth_Muscle_Cells_-_Pulmonary_Artery_c_donor1.bam.bai\r",
      "\r\n",
      "[mpileup] 512 samples in 512 input files\r",
      "\r\n",
      "Warning: Potential memory hog, up to 51200000M reads in the pileup!\r",
      "\r\n",
      "Done with 28S \r",
      "\r\n",
      "chr13:1007948-1013018\r",
      "\r\n",
      "upload: ./hCAGE_cell_lines.18SE.gvcf to s3://crownproject/hCAGE/gvcf/hCAGE_cell_lines.18SE.gvcf\r",
      "\r\n",
      "Completed 1 of 12 part(s) with 5 file(s) remaining\r",
      "Completed 2 of 12 part(s) with 5 file(s) remaining\r",
      "upload: ./hCAGE_cell_lines.5.8S.gvcf to s3://crownproject/hCAGE/gvcf/hCAGE_cell_lines.5.8S.gvcf\r",
      "\r\n",
      "Completed 3 of 12 part(s) with 4 file(s) remaining\r",
      "upload: ./hCAGE_cell_lines.5S.gvcf to s3://crownproject/hCAGE/gvcf/hCAGE_cell_lines.5S.gvcf\r",
      "\r\n",
      "Completed 4 of 12 part(s) with 3 file(s) remaining\r",
      "Completed 5 of 12 part(s) with 3 file(s) remaining\r",
      "upload: ./hCAGE_cell_lines.bamlist to s3://crownproject/hCAGE/gvcf/hCAGE_cell_lines.bamlist\r",
      "\r\n",
      "Completed 6 of 12 part(s) with 2 file(s) remaining\r",
      "Completed 7 of 12 part(s) with 2 file(s) remaining\r",
      "Completed 8 of 12 part(s) with 2 file(s) remaining\r",
      "Completed 9 of 12 part(s) with 2 file(s) remaining\r",
      "Completed 10 of 12 part(s) with 2 file(s) remaining\r",
      "Completed 11 of 12 part(s) with 2 file(s) remaining\r",
      "Completed 12 of 12 part(s) with 2 file(s) remaining\r",
      "upload: ./hCAGE_cell_lines.18S.gvcf to s3://crownproject/hCAGE/gvcf/hCAGE_cell_lines.18S.gvcf\r",
      "\r\n",
      "Completed 12 of 12 part(s) with 1 file(s) remaining\r",
      "upload: ./hCAGE_cell_lines.28S.gvcf to s3://crownproject/hCAGE/gvcf/hCAGE_cell_lines.28S.gvcf\r",
      "\r\n",
      "Run Complete -- Instance is online.\r",
      "\r\n",
      "\u001b[01;32mubuntu@ip-172-31-17-124\u001b[00m:\u001b[01;34m~/hCAGE\u001b[00m$ exit\r",
      "\r\n",
      "exit\r",
      "\r\n"
     ]
    }
   ],
   "source": [
    "# aws s3 cp screenlog.0 s3://crownproject/hCAGE/hCAGE_gvcf.log\n",
    "\n",
    "aws s3 cp s3://crownproject/hCAGE/hCAGE_gvcf.log ./\n",
    "\n",
    "cat hCAGE_gvcf.log\n",
    "\n",
    "## shit, the primary cell data was misnamed (line 84) as cell line and overwrote\n",
    "## the original run. Will have to re-run for cell lines and rename primary hCAGE data\n",
    "# on AWS now"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Completed 89.7 KiB/89.8 MiB with 12 file(s) remaining\r",
      "download: s3://crownproject/hCAGE/gvcf/hCAGE_cell_lines.5S.gvcf to gvcf/hCAGE_cell_lines.5S.gvcf\r\n",
      "Completed 89.7 KiB/89.8 MiB with 11 file(s) remaining\r",
      "Completed 345.7 KiB/89.8 MiB with 11 file(s) remaining\r",
      "Completed 601.7 KiB/89.8 MiB with 11 file(s) remaining\r",
      "Completed 613.8 KiB/89.8 MiB with 11 file(s) remaining\r",
      "download: s3://crownproject/hCAGE/gvcf/hCAGE_cell_lines.bamlist to gvcf/hCAGE_cell_lines.bamlist\r\n",
      "Completed 613.8 KiB/89.8 MiB with 10 file(s) remaining\r",
      "Completed 869.8 KiB/89.8 MiB with 10 file(s) remaining\r",
      "Completed 1.1 MiB/89.8 MiB with 10 file(s) remaining  \r",
      "Completed 1.3 MiB/89.8 MiB with 10 file(s) remaining  \r",
      "Completed 1.6 MiB/89.8 MiB with 10 file(s) remaining  \r",
      "Completed 1.8 MiB/89.8 MiB with 10 file(s) remaining  \r",
      "Completed 2.1 MiB/89.8 MiB with 10 file(s) remaining  \r",
      "Completed 2.3 MiB/89.8 MiB with 10 file(s) remaining  \r",
      "Completed 2.6 MiB/89.8 MiB with 10 file(s) remaining  \r",
      "Completed 2.8 MiB/89.8 MiB with 10 file(s) remaining  \r",
      "Completed 3.1 MiB/89.8 MiB with 10 file(s) remaining  \r",
      "Completed 3.3 MiB/89.8 MiB with 10 file(s) remaining  \r",
      "Completed 3.6 MiB/89.8 MiB with 10 file(s) remaining  \r",
      "Completed 3.8 MiB/89.8 MiB with 10 file(s) remaining  \r",
      "Completed 4.1 MiB/89.8 MiB with 10 file(s) remaining  \r",
      "Completed 4.3 MiB/89.8 MiB with 10 file(s) remaining  \r",
      "Completed 4.6 MiB/89.8 MiB with 10 file(s) remaining  \r",
      "Completed 4.8 MiB/89.8 MiB with 10 file(s) remaining  \r",
      "Completed 5.1 MiB/89.8 MiB with 10 file(s) remaining  \r",
      "Completed 5.2 MiB/89.8 MiB with 10 file(s) remaining  \r",
      "download: s3://crownproject/hCAGE/gvcf/hCAGE_cell_lines.18SE.gvcf to gvcf/hCAGE_cell_lines.18SE.gvcf\r\n",
      "Completed 5.2 MiB/89.8 MiB with 9 file(s) remaining\r",
      "Completed 5.5 MiB/89.8 MiB with 9 file(s) remaining\r",
      "Completed 5.7 MiB/89.8 MiB with 9 file(s) remaining\r",
      "Completed 6.0 MiB/89.8 MiB with 9 file(s) remaining\r",
      "Completed 6.2 MiB/89.8 MiB with 9 file(s) remaining\r",
      "Completed 6.5 MiB/89.8 MiB with 9 file(s) remaining\r",
      "Completed 6.7 MiB/89.8 MiB with 9 file(s) remaining\r",
      "Completed 7.0 MiB/89.8 MiB with 9 file(s) remaining\r",
      "Completed 7.2 MiB/89.8 MiB with 9 file(s) remaining\r",
      "Completed 7.5 MiB/89.8 MiB with 9 file(s) remaining\r",
      "Completed 7.7 MiB/89.8 MiB with 9 file(s) remaining\r",
      "Completed 7.8 MiB/89.8 MiB with 9 file(s) remaining\r",
      "download: s3://crownproject/hCAGE/gvcf/hCAGE_primary.5.8S.gvcf to gvcf/hCAGE_primary.5.8S.gvcf\r\n",
      "Completed 7.8 MiB/89.8 MiB with 8 file(s) remaining\r",
      "Completed 8.0 MiB/89.8 MiB with 8 file(s) remaining\r",
      "download: s3://crownproject/hCAGE/gvcf/hCAGE_primary.18SE.gvcf to gvcf/hCAGE_primary.18SE.gvcf\r\n",
      "Completed 8.0 MiB/89.8 MiB with 7 file(s) remaining\r",
      "Completed 8.3 MiB/89.8 MiB with 7 file(s) remaining\r",
      "Completed 8.5 MiB/89.8 MiB with 7 file(s) remaining\r",
      "Completed 8.8 MiB/89.8 MiB with 7 file(s) remaining\r",
      "Completed 8.8 MiB/89.8 MiB with 7 file(s) remaining\r",
      "download: s3://crownproject/hCAGE/gvcf/hCAGE_cell_lines.5.8S.gvcf to gvcf/hCAGE_cell_lines.5.8S.gvcf\r\n",
      "Completed 8.8 MiB/89.8 MiB with 6 file(s) remaining\r",
      "Completed 9.1 MiB/89.8 MiB with 6 file(s) remaining\r",
      "Completed 9.3 MiB/89.8 MiB with 6 file(s) remaining\r",
      "Completed 9.6 MiB/89.8 MiB with 6 file(s) remaining\r",
      "Completed 9.8 MiB/89.8 MiB with 6 file(s) remaining\r",
      "Completed 10.1 MiB/89.8 MiB with 6 file(s) remaining\r",
      "Completed 10.3 MiB/89.8 MiB with 6 file(s) remaining\r",
      "Completed 10.6 MiB/89.8 MiB with 6 file(s) remaining\r",
      "Completed 10.8 MiB/89.8 MiB with 6 file(s) remaining\r",
      "Completed 11.1 MiB/89.8 MiB with 6 file(s) remaining\r",
      "Completed 11.3 MiB/89.8 MiB with 6 file(s) remaining\r",
      "Completed 11.6 MiB/89.8 MiB with 6 file(s) remaining\r",
      "Completed 11.8 MiB/89.8 MiB with 6 file(s) remaining\r",
      "Completed 12.1 MiB/89.8 MiB with 6 file(s) remaining\r",
      "Completed 12.3 MiB/89.8 MiB with 6 file(s) remaining\r",
      "Completed 12.6 MiB/89.8 MiB with 6 file(s) remaining\r",
      "Completed 12.8 MiB/89.8 MiB with 6 file(s) remaining\r",
      "Completed 13.1 MiB/89.8 MiB with 6 file(s) remaining\r",
      "Completed 13.3 MiB/89.8 MiB with 6 file(s) remaining\r",
      "Completed 13.6 MiB/89.8 MiB with 6 file(s) remaining\r",
      "Completed 13.8 MiB/89.8 MiB with 6 file(s) remaining\r",
      "Completed 14.1 MiB/89.8 MiB with 6 file(s) remaining\r",
      "Completed 14.3 MiB/89.8 MiB with 6 file(s) remaining\r",
      "Completed 14.6 MiB/89.8 MiB with 6 file(s) remaining\r",
      "Completed 14.8 MiB/89.8 MiB with 6 file(s) remaining\r",
      "Completed 15.1 MiB/89.8 MiB with 6 file(s) remaining\r",
      "Completed 15.1 MiB/89.8 MiB with 6 file(s) remaining\r",
      "download: s3://crownproject/hCAGE/gvcf/hCAGE_primary.5S.gvcf to gvcf/hCAGE_primary.5S.gvcf\r\n",
      "Completed 15.1 MiB/89.8 MiB with 5 file(s) remaining\r",
      "Completed 15.3 MiB/89.8 MiB with 5 file(s) remaining\r",
      "Completed 15.6 MiB/89.8 MiB with 5 file(s) remaining\r",
      "Completed 15.8 MiB/89.8 MiB with 5 file(s) remaining\r",
      "Completed 16.1 MiB/89.8 MiB with 5 file(s) remaining\r",
      "Completed 16.3 MiB/89.8 MiB with 5 file(s) remaining\r",
      "Completed 16.6 MiB/89.8 MiB with 5 file(s) remaining\r",
      "Completed 16.8 MiB/89.8 MiB with 5 file(s) remaining\r",
      "Completed 17.1 MiB/89.8 MiB with 5 file(s) remaining\r",
      "Completed 17.3 MiB/89.8 MiB with 5 file(s) remaining\r",
      "Completed 17.6 MiB/89.8 MiB with 5 file(s) remaining\r",
      "Completed 17.8 MiB/89.8 MiB with 5 file(s) remaining\r",
      "Completed 18.1 MiB/89.8 MiB with 5 file(s) remaining\r",
      "Completed 18.3 MiB/89.8 MiB with 5 file(s) remaining\r",
      "Completed 18.6 MiB/89.8 MiB with 5 file(s) remaining\r",
      "Completed 18.8 MiB/89.8 MiB with 5 file(s) remaining\r",
      "Completed 19.1 MiB/89.8 MiB with 5 file(s) remaining\r",
      "Completed 19.3 MiB/89.8 MiB with 5 file(s) remaining\r",
      "Completed 19.6 MiB/89.8 MiB with 5 file(s) remaining\r",
      "Completed 19.8 MiB/89.8 MiB with 5 file(s) remaining\r",
      "Completed 20.1 MiB/89.8 MiB with 5 file(s) remaining\r",
      "Completed 20.3 MiB/89.8 MiB with 5 file(s) remaining\r",
      "Completed 20.6 MiB/89.8 MiB with 5 file(s) remaining\r",
      "Completed 20.8 MiB/89.8 MiB with 5 file(s) remaining\r",
      "Completed 21.1 MiB/89.8 MiB with 5 file(s) remaining\r",
      "Completed 21.3 MiB/89.8 MiB with 5 file(s) remaining\r",
      "Completed 21.6 MiB/89.8 MiB with 5 file(s) remaining\r",
      "Completed 21.8 MiB/89.8 MiB with 5 file(s) remaining\r",
      "Completed 22.1 MiB/89.8 MiB with 5 file(s) remaining\r",
      "Completed 22.3 MiB/89.8 MiB with 5 file(s) remaining\r",
      "Completed 22.6 MiB/89.8 MiB with 5 file(s) remaining\r",
      "Completed 22.8 MiB/89.8 MiB with 5 file(s) remaining\r",
      "Completed 23.1 MiB/89.8 MiB with 5 file(s) remaining\r",
      "Completed 23.3 MiB/89.8 MiB with 5 file(s) remaining\r",
      "Completed 23.6 MiB/89.8 MiB with 5 file(s) remaining\r",
      "Completed 23.8 MiB/89.8 MiB with 5 file(s) remaining\r",
      "Completed 24.0 MiB/89.8 MiB with 5 file(s) remaining\r",
      "Completed 24.3 MiB/89.8 MiB with 5 file(s) remaining\r",
      "Completed 24.5 MiB/89.8 MiB with 5 file(s) remaining\r",
      "Completed 24.8 MiB/89.8 MiB with 5 file(s) remaining\r",
      "Completed 25.0 MiB/89.8 MiB with 5 file(s) remaining\r",
      "Completed 25.0 MiB/89.8 MiB with 5 file(s) remaining\r",
      "download: s3://crownproject/hCAGE/gvcf/hCAGE_primary.bamlist to gvcf/hCAGE_primary.bamlist\r\n",
      "Completed 25.0 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 25.3 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 25.5 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 25.8 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 26.0 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 26.3 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 26.5 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 26.8 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 27.0 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 27.3 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 27.5 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 27.8 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 28.0 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 28.3 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 28.5 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 28.8 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 29.0 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 29.3 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 29.5 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 29.8 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 30.0 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 30.3 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 30.5 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 30.8 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 31.0 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 31.3 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 31.5 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 31.8 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 32.0 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 32.3 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 32.5 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 32.8 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 33.0 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 33.3 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 33.5 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 33.8 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 34.0 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 34.3 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 34.5 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 34.8 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 35.0 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 35.3 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 35.5 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 35.8 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 36.0 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 36.3 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 36.5 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 36.8 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 37.0 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 37.3 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 37.5 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 37.8 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 38.0 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 38.3 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 38.5 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 38.8 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 39.0 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 39.3 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 39.5 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 39.8 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 40.0 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 40.3 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 40.5 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 40.8 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 41.0 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 41.3 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 41.5 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 41.8 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 42.0 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 42.3 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 42.5 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 42.8 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 43.0 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 43.3 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 43.5 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 43.8 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 44.0 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 44.3 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 44.5 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 44.8 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 45.0 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 45.3 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 45.5 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 45.8 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 46.0 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 46.3 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 46.5 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 46.8 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 47.0 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 47.3 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 47.5 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 47.8 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 48.0 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 48.3 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 48.5 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 48.8 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 49.0 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 49.3 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 49.5 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 49.8 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 50.0 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 50.3 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 50.5 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 50.8 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 51.0 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 51.3 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 51.5 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 51.8 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 52.0 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 52.3 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 52.5 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 52.8 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 53.0 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 53.3 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 53.5 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 53.8 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 54.0 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 54.3 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 54.5 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 54.8 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 55.0 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 55.3 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 55.5 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 55.8 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 56.0 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 56.3 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 56.5 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 56.8 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 57.0 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 57.3 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 57.5 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 57.8 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 58.0 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 58.3 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 58.5 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 58.8 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 59.0 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 59.3 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 59.5 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 59.8 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 60.0 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 60.3 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 60.5 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 60.8 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 61.0 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 61.3 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 61.5 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 61.8 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 62.0 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 62.3 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 62.5 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 62.8 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 63.0 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 63.3 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 63.5 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 63.8 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 64.0 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 64.3 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 64.5 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 64.8 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 65.0 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 65.3 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 65.5 MiB/89.8 MiB with 4 file(s) remaining\r",
      "Completed 65.6 MiB/89.8 MiB with 4 file(s) remaining\r",
      "download: s3://crownproject/hCAGE/gvcf/hCAGE_cell_lines.18S.gvcf to gvcf/hCAGE_cell_lines.18S.gvcf\r\n",
      "Completed 65.6 MiB/89.8 MiB with 3 file(s) remaining\r",
      "Completed 65.9 MiB/89.8 MiB with 3 file(s) remaining\r",
      "Completed 66.1 MiB/89.8 MiB with 3 file(s) remaining\r",
      "Completed 66.4 MiB/89.8 MiB with 3 file(s) remaining\r",
      "Completed 66.6 MiB/89.8 MiB with 3 file(s) remaining\r",
      "Completed 66.9 MiB/89.8 MiB with 3 file(s) remaining\r",
      "Completed 67.1 MiB/89.8 MiB with 3 file(s) remaining\r",
      "Completed 67.4 MiB/89.8 MiB with 3 file(s) remaining\r",
      "Completed 67.6 MiB/89.8 MiB with 3 file(s) remaining\r",
      "Completed 67.9 MiB/89.8 MiB with 3 file(s) remaining\r",
      "Completed 68.1 MiB/89.8 MiB with 3 file(s) remaining\r",
      "Completed 68.4 MiB/89.8 MiB with 3 file(s) remaining\r",
      "Completed 68.6 MiB/89.8 MiB with 3 file(s) remaining\r",
      "Completed 68.9 MiB/89.8 MiB with 3 file(s) remaining\r",
      "Completed 69.1 MiB/89.8 MiB with 3 file(s) remaining\r",
      "Completed 69.4 MiB/89.8 MiB with 3 file(s) remaining\r",
      "Completed 69.6 MiB/89.8 MiB with 3 file(s) remaining\r",
      "Completed 69.9 MiB/89.8 MiB with 3 file(s) remaining\r",
      "Completed 70.1 MiB/89.8 MiB with 3 file(s) remaining\r",
      "Completed 70.4 MiB/89.8 MiB with 3 file(s) remaining\r",
      "Completed 70.6 MiB/89.8 MiB with 3 file(s) remaining\r",
      "Completed 70.9 MiB/89.8 MiB with 3 file(s) remaining\r",
      "Completed 71.1 MiB/89.8 MiB with 3 file(s) remaining\r",
      "Completed 71.4 MiB/89.8 MiB with 3 file(s) remaining\r",
      "Completed 71.6 MiB/89.8 MiB with 3 file(s) remaining\r",
      "Completed 71.9 MiB/89.8 MiB with 3 file(s) remaining\r",
      "Completed 72.1 MiB/89.8 MiB with 3 file(s) remaining\r",
      "Completed 72.4 MiB/89.8 MiB with 3 file(s) remaining\r",
      "Completed 72.6 MiB/89.8 MiB with 3 file(s) remaining\r",
      "Completed 72.9 MiB/89.8 MiB with 3 file(s) remaining\r",
      "Completed 73.1 MiB/89.8 MiB with 3 file(s) remaining\r",
      "Completed 73.4 MiB/89.8 MiB with 3 file(s) remaining\r",
      "Completed 73.5 MiB/89.8 MiB with 3 file(s) remaining\r",
      "download: s3://crownproject/hCAGE/gvcf/hCAGE_primary.18S.gvcf to gvcf/hCAGE_primary.18S.gvcf\r\n",
      "Completed 73.5 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 73.8 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 74.0 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 74.3 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 74.5 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 74.8 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 75.0 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 75.3 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 75.5 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 75.8 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 76.0 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 76.3 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 76.5 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 76.8 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 77.0 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 77.3 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 77.5 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 77.8 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 78.0 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 78.3 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 78.3 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 78.6 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 78.8 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 79.1 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 79.3 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 79.6 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 79.8 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 80.1 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 80.3 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 80.6 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 80.8 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 81.1 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 81.3 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 81.6 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 81.8 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 82.1 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 82.3 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 82.6 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 82.8 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 83.1 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 83.3 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 83.6 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 83.8 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 84.1 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 84.3 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 84.6 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 84.8 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 85.1 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 85.3 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 85.6 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 85.8 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 86.1 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 86.3 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 86.6 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 86.8 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 87.1 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 87.3 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 87.6 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 87.8 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 88.1 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 88.3 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 88.6 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 88.8 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 89.1 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 89.3 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 89.6 MiB/89.8 MiB with 2 file(s) remaining\r",
      "Completed 89.8 MiB/89.8 MiB with 2 file(s) remaining\r",
      "download: s3://crownproject/hCAGE/gvcf/hCAGE_primary.28S.gvcf to gvcf/hCAGE_primary.28S.gvcf\r\n",
      "Completed 89.8 MiB/89.8 MiB with 1 file(s) remaining\r",
      "download: s3://crownproject/hCAGE/gvcf/hCAGE_cell_lines.28S.gvcf to gvcf/hCAGE_cell_lines.28S.gvcf\r\n"
     ]
    }
   ],
   "source": [
    "mkdir -p gvcf\n",
    "aws s3 cp --recursive s3://crownproject/hCAGE/gvcf/ ./gvcf/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Materials and Methods -- CRC\n",
    "\n",
    "Launch an AMI instance and run ADcalc script on data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# DNS:XXX\n",
    "# AMI: ami-0b375c9c58cb4a7a2 (TCGA aligner)\n",
    "# Instance: m4.4xlarge\n",
    "# Storage: 400 Gb\n",
    "\n",
    "# ON REMOTE:\n",
    "\n",
    "## Copy CRC files into it's dir\n",
    "#mkdir -p ~/crc; cd crc;\n",
    "#aws s3 cp --recursive s3://crownproject/crc/ ./\n",
    "\n",
    "## PATHS:\n",
    "## ~/hCAGE/cell_lines/BAM/*\n",
    "## ~/hCAGE/primary/BAM/*\n",
    "\n",
    "# cd ~/hCAGE/cell_lines/BAM/\n",
    "\n",
    "## Run ADcalc script.sh\n",
    "# screen -L\n",
    "# bash ADcalc_tcga.sh\n",
    "\n",
    "# aws s3 cp screenlog.0 s3://crownproject/ccle/logs/ccle.gvcf.log\n",
    "\n",
    "aws s3 cp s3://crownproject/ccle/logs/ccle.gvcf.log ./\n",
    "cat ccle.gvcf.log\n",
    "\n",
    "## DONE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "# ADcalc_crc.sh\n",
    "# Allelic Depth Calculator\n",
    "# for a position\n",
    "#\n",
    "# s3://crownproject/hCAGE/ADcalc_crc.sh\n",
    "\n",
    "# Controls -----------------\n",
    "DEPTH='100000' #Max per file DP\n",
    "\n",
    "# Regions in hgr1.fa reference genome\n",
    "REGIONS=('chr13:1003660-1005529' 'chr13:1005529-1005629' \\\n",
    "        'chr13:10219-10340' 'chr13:1006622-1006779' 'chr13:1007948-1013018')\n",
    "\n",
    "# Corresponding region/gene names\n",
    "GENES=('18S' '18SE' '5S' '5.8S' '28S')\n",
    "\n",
    "# 18S  1870\n",
    "# 18SE 101\n",
    "# 5S   122\n",
    "# 5.8S 158\n",
    "# 28S  5071\n",
    "\n",
    "# Terminate instances upon completion (for debugging)\n",
    "TERMINATE='FALSE'\n",
    "\n",
    "# S3 Output directory\n",
    "S3DIR='s3://crownproject/crc/gvcf/'\n",
    "BAMLIST='bam.list.tmp'\n",
    "\n",
    "# Script ------------------ ------------------------------\n",
    "cd ~/crc/\n",
    "mkdir -p GVCF #Output Folder\n",
    "TYPE='crc1' # hardcode single ccle run\n",
    "cd crc1/hgr1/\n",
    "\n",
    "#for TYPE in $(echo \"hgr1\")\n",
    "#do\n",
    "    echo Analyzing $TYPE...\n",
    "    #cd $TYPE\n",
    "\n",
    "    ls *.bam > bam.list.tmp\n",
    "    ls *.bam > ../../GVCF/$TYPE.bamlist\n",
    "          \n",
    "    for index in ${!GENES[*]}\n",
    "    do\n",
    "      printf \"Started processing %s\\n\" ${GENES[$index]}\n",
    "      OUTPUT=\"../../GVCF/$TYPE.${GENES[$index]}.gvcf\"\n",
    "\n",
    "      # Iterate through every bam file in directory\n",
    "      # look-up position and return VCF\n",
    "      bcftools mpileup -f ~/resources/hgr1/hgr1.fa \\\n",
    "        --max-depth $DEPTH -A --min-BQ 30 \\\n",
    "        -a FORMAT/DP,AD \\\n",
    "        -r ${REGIONS[$index]} \\\n",
    "        --ignore-RG \\\n",
    "        -b $BAMLIST | \\\n",
    "        bcftools annotate -x INFO,FORMAT/PL - | \\\n",
    "        bcftools view -O v - \\\n",
    "        >> $OUTPUT\n",
    "\n",
    "      RESULTS+=(\"$OUTPUT\")\n",
    "      printf \"Done with %s \\n\" ${GENES[$index]}\n",
    "      printf \"%s\\n\" ${REGIONS[$index]}\n",
    "\n",
    "    done\n",
    "\n",
    "    rm bam.list.tmp\n",
    "\n",
    "#    cd .. # move to tcga folder to reset\n",
    "#done\n",
    "\n",
    "# Copy GVCF output to AWS S3\n",
    "cd ../../GVCF\n",
    "aws s3 cp --recursive ./ $S3DIR\n",
    "\n",
    "# -------------------------------------------\n",
    "cd ~/crc/\n",
    "mkdir -p GVCF #Output Folder\n",
    "TYPE='crc4' # hardcode single ccle run\n",
    "cd crc4/hgr1/\n",
    "\n",
    "#for TYPE in $(echo \"hgr1\")\n",
    "#do\n",
    "    echo Analyzing $TYPE...\n",
    "    #cd $TYPE\n",
    "\n",
    "    ls *.bam > bam.list.tmp\n",
    "    ls *.bam > ../../GVCF/$TYPE.bamlist\n",
    "          \n",
    "    for index in ${!GENES[*]}\n",
    "    do\n",
    "      printf \"Started processing %s\\n\" ${GENES[$index]}\n",
    "      OUTPUT=\"../../GVCF/$TYPE.${GENES[$index]}.gvcf\"\n",
    "\n",
    "      # Iterate through every bam file in directory\n",
    "      # look-up position and return VCF\n",
    "      bcftools mpileup -f ~/resources/hgr1/hgr1.fa \\\n",
    "        --max-depth $DEPTH -A --min-BQ 30 \\\n",
    "        -a FORMAT/DP,AD \\\n",
    "        -r ${REGIONS[$index]} \\\n",
    "        --ignore-RG \\\n",
    "        -b $BAMLIST | \\\n",
    "        bcftools annotate -x INFO,FORMAT/PL - | \\\n",
    "        bcftools view -O v - \\\n",
    "        >> $OUTPUT\n",
    "\n",
    "      RESULTS+=(\"$OUTPUT\")\n",
    "      printf \"Done with %s \\n\" ${GENES[$index]}\n",
    "      printf \"%s\\n\" ${REGIONS[$index]}\n",
    "\n",
    "    done\n",
    "\n",
    "    rm bam.list.tmp\n",
    "\n",
    "#    cd .. # move to tcga folder to reset\n",
    "#done\n",
    "\n",
    "# Copy GVCF output to AWS S3\n",
    "cd ../../GVCF\n",
    "aws s3 cp --recursive ./ $S3DIR\n",
    "\n",
    "\n",
    "# Shutdown and Terminate instance\n",
    "EC2ID=$(ec2metadata --instance-id)\n",
    "sleep 20s # to catch errors\n",
    "\n",
    "if [ \"$TERMINATE\" = TRUE ]\n",
    "then\n",
    "  echo \"Run Complete -- Shutting down instance.\"\n",
    "  aws ec2 terminate-instances --instance-ids $EC2ID\n",
    "else\n",
    "  echo \"Run Complete -- Instance is online.\"\n",
    "fi\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Discussion\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
