{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# TCGA Run Series 2 - Non-matched\n",
    "```\n",
    "pi:ababaian\n",
    "files: ~/Crown/data2/tcag2_pilot/\n",
    "start: 2019 05 06\n",
    "complete : YYYY MM DD\n",
    "```\n",
    "## Introduction\n",
    "\n",
    "I've been working with G. Novakovsky the last few months on a derivative of `Crown` called `Goat` (https://github.com/ababaian/Goat) which is focused on 3-D clustering of rRNA mutations/variations and RP mutations (from COSMIC) in the cryo-EM structure.\n",
    "\n",
    "We've had quite a bit of interesting hits arise in the last few weeks which need to be validated.\n",
    "\n",
    "### 28S.G2407\n",
    "\n",
    "The most notable mutation in rRNA is 28S.G2407C. This variation is recurrent in three patients (out of 649), two of which are Lung Adenocarcinoma.\n",
    "\n",
    "\n",
    "| hgr1_coord | gene_position | gene | cancer    | ref_allele | alt_allele | VAFcancer   | VAFnorm     | MutOrSNP | DPcancer | DPnorm | patient | patientID    | patientAll | Mut/Patient | Mut/Site | \n",
    "|------------|---------------|------|-----------|------------|------------|-------------|-------------|----------|----------|--------|---------|--------------|------------|-------------|----------| \n",
    "| 1010354    | 2407          | 28S  | TCGA-LUAD | G          | C,T,A      | 0.167116564 | 0.000470035 | MUT      | 4075     | 8510   | 8       | TCGA-44-2662 | 392        | 7           | 3        | \n",
    "| 1010354    | 2407          | 28S  | TCGA-BLCA | G          | C,T,A      | 0.192732455 | 0           | MUT      | 5614     | 3240   | 1       | TCGA-BL-A13J | 1          | 2           | 3        | \n",
    "| 1010354    | 2407          | 28S  | TCGA-LUAD | G          | C,T,A      | 0.998996488 | 0           | MUT      | 63776    | 6231   | 15      | TCGA-44-6147 | 399        | 1           | 3        | \n",
    "\n",
    "1. The variant allele frequency (VAF) of these variants is high in the cancers which have them (16.7 - 99.8%). Considering there are between 50-800 rDNA copies, a high VAF is consistent with a *sweep* for this allele when the mutation arises. **Note:** This is still confounded by incomplete cancer-cell frequencies in the samples, so the actual allele frequency may be higher.\n",
    "2. These positions are *de novo* mutations, they are absent from the patient-matched normal control RNA-seq (VAFnorm < 0.001). There is such mutation/variation in any of the remaining normal controls (~645).\n",
    "\n",
    "![28S.G2407 VAF_Cancer vs. VAF_Normal for 649 patients](../figure2/190506_28S.G2407_VAF.png)\n",
    "\n",
    "\n",
    "`INCLUDE EXAMPLE OF IGV READ STATS SHOWING BI_DIRECTIONAL MUTATION IN PATIENT`\n",
    "\n",
    "I will be expanding the TCGA RNAseq analysis to include DNAseq, as well as the remaining patients without matched cancer-normal RNAseq. This will bring up the discovery set to N = ~10,000 cancers. We will use normal DNAseq to identify mutations (vs rDNA SNPs with high allele frequency).\n",
    "\n",
    "Initially I would like to test `TCGA-44-2662`, `TCGA-BL-A13J`, `TCGA-44-6147` replicate RNA-seq, as well as DNA-seq to validate the presense/absense of G2407C at similiar allele frequency, and validate this mutation.\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "## Objective\n",
    "\n",
    "1. Re-initiate AWS S3 based TCGA analysis.\n",
    "2. Run a pilot alignment on all seq data from `TCGA-44-2662`, `TCGA-BL-A13J`, `TCGA-44-6147` to `hgr1`.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Matererials and Methods"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "### RNA-seq and Input Table"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "#### RNAseq\n",
    "\n",
    "| File ID                              | File Name                                                 | Data Category       | Data Type     | Project ID | Case ID      | Sample ID        | Sample Type   | Case_Match | MatchedNormal | CanNorm_Pair1 | CanNorm_Pair2 | IncludeSample | \n",
    "|--------------------------------------|-----------------------------------------------------------|---------------------|---------------|------------|--------------|------------------|---------------|------------|---------------|---------------|---------------|---------------| \n",
    "| 10cdc012-492d-4090-83ab-ff73ac0ee536 | f7f51052-0994-49aa-b55a-f84b21429e80_gdc_realn_rehead.bam | Raw Sequencing Data | Aligned Reads | TCGA-LUAD  | TCGA-44-2662 | TCGA-44-2662-01A | Primary Tumor | 1          | 0             | 0             | 0             | Exclude       | \n",
    "| 320b2265-20e7-40a8-a554-f9231f6ea914 | 74be9457-39ce-4029-9076-53b0e57d2067_gdc_realn_rehead.bam | Raw Sequencing Data | Aligned Reads | TCGA-LUAD  | TCGA-44-2662 | TCGA-44-2662-01A | Primary Tumor | 1          | 0             | 0             | 0             | Exclude       | \n",
    "| fd1ec26a-64e6-4bc4-b923-182bcf3446a0 | 2775d5a7-9663-4169-83e3-937032a27d78_gdc_realn_rehead.bam | Raw Sequencing Data | Aligned Reads | TCGA-BLCA  | TCGA-BL-A13J | TCGA-BL-A13J-01A | Primary Tumor | 1          | 0             | 0             | 0             | Exclude       | \n",
    "| 26a1a274-2b68-4d22-8942-321c718e5bd7 | 786870b0-b32a-4e46-90bf-31792658a685_gdc_realn_rehead.bam | Raw Sequencing Data | Aligned Reads | TCGA-BLCA  | TCGA-BL-A13J | TCGA-BL-A13J-01A | Primary Tumor | 0          | 0             | 0             | 0             | Exclude       | \n",
    "| 16dfa0e8-e7cf-4589-bd99-82944d703695 | 6cb90565-9ca5-48ee-a258-40ea601c7380_gdc_realn_rehead.bam | Raw Sequencing Data | Aligned Reads | TCGA-LUAD  | TCGA-44-6147 | TCGA-44-6147-01A | Primary Tumor | 1          | 0             | 0             | 0             | Exclude       | \n",
    "| 056c3eb5-5317-4cb2-9f33-0e93b4e54be5 | f493fc9c-abd8-4623-b03b-6300c13a3198_gdc_realn_rehead.bam | Raw Sequencing Data | Aligned Reads | TCGA-LUAD  | TCGA-44-6147 | TCGA-44-6147-01A | Primary Tumor | 0          | 0             | 0             | 0             | Exclude       | \n",
    "\n",
    "\n",
    "#### Input Table\n",
    "\n",
    "| Sample ID        | Project ID | File ID                              | \n",
    "|------------------|------------|--------------------------------------| \n",
    "| TCGA-44-2662-01A | TCGA-LUAD  | 10cdc012-492d-4090-83ab-ff73ac0ee536 | \n",
    "| TCGA-44-2662-01A | TCGA-LUAD  | 320b2265-20e7-40a8-a554-f9231f6ea914 | \n",
    "| TCGA-BL-A13J-01A | TCGA-BLCA  | fd1ec26a-64e6-4bc4-b923-182bcf3446a0 | \n",
    "| TCGA-BL-A13J-01A | TCGA-BLCA  | 26a1a274-2b68-4d22-8942-321c718e5bd7 | \n",
    "| TCGA-44-6147-01A | TCGA-LUAD  | 16dfa0e8-e7cf-4589-bd99-82944d703695 | \n",
    "| TCGA-44-6147-01A | TCGA-LUAD  | 056c3eb5-5317-4cb2-9f33-0e93b4e54be5 | \n",
    "\n",
    "\n",
    "**Note:** When running all RNA-seq samples, there will be collisions in some replciates as is the case for these three samples. Add a suffix `x` from `a-z` in these cases to distinguish replicate files.\n",
    "\n",
    "| Sample ID         | Project ID | File ID                              | \n",
    "|-------------------|------------|--------------------------------------| \n",
    "| TCGA-44-2662-01Aa | TCGA-LUAD  | 10cdc012-492d-4090-83ab-ff73ac0ee536 | \n",
    "| TCGA-44-2662-01Ab | TCGA-LUAD  | 320b2265-20e7-40a8-a554-f9231f6ea914 | \n",
    "| TCGA-BL-A13J-01Aa | TCGA-BLCA  | fd1ec26a-64e6-4bc4-b923-182bcf3446a0 | \n",
    "| TCGA-BL-A13J-01Ab | TCGA-BLCA  | 26a1a274-2b68-4d22-8942-321c718e5bd7 | \n",
    "| TCGA-44-6147-01Aa | TCGA-LUAD  | 16dfa0e8-e7cf-4589-bd99-82944d703695 | \n",
    "| TCGA-44-6147-01Ab | TCGA-LUAD  | 056c3eb5-5317-4cb2-9f33-0e93b4e54be5 | \n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Scripts and Localization"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "droneB.sh  hgr1_align_v3.tcga.sh  input_tcga2.txt  pilot_files.xlsx  queenB.sh\r\n"
     ]
    }
   ],
   "source": [
    "WORKDIR='/home/artem/Crown/data2/tcga2_pilot'\n",
    "cd $WORKDIR\n",
    "ls\n",
    "\n",
    "# Amazon AWS S3 Home URL\n",
    "S3URL='s3://crownproject/tcga2'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TCGA-44-2662-01Aa\tTCGA-LUAD\t10cdc012-492d-4090-83ab-ff73ac0ee536\r\n",
      "TCGA-44-2662-01Ab\tTCGA-LUAD\t320b2265-20e7-40a8-a554-f9231f6ea914\r\n",
      "TCGA-BL-A13J-01Aa\tTCGA-BLCA\tfd1ec26a-64e6-4bc4-b923-182bcf3446a0\r\n",
      "TCGA-BL-A13J-01Ab\tTCGA-BLCA\t26a1a274-2b68-4d22-8942-321c718e5bd7\r\n",
      "TCGA-44-6147-01Aa\tTCGA-LUAD\t16dfa0e8-e7cf-4589-bd99-82944d703695\r\n",
      "TCGA-44-6147-01Ab\tTCGA-LUAD\t056c3eb5-5317-4cb2-9f33-0e93b4e54be5"
     ]
    }
   ],
   "source": [
    "INPUT='input_tcga2.txt'\n",
    "\n",
    "cat $INPUT"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "#### Scripts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#!/bin/bash\r\n",
      "# 1kg_align_v2.tcga.sh\r\n",
      "# rDNA alignment pipeline\r\n",
      "# 190506 build -- TCGA\r\n",
      "# AMI: crown-180813 - ami-0031fd61f932bdef9\r\n",
      "# EC2: c4.2xlarge (8cpu / 15 gb)\r\n",
      "# EC2: c4.xlarge  (4cpu / 8  gb)\r\n",
      "# Storage: 200 Gb\r\n",
      "#\r\n",
      "\r\n",
      "# Input Requirements --------------------------\r\n",
      "\r\n",
      "# $1 : Library name and file-output name (unique)\r\n",
      "# $2 : Library population/analysis set\r\n",
      "# $3 : Library UUID\r\n",
      "\r\n",
      "# Control Panel -------------------------------\r\n",
      "# Amazon AWS S3 Home URL\r\n",
      "  S3URL='s3://crownproject/tcga2'\r\n",
      "\r\n",
      "# CPU\r\n",
      "\tTHREADS='3'\r\n",
      "\r\n",
      "# Terminate instances upon completion (for debuggin)\r\n",
      "  TERMINATE='FALSE'\r\n",
      "\r\n",
      "# Sequencing Data\r\n",
      "\tLIBRARY=$1 # Library/ File name\r\n",
      "\r\n",
      "# TCGA FILE UUID\r\n",
      "  UUID=$3\r\n",
      "\r\n",
      " # FastQ File-names\r\n",
      "    FQ0=\"$LIBRARY.tmp.sort.0.fq\"\r\n",
      "    FQ1=\"$LIBRARY.tmp.sort.1.fq\"\r\n",
      "    FQ2=\"$LIBRARY.tmp.sort.2.fq\"\r\n",
      "    \r\n",
      "# Read Group Data\r\n",
      "# Extract from downloaded BAM file / input\r\n",
      "\tRGPO=$2 # Patient Population\r\n",
      "\r\n",
      "\t#RGSM= # Sample. Patient Identifer\r\n",
      "\t#RGID= # Read Group ID. Accession Number\r\n",
      "    \r\n",
      "\tRGLB=$LIBRARY # Library Name. Accession Number\r\n",
      "\tRGPL='ILLUMINA'  # Sequencing Platform.\r\n",
      "    \r\n",
      "\t# Extract Sequencing Run Info\r\n",
      "\t#  RGPU=$(gzip -dc $FQ1 | head -n1 - | cut -f1 -d':' | cut -f2 -d' ')\r\n",
      "\r\n",
      "# Initialize wordir ---------------------------\r\n",
      "\r\n",
      "# Make working directory\r\n",
      "  mkdir -p align; cd align\r\n",
      "\r\n",
      "# Copy hgrX genome and create bowtie2 index\r\n",
      "  cp ~/resources/hgr1/* ./\r\n",
      "\r\n",
      "# Load updated GDC token to instance\r\n",
      "aws s3 cp $S3URL/scripts/gdc.token gdc.token\r\n",
      "chmod 600 gdc.token\r\n",
      "  \r\n",
      "# Download RNAseq BAM file\r\n",
      "# with a UUID as input\r\n",
      "  ~/bin/gdc-client download -t gdc.token -d ./ \\\r\n",
      "  -n $THREADS $UUID\r\n",
      "  \r\n",
      "# Link the RNA-seq bamfile which is called by its UID to workdir\r\n",
      "  ln -s */*.bam input.bam\r\n",
      "  \r\n",
      "# Extract ReadGroup Sample Name (SM)\r\n",
      "  RGSM=$(~/bin/samtools view -H input.bam | grep '^@RG' | sed \"s/.*SM:\\([^\\t]*\\).*/\\1/g\" | uniq )\r\n",
      "\r\n",
      "# Extract ReadGroup identifier (ID)\r\n",
      "  #RGID=$(~/bin/samtools view -H input.bam | grep '^@RG' | sed \"s/.*ID:\\([^\\t]*\\).*/\\1/g\" | uniq )\r\n",
      "  RGID=$(~/bin/samtools view -H input.bam | grep '^@RG' | sed \"s/.*ID:\\([^\\t]*\\).*/\\1/g\" | head -n1 - )\r\n",
      "\r\n",
      "# Convert input bam file to fastq files for re-alignment\r\n",
      "~/bin/samtools sort -@ $THREADS-n input.bam | \\\r\n",
      "    ~/bin/samtools fastq -@ $THREADS -O \\\r\n",
      "    -0 $FQ0 \\\r\n",
      "    -1 $FQ1 \\\r\n",
      "    -2 $FQ2 -\r\n",
      "\r\n",
      "# SINGLE END READS ====================================================\r\n",
      "\r\n",
      "if [ -s $FQ0 ]\r\n",
      "then\r\n",
      "    # Single-End Extracted Reads Alignment\r\n",
      "\r\n",
      "    # Extract Sequencing Run Info\r\n",
      "    #RGPU=$(gzip -dc $FQ0| head -n1 - | cut -f1 -d':' | cut -f2 -d' ')\r\n",
      "    RGPU=$(head -n1 $FQ0 | cut -f1 -d':' | cut -f2 -d' ')\r\n",
      "\r\n",
      "    # Bowtie2: align to genome\r\n",
      "    bowtie2 --very-sensitive-local -p $THREADS \\\r\n",
      "      --rg-id $RGID --rg LB:$RGLB --rg SM:$RGSM \\\r\n",
      "      --rg PL:$RGPL --rg PU:$RGPU \\\r\n",
      "      -x hgr1 -U $FQ0 | \\\r\n",
      "      ~/bin/samtools view -bS - > aligned_unsorted.bam\r\n",
      "     \r\n",
      "    # Calculate library flagstats\r\n",
      "      ~/bin/samtools flagstat aligned_unsorted.bam > aligned_unsorted.flagstat\r\n",
      "      rm $FQ0 # Remove fastq files to save space\r\n",
      "\r\n",
      "    # Read Subset ------------------------------\r\n",
      "    # Extract mapped reads, and their unmapped pairs\r\n",
      "\r\n",
      "      # Extract Header\r\n",
      "      ~/bin/samtools view -H aligned_unsorted.bam > align.header.tmp\r\n",
      "\r\n",
      "      # Extract Mapped Reads\r\n",
      "      ~/bin/samtools view -b -F 4 aligned_unsorted.bam | \\\r\n",
      "      ~/bin/samtools sort -@ $THREADS -O BAM - > align.hgr1.bam #mapped\r\n",
      "      \r\n",
      "    # Calcualte library flagstats\r\n",
      "    ~/bin/samtools index align.hgr1.bam\r\n",
      "    ~/bin/samtools flagstat align.hgr1.bam > align.hgr1.flagstat\r\n",
      "\r\n",
      "    # Rename the total Bam Files\r\n",
      "      #mv aligned_unsorted.bam $LIBRARY.se.bam\r\n",
      "      #mv aligned_unsorted.bam.bai $LIBRARY.se.bam.bai\r\n",
      "      mv aligned_unsorted.flagstat $LIBRARY.se.flagstat\r\n",
      "\r\n",
      "    # Rename the hgr-aligned Bam files\r\n",
      "      mv align.hgr1.bam $LIBRARY.hgr1.se.bam\r\n",
      "      mv align.hgr1.bam.bai $LIBRARY.hgr1.se.bam.bai\r\n",
      "      mv align.hgr1.flagstat $LIBRARY.hgr1.se.flagstat\r\n",
      "      \r\n",
      "    # Alignments (Full)\r\n",
      "    aws s3 cp $LIBRARY.se.flagstat $S3URL/$RGPO/\r\n",
      "\r\n",
      "    # Alignments (Aligned)\r\n",
      "    aws s3 cp $LIBRARY.hgr1.se.bam $S3URL/$RGPO/\r\n",
      "    aws s3 cp $LIBRARY.hgr1.se.bam.bai $S3URL/$RGPO/\r\n",
      "    aws s3 cp $LIBRARY.hgr1.se.flagstat $S3URL/$RGPO/\r\n",
      "\r\n",
      "\r\n",
      "fi\r\n",
      "\r\n",
      "# PAIRED END READS ====================================================\r\n",
      "\r\n",
      "\r\n",
      "if [ -s $FQ1 ]\r\n",
      "then\r\n",
      "    # Paired-End Extracted Reads Alignment\r\n",
      "\r\n",
      "    # Extract Sequencing Run Info\r\n",
      "    #RGPU=$(gzip -dc $FQ1| head -n1 $FQ1 | cut -f1 -d':' | cut -f2 -d' ')\r\n",
      "    RGPU=$(head -n1 $FQ1 | cut -f1 -d':' | cut -f2 -d' ')\r\n",
      "    \r\n",
      "    # Bowtie2: align to genome\r\n",
      "    bowtie2 --very-sensitive-local -p $THREADS \\\r\n",
      "      --rg-id $RGID --rg LB:$RGLB --rg SM:$RGSM \\\r\n",
      "      --rg PL:$RGPL --rg PU:$RGPU \\\r\n",
      "      -x hgr1 -1 $FQ1 -2 $FQ2 | \\\r\n",
      "      ~/bin/samtools view -bS - > aligned_unsorted.bam\r\n",
      "      \r\n",
      "    # Calcualte library flagstats\r\n",
      "      ~/bin/samtools flagstat aligned_unsorted.bam > aligned_unsorted.flagstat\r\n",
      "      \r\n",
      "      rm $FQ1 $FQ2 # Remove fastq files to save space\r\n",
      "\r\n",
      "      \r\n",
      "    # Read Subset ------------------------------\r\n",
      "    # Extract mapped reads, and their unmapped pairs\r\n",
      "\r\n",
      "      # Extract Header\r\n",
      "      ~/bin/samtools view -H aligned_unsorted.bam > align.header.tmp\r\n",
      "\r\n",
      "      # Unmapped reads with mapped pairs\r\n",
      "      # Extract Mapped Reads\r\n",
      "      # and their unmapped pairs\r\n",
      "      ~/bin/samtools view -b -F 4 aligned_unsorted.bam > align.F4.bam #mapped\r\n",
      "      ~/bin/samtools view -b -f 4 -F 8 aligned_unsorted.bam > align.f4F8.bam #unmapped pairs\r\n",
      "\r\n",
      "      # Re-compile bam file\r\n",
      "      ~/bin/samtools cat -h align.header.tmp -o align.hgr1.tmp.bam align.F4.bam align.f4F8.bam\r\n",
      "        ~/bin/samtools sort -@ $THREADS -O BAM align.hgr1.tmp.bam > align.hgr1.bam\r\n",
      "        ~/bin/samtools index align.hgr1.bam\r\n",
      "        ~/bin/samtools flagstat align.hgr1.bam > align.hgr1.flagstat\r\n",
      "\r\n",
      "      # Clean up \r\n",
      "      rm *tmp* align.F4.bam align.f4F8.bam\r\n",
      "\r\n",
      "    # Rename the total Bam Files\r\n",
      "      mv aligned_unsorted.bam $LIBRARY.bam\r\n",
      "      #mv aligned_unsorted.bam.bai $LIBRARY.bam.bai\r\n",
      "      mv aligned_unsorted.flagstat $LIBRARY.flagstat\r\n",
      "\r\n",
      "    # Rename the hgr-aligned Bam files\r\n",
      "      mv align.hgr1.bam $LIBRARY.hgr1.bam\r\n",
      "      mv align.hgr1.bam.bai $LIBRARY.hgr1.bam.bai\r\n",
      "      mv align.hgr1.flagstat $LIBRARY.hgr1.flagstat\r\n",
      "      \r\n",
      "    # Copy output to AWS S3 \r\n",
      "    # Alignments (Full)\r\n",
      "      aws s3 cp $LIBRARY.flagstat $S3URL/$RGPO/\r\n",
      "\r\n",
      "    # Alignments (Aligned)\r\n",
      "      aws s3 cp $LIBRARY.hgr1.bam $S3URL/$RGPO/\r\n",
      "      aws s3 cp $LIBRARY.hgr1.bam.bai $S3URL/$RGPO/\r\n",
      "      aws s3 cp $LIBRARY.hgr1.flagstat $S3URL/$RGPO/\r\n",
      "\r\n",
      "fi\r\n",
      " \r\n",
      "\r\n",
      "# Copy screen log file to AWS S3\r\n",
      "cp ~/screenlog.0 ./$LIBRARY.screenlog\r\n",
      "aws s3 cp $LIBRARY.screenlog $S3URL/logs/\r\n",
      "\r\n",
      "# Shutdown and Terminate instance\r\n",
      "EC2ID=$(ec2metadata --instance-id)\r\n",
      "sleep 20s # to catch errors\r\n",
      "\r\n",
      "if [ \"$TERMINATE\" = TRUE ]\r\n",
      "then\r\n",
      "  aws ec2 terminate-instances --instance-ids $EC2ID\r\n",
      "fi\r\n",
      "\r\n",
      "# Script complete\r\n",
      "\r\n",
      "\r\n",
      "#!/bin/bash\r\n",
      "# queenB.sh\r\n",
      "# 20180814 build\r\n",
      "# EC2 Launch / Control Script\r\n",
      "#\r\n",
      "\r\n",
      "# 1. queenB script is initialized locally and input files\r\n",
      "#    are parsed ready for cluster analaysis\r\n",
      "# 2. queenB launches instances, logs in to it and runs the\r\n",
      "#    droneB.sh script remotely.\r\n",
      "# 3. The droneB script is executed on the instance and it\r\n",
      "#    launches a `screen` on the instance and loads and \r\n",
      "#    starts to perform the $TASK (gather.sh) script.\r\n",
      "# 4. TASK script should include a instance shut-down\r\n",
      "#    command to close instance upon completion.\r\n",
      "#\r\n",
      "\r\n",
      "# Control Panel =========================\r\n",
      "# Amazon AWS S3 Home URL\r\n",
      "S3URL='s3://crownproject/tcga2'\r\n",
      "\r\n",
      "# EC2 TASK Script - script for droneB to execute\r\n",
      "TASK=\"$S3URL/scripts/hgr1_align_v3.tcga.sh\"\r\n",
      "\r\n",
      "# Parameter file:\r\n",
      "# Each line of PARAMETERS will be input to STDIN of\r\n",
      "# the droneB script which can then be used to run the\r\n",
      "# TASK script.\r\n",
      "# i.e. bash droneB.sh <line_N_of_PARAMETERS>\r\n",
      "# PARAMETERS=\"tcga0_input.txt\"\r\n",
      "PARAMETERS=$1\r\n",
      "\r\n",
      "# EC2 Set-up\r\n",
      "instanceTYPE='c4.xlarge'\r\n",
      "imageID='ami-0031fd61f932bdef9' #AMI TCGA\r\n",
      "\r\n",
      "devNAME='/dev/sda1' # /dev/sda1 for Crown-AMI\r\n",
      "volSIZE='120' # in Gb\r\n",
      "\r\n",
      "# Number of instances to launch\r\n",
      "#COUNT=2 # predetermined number\r\n",
      "COUNT=$(awk 'END{print NR}' $PARAMETERS) # for each input argument\r\n",
      "\r\n",
      "# Security\r\n",
      "keyNAME='CrownKey'\r\n",
      "keyPATH=\"/home/ec2-user/.ssh/CrownKey.pem\"\r\n",
      "secGROUP='crown-group'\r\n",
      "\r\n",
      "# Input Validation ======================\r\n",
      "# Test input file for duplicate sample ID (which will cause file collisions downstream)\r\n",
      "# exit with error 1 if collision exists\r\n",
      "\r\n",
      "SAMPLEIDS=$(cut -f1 $1 | uniq -d )\r\n",
      "\r\n",
      "if [ -n \"$SAMPLEIDS\" ]\r\n",
      "then\r\n",
      "  # If there are duplicate sample IDs exit\r\n",
      "  echo \"Error 02 - Duplicate Sample ID in input\"\r\n",
      "  echo \" add identifiers to replicate bams\"\r\n",
      "  echo \"\"\r\n",
      "  echo \"$SAMPLEIDS\"\r\n",
      "  exit 2 \r\n",
      "fi\r\n",
      "\r\n",
      "# Script Core ===========================\r\n",
      "\r\n",
      "for ITER in $(seq 1 $COUNT)\r\n",
      "do\r\n",
      "\r\n",
      "  # Count the total number of instances\r\n",
      "  running_instance=$(aws ec2 describe-instances |\\\r\n",
      "    grep '\"Name\": \"running\"' - |\\\r\n",
      "    wc -l - | cut -f1 -d' ' - )\r\n",
      "\r\n",
      "  while [ $running_instance -ge 25 ];\r\n",
      "  do\r\n",
      "\r\n",
      "  \techo \"Reached max of $running_instance, wait.\"\r\n",
      "  \techo ''\r\n",
      "\r\n",
      "  \t# If there are greater then 25 instances\r\n",
      "  \t# wait for instance to close\r\n",
      "  \tsleep 5m\r\n",
      "\r\n",
      "  \trunning_instance=$(aws ec2 describe-instances |\\\r\n",
      "    grep '\"Name\": \"running\"' - |\\\r\n",
      "    wc -l - | cut -f1 -d' ' - )\r\n",
      "  done\r\n",
      "\r\n",
      "  # Extract Parameters/Arguments ----------\r\n",
      "\r\n",
      "  ARGS=$(sed -n \"$ITER\"p $PARAMETERS | sed 's/\\t/ /g' - )\r\n",
      "\r\n",
      "  echo \"Launch instance # $ITER\"\r\n",
      "  date\r\n",
      "  echo \"Instance Type: $instanceTYPE\"\r\n",
      "  echo \"AMI Image: $imageID\"\r\n",
      "  echo \"Run Script: $TASK\"\r\n",
      "  echo \"Parameters: $ARGS\"\r\n",
      "\r\n",
      "  # Launch an instance --------------------\r\n",
      "  # NOTE: each iteration of the for loop launches one instance\r\n",
      "  # therefore each loop launches only one instance\r\n",
      "  aws ec2 run-instances --image-id $imageID --count 1 \\\r\n",
      "   --instance-type $instanceTYPE --key-name $keyNAME \\\r\n",
      "   --block-device-mappings DeviceName=$devNAME,Ebs={VolumeSize=$volSIZE} \\\r\n",
      "   --security-groups $secGROUP > launch.tmp\r\n",
      "\r\n",
      "  # Another alternative is to use --user-data droneB.sh \r\n",
      "  # which will run at instance boot-up\r\n",
      "  # passing arguments to it may be challenging\r\n",
      "\r\n",
      "  # Retrieve instance ID\r\n",
      "  instanceID=$(cat launch.tmp | \\\r\n",
      "    egrep -o -e 'InstanceId[\":/A-Za-z0-9_ \\\\-]*' - |\\\r\n",
      "    cut -f2 -d' ' - | xargs)\r\n",
      "\r\n",
      "  echo \"Instance ID: $instanceID\"\r\n",
      "\r\n",
      "\r\n",
      "  # Add a few minute wait here to allow for Public DNS to be assigned\r\n",
      "  # otherwise ssh doesn't work\r\n",
      "  sleep 180s\r\n",
      "\r\n",
      "  # Check that the instance is 'running' and not pending\r\n",
      "  pending_instance=$(aws ec2 describe-instances |\\\r\n",
      "   grep '\"Name\": \"pending\"' - |\\\r\n",
      "   wc -l - | cut -f1 -d' ' - )\r\n",
      "\r\n",
      "  while [ $pending_instance -gt 0 ];\r\n",
      "  do\r\n",
      "\r\n",
      "  \techo \"Launchign instance not ready yet, chill\"\r\n",
      "  \t# Instance still pending\r\n",
      "  \tsleep 180s\r\n",
      "\r\n",
      "    pending_instance=$(aws ec2 describe-instances |\\\r\n",
      "     grep '\"Name\": \"pending\"' - |\\\r\n",
      "     wc -l - | cut -f1 -d' ' - )\r\n",
      "  done\r\n",
      "\r\n",
      "\r\n",
      "  # Retrieve public DNS\r\n",
      "  aws ec2 describe-instances --instance-ids $instanceID > launch2.tmp\r\n",
      "\r\n",
      "  pubDNS=$(cat launch2.tmp | \\\r\n",
      "    egrep -o -m 1 -e 'PublicDnsName[.\":/A-Za-z0-9_ \\\\-]*' - |\\\r\n",
      "    cut -f2 -d' ' - | xargs)\r\n",
      "\r\n",
      "  echo \"Public DNS: $pubDNS\"\r\n",
      "\r\n",
      "  # Access the instance -------------------\r\n",
      "\r\n",
      "  LOGIN=\"ubuntu@$pubDNS\" \r\n",
      "\r\n",
      "  ssh -i $keyPATH \\\r\n",
      "    -o StrictHostKeyChecking=no \\\r\n",
      "    $LOGIN 'bash -s' < droneB.sh $TASK $(echo $ARGS)\r\n",
      "\r\n",
      "  # Cleanup\r\n",
      "  rm *.tmp\r\n",
      "\r\n",
      "  echo ''\r\n",
      "  echo ''\r\n",
      "\r\n",
      "done\r\n",
      "\r\n",
      "# end of script\r\n",
      "\r\n",
      "#!/bin/bash\r\n",
      "# droneB.sh\r\n",
      "#\r\n",
      "\r\n",
      "# This script-layer is neccesary to launch a screen session\r\n",
      "# on each ec2-machine. The pipeline is run within that session\r\n",
      "# and the output is logged. This allows 'looking in' on sessions\r\n",
      "# as they are running.\r\n",
      "\r\n",
      "# Commands to run on server-side\r\n",
      "# ===============================================================\r\n",
      "\r\n",
      "SCRIPTPATH=$1\r\n",
      "\r\n",
      "SCRIPT=$(basename $1)\r\n",
      "\r\n",
      "shift # drop first (TASK or SCRIPT variable)\r\n",
      "\r\n",
      "# Download pipeline / droneB's function\r\n",
      "  aws s3 cp $SCRIPTPATH ./\r\n",
      "\r\n",
      "  chmod 777 *.sh\r\n",
      "\r\n",
      "# open screen; run gather.sh function. -L logged\r\n",
      "  screen -Ldmt sh ~/$SCRIPT $@\r\n",
      "\r\n",
      "\r\n",
      "# ===============================================================\r\n",
      "\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "cd $WORKDIR\n",
    "# Echo \n",
    "\n",
    "cat hgr1_align_v3.tcga.sh\n",
    "echo \n",
    "echo\n",
    "cat queenB.sh\n",
    "echo \n",
    "echo\n",
    "cat droneB.sh\n",
    "echo \n",
    "echo "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Results - TCGA2 Pilot Run\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Completed 4.2 KiB/4.2 KiB with 1 file(s) remaining\r",
      "upload: ./queenB.sh to s3://crownproject/tcga2/scripts/queenB.sh\r\n",
      "Completed 657 Bytes/657 Bytes with 1 file(s) remaining\r",
      "upload: ./droneB.sh to s3://crownproject/tcga2/scripts/droneB.sh\r\n",
      "Completed 6.5 KiB/6.5 KiB with 1 file(s) remaining\r",
      "upload: ./hgr1_align_v3.tcga.sh to s3://crownproject/tcga2/scripts/hgr1_align_v3.tcga.sh\r\n",
      "Completed 389 Bytes/389 Bytes with 1 file(s) remaining\r",
      "upload: ./input_tcga2.txt to s3://crownproject/tcga2/scripts/input_tcga2.txt\r\n",
      "Completed 1.0 KiB/1.0 KiB with 1 file(s) remaining\r",
      "upload: ../../gdc.token.txt to s3://crownproject/tcga2/scripts/gdc.token\r\n"
     ]
    }
   ],
   "source": [
    "# Local Folder Operations -----------------------------\n",
    "# LOCAL:\n",
    "cd $WORKDIR\n",
    "\n",
    "#NOTE For pilot run, AWS s3 shutdown commented out. Re-upload hgr1 script upon full run\n",
    "\n",
    "aws s3 cp queenB.sh $S3URL/scripts/\n",
    "aws s3 cp droneB.sh $S3URL/scripts/\n",
    "aws s3 cp hgr1_align_v3.tcga.sh $S3URL/scripts/\n",
    "aws s3 cp $INPUT $S3URL/scripts/\n",
    "aws s3 cp ../../gdc.token.txt $S3URL/scripts/gdc.token\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Completed 3.7 KiB/3.7 KiB with 1 file(s) remaining\r",
      "download: s3://crownproject/tcga2/logs/tcga2_pilot.log to ./tcga2_pilot.log\r\n",
      "\u001bkec2-user@ip-172-31-31-231:~\u001b\\\u001b[?1034h[ec2-user@ip-172-31-31-231 ~]$ ls\r",
      "\r\n",
      "droneB.sh  gdc.token  hgr1_align_v3.tcga.sh  input_tcga2.txt  queenB.sh  screenlog.0\r",
      "\r\n",
      "\u001bkec2-user@ip-172-31-31-231:~\u001b\\[ec2-user@ip-172-31-31-231 ~]$ bash queenB.sh input_tcga2.txt\r",
      "\r\n",
      "Launch instance # 1\r",
      "\r\n",
      "Mon May  6 19:37:19 UTC 2019\r",
      "\r\n",
      "Instance Type: c4.xlarge\r",
      "\r\n",
      "AMI Image: ami-0031fd61f932bdef9\r",
      "\r\n",
      "Run Script: s3://crownproject/tcga2/scripts/hgr1_align_v3.tcga.sh\r",
      "\r\n",
      "Parameters: TCGA-44-2662-01Aa TCGA-LUAD 10cdc012-492d-4090-83ab-ff73ac0ee536\r",
      "\r\n",
      "Instance ID: i-06cea275db49daf32\r",
      "\r\n",
      "Public DNS: ec2-18-236-102-47.us-west-2.compute.amazonaws.com\r",
      "\r\n",
      "Warning: Permanently added 'ec2-18-236-102-47.us-west-2.compute.amazonaws.com,172.31.20.76' (ECDSA) to the list of known hosts.\r",
      "\r",
      "\r\n",
      "download: s3://crownproject/tcga2/scripts/hgr1_align_v3.tcga.sh to ./hgr1_align_v3.tcga.sh\r",
      "\r\n",
      "\r",
      "\r\n",
      "\r",
      "\r\n",
      "Launch instance # 2\r",
      "\r\n",
      "Mon May  6 19:40:26 UTC 2019\r",
      "\r\n",
      "Instance Type: c4.xlarge\r",
      "\r\n",
      "AMI Image: ami-0031fd61f932bdef9\r",
      "\r\n",
      "Run Script: s3://crownproject/tcga2/scripts/hgr1_align_v3.tcga.sh\r",
      "\r\n",
      "Parameters: TCGA-44-2662-01Ab TCGA-LUAD 320b2265-20e7-40a8-a554-f9231f6ea914\r",
      "\r\n",
      "Instance ID: i-010c25d314e03e73d\r",
      "\r\n",
      "Public DNS: ec2-54-200-177-86.us-west-2.compute.amazonaws.com\r",
      "\r\n",
      "Warning: Permanently added 'ec2-54-200-177-86.us-west-2.compute.amazonaws.com,172.31.22.191' (ECDSA) to the list of known hosts.\r",
      "\r",
      "\r\n",
      "download: s3://crownproject/tcga2/scripts/hgr1_align_v3.tcga.sh to ./hgr1_align_v3.tcga.sh\r",
      "\r\n",
      "\r",
      "\r\n",
      "\r",
      "\r\n",
      "Launch instance # 3\r",
      "\r\n",
      "Mon May  6 19:43:35 UTC 2019\r",
      "\r\n",
      "Instance Type: c4.xlarge\r",
      "\r\n",
      "AMI Image: ami-0031fd61f932bdef9\r",
      "\r\n",
      "Run Script: s3://crownproject/tcga2/scripts/hgr1_align_v3.tcga.sh\r",
      "\r\n",
      "Parameters: TCGA-BL-A13J-01Aa TCGA-BLCA fd1ec26a-64e6-4bc4-b923-182bcf3446a0\r",
      "\r\n",
      "Instance ID: i-040cf7268f05773ba\r",
      "\r\n",
      "Public DNS: ec2-34-222-32-155.us-west-2.compute.amazonaws.com\r",
      "\r\n",
      "Warning: Permanently added 'ec2-34-222-32-155.us-west-2.compute.amazonaws.com,172.31.18.182' (ECDSA) to the list of known hosts.\r",
      "\r",
      "\r\n",
      "download: s3://crownproject/tcga2/scripts/hgr1_align_v3.tcga.sh to ./hgr1_align_v3.tcga.sh\r",
      "\r\n",
      "\r",
      "\r\n",
      "\r",
      "\r\n",
      "Launch instance # 4\r",
      "\r\n",
      "Mon May  6 19:46:45 UTC 2019\r",
      "\r\n",
      "Instance Type: c4.xlarge\r",
      "\r\n",
      "AMI Image: ami-0031fd61f932bdef9\r",
      "\r\n",
      "Run Script: s3://crownproject/tcga2/scripts/hgr1_align_v3.tcga.sh\r",
      "\r\n",
      "Parameters: TCGA-BL-A13J-01Ab TCGA-BLCA 26a1a274-2b68-4d22-8942-321c718e5bd7\r",
      "\r\n",
      "Instance ID: i-0f801df17b5bccca5\r",
      "\r\n",
      "Public DNS: ec2-52-40-67-153.us-west-2.compute.amazonaws.com\r",
      "\r\n",
      "Warning: Permanently added 'ec2-52-40-67-153.us-west-2.compute.amazonaws.com,172.31.27.52' (ECDSA) to the list of known hosts.\r",
      "\r",
      "\r\n",
      "download: s3://crownproject/tcga2/scripts/hgr1_align_v3.tcga.sh to ./hgr1_align_v3.tcga.sh\r",
      "\r\n",
      "\r",
      "\r\n",
      "\r",
      "\r\n",
      "Launch instance # 5\r",
      "\r\n",
      "Mon May  6 19:49:52 UTC 2019\r",
      "\r\n",
      "Instance Type: c4.xlarge\r",
      "\r\n",
      "AMI Image: ami-0031fd61f932bdef9\r",
      "\r\n",
      "Run Script: s3://crownproject/tcga2/scripts/hgr1_align_v3.tcga.sh\r",
      "\r\n",
      "Parameters: TCGA-44-6147-01Aa TCGA-LUAD 16dfa0e8-e7cf-4589-bd99-82944d703695\r",
      "\r\n",
      "Instance ID: i-0cf7a9d9dacc77848\r",
      "\r\n",
      "Public DNS: ec2-54-244-3-125.us-west-2.compute.amazonaws.com\r",
      "\r\n",
      "Warning: Permanently added 'ec2-54-244-3-125.us-west-2.compute.amazonaws.com,172.31.24.135' (ECDSA) to the list of known hosts.\r",
      "\r",
      "\r\n",
      "download: s3://crownproject/tcga2/scripts/hgr1_align_v3.tcga.sh to ./hgr1_align_v3.tcga.sh\r",
      "\r\n",
      "\r",
      "\r\n",
      "\r",
      "\r\n",
      "Launch instance # 6\r",
      "\r\n",
      "Mon May  6 19:53:00 UTC 2019\r",
      "\r\n",
      "Instance Type: c4.xlarge\r",
      "\r\n",
      "AMI Image: ami-0031fd61f932bdef9\r",
      "\r\n",
      "Run Script: s3://crownproject/tcga2/scripts/hgr1_align_v3.tcga.sh\r",
      "\r\n",
      "Parameters: TCGA-44-6147-01Ab TCGA-LUAD 056c3eb5-5317-4cb2-9f33-0e93b4e54be5\r",
      "\r\n",
      "Instance ID: i-02b70709bd0b30e98\r",
      "\r\n",
      "Public DNS: ec2-34-221-181-210.us-west-2.compute.amazonaws.com\r",
      "\r\n",
      "Warning: Permanently added 'ec2-34-221-181-210.us-west-2.compute.amazonaws.com,172.31.25.111' (ECDSA) to the list of known hosts.\r",
      "\r",
      "\r\n",
      "download: s3://crownproject/tcga2/scripts/hgr1_align_v3.tcga.sh to ./hgr1_align_v3.tcga.sh\r",
      "\r\n",
      "\r",
      "\r\n",
      "\r",
      "\r\n",
      "\u001bkec2-user@ip-172-31-31-231:~\u001b\\[ec2-user@ip-172-31-31-231 ~]$ "
     ]
    }
   ],
   "source": [
    "# Remote EC2 Instance Operations ----------------------\n",
    "\n",
    "# Remote:\n",
    "# Manually open an Amazon Linux 2 AMI\n",
    "# ami-061392db613a6357b\n",
    "# t2.micro\n",
    "#\n",
    "# ssh login:\n",
    "# ssh -i \"crown.pem\" ec2-user@PUBLICDNS\n",
    "#\n",
    "\n",
    "# Commands on EC2 machine to set-up AWS\n",
    "# enter personal login info:\n",
    "\n",
    "# REMOTE:\n",
    "#aws configure\n",
    "  # AWS Key ID\n",
    "  # AWS Secret Key ID\n",
    "  # Region: us-west-2\n",
    "  \n",
    "# Copy local run files to S3 and download them on EC2\n",
    "\n",
    "# REMOTE:\n",
    "# aws s3 cp --recursive s3://crownproject/tcga2/scripts/ ./\n",
    "#\n",
    "# mv <KEY>.pem ~/.ssh/\n",
    "# chmod 400 ~/.ssh/<KEY>.pem\n",
    "\n",
    "# REMOTE:\n",
    "# Open logging screen and being launchign EC2 instances\n",
    "# screen -L\n",
    "# \n",
    "# bash queenB.sh input_tcga2.txt\n",
    "#\n",
    "# aws s3 cp screenlog.0 s3://crownproject/tcga2/logs/tcga2_pilot.log\n",
    "\n",
    "aws s3 cp s3://crownproject/tcga2/logs/tcga2_pilot.log ./\n",
    "cat tcga2_pilot.log\n",
    "\n",
    "# Run completed successfully"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Discussion\n",
    "\n",
    "### TCGA-44-2662\n",
    "The variant did not replicate at 2407 in the other sample in this patient.\n",
    "\n",
    "```\n",
    "Replicate 01Aa\n",
    "chr13:1,010,354\n",
    "<hr>Total count: 729118\n",
    "A      : 675  (0%,     584+,   91- )\n",
    "C      : 18901  (3%,     12835+,   6066- )\n",
    "G      : 708504  (97%,     428912+,   279592- )\n",
    "T      : 600  (0%,     415+,   185- )\n",
    "N      : 438  (0%,     392+,   46- )\n",
    "---------------\n",
    "DEL: 0\n",
    "INS: 130\n",
    "```\n",
    "```\n",
    "Replicate 01Ab\n",
    "chr13:1,010,354\n",
    "<hr>Total count: 22765\n",
    "A      : 143  (1%,     139+,   4- )\n",
    "C      : 160  (1%,     132+,   28- )\n",
    "G      : 22315  (98%,     18903+,   3412- )\n",
    "T      : 50  (0%,     22+,   28- )\n",
    "N      : 97  (0%,     68+,   29- )\n",
    "---------------\n",
    "DEL: 0\n",
    "INS: 1\n",
    "```\n",
    "\n",
    "```\n",
    "Replicate 01B\n",
    "chr13:1,010,354\n",
    "<hr>Total count: 4310\n",
    "A      : 0\n",
    "C      : 731  (17%,     482+,   249- )\n",
    "G      : 3570  (83%,     1672+,   1898- )\n",
    "T      : 9  (0%,     0+,   9- )\n",
    "N      : 0\n",
    "---------------\n",
    "```\n",
    "\n",
    "```\n",
    "Replicate 11A (normal)\n",
    "chr13:1,010,354\n",
    "<hr>Total count: 9251\n",
    "A      : 7  (0%,     6+,   1- )\n",
    "C      : 7  (0%,     4+,   3- )\n",
    "G      : 9217  (100%,     7425+,   1792- )\n",
    "T      : 8  (0%,     4+,   4- )\n",
    "N      : 12  (0%,     7+,   5- )\n",
    "---------------\n",
    "```\n",
    "\n",
    "Exploring reasons or differences between samples on the GDC website I noted that 01Ax is not FFPE slide while 01B is an FFPE slide. This was also true for the other three samples with this variant at 2407. This may perhaps explain some of the variance observed for the alternative allele.\n",
    "\n",
    "\n",
    "### FFPE Samples\n",
    "`/home/artem/Desktop/Crown/data2/tcga2_pilot/tcga-ffpe`\n",
    "\n",
    "I was able to filter for all FFPE samples in TCGA cohorts on the GDC using the filter:\n",
    "` cases.samples.is_ffpe in [true] and files.data_format in [\"BAM\"] and files.data_type in [\"Aligned Reads\"] and files.experimental_strategy in [\"RNA-Seq\"] `\n",
    "\n",
    "Which yielded the following list of files:\n",
    "```\n",
    "Sample ID\n",
    "TCGA-BL-A13I-01B\n",
    "TCGA-A7-A26I-01B\n",
    "TCGA-AC-A3QQ-01B\n",
    "TCGA-PL-A8LX-01A\n",
    "TCGA-A7-A26E-01B\n",
    "TCGA-A6-2684-01C\n",
    "TCGA-A7-A26F-01B\n",
    "TCGA-BK-A0CA-01B\n",
    "TCGA-BK-A26L-01C\n",
    "TCGA-AC-A2QH-01B\n",
    "TCGA-A7-A13G-01B\n",
    "TCGA-PL-A8LY-01A\n",
    "TCGA-A7-A0DB-01C\n",
    "TCGA-44-4112-01B\n",
    "TCGA-A7-A0DC-01B\n",
    "TCGA-44-3918-01B\n",
    "TCGA-AC-A3OD-01B\n",
    "TCGA-44-2666-01B\n",
    "TCGA-44-6775-01C\n",
    "TCGA-44-6147-01B\n",
    "TCGA-BK-A0CC-01B\n",
    "TCGA-A6-2674-01B\n",
    "TCGA-44-6146-01B\n",
    "TCGA-A6-2677-01B\n",
    "TCGA-A6-5665-01B\n",
    "TCGA-PL-A8LZ-01A\n",
    "TCGA-HC-8258-01B\n",
    "TCGA-A6-6780-01B\n",
    "TCGA-B2-5633-01B\n",
    "TCGA-B2-3923-01B\n",
    "TCGA-BL-A13J-01B\n",
    "TCGA-A7-A13D-01B\n",
    "TCGA-B2-3924-01B\n",
    "TCGA-A7-A26J-01B\n",
    "TCGA-A6-3810-01B\n",
    "TCGA-A6-5659-01B\n",
    "TCGA-A6-3809-01B\n",
    "TCGA-44-2656-01B\n",
    "TCGA-HC-8265-01B\n",
    "TCGA-44-2662-01B\n",
    "TCGA-44-2665-01B\n",
    "TCGA-PL-A8LV-01A\n",
    "TCGA-A6-2672-01B\n",
    "TCGA-A6-6781-01B\n",
    "TCGA-B2-5635-01B\n",
    "TCGA-PN-A8MA-01A\n",
    "TCGA-BK-A139-01C\n",
    "TCGA-A7-A13E-01B\n",
    "TCGA-44-2668-01B\n",
    "TCGA-A6-6650-01B\n",
    "TCGA-A6-5661-01B\n",
    "TCGA-HC-7740-01B\n",
    "TCGA-BL-A0C8-01B\n",
    "TCGA-44-3917-01B\n",
    "TCGA-44-5645-01B\n",
    "TCGA-A6-5656-01B\n",
    "```\n",
    "\n",
    "Note that all of except 3 (all of which are `TCGA-PL`) of these files are `01B/C` replicates and not `01A` replicates. All further initial variant analyses should focus on comparing `11A` normals to `01A` tumour samples. It is currently unknown how many of these samples are included in the previous analysis cohorts but moving forward all analysis will be on strictly NON-FFPE samples only.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### TCGA Run 2 List\n",
    "\n",
    "The pipeline is working as it's set-up. Open up the analysis to entire cohorts of patients, avoid repeating samples which have already been performed.\n",
    "\n",
    "**Master List COAD:**\n",
    "`cases.project.project_id in [\"TCGA-COAD\"] and files.data_format in [\"BAM\"] and files.data_type in [\"Aligned Reads\"] and files.experimental_strategy in [\"RNA-Seq\"] and cases.samples.is_ffpe in [false]`\n",
    "\n",
    "**List of all previously aligned files (1420 files):**\n",
    "`~/Crown/data2/tcga2_pilot/tcga1_filelist.bamlist`"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
