{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Initial Variant Calling - total RNA (Adaconis)\n",
    "```\n",
    "pi:ababaian\n",
    "start: 2016 04 28\n",
    "complete : 2016 05 04\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "## Objective\n",
    "Do a first pass analysis of the variants, get a rough idea of which positions to look at and then work on methods to 'better' describe a variant"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Materials and Methods\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# Move to working directory\n",
    "cd ~/Crown/data/adaconis/totalRNA1\n",
    "mkdir -p vcf\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "\n",
    "# samtools mpileup -uf ref.fa aln1.bam aln2.bam | bcftools view -bvcg - > var.raw.bcf  \n",
    "#  bcftools view var.raw.bcf | vcfutils.pl varFilter -D100 > var.flt.vcf  \n",
    "\n",
    "samtools mpileup -f ~/Crown/resources/hgr/hgr.fa -d 10000 -uv accepted_hits.bam > vcf/total1.uv.vcf"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Results\n",
    "\n",
    "VCF is quite detailed and not super useful at the moment. I'll go over the results using IGV and simply look at positions which are obviously variable. Low hanging fruit by IGV inspection\n",
    "\n",
    "### Total RNA 2 - 18S Variants\n",
    "\n",
    "#### U1248C | CU1248--\n",
    "TotalRNA2\n",
    "```\n",
    "chr13:1,004,904\n",
    "Total count: 118194\n",
    "A      : 1749  (1%,     1366+,   383- )\n",
    "C      : 32288  (27%,     20271+,   12017- )\n",
    "G      : 537  (0%,     411+,   126- )\n",
    "T      : 83620  (71%,     39198+,   44422- )\n",
    "N      : 0\n",
    "DEL: 4083\n",
    "INS: 0\n",
    "```\n",
    "TotalRNA1 \n",
    "```\n",
    "chr13:1,004,904\n",
    "Total count: 129476\n",
    "A      : 1943  (2%,     1530+,   413- )\n",
    "C      : 35441  (27%,     21938+,   13503- )\n",
    "G      : 568  (0%,     416+,   152- )\n",
    "T      : 91524  (71%,     42285+,   49239- )\n",
    "N      : 0\n",
    "---------------\n",
    "DEL: 4470\n",
    "INS: 0\n",
    "```\n",
    "![U1248C Conservation](../figure/20160428_18S_U1248C.png)\n",
    "- U1248 is \"C\" in two other species; not perfectly evolutionary conserved\n",
    "\n",
    "- Also present is a deletion of C1247 and U124 at about 3-4%\n",
    "\n",
    "- 4UG0 Sequence: ...UUUGAC**U**CAACACGGG...\n",
    "\n",
    "![U1248C Secondary Structure](../figure/20160502_secondary_18S_U1248C.png)\n",
    "   \n",
    "- Helix 31: UUUG AC**U**CAACA CGGG\n",
    "\n",
    "- There aren't any base-stacking or base-interactions with U1248 from the Petrov SSU (RiboZone) structural data.\n",
    "\n",
    "- While helix 30 assocaites with eIF1 below P-site for start-codon selection\n",
    "\n",
    "This likely means that this variant is a 'normal variant', possibly present in many species at a polymorphic level. If it's a mutation then it's incredibly interesting to see that it 'reverts' or 'converts' to a form (C) that is present in some other species like Lung Fish. This could mean that the 'deregulation' occurs within an acceptable mutation-space for 18S. Needs human DNA data from normal people to confirm or refute. Worth following up on.\n",
    "\n",
    "Note: This position may be a hyper-modified pseudouridine which means that the \"variant\" being observed could be an artifact of faulty reverse transcription at such a base.\n",
    "\n",
    "#### -1638G or G1638-\n",
    "```\n",
    "chr13:1,005,294\n",
    "Total count: 36962\n",
    "A      : 490  (1%,     162+,   328- )\n",
    "C      : 45  (0%,     0+,   45- )\n",
    "G      : 36369  (98%,     194+,   36175- )\n",
    "T      : 31  (0%,     1+,   30- )\n",
    "N      : 27  (0%,     1+,   26- )\n",
    "---------------\n",
    "DEL: 111134\n",
    "INS: 0\n",
    "```\n",
    "- Secondary Structure \n",
    "Note: In Petrov Alignments this 'G' at 1638 is absent. It looks like the reference sequence and this one are not the same. It will be worth aligning different rRNA sequences from different sources to check these. The above positions are from Petrov alignment for human\n",
    "\n",
    "#### G1647-\n",
    "```\n",
    "chr13:1,005,304\n",
    "Total count: 14953\n",
    "A      : 11640  (78%,     222+,   11418- )\n",
    "C      : 3081  (21%,     705+,   2376- )\n",
    "G      : 180  (1%,     12+,   168- )\n",
    "T      : 40  (0%,     1+,   39- )\n",
    "N      : 12  (0%,     0+,   12- )\n",
    "---------------\n",
    "DEL: 86629\n",
    "INS: 0\n",
    "```\n",
    "\n",
    "### Total RNA 2 - 28S Variants\n",
    "chr:1,007,935\n",
    "\n",
    "####  A60G (A217G)\n",
    "Note: In Petrov LSU alignment, this base is a 'G'.\n",
    "In the Petrov secondary structure this 'G' is at position 60.\n",
    "In the genome browswer this is at position 60 of the 28S\n",
    "```\n",
    "chr13:1,007,994\n",
    "Total count: 997220\n",
    "A      : 635482  (64%,     632964+,   2518- )\n",
    "C      : 117  (0%,     117+,   0- )\n",
    "G      : 361491  (36%,     360145+,   1346- )\n",
    "T      : 101  (0%,     100+,   1- )\n",
    "N      : 29  (0%,     29+,   0- )\n",
    "---------------\n",
    "DEL: 8\n",
    "INS: 3\n",
    "```\n",
    "\n",
    "#### G772 (?)\n",
    "'G' insertion at 772\n",
    "```\n",
    "chr13:1,008,706\n",
    "Total count: 5325\n",
    "A      : 1  (0%,     1+,   0- )\n",
    "C      : 5262  (99%,     3941+,   1321- )\n",
    "G      : 62  (1%,     4+,   58- )\n",
    "T      : 0\n",
    "N      : 0\n",
    "---------------\n",
    "DEL: 0\n",
    "INS: 1301\n",
    "```\n",
    "```\n",
    "chr13:1,008,707\n",
    "Total count: 4968\n",
    "A      : 4  (0%,     1+,   3- )\n",
    "C      : 20  (0%,     15+,   5- )\n",
    "G      : 536  (11%,     323+,   213- )\n",
    "T      : 4408  (89%,     3298+,   1110- )\n",
    "N      : 0\n",
    "---------------\n",
    "DEL: 0\n",
    "INS: 1300\n",
    "```\n",
    "\n",
    "#### A1479T\n",
    "\n",
    "- A1322 on Petrov Structure\n",
    "- Yeas A645 or A1309 by [Waku et al.,](http://www.ncbi.nlm.nih.gov/pubmed/?term=27149924)\n",
    "- N1 - MethylAdenosine modification at this position according to PMID:27149924\n",
    "- Nucleomethylin mediated modification; deficiency inhibits cell proliferation.\n",
    "- Waku also shows methylation at A1136 by their numbering; should be ~**A1652** on my alignment\n",
    "\n",
    "\n",
    "```\n",
    "chr13:1,009,243\n",
    "Total count: 6341\n",
    "A      : 4779  (75%,     3296+,   1483- )\n",
    "C      : 36  (1%,     34+,   2- )\n",
    "G      : 360  (6%,     304+,   56- )\n",
    "T      : 1166  (18%,     1015+,   151- )\n",
    "N      : 0\n",
    "---------------\n",
    "DEL: 24\n",
    "INS: 0\n",
    "```\n",
    "- A1322 on secondary structure\n",
    "- Helix 25a: CCGU CUUG**A**AAC ACGG\n",
    "![A1479T Consevartion](../figure/20160504_A1479_Conservation.png)\n",
    "\n",
    "\n",
    "#### G4641A\n",
    "```\n",
    "chr13:1,012,388\n",
    "Total count: 143874\n",
    "A      : 143382  (100%,     97285+,   46097- )\n",
    "C      : 274  (0%,     69+,   205- )\n",
    "G      : 110  (0%,     35+,   75- )\n",
    "T      : 104  (0%,     17+,   87- )\n",
    "N      : 4  (0%,     3+,   1- )\n",
    "---------------\n",
    "DEL: 6\n",
    "INS: 0\n",
    "```\n",
    " - 100% of rRNA is A-variant\n",
    " - Petrov alignment sequence is 'A' not 'G'\n",
    " - In some species like Yeast, it's a G, human/chimp it's A\n",
    " - Secondary Structure: 4484 Helix 91: GUGAA GCAGAA UUC**A**C\n",
    "![AG4641A Conservation](../figure/20160504_G4641A_Conservation.png)\n",
    "\n",
    "\n",
    "#### T4687A/G\n",
    "```\n",
    "chr13:1,012,434\n",
    "Total count: 163870\n",
    "A      : 31777  (19%,     20843+,   10934- )\n",
    "C      : 779  (0%,     571+,   208- )\n",
    "G      : 26071  (16%,     16590+,   9481- )\n",
    "T      : 105243  (64%,     69732+,   35511- )\n",
    "N      : 0\n",
    "---------------\n",
    "DEL: 2063\n",
    "INS: 4\n",
    "```\n",
    "- Secondary Structure U4530, between helix 90,93: 90GGG **U**UUA GACC93\n",
    "- Strong evolutionary conservation (perfect) around this region\n",
    "![U4687A Conservation](../figure/20160504_U4687A_Conservation.png)\n",
    "\n",
    "#### C5063U (Expansion Loop Area)\n",
    "```\n",
    "chr13:1,012,805\n",
    "Total count: 445143\n",
    "A      : 272  (0%,     1+,   271- )\n",
    "C      : 267568  (60%,     1557+,   266011- )\n",
    "G      : 944  (0%,     0+,   944- )\n",
    "T      : 176357  (40%,     1114+,   175243- )\n",
    "N      : 2  (0%,     0+,   2- )\n",
    "---------------\n",
    "DEL: 0\n",
    "INS: 4\n",
    "```\n",
    "\n",
    "- Secondary Structure Stem 98ES39b: C4906 \n",
    "\n",
    "\n",
    "#### Variations in CCG / poly-C / simple / expansion Loops\n",
    "U1025C (?)\n",
    "```\n",
    "chr13:1,008,804\n",
    "Total count: 16469\n",
    "A      : 10  (0%,     9+,   1- )\n",
    "C      : 15396  (93%,     4558+,   10838- )\n",
    "G      : 1049  (6%,     464+,   585- )\n",
    "T      : 14  (0%,     7+,   7- )\n",
    "N      : 0\n",
    "---------------\n",
    "DEL: 2\n",
    "INS: 8\n",
    "\n",
    "chr13:1,008,921\n",
    "Total count: 8594\n",
    "A      : 5788  (67%,     2524+,   3264- )\n",
    "C      : 2623  (31%,     208+,   2415- )\n",
    "G      : 131  (2%,     2+,   129- )\n",
    "T      : 52  (1%,     2+,   50- )\n",
    "N      : 0\n",
    "---------------\n",
    "DEL: 2\n",
    "INS: 2\n",
    "\n",
    "chr13:1,008,935\n",
    "Total count: 13497\n",
    "A      : 56  (0%,     34+,   22- )\n",
    "C      : 2656  (20%,     12+,   2644- )\n",
    "G      : 171  (1%,     15+,   156- )\n",
    "T      : 10614  (79%,     2543+,   8071- )\n",
    "N      : 0\n",
    "---------------\n",
    "DEL: 1\n",
    "INS: 2\n",
    "\n",
    "chr13:1,008,953\n",
    "Total count: 25663\n",
    "A      : 61  (0%,     2+,   59- )\n",
    "C      : 4468  (17%,     7+,   4461- )\n",
    "G      : 208  (1%,     6+,   202- )\n",
    "T      : 20926  (82%,     2948+,   17978- )\n",
    "N      : 0\n",
    "---------------\n",
    "DEL: 0\n",
    "INS: 1\n",
    "\n",
    "chr13:1,009,035\n",
    "Total count: 8884\n",
    "A      : 89  (1%,     0+,   89- )\n",
    "C      : 1321  (15%,     14+,   1307- )\n",
    "G      : 50  (1%,     2+,   48- )\n",
    "T      : 7424  (84%,     1885+,   5539- )\n",
    "N      : 0\n",
    "---------------\n",
    "DEL: 4\n",
    "INS: 49\n",
    "\n",
    "chr13:1,010,959\n",
    "Total count: 11440\n",
    "A      : 8817  (77%,     1166+,   7651- )\n",
    "C      : 1364  (12%,     14+,   1350- )\n",
    "G      : 1222  (11%,     241+,   981- )\n",
    "T      : 37  (0%,     0+,   37- )\n",
    "N      : 0\n",
    "---------------\n",
    "DEL: 0\n",
    "INS: 1\n",
    "```\n",
    "\n",
    "CG --> CG to make poly-CGG\n",
    "```\n",
    "chr13:1,011,149\n",
    "Total count: 2461\n",
    "A      : 0\n",
    "C      : 10  (0%,     0+,   10- )\n",
    "G      : 2442  (99%,     82+,   2360- )\n",
    "T      : 9  (0%,     0+,   9- )\n",
    "N      : 0\n",
    "---------------\n",
    "chr13:1,011,150\n",
    "Total count: 2497\n",
    "A      : 2  (0%,     1+,   1- )\n",
    "C      : 2451  (98%,     90+,   2361- )\n",
    "G      : 39  (2%,     0+,   39- )\n",
    "T      : 5  (0%,     0+,   5- )\n",
    "N      : 0\n",
    "---------------\n",
    "DEL: 0\n",
    "INS: 1\n",
    "\n",
    "chr13:1,012,038\n",
    "Total count: 25792\n",
    "A      : 22000  (85%,     4177+,   17823- )\n",
    "C      : 411  (2%,     35+,   376- )\n",
    "G      : 3275  (13%,     0+,   3275- )\n",
    "T      : 93  (0%,     0+,   93- )\n",
    "N      : 13  (0%,     0+,   13- )\n",
    "---------------\n",
    "\n",
    "chr13:1,012,057\n",
    "Total count: 31855\n",
    "A      : 98  (0%,     1+,   97- )\n",
    "C      : 20147  (63%,     2023+,   18124- )\n",
    "G      : 11295  (35%,     101+,   11194- )\n",
    "T      : 315  (1%,     8+,   307- )\n",
    "N      : 0\n",
    "---------------\n",
    "DEL: 3\n",
    "INS: 167\n",
    "```\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Discussion\n",
    "\n",
    "### Variant Summary\n",
    "The most interesting variant is the 18S U1248C in helix 31. From E. coli, this position.\n",
    "\n",
    "Many variants occured in the so called Expansion Loops, especially at 'CGG' positions. These seem to be in general quite variable and may represent a good chunk of the 'difficult' parts of rRNA to sequence. I won't focus all that much on them initially since if a mutation is oncogenic it probably will be variable at a more \"important\" place.\n",
    "\n",
    "\n",
    "### The Irony\n",
    "\n",
    "I've essentially shalved this project since before I found the variation too difficult to deal with. The irony is that in 2013 my 'sketchier' analysis where I was less confident in the results (based on mRNA-seq), the example I was dealing with was r.U1248 or 18S U1248, the same position I \"found\" from the initial analysis. Direct quote from May 2013 below\n",
    "\n",
    "```\n",
    "May 13 2013\n",
    " \tThis is cool because it means that unless you're certain there\n",
    "\tare variants then automated processes will miss this.\n",
    "\n",
    "Example: Position r.4904 (chr13:1004904)\n",
    "\tIn the manual annotation I had\n",
    "                r.4904 del      0.036   *\n",
    "                r.4904 T > C    0.34    *\n",
    "                r.4904 T > A    0.030   *\n",
    "```\n",
    "\n",
    "and May 25th 2013\n",
    "```\n",
    "In the human Ribosomal structure paper by Anger et al,. (nature, may 2013)\n",
    "\n",
    "I have looked at position r.4904 which is variable in Hodgkin's lymphoma and\n",
    "K562 to see how it fits into the structure of the ribosome.\n",
    "\t\n",
    "\t1) The paper refers to this position as 18S 1248 U\n",
    "\t2) 1248 U is within stem-loop H31 of the 18S human rRNA\n",
    "\t3) 18S:1246-1248 [ACU] interact with residues 149,151 of SERPB1 (serpine1)\n",
    "\n",
    "I'm not sure as of right now how the U > C change of the rRNA will effect the interaction\n",
    "but if it's recurently there then that may suggest significance.\n",
    "I need to run the normal B-cells immediatly and see if the variation is absent in\n",
    "those cells. Anger used human peripheral blood as the source of 80S ribosome\n",
    "so we may be on to something!!!!!!\n",
    "```\n",
    "\n",
    "### IGV at Colorectal Cancer\n",
    "\n",
    "For LIONS I have access to some colorectal cancer transcriptomes from a patient (opposed to cell line). Since hg19r is the standard transcriptome I use, I have information about this position there.\n",
    "```\n",
    "chr13:1,004,904\n",
    "Total count: 53585\n",
    "A      : 2397  (4%,     1159+,   1238- )\n",
    "C      : 23565  (44%,     11707+,   11858- )\n",
    "G      : 3402  (6%,     1495+,   1907- )\n",
    "T      : 24221  (45%,     11718+,   12503- )\n",
    "N      : 0\n",
    "---------------\n",
    "DEL: 2188\n",
    "INS: 1\n",
    "```\n",
    "Which is also variable but at a higher level (45% C) and there are 4% and 6% variants of A and G. This isn't background noise, when you look at variation in the immediatly surrounding area it's substantially lower.\n",
    "\n",
    "![Colorectal Cancer. 587294.bam 18S U1248 in IGV](../figure/20160504_Colorectal_18S_U1248.png)\n",
    "\n",
    "### IGV at Hodgkins\n",
    "\n",
    "I have A05254r.bam on my local computer. Took a look on IGV. Same general pattern, only a bit lower C at 30%.\n",
    "```\n",
    "chr13:1,004,904\n",
    "Total count: 12174\n",
    "A      : 339  (3%,     186+,   153- )\n",
    "C      : 3704  (30%,     1731+,   1973- )\n",
    "G      : 463  (4%,     237+,   226- )\n",
    "T      : 7667  (63%,     3682+,   3985- )\n",
    "N      : 1  (0%,     1+,   0- )\n",
    "---------------\n",
    "DEL: 504\n",
    "INS: 0\n",
    "```\n",
    "\n",
    "### IGV At B-Cell\n",
    "\n",
    "Downloaded the HGR region from the normal B-cell A05247r.bam and HS2254r.bam transcriptomes.\n",
    "```\n",
    "chr13:1,004,904\n",
    "Total count: 38694\n",
    "A      : 1326  (3%,     626+,   700- )\n",
    "C      : 15820  (41%,     7501+,   8319- )\n",
    "G      : 1867  (5%,     948+,   919- )\n",
    "T      : 19680  (51%,     9413+,   10267- )\n",
    "N      : 1  (0%,     0+,   1- )\n",
    "---------------\n",
    "DEL: 2452\n",
    "INS: 0\n",
    "\n",
    "Total count: 32550\n",
    "A      : 1237  (4%,     734+,   503- )\n",
    "C      : 14700  (45%,     8292+,   6408- )\n",
    "G      : 1532  (5%,     909+,   623- )\n",
    "T      : 15081  (46%,     8371+,   6710- )\n",
    "N      : 0\n",
    "---------------\n",
    "DEL: 1465\n",
    "INS: 0\n",
    "```\n",
    "\n",
    "It's possible this variant is present in many cells. But the base is also 'hypermodified' which may mean that this is simply a reverse-transcritase artifact. If it is that may be interesting since the 'error profile' at this position could be used to 'predict' other modified sites in RNA samples.\n",
    "\n",
    "### U1248 in E. Coli\n",
    "\n",
    "- G966 in E. Coli or U1191 in Yeast \n",
    "- Secondary Structure: 16S helix 31 (also 970 loop)\n",
    "- 960... UUCG AU**G**CAACG CGAA\n",
    "- 2me-G966 and 5me-C967 are methylated in E. coli [PMID:21278676](http://www.ncbi.nlm.nih.gov/pubmed/21278676)\n",
    "- Authors report that 'corresponding base in eukarya is modified' 1-methyl-3-(3-amino-3-carboxypropyl)pseudouridine.\n",
    "- \"Single mutations at m2-G966 or m5-C967 produce more protein in vivo than wt ribosomes\"\n",
    "- \"Binding site for IF3\" (eukaryotic eIF1)\n",
    "\n",
    "#### h31-tRNA Interaction\n",
    "h31 and more specifically me2-G966 interacts with the Anti-Codon of tRNA at position 34. This means that the first base of the genetic code may somehow be involved at the P-site if these are really variant rRNA molecules.\n",
    "\n",
    "E.coli helix 31 structure. From PMID:21278676\n",
    "![E.coli helix 31 structure. From PMID:21278676)](../figure/20160504_Ecoli_h31.png)\n",
    "P-site structure. H31 - AntiCodon Interaction. PMID:11283358\n",
    "![P-site structure. H31 - AntiCodon Interaction. PMID:11283358](../figure/20160504_Ecoli_Psite_5.5A.png)\n",
    "Yeast tRNA-Phe. PMID:10943889\n",
    "![Yeast tRNA-Phe. PMID:10943889](../figure/20160504_Yeast_tRNA_Phe.png)\n",
    "\n",
    "- U1248 is U1191 in Yeast\n",
    "\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
