{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# rDNA Conservation and Statistics\n",
    "```\n",
    "pi:ababaian\n",
    "start: 2016 10 31\n",
    "complete : 2016 11 06\n",
    "addendum : 2017 12 26 (hgr0)\n",
    "```\n",
    "## Introduction\n",
    "\n",
    "It's difficult to interpret rDNA variation at times, I'd like to make conservation / GC / other information graphs to IGV or USCS so that I can look at the relationship between conservation  and variation for information.\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "## Objective\n",
    "\n",
    "- Generate 'Conservation' plots for SSU and LSU in in wig/bigwig format\n",
    "- This will let variants to be intepreted based on their conservation quickly\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Materials and Methods\n",
    "\n",
    "\n",
    "- riboZone alignment files\n",
    "```\n",
    "euk_LSU_rRNA.fa\n",
    "euk_SSU_rRNA.fa\n",
    "LSU_rRNA_alignment.fa\n",
    "SSU_rRNA_alignment.fa\n",
    "```\n",
    "\n",
    "\n",
    "#### Getting Sequence and Consensus Values\n",
    "- Jalview was used to quickly get a value for 'consensus' across the multiple alignments\n",
    "\n",
    "SSU_align.fa:\n",
    "    This file is a gapped alignment extracted from Homo Sapiens \n",
    "    \n",
    "SSU_cons.csv\n",
    "```\n",
    "Annotations > Autocalculated Annotations > Consensus\n",
    "\n",
    "Right click \"Consensus\" > Export Annotation\n",
    "```\n",
    "\n",
    "\n",
    "- `cons.sh` script below ran for SSU and LSU subsets...\n",
    "- Life: All aligned sequences\n",
    "- Euk: Eukaryotes\n",
    "- Vert: Vertebrates (to xenopus)\n",
    "\n",
    "![Example of SSU Conservation](../figure/20161031_SSU_conservation_example.png)\n",
    "\n",
    "#### Differences between hgr 28s and riboZone 28s\n",
    "\n",
    "In the process of calculating conensus and applying it to hgr the 28S sequences did not add up. The riboZone sequence is longer then the hgr sequence. I aligned the two sequences to one another. Output is in CROWN/data/riboZone/SSU_rRNA_alignment.fa\n",
    "\n",
    "The first 157 nt missing in hgr is the 5.8S rDNA, otherwise it's differences between the 'human' rRNA.\n",
    "\n",
    "#### Deleted in hgr (28S) relative to riboZone\n",
    "1,157\n",
    "293,295\n",
    "407,409\n",
    "417,419\n",
    "939,942\n",
    "3119,3120\n",
    "3137\n",
    "3274\n",
    "3319\n",
    "3332,3335\n",
    "3348,3349\n",
    "3539\n",
    "3730\n",
    "4182\n",
    "4260\n",
    "4297,4303\n",
    "4879\n",
    "4963\n",
    "4978,4980\n",
    "\n",
    "#### Insert in hgr (28S) relative to riboZone\n",
    "3309\n",
    "3367\n",
    "3481\n",
    "3494\n",
    "3512\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "# Converts a gapped ALIGN and its CONSERVATION to a wig file\n",
    "ALIGN='lsu_vert_align.fa' # Gapped Fasta Alignment\n",
    "CONS='lsu_vert_cons.csv' # JalView Consensus Output\n",
    "OUTPUT='VertCons_LSU.wig'\n",
    "\n",
    "# Start position of SSU (1003657) or 5.8S (1006623) or LSU (1007935)\n",
    "\n",
    "# Convert ALIGN.fa into one nucleotide per row\n",
    "\n",
    "sed 1d $ALIGN |\n",
    "sed 's/\\./\\-\\n/g' - |\n",
    "sed 's/[ATGCU]/\\N\\n/g' - |\n",
    "sed '/^\\s*$/d' - > align.csv\n",
    "\n",
    "\n",
    "sed 2d $CONS |\n",
    "sed 's/,/\\n/g' - |\n",
    "sed 1d - |\n",
    "head -n -1 - > cons.csv\n",
    " \n",
    "paste align.csv cons.csv |\n",
    "grep -e '^N' - |\n",
    "cut -f 2 - |\n",
    "sed 's/ //g' - > cons.values\n",
    "\n",
    "\n",
    "# SSU Alignments\n",
    "\n",
    "\t#echo \"fixedStep chrom=chr13 start=1003657 step=1\" > $OUTPUT\n",
    "\t#cat cons.values >> $OUTPUT\n",
    "\t# rm align.csv cons.csv cons.values \n",
    "\n",
    "# LSU Alignments\n",
    "# there are differences between hgr and riboZone LSU\n",
    "\n",
    "\t# Delete in hgr (28S) relative to riboZone\n",
    "\t# 40 nucleotides removed\n",
    "\tsed 4978,4980d cons.values |\n",
    "\tsed 4963d - |\n",
    "\tsed 4879d - |\n",
    "\tsed 4297,4303d - |\n",
    "\tsed 4260d - |\n",
    "\tsed 4182d - |\n",
    "\tsed 3730d - |\n",
    "\tsed 3539d - |\n",
    "\tsed 3348,3349d - |\n",
    "\tsed 3332,3335d - |\n",
    "\tsed 3319d - |\n",
    "\tsed 3274d - |\n",
    "\tsed 3137d - |\n",
    "\tsed 3119,3120d - |\n",
    "\tsed 939,942d - |\n",
    "\tsed 417,419d - |\n",
    "\tsed 407,409d - |\n",
    "\tsed 293,295d - > cons.d.values\n",
    "\n",
    "\t# Insert in hgr (28S) relative to riboZone (offset by 40nt)\n",
    "\tsed 3472i0.1 cons.d.values |\n",
    "\tsed 3454i0.1 - |\n",
    "\tsed 3441i0.1 - |\n",
    "\tsed 3327i0.1 - |\n",
    "\tsed 3269i0.1 - > cons.di.values\n",
    "\n",
    "\t# 5.8S subunit (1-157 nt)\n",
    "\tsed -i '158ifixedStep chrom=chr13 start=1007935 step=1' cons.di.values\n",
    "\tsed -i '1ifixedStep chrom=chr13 start=1006623 step=1' cons.di.values\n",
    "\tmv cons.di.values $OUTPUT\n",
    "\n",
    "\trm align.csv cons.csv cons.values cons.d.values\n",
    "\n",
    "\n",
    "# NOTE: There seems to be about a 2bp mis-alignment with the above script\n",
    "# but it's kind of hit and miss. Don't take the bp value directly (or refer\n",
    "# back to riboZone) but the general consensus trend does hold pretty well."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": []
    }
   ],
   "source": [
    "#!/bin/bash\n",
    "# gcContent Calculator\n",
    "# for rDNA\n",
    "#\n",
    "# Calculated gc of rDNA for 30,50,75 bp windows\n",
    "CROWN='/home/artem/Crown'\n",
    "\n",
    "cd $CROWN/resources/rDNA/\n",
    "\n",
    "WINDOW='75'\n",
    "SLIDE='1'\n",
    "NAME=\"w$WINDOW.s$SLIDE\"\n",
    "\n",
    "bedtools makewindows -g rDNA.fa.idx -w $WINDOW -s $SLIDE > rDNA.$NAME.bed\n",
    "\n",
    "# make start=1,000,000 + (0.5 * $WINDOW)\n",
    "echo \"fixedStep chrom=chr13 start=1000037 step=$SLIDE\" > rDNA.gc.$NAME.wig\n",
    "bedtools nuc -fi rDNA.fa -bed rDNA.$NAME.bed | cut -f 5 - | sed 1d - >> rDNA.gc.$NAME.wig\n",
    "\n",
    "rm rDNA.$NAME.bed\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "code_folding": [],
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "fastaToRead: command not found\r\n",
      "Could not locate a Bowtie index corresponding to basename \"hgr\"\r\n",
      "Command: bowtie --wrapper basic-0 -S -k 10 -f hgr rDNA.w75.fa \r\n",
      "sort: invalid option -- 't'\r\n",
      "Usage: samtools sort [options...] [in.bam]\r\n",
      "Options:\r\n",
      "  -l INT     Set compression level, from 0 (uncompressed) to 9 (best)\r\n",
      "  -m INT     Set maximum memory per thread; suffix K/M/G recognized [768M]\r\n",
      "  -n         Sort by read name\r\n",
      "  -o FILE    Write final output to FILE rather than standard output\r\n",
      "  -O FORMAT  Write output as FORMAT ('sam'/'bam'/'cram')   (either -O or\r\n",
      "  -T PREFIX  Write temporary files to PREFIX.nnnn.bam       -T is required)\r\n",
      "  -@ INT     Set number of sorting and compression threads [1]\r\n",
      "\r\n",
      "Legacy usage: samtools sort [options...] <in.bam> <out.prefix>\r\n",
      "Options:\r\n",
      "  -f         Use <out.prefix> as full final filename rather than prefix\r\n",
      "  -o         Write final output to stdout rather than <out.prefix>.bam\r\n",
      "  -l,m,n,@   Similar to corresponding options above\r\n",
      "samtools: error closing standard output: -1\r\n"
     ]
    }
   ],
   "source": [
    "# rDNA Mappability\n",
    "# rDNA aligned to hgr genome\n",
    "\n",
    "# On AWS\n",
    "# Requires HMMcopy bins to run\n",
    "# - fastaToRead\n",
    "\n",
    "# Convert fasta to reads; 75 bp slide by 1\n",
    "fastaToRead -w rDNA.fa > rDNA.w75.fa\n",
    "\n",
    "# Re-align rDNA to HGR \n",
    "bowtie hgr -S -k 10 -f rDNA.w75.fa > rDNA.w75.k10.sam\n",
    "\n",
    "# Downloaded and converted to bam\n",
    "samtools view -bh rDNA.w75.k10.sam | samtools sort -t temp -O bam -o rDNA.w75k10.bam -\n",
    "\n",
    "# Moved to Crown directory\n",
    "# ~/Crown/data/rDNA_stats/map/\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Results\n",
    "\n",
    "#### Conservation and low mapping of reads\n",
    "\n",
    "![rDNA Conservation and Alignment](../figure/20161031_rDNA_conservation_alignment.png)\n",
    "\n",
    "Conservations shown for all Life, Eukaroytes and Vertebrates (top to bottom) and alignment from NA19240.\n",
    "\n",
    "There is a really exciting correlation here. Regions which are less conserved are showing less alignment (and generally a more even GC% as well)\n",
    "\n",
    "![rDNA Conservation and Alignment and GC Contrent](../figure/20161031_rDNA_conservation_alignment2.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Discussion\n",
    "\n",
    "\n",
    "- These look good. A very obvious correlation is visible where high GC regions are less conserved throughout life. If that's where lots of the variation will be it'll be difficult to analyze =D\n",
    "\n",
    "\n",
    "![Deletion at 1,005,294 in NA19240](../figure/20161104_g.5294del_NA19240.png)\n",
    "\n",
    "- That being said some of the variation (see above) falls into mid-GC ranges in areas with low/moderate conservation. This is actually really informative for interpreting\n",
    "\n",
    "NOTE: the 'mappability' track generated here had a second copy of rDNA at chr13:2,000,000. This was fixed in the genome file but not going to remake the alignment since it won't influence the output."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Addendum - hgr0\n",
    "\n",
    "Using the same data as above I re-generated rDNA statistics/annotation tracks for the hgr0 reference sequence.\n",
    "\n",
    "- GC content, 30, 50 and 70 bp windows\n",
    "- Shannon Entropy\n",
    "- Protein Contact Points (rvis)\n",
    "- Nucleotide (rRNA) base pairing (rvis).\n",
    "- Ribosomal Domains\n",
    "\n",
    "Files stored in ~/Crown/data/rDNA_stats/hgr0/\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
