{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# rDNA Haplotype Discovery\n",
    "```\n",
    "pi:ababaian\n",
    "start: 2016 11 25\n",
    "complete : 2016 12 08\n",
    "```\n",
    "## Introduction\n",
    "\n",
    "To date, rDNA alignments are based on a reference sequence [GenBank: U13369.1](http://www.ncbi.nlm.nih.gov/nuccore/555853). One of the main issues I've anticipated and appears to hold true in the data is that if there are clusters of variants in this consensus sequence then alignment is likely to be difficult since reads mapping to those variants will be dropped out.\n",
    "\n",
    "## Discovery - A3838\n",
    "\n",
    "During the 'automated VCF' experiment exactly this case was found and standard VCF calling pipeline fails to call these variants since the reads are 'poorly aligned'. In NA19240, at position 1,003,838 there is a variant allele C to A at 5% frequency, with reads coming from both strands. Keep in mind if one estimates 300 rDNA copies then having a minor frequency of \\<0.3% is biologically pertinant, although may be below detection thresholds. Interpreted another way ~15/300 copies of rDNA are variant in NA19240.\n",
    "\n",
    "```\n",
    "chr13:1,003,838\n",
    "<hr>Total count: 652\n",
    "A      : 34  (5%,     15+,   19- )\n",
    "C      : 617  (95%,     283+,   334- )\n",
    "G      : 0\n",
    "T      : 1  (0%,     0+,   1- )\n",
    "N      : 0\n",
    "---------------\n",
    "```\n",
    "In IGV I sorted reads by 'base' which shows that the A variant is associated with more neighbouring variants. Together this leads to a very low MAPQ score for these reads and thus exclusion from VCF calling. This variant occurs in a conserved sequence of 18S.\n",
    "\n",
    "![20161115_na19240_igv_altHaplo_A3838.png](../figure/20161115_na19240_igv_altHaplo_A3838.png)\n",
    "\n",
    "\n",
    "I selected the reads with the A3838 variant, and the original consensus sequence of the rDNA and multiple-aligned them with standard paramters using `clustalX`. I then manually fixed the alignment in Jalview. Output of that is...\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">NHIST_0024/1-100\r\n",
      "........................................................................\r\n",
      "........................................................................\r\n",
      ".......GCTCCTCTCCTACTTGGAAAACTGTGGTAATTCTAGAGCTATTACATGCCGAAGGGCACTGACCC\r\n",
      "CCTTCACGGGGAAGATGCGTGCATTTATCAGATTA.....................................\r\n",
      "........................................................................\r\n",
      "...........................................\r\n",
      ">NHIST_0004/1-100\r\n",
      "........................................................................\r\n",
      "....................................TGTGAATGGCTCATTAAATCAGTTATGGTTCTTTTG\r\n",
      "GTTGCTGGCTCCTCTCCTACTTGGAAAACTGTGGTAATTCTAGAGCTATTACATGCCGAAGGGC........\r\n",
      "........................................................................\r\n",
      "........................................................................\r\n",
      "...........................................\r\n",
      ">NHIST_0015/1-100\r\n",
      "........................................................................\r\n",
      "...............................................................GTTCTTTTG\r\n",
      "GTCGCTGGCTCCTCTCCTACTTGGAAAACTGTGGTAATTCTAGAGCTATTACATGCCGAAGGGCACTGACCC\r\n",
      "CCTTCACAGGGAAGATGCG.....................................................\r\n",
      "........................................................................\r\n",
      "...........................................\r\n",
      ">NHIST_0005/1-100\r\n",
      "........................................................................\r\n",
      "...................................CTGTGAATGGCTCATTAAATCAGTTATGGTTCTTTTG\r\n",
      "GTCGCTGGCTCCTCTTCTACTTGGAAAACTGTGGTAATTCTAGAGCTATTACATGCCGAAGGG.........\r\n",
      "........................................................................\r\n",
      "........................................................................\r\n",
      "...........................................\r\n",
      ">NHIST_0016/1-100\r\n",
      "........................................................................\r\n",
      "..........................................TGGCTCATTAAATCAGTTATGGTTCTTTTG\r\n",
      "GTCGCTGGCTCCTCTTCTACTTGGAAAACTGTGGTAATTCTAGAGCTATTACATGCCGAAGGGCACTGAC..\r\n",
      "........................................................................\r\n",
      "........................................................................\r\n",
      "...........................................\r\n",
      ">chr13_1003788-1003887/1-100\r\n",
      "........................................................................\r\n",
      "........................................................................\r\n",
      ".........TCCTCTCCTACTTGGATAACTGTGGTAATTCTAGAGCTAATACATGCCGACGGGCGCTGACCC\r\n",
      "CCTTCGCGGGGGGGATGCGTGCATTTATCAGATCAAA...................................\r\n",
      "........................................................................\r\n",
      "...........................................\r\n",
      ">NHIST_0027/1-100\r\n",
      "........................................................................\r\n",
      ".................................................TTAAATCAGTTATGGTTCTTTTG\r\n",
      "TTTGCTCGCTCCTCTCCTACTTGGAAAACTGTGGTAATTCTAGAGCTAATACATGCCAAAGGGCGCTGACCC\r\n",
      "CCTTC...................................................................\r\n",
      "........................................................................\r\n",
      "...........................................\r\n",
      ">NHIST_0022/1-100\r\n",
      "........................................................................\r\n",
      "....................................................AATCAGTTATGGTTCTTTTG\r\n",
      "GTCGCTCGCTCCTCTCCTACTTGGAAAACTGTGGTAATTCTAGAGCTAATGCATGCCGAAGGGTGCTGACCC\r\n",
      "CCTTCGCG................................................................\r\n",
      "........................................................................\r\n",
      "...........................................\r\n",
      ">NHIST_0026/1-100\r\n",
      "........................................................................\r\n",
      "........................................................AGTTATGGTTCTTTTG\r\n",
      "TTTGCTCGTTCCTCTCCTACTTGGAAAACTGTGGTAATTCTAGAGCTAATACATGCCAAAGGGCGCTGACCC\r\n",
      "CCTTCGCGGGGA............................................................\r\n",
      "........................................................................\r\n",
      "...........................................\r\n",
      ">NHIST_0025/1-100\r\n",
      "........................................................................\r\n",
      ".............................................................TGGTTCTTTCA\r\n",
      "TTTGCTCGCTCCTCTCCTACTTGGAAAACTGTGGTAATTCTAGAGCTAATACATGCCAAAGGGCGCTGACCC\r\n",
      "CCTTCGCGGGGAAGATG.......................................................\r\n",
      "........................................................................\r\n",
      "...........................................\r\n",
      ">NHIST_0012/1-100\r\n",
      "........................................................................\r\n",
      "...............................................................GTTCTTTTG\r\n",
      "ATCGCTCGCTCCTCTCCTACTTTGAAAACTGTGGTAATTCTAGAGCTAATGCATGCCGAAGGGCGCTGACCC\r\n",
      "CCTTCGTGGGGAAGATGCG.....................................................\r\n",
      "........................................................................\r\n",
      "...........................................\r\n",
      ">NHIST_0009/1-100\r\n",
      "........................................................................\r\n",
      ".................................................TTAAATCAGTTATGGTTCTTTCA\r\n",
      "TTTGCTCGCTCCTCTCCTACTTGGAAAACTGTGGTAATTCTAGAGCTAATACATGCCAAAGGGCGCTGACCC\r\n",
      "CCTTC...................................................................\r\n",
      "........................................................................\r\n",
      "...........................................\r\n",
      ">NHIST_0017/1-100\r\n",
      "........................................................................\r\n",
      "........................................AATGGCTCATTAAATCAGTTATGGTTCTTTTG\r\n",
      "ATCGCTCGCTCCTCTCCTACTTTGAAAACTGTGGTAATTCTAGAGCTAATGCATGCCGAAGGGCGCTG....\r\n",
      "........................................................................\r\n",
      "........................................................................\r\n",
      "...........................................\r\n",
      ">NHIST_0013/1-100\r\n",
      "........................................................................\r\n",
      "....................................TGCGAATGGCTCATTAAATCAGTTATGGTTCTTTTG\r\n",
      "GTCGCTCGCTCCTCTCCTACTTGGAAAACTGTGGTAATTCTAGAGCTAATGCATGCCGAAGGGT........\r\n",
      "........................................................................\r\n",
      "........................................................................\r\n",
      "...........................................\r\n",
      ">NHIST_0021/1-100\r\n",
      "........................................................................\r\n",
      ".....................................................................TTG\r\n",
      "GTCACTCTCTCCTCTCCTACTTGCAAAACTGTGGTAATTATAGAGCTAATACATGCCGAAGGGCACTGATCC\r\n",
      "CCTTCGCGGGGAAGATGCGTGCATT...............................................\r\n",
      "........................................................................\r\n",
      "...........................................\r\n",
      ">NHIST_0028/1-100\r\n",
      "........................................................................\r\n",
      "................................................ATTAAATCAGTTATGGTTCTTTTG\r\n",
      "ATCGCTCGCTCCTCTCCTACTTTGCAAACTGTGGTAATTCTAGAGCTAATGCATGCCGAAGGGCGCTGACCC\r\n",
      "CCTT....................................................................\r\n",
      "........................................................................\r\n",
      "...........................................\r\n",
      ">chr13_1003638-1004037/1-400\r\n",
      "...CGCCGCGCTCTACCTTACCTACCTGGTTGATCCTGCCAGTAGCATATGCTTGTCTCAAAGATTAAGCCA\r\n",
      "TGCATGTCTAAGTACGCACGGCCGGTACAGTGAAACTGCGAATGGCTCATTAAATCAGTTATGGTTCCTTTG\r\n",
      "GTCGCTCGCTCCTCTCCTACTTGGATAACTGTGGTAATTCTAGAGCTAATACATGCCGACGGGCGCTGACCC\r\n",
      "CCTTCGCGGGGGGGATGCGTGCATTTATCAGATCAAAACCAACCCGGTCAGCCCCTCTCCGGCCCCGGCCGG\r\n",
      "GGGGCGGGCGCCGGCGGCTTTGGTGACTCTAGATAACCTCGGGCCGATCGCACGCCCCCCGTGGCGGCGACG\r\n",
      "ACCCATTCGAACGTCTGCCCTATCAACTTTCGATGGTAGTCGC\r\n",
      ">NHIST_0003/1-100\r\n",
      "........................................................................\r\n",
      "...................................................AAATCAGTTATGGTTCTTTTG\r\n",
      "ATCGCTCGCTCCTCTCCTACTTTGCAAACTGTGGTAATTCTAGAGCTAATGCATGCCGAAGGGCGCTGACCC\r\n",
      "CCTTCGT.................................................................\r\n",
      "........................................................................\r\n",
      "...........................................\r\n",
      ">NHIST_0014/1-100\r\n",
      "........................................................................\r\n",
      "........................................................................\r\n",
      ".......................................................CCGAAGGGCACTGATCC\r\n",
      "CCTTCGCGGGGAAGATGCGTGCATTTATCAGATCAAAACCAACCCGGTCAGCCCCTCTTCAGCCCAGGCCGG\r\n",
      "GGGAG.GGGTTC............................................................\r\n",
      "...........................................\r\n",
      ">NHIST_0000/1-100\r\n",
      "........................................................................\r\n",
      "........................................................................\r\n",
      ".........................................................GAAGGGCACTGACCC\r\n",
      "CCTTCACAGGGAAGATGCGTGCATTTATCAGATCAAAACCACCCCAGTCAGCCCCTCTCCGTCCCCGGCCGG\r\n",
      "GGGGTCGGGTGCC...........................................................\r\n",
      "...........................................\r\n",
      ">NHIST_0002/1-100\r\n",
      "........................................................................\r\n",
      "........................................................................\r\n",
      ".........................................................GAAGGGCACTGACCC\r\n",
      "CCTTCACGGGGAAGATGCGTGCATTTATCAGATTAAAACCAACCCAGTCAGCCCCTCTCCGGCCCCGGCCGG\r\n",
      "GGGGTCGGGTACC...........................................................\r\n",
      "...........................................\r\n",
      ">NHIST_0018/1-100\r\n",
      "........................................................................\r\n",
      "........................................................................\r\n",
      ".................................................TGCATGCCGAAGGGTGCTGACCC\r\n",
      "CCTTCGCGGGGAAGATGCCTGCATTTATCAGATCAAAACCAACCCGGTCAGCCCCTCTCCGGCCCCTGCCTA\r\n",
      "GGGGT...................................................................\r\n",
      "...........................................\r\n",
      ">NHIST_0023/1-100\r\n",
      "........................................................................\r\n",
      "........................................................................\r\n",
      "............................................GCTAATGCATGCCGAAGGGCGCTGACCC\r\n",
      "CCTTCGTGGGGAAGATGCGTGCATTTATCAGATCAAAACCAACCCGGTCAGCCCCTCTCTGGCCCCGGCCAG\r\n",
      "........................................................................\r\n",
      "...........................................\r\n",
      ">NHIST_0006/1-100\r\n",
      "........................................................................\r\n",
      "........................................................................\r\n",
      ".................................GTAATTATAGAGCTAATACATGCCGAAGGGCACTGATCC\r\n",
      "CCTTCGCGGGGAAGATGCGTGCATTTATCAGATCAAAACCAACCCGGTCAGCCCCTCTTCA...........\r\n",
      "........................................................................\r\n",
      "...........................................\r\n",
      ">NHIST_0001/1-100\r\n",
      "........................................................................\r\n",
      "........................................................................\r\n",
      "....................................ATTCTAGAGCTATTACATGCGGAAGGGCACTGACCC\r\n",
      "CCTTCACAGGGAAGATGCGTGCATTTATCAGATCAAAACCAACCCAGTCAGCCCCTCTCCGTCC........\r\n",
      "........................................................................\r\n",
      "...........................................\r\n",
      ">NHIST_0020/1-100\r\n",
      "........................................................................\r\n",
      "........................................................................\r\n",
      "...........CTCTCCTACTTGCAAAACTGTGGTAATTATAGAGCTAATACATGCCGAAGGGCACTGATCC\r\n",
      "CCTTCGCGGGGAAGATGCGTGCATTTATCAGATCAAAAC.................................\r\n",
      "........................................................................\r\n",
      "...........................................\r\n",
      ">NHIST_0010/1-100\r\n",
      "........................................................................\r\n",
      "........................................................................\r\n",
      "................................GGTAATTATAGAGCTAATACATGCCGAAGGGCACTGATCC\r\n",
      "CCTTCGCGGGGAAGATGCGTGCATTTATCAGAACAAAACCAACCCGGTCAGCCCCTCTTC............\r\n",
      "........................................................................\r\n",
      "...........................................\r\n",
      ">NHIST_0029/1-100\r\n",
      "........................................................................\r\n",
      "........................................................................\r\n",
      ".................TACTTGCAAAACTGTGGTAATTATAGAGCTAATACATGCCGAAGGGCACTGATCC\r\n",
      "CCTTCGCGGGGAAGATGCGTGCATTTATCAGATCAAAACCAACCC...........................\r\n",
      "........................................................................\r\n",
      "...........................................\r\n",
      ">NHIST_0008/1-100\r\n",
      "........................................................................\r\n",
      "........................................................................\r\n",
      "...................CTTGGAAAACTGTGGTAATTCTAGAGCTAATGCATGCCGAAGGGTGCTGACCC\r\n",
      "CCTTCGCGGGGAAGATGCCTGCATTTATCAGATCAAAACCAACCCGG.........................\r\n",
      "........................................................................\r\n",
      "...........................................\r\n",
      ">NHIST_0019/1-100\r\n",
      "........................................................................\r\n",
      "........................................................................\r\n",
      "........................AAAACTGTGGTAATTATAGAGCTAATACATGCCGAAGGGCACTGATCC\r\n",
      "CCTTCGCGGGGAAGATGCGTGCATTTATCAGAACAAAACCAACCCGGTCAGC....................\r\n",
      "........................................................................\r\n",
      "...........................................\r\n",
      ">NHIST_0007/1-100\r\n",
      "........................................................................\r\n",
      "........................................................................\r\n",
      "......................TGAAAACTGTGGTAATTCTAGAGCTAATGCATGCCGAAGGGCGCTGACCC\r\n",
      "CCTTCGTGGGGAAGATGCGTGCATTTATCAGATCAAAACCAACCCGGTCA......................\r\n",
      "........................................................................\r\n",
      "...........................................\r\n",
      ">NHIST_0011/1-100\r\n",
      "........................................................................\r\n",
      "........................................................................\r\n",
      ".......................GAAAACTGTGGTAATTCTAGAGCTAATGCATGCCGAAGGGTGCTGACCC\r\n",
      "CCTTCGCGGGGAAGATGCCTGCATTTATCAGATCAAAACCAACCCGGTCAG.....................\r\n",
      "........................................................................\r\n",
      "...........................................\r\n"
     ]
    }
   ],
   "source": [
    "cd ~/Crown/data/haplo_discovery\n",
    "cat 20161125_altHaplo_A3838.mfa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![20161115_rDNA_haplo_A3838_realigned](../figure/20161115_rDNA_haplo_A3838_realigned.png)\n",
    "\n",
    "Which you can see appears to be a variant haplotype of 18S with multiple neighbouring variations.\n",
    "```\n",
    "\\>Standard_rDNA_around_C3838\n",
    "TGCGAATGGCTCATTAAATCAGTTATGGTTCCTTTGGTCGCTCGCTCCTCTCCTACTTGGATAACTGTGGTA\n",
    "ATTCTAGAGCTAATACATGCCGACGGGCGCTGACCCCCTTCGCGGGGGGGATGCGTGCATTTATCAGATCAA\n",
    "AACCAACCCGGTCAGCCCCTCTCCGGCCCCGGCCGGGGGGCGGGCGCCG\n",
    "\\>Variant_rDNA_around_A3838\n",
    "TGTGAATGGCTCATTAAATCAGTTATGGTTCTTTTGGTCGCTCGCTCCTCTCCTACTTGGAAAACTGTGGTA\n",
    "ATTCTAGAGCTAATACATGCCGAAGGGCGCTGACCCCCTTCGCGGGGAAGATGCGTGCATTTATCAGATCAA\n",
    "AACCAACCCGGTCAGCCCCTCTCCGGCCCCGGCCGGGGGGTCGGGTACC\n",
    "```\n",
    "\n",
    "Blat(UCSC) A3838 to hg19\n",
    "```\n",
    "Side by Side Alignment\n",
    "\n",
    "\n",
    "00001 tgtgaatggctcattaaatcagttatggttcttttggtcgctcgctcctc 00050\n",
    ">>>>> || ||||||||||||||||||||||||||||||||| ||||||||||||| >>>>>\n",
    "18257 tgcgaatggctcattaaatcagttatggttcttttgatcgctcgctcctc 18306\n",
    "\n",
    "00051 tcctacttggaaaactgtggtaattctagagctaatacatgccgaagggc 00100\n",
    ">>>>> |||||||| ||||||||||||||||||||||||||| ||||||||||||| >>>>>\n",
    "18307 tcctactttgaaaactgtggtaattctagagctaatgcatgccgaagggc 18356\n",
    "\n",
    "00101 gctgacccccttcgcggggaagatgcgtgcatttatcagatcaaaaccaa 00150\n",
    ">>>>> |||||||||||||| ||||||||||||||||||||||||||||||||||| >>>>>\n",
    "18357 gctgacccccttcgtggggaagatgcgtgcatttatcagatcaaaaccaa 18406\n",
    "\n",
    "00151 cccggtcagcccctctccggccccggccggggggtcgggtacc 00193\n",
    ">>>>> ||||||||||||||||| |||||||||| ||||||||||| || >>>>>\n",
    "18407 cccggtcagcccctctctggccccggccagggggtcgggtgcc 18449\n",
    "\n",
    "```\n",
    "BLAST (NCBI)\n",
    "There are actually really good hits to match the variant consensus sequence:\n",
    "\n",
    "```\n",
    "Score\tExpect\tIdentities\tGaps\tStrand\n",
    "318 bits(172) \t2e-83 \t186/193(96%) \t0/193(0%) \tPlus/Plus\n",
    "\n",
    ">AC018688.9 Homo sapiens BAC clone RP11-462H3 from UL, complete sequence\n",
    "TGTGAATGGCTCATTAAATCAGTTATGGTTCTTTTGGTTGCTGGCTCCTCTCCTACTTGGAAAACTGTGGTAATTCTAGA\n",
    "GCTATTACATGCCGAAGGGCACTGACCCCCTTCACGGGGAAGATGCGTGCATTTATCAGATTAAAACCAACCCAGTCAGC\n",
    "CCCTCTCCGGCCCCGGCCGGGGGGTCGGGTACC\n",
    "```\n",
    "Downloaded that BAC clone sequence to `~/Crown/data/haplo_discovery`\n",
    "\n",
    "# T3774\n",
    "\"Walking\" down the fragments there looks to be more variants at about 5% frequency and paired-end reads link the A3838 variations with these ones. Same Sort-->Align strategy was used to pull out T3774 variant reads:\n",
    "```\n",
    "chr13:1,003,774\n",
    "<hr>Total count: 771\n",
    "A      : 0\n",
    "C      : 733  (95%,     357+,   376- )\n",
    "G      : 1  (0%,     1+,   0- )\n",
    "T      : 37  (5%,     18+,   19- )\n",
    "N      : 0\n",
    "---------------\n",
    "```\n",
    "![20161125_T3447_aligned.png](../figure/20161125_T3447_aligned.png)\n",
    "\n",
    "\n",
    "## Replicating A3838 + T3774 in NA12878\n",
    "\n",
    "### Using NA12878 PacBio - rep2\n",
    "```\n",
    "chr13:1,003,774\n",
    "<hr>Total count: 500\n",
    "A      : 2  (0%,     1+,   1- )\n",
    "C      : 480  (96%,     206+,   274- )\n",
    "G      : 1  (0%,     0+,   1- )\n",
    "T      : 17  (3%,     10+,   7- )\n",
    "N      : 0\n",
    "---------------\n",
    "DEL: 41\n",
    "INS: 98\n",
    "```\n",
    "```\n",
    "chr13:1,003,838\n",
    "<hr>Total count: 521\n",
    "A      : 13  (2%,     5+,   8- )\n",
    "C      : 506  (97%,     208+,   298- )\n",
    "G      : 2  (0%,     1+,   1- )\n",
    "T      : 0\n",
    "N      : 0\n",
    "---------------\n",
    "DEL: 25\n",
    "INS: 78\n",
    "```\n",
    "### Using NA12878 Ultra-deep DNA library\n",
    "```\n",
    "chr13:1,003,774\n",
    "<hr>Total count: 9314\n",
    "A      : 40  (0%,     40+,   0- )\n",
    "C      : 8972  (96%,     8888+,   84- )\n",
    "G      : 18  (0%,     18+,   0- )\n",
    "T      : 284  (3%,     256+,   28- )\n",
    "N      : 0\n",
    "---------------\n",
    "```\n",
    "```\n",
    "chr13:1,003,838\n",
    "<hr>Total count: 9489\n",
    "A      : 321  (3%,     276+,   45- )\n",
    "C      : 9096  (96%,     8743+,   353- )\n",
    "G      : 37  (0%,     29+,   8- )\n",
    "T      : 35  (0%,     35+,   0- )\n",
    "N      : 0\n",
    "---------------\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Discussion\n",
    "\n",
    "This probably needs to be expanded but my take away from this is that there are alternative rDNA regions, this could very likely be pseudogenes which needs to be ruled out. I'm not satisfied with the alignment/detection of these variants by GATK/VCF so what I pivoted to is to see if there are alternative aligners which could do a better job at aligning divergent reads.\n",
    "\n",
    "This section really wasn't 'Hypothesis' driven to be honest, it was partly informative but I'm unsure if it was worth the effort, I need to focus my efforts on asking dis-provable questions instead of tugging at every thread which shows itself on this project."
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
