{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# EC2 Make Script\n",
    "```\n",
    "pi:ababaian\n",
    "start: 2016 11 06\n",
    "last completion : 2016 12 29 \n",
    "```\n",
    "## Introduction\n",
    "\n",
    "On EC2, a standard system image can be saved. This script defines how the standard EC2 system was made and includes a version change-log as it's updated. Current major features are:\n",
    "\n",
    "### Resources\n",
    "* hg38 genome (fasta)\n",
    "* hgr genome, single copy rDNA at chr13:1000000. (fa)\n",
    "* hgr0 reference sequence\n",
    "* rDNA (fa)\n",
    "\n",
    "### Software\n",
    "* blasr\n",
    "* bowtie 1 & 2\n",
    "* Cufflinks v2.2.1\n",
    "* GATK 3.6\n",
    "* samtools\n",
    "* sl\n",
    "* tophat\n",
    "* UCSC utilities\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Things to add in next update\n",
    "\n",
    "* picard\n",
    "* hgr1 reference sequence\n",
    "* bcftools v.1.4"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Make Scripts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "code_folding": [],
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#!/bin/bash/\n",
    "# crown-161106\n",
    "# ami-59b71739\n",
    "# Primary Make Script\n",
    "# Initialization script for an Ubuntu 16.04 LTS instance\n",
    "# Run with at least 4 Gb of memory to compile successfully\n",
    "\n",
    "# Update\n",
    "sudo apt-get update\n",
    "\n",
    "# Bioinformatics Software\n",
    "sudo apt-get install samtools # v. 0.1.19\n",
    "sudo apt-get install bowtie # v. 1.1.2\n",
    "sudo apt-get install bowtie2 # v. 2.2.6\n",
    "sudo apt-get install tophat\n",
    "#sudo apt-get install blasr # v.\n",
    "\n",
    "sudo apt-get install docker.io\n",
    "sudo service docker start\n",
    "sudo usermod -a -G docker ubuntu\n",
    "# Need to re-login here. Possible split into multiple tasks\n",
    "\n",
    "# AWS Command Line\n",
    "sudo apt-get install awscli\n",
    "aws configure # ENTER CREDENTIALS MANUALLY\n",
    "\n",
    "# Small Binary Utilities Download (NCBI)\n",
    "mkdir ~/bin; cd ~/bin\n",
    "\n",
    "\twget http://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/faToTwoBit\n",
    "\twget http://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/twoBitToFa\n",
    "\twget http://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/wigToBigWig\n",
    "\twget http://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/fastqToFa\n",
    "\twget http://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/faSplit\n",
    "\n",
    "\tchmod 755 *\n",
    "cd ~\n",
    "\n",
    "# Compiler Software\n",
    "\tsudo apt-get install build-essential\n",
    "\tsudo apt-get install gfortran\n",
    "\tsudo apt-get install graphviz\n",
    "\tsudo apt-get install libjpeg-dev\n",
    "\tsudo apt-get install libfreetype6-dev\n",
    "\tsudo apt-get install python\n",
    "\n",
    "\n",
    "# Make Blastr from source\n",
    "mkdir software; cd software\n",
    "git clone git://github.com/PacificBiosciences/pitchfork\n",
    "\n",
    "cd pitchfork\n",
    "make init\n",
    "cd deployment\n",
    "sh setup-env.sh\n",
    "cd ..\n",
    "make blasr\n",
    "\n",
    "cd workspace\n",
    "ln -s hdf5-1.8.16 hdf5\n",
    "ln -s blasr blasr_install\n",
    "cat deployment/setup-env.sh >> /home/ubuntu/.bashrc\n",
    "\n",
    "# Download hg38 (from UCSC)\n",
    "\tmkdir ~/resources; cd ~/resources\n",
    "\twget ftp://hgdownload.cse.ucsc.edu/goldenPath/hg38/bigZips/hg38.2bit\n",
    "\ttwoBitToFa hg38.2bit hg38.fa\n",
    "\tsamtools faidx hg38.fa\n",
    "\trm hg38.2bit\n",
    "\n",
    "# Download hgr and rDNA (from S3)\n",
    "\taws s3 cp s3://crownproject/resources/hgr.fa ./\n",
    "\taws s3 cp s3://crownproject/resources/hgr_main.fa ./\n",
    "\taws s3 cp s3://crownproject/resources/rDNA.fa ./\n",
    "\n",
    "\n",
    "## Download Biocontainers\n",
    "# git clone https://github.com/BioContainers/containers\n",
    "\n",
    "# Install Git LFS\n",
    "#\twget https://github.com/github/git-lfs/releases/download/v1.4.4/git-lfs-linux-amd64-1.4.4.tar.gz\n",
    "#\n",
    "#\ttar -xvf git-lfs-linux-amd64-1.4.4.tar.gz\n",
    "#\tsudo sh git-lfs-1.4.4/install.sh\t\n",
    "#\trm git*\n",
    "\n",
    "\n",
    "\n",
    "# Build Docker Container\n",
    "\t# docker run [OPTIONS] <IMAGE> \t<command> <arguments>\n",
    "\t# -V : create symbolic link between <env dir>:<container dir>\n",
    "\t# becomes\n",
    "\t# <Command> <Argument>\n",
    "\t\n",
    "\t# Build bowtie container\n",
    "\t# cd containers/bowtie/1.1.2/\n",
    "\t# docker build -t bowtie . #builds bowtie image\n",
    "\t# alias bowtie='docker run -V /home/ec2-user:/home/ bowtie bowtie'\n",
    "\n",
    "\t# Build Samtools 1.3 Container\n",
    "\t# cd containers/samtools/1.3.1/\n",
    "\t# docker build -t samtools . #builds samtools image\n",
    "\n",
    "\n",
    "# Download Crown Project Files\n",
    "#\tgit clone https://github.com/ababaian/Crown.git\n",
    "\n",
    "##\n",
    "##\n",
    "## CROWN_INIT INSTANCE SCREENSHOT HERE\n",
    "##\n",
    "##"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#!/bin/bash/\n",
    "# crown-161229\n",
    "# ami-22ca7b42\n",
    "\n",
    "## HGR Reference Update ----------------------------------------------------\n",
    "# There is an error on the standard hgr.fa and hgr_main.fa\n",
    "# where the rDNA array is duplicated twice. Corrected version re-uploaded\n",
    "\n",
    "# rm old references\n",
    "cd ~/resources/\n",
    "rm hgr.fa; hgr_main.fa\n",
    "\n",
    "# Download new references\n",
    "    # Single copy rDNA on chr13\n",
    "    aws s3 cp s3://crownproject/resources/hgr.fa ./\n",
    "        samtools faidx\n",
    "        \n",
    "    # Just the region of rDNA upto the end of 28S (45s)    \n",
    "    aws s3 cp s3://crownproject/resources/hgr_45s.fa ./\n",
    "    \n",
    "    # hg38 genome with rDNA injection\n",
    "    aws s3 cp s3://crownproject/resources/hg38r.2bit ./\n",
    "\n",
    "# New Software ------------------------------------------------------------\n",
    "# Genome Analysis Tookit (GATK) 3.6-0-g89b7209\n",
    "# Requires Java (OpenDJK 1.8.0_111)\n",
    "    sudo apt-get install default-jre\n",
    "    sudo apt-get install default-jdk\n",
    "    \n",
    "# Download gatk from s3    \n",
    "cd ~/software\n",
    "    aws s3 cp s3://crownproject/software/GenomeAnalysisTK-3.6.tar.bz2 ./\n",
    "    tar -xvf GenomeAnalysisTK-3.6.tar.bz2\n",
    "\n",
    "# Build java running script 'gatk'\n",
    "cd ~/bin\n",
    "    echo '#!/bin/bash' > gatk\n",
    "    echo '' >> gatk\n",
    "    echo 'echo Running GATK' >> gatk\n",
    "    echo 'java -Xmx4G -jar /home/ubuntu/software/GenomeAnalysisTK.jar $@' >> gatk"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "# crown-170220\n",
    "# ami-66129306\n",
    "\n",
    "# Update from crown-161229 / ami-22ca7b42\n",
    "\n",
    "# hgr0 Reference Update ------------------------------------------------\n",
    "aws s3 cp s3://crownproject/resources/hgr0.fa ./\n",
    "aws s3 cp s3://crownproject/resources/hgr0.gatk.fa ./\n",
    "aws s3 cp s3://crownproject/resources/hgr0.gatk.dict ./\n",
    "\n",
    "samtools faidx hgr0.fa\n",
    "samtools faidx hgr0.gatk.fa\n",
    "\n",
    "\n",
    "# New Software ---------------------------------------------------------\n",
    "# Cufflinks\n",
    "cd software\n",
    "  aws s3 cp s3://mouseproject/cufflinks-2.2.1.Linux_x86_64.tar.gz ./\n",
    "  tar -xvf cufflinks-2.2.1.Linux_x86_64.tar.gz\n",
    "  mv cufflinks-2.2.1.Linux_x86_64/* ~/bin/\n",
    "  rmdir cufflinks-2.2.1.Linux_x86_64/\n",
    "\n",
    "# sl\n",
    "  sudo apt-get install sl\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Holder space (Non Implemented Commands)\n",
    "\n",
    "# BWA\n",
    "# wget https://sourceforge.net/projects/bio-bwa/files/bwakit/bwakit-0.7.15_x64-linux.tar.bz2/download\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
