{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 10 Genomes Pilot Alignment\n",
    "```\n",
    "pi:ababaian\n",
    "start: 2016 12 23\n",
    "complete : 2017 01 13\n",
    "\n",
    "addendum: 2017 02 11 - NA12878\n",
    "```\n",
    "## Introduction\n",
    "\n",
    "\n",
    "I've faffed around with techincal matters the last month and need to re-focus on the goal of the project; define the global variation in rDNA. I have a much better understanding of *how* the variation can be detected but the issues with standardized VCF and structural variation detection still exists. I've decided to take a short break and get to aligning more genomes and building the infrastructure for a larger alignment run.\n",
    "\n",
    "I'd like to expand to some more diverse populations and include a trio:\n",
    "\n",
    "1. [YRI Trio: Yoruba](ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/working/20130606_sample_info/20130606_g1k.ped)\n",
    "    * Daughter: NA19240 [index]\n",
    "    * Mother: NA19238\n",
    "    * Father: NA19239\n",
    "    * Mat. Half Sibling: NA18913 [No genome data]\n",
    "    \n",
    "2. CEU - Utah\n",
    "    * Female: NA12878 (Previously analyzed. Use PCR-Free Seq)\n",
    "    \n",
    "3. GIH - Gujarati Indian, TX\n",
    "    * Male: NA20845\n",
    "    \n",
    "3. CHB - China\n",
    "    * Female: NA18525\n",
    "    \n",
    "4. FIN - Finland\n",
    "    * Female: HG00268\n",
    "    \n",
    "5. PUR - Puerto Rican\n",
    "    * Male: HG00731 (PCR-Free Sequencing)\n",
    "    \n",
    "6. GBR - Brit\n",
    "    * Male: HG00096\n",
    "    \n",
    "7. LWK - Kenya\n",
    "    * Female: NA19017\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "## Objective\n",
    "\n",
    "   * Build a EC2 / Genesis compatible pipeline to perform rapid alignment on an input fastq file-set from 1kg. This can then be customized and refined in the future as better alignment methods are developed.\n",
    "    \n",
    "   * Run the above method on 10 pilot genomes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Materials and Methods\n",
    "\n",
    "### Fastq Files\n",
    "\n",
    "`~/Crown/data/1kg_pilot/1kg_pilot.ods`\n",
    "\n",
    "| Sample | Population | Sex | Notes       | Fastq | Size (KB) |\n",
    "|--------|:----------:|:---:|-------------|-------|-----------|\n",
    "|NA19240|YRI|F|Low-Coverage, WGS| ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/SRR794330/SRR794330_1.fastq.gz | 2940409 |\n",
    "|NA19240|YRI|F|Low-Coverage, WGS| ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/SRR794330/SRR794330_2.fastq.gz | 2908844 |\n",
    "|HG03118|ESN|M|Low-Coverage, WGS| ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR187/ERR187387/ERR187387_1.fastq.gz | 2155306 |\n",
    "|HG03118|ESN|M|Low-Coverage, WGS| ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR187/ERR187387/ERR187387_2.fastq.gz | 2234829 |\n",
    "|HG02716|GWD|F|V. Low-Coverage, WGS| ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR583/SRR583743/SRR583743_1.fastq.gz | 289380 |\n",
    "|HG02716|GWD|F|V. Low-Coverage, WGS| ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR583/SRR583743/SRR583743_2.fastq.gz| 298947 |\n",
    "| | | | | | |\n",
    "|NA19240|YRI|F|WGS, PCR-free|  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR894/ERR894723/ERR894723_1.fastq.gz|24068394|\n",
    "|NA19240|YRI|F|WGS, PCR-free| ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR894/ERR894723/ERR894723_2.fastq.gz|24313932|\n",
    "|NA19238|YRI|F|WGS, PCR-free| ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR899/ERR899706/ERR899706_1.fastq.gz|5423970|\n",
    "|NA19238|YRI|F|WGS, PCR-free| ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR899/ERR899706/ERR899706_2.fastq.gz|5554683|\n",
    "|NA19239|YRI|M|WGS, PCR-free| ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR899/ERR899707/ERR899707_1.fastq.gz|8052464|\n",
    "|NA19239|YRI|M|WGS, PCR-free| ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR899/ERR899707/ERR899707_2.fastq.gz|8192156|\n",
    "|NA12878|CEU|F|WGS| ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR622/SRR622457/SRR622457_1.fastq.gz|49355706|\n",
    "|NA12878|CEU|F|WGS| ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR622/SRR622457/SRR622457_2.fastq.gz|69689596|\n",
    "|NA20845|GIH|M|WGS| ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR129/002/SRR1295542/SRR1295542_1.fastq.gz|27606434|\n",
    "|NA20845|GIH|M|WGS| ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR129/002/SRR1295542/SRR1295542_2.fastq.gz|28326194|\n",
    "|NA18525|CHB|F|WGS| ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR129/009/SRR1295539/SRR1295539_1.fastq.gz|22751767|\n",
    "|NA18525|CHB|F|WGS| ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR129/009/SRR1295539/SRR1295539_2.fastq.gz|22822761|\n",
    "|HG00268|FIN|F|WGS| ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR129/002/SRR1291262/SRR1291262_1.fastq.gz|5291105|\n",
    "|HG00268|FIN|F|WGS| ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR129/002/SRR1291262/SRR1291262_2.fastq.gz|4697590|\n",
    "|HG00731|PUR|M|WGS, PCR-free| ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR903/ERR903028/ERR903028_1.fastq.gz|24996442|\n",
    "|HG00731|PUR|M|WGS, PCR-free| ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR903/ERR903028/ERR903028_2.fastq.gz|24996442|\n",
    "|HG00096|GBR|M|WGS| ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR129/006/SRR1291026/SRR1291026_1.fastq.gz|23936288|\n",
    "|HG00096|GBR|M|WGS| ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR129/006/SRR1291026/SRR1291026_2.fastq.gz|25011312|\n",
    "|NA19017|LWK|F|WGS| ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR129/004/SRR1295544/SRR1295544_1.fastq.gz|23968743|\n",
    "|NA19017|LWK|F|WGS| ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR129/004/SRR1295544/SRR1295544_2.fastq.gz|25660691|\n",
    "\n",
    "\n",
    "## Pilot Pilot Run\n",
    "\n",
    "To set-up that analysis; I'm going to have to run a pilot of the pilot run. To make this infomrative and quick, I'll use a low-coverage WGS data-set which should show the difference between high coverage and low-coverage WGS over rDNA.\n",
    "\n",
    "#### Parameters\n",
    "* Reads\n",
    "    * [NA19240 Low Coverage Read 1](ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/SRR794330/SRR794330_1.fastq.gz)\n",
    "    * [NA19240 Low Coverage Read 2](ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/SRR794330/SRR794330_1.fastq.gz)\n",
    "* Genome: hgr.fa\n",
    "* AMI: crown-161229\n",
    "* EC2 CPU: c4.xlarge / m4.xlarge\n",
    "* Storage: 80 Gb\n",
    "* spot bid: 0.06\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "code_folding": [
     0
    ],
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# EC2 Command History\n",
    "\n",
    "# Make working directory\n",
    "    mkdir align; cd align\n",
    "\n",
    "# Download Genoem Sequencing Data\n",
    "  wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/SRR794330/SRR794330_1.fastq.gz\n",
    "  wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR794/SRR794330/SRR794330_2.fastq.gz\n",
    "\n",
    "# Copy hgr genome and create bowtie2 index\n",
    "  cp ~/resources/hgr.fa ./\n",
    "  cp ~/resources/hgr.fa.fai ./\n",
    "  \n",
    "  bowtie2-build hgr.fa hgr\n",
    "  \n",
    "# Read Group Data\n",
    "RGID='SRR794330' # Read Group ID. Accession Number\n",
    "RGSM='NA19240'   # Sample. Patient Identifer\n",
    "RGLB='SRR794330' # Library Name. Accession Number\n",
    "RGPL='ILLUMINA'  # Sequencing Platform.\n",
    "# Extract Sequencing Run Info\n",
    "RGPU=$(gzip -dc SRR794330_1.fastq.gz | head -n1 - | cut -f1 -d':' | cut -f2 -d' ')\n",
    "\n",
    "\n",
    "# Bowtie2: align to genome\n",
    "bowtie2 \\\n",
    "--very-sensitive-local -p 2 \\\n",
    "--rg-id $RGID --rg LB:$RGLB \\\n",
    "--rg SM:$RGSM --rg PL:$RGPL \\\n",
    "--rg PU:$RGPU \\\n",
    "-x hgr -1 SRR794330_1.fastq.gz -2 SRR794330_2.fastq.gz\n",
    "| samtools view -bS - > aligned_unsorted.bam\n",
    "\n",
    "bowtie2 --very-sensitive-local -p 2 --rg-id $RGID --rg LB:$RGLB --rg SM:$RGSM --rg PL:$RGPL --rg PU:$RGPU -x hgr -1 SRR794330_1.fastq.gz -2 SRR794330_2.fastq.gz -S aligned_reads.sam\n",
    "\n",
    "# samtools view -bS aligned_reads.sam > aligned_unsorted.bam\n",
    "# rm aligned_reads.sam\n",
    "\n",
    "samtools sort aligned_unsorted.bam aligned\n",
    "samtools index aligned.bam\n",
    "rm aligned_unsorted.bam\n",
    "\n",
    "# Re-opened in AM: changed to EC2 m4.xlarge instance\n",
    "\n",
    "# Extract mapped reads, and their unmapped pairs\n",
    "# surprisingly difficult...\n",
    "\n",
    "# Unmapped reads with mapped pairs\n",
    "\n",
    "    # Extract Mapped Reads\n",
    "    # and their unmapped pairs\n",
    "    samtools view -b -F 4 aligned.bam > align.F4.bam #mapped\n",
    "    samtools view -b -f 4 -F 8 aligned.bam > align.f4F8.bam #unmapped pairs\n",
    "    \n",
    "    # Read Counts: align.F4 and align.f4F8\n",
    "    # 4216876 Reads: 2098501 read1  2118375 read2\n",
    "    # 1045242 Reads:  532558 read1   512684 read2\n",
    "    # 5262118        2631059 read1  2631059 read2    \n",
    "    \n",
    "    # Extract Header\n",
    "    samtools view -H align.F4.bam > align.header.tmp\n",
    "        \n",
    "    # Extract just the 45S unit\n",
    "    echo -e \"chr13\\t999000\\t1013500\" > rDNA_45s.bed\n",
    "    samtools view -b -L rDNA-45s.bed align.F4.bam > align.F4.45s.bam\n",
    "    \n",
    "    # What are the 45S mapped readnames\n",
    "    samtools view align.F4.45s.bam | cut -f1 - > read.names.45s.tmp\n",
    "    \n",
    "    # Extract cases of read pairs mapped on edge of cut-off\n",
    "    samtools view align.F4.bam | grep -Ff read.names.45s.txt - > align.F4.tmp.sam\n",
    "\n",
    "    # Complete mapped reads list\n",
    "    cut -f1 align.F4.tmp.sam > read.names.45s.long.tmp\n",
    "\n",
    "    # Extract unmapped reads with a mapped pair in 45s.long\n",
    "    samtools view align.f4F8.bam | grep -Ff read.names.45s.long.tmp - > align.f4F8.tmp.sam\n",
    "\n",
    "    # Re-compile bam file\n",
    "    cat align.header.tmp align.F4.tmp.sam align.f4F8.tmp.sam | samtools view -bS - > align.45s.tmp.bam\n",
    "      samtools sort align.45s.tmp.bam align.45s\n",
    "      samtools index align.45s.bam\n",
    "      rm align.45s.tmp.bam\n",
    "      \n",
    "    # Read Counts: align.45s\n",
    "    # 66736 Reads\n",
    "    # 60877 mapped (91.22%)\n",
    "    # 33368 read1; 33368 read2\n",
    "    \n",
    "    # Recompile the total rDNA mapped read pairs\n",
    "    samtools cat -h align.header.tmp -o align.rDNA.bam align.F4.bam align.f4F8.bam\n",
    "    samtools index align.rDNA.bam\n",
    "    \n",
    "    \n",
    "    rm *tmp* # Clean-up\n",
    "\n",
    "# Rename the final Bam Files\n",
    "    mv aligned.bam NA19240_lowcov.bam\n",
    "    mv aligned.bam.bai NA19240_lowcov.bam.bai\n",
    "\n",
    "    mv align.rDNA.bam NA19240_lowcov.rDNA.bam\n",
    "    mv align.rDNA.bam.bai NA19240_lowcov.rDNA.bam.bai\n",
    "    \n",
    "    mv align.45s.bam NA19240_lowcov.45s.bam\n",
    "    mv align.45s.bam.bai NA19240_lowcov.45s.bam.bai\n",
    "\n",
    "# aws s3 cp aligned.bam s3://crownproject/1kg_pilot/NA19240_lowcov.bam\n",
    "# aws s3 cp aligned.bam.bai s3://crownproject/1kg_pilot/NA19240_lowcov.bam.bai\n",
    "# Closed instance for the night\n",
    "\n",
    "# GATK variant calling over 45S region\n",
    "  aws s3 cp s3://crownproject/resources/hgr.gatk.fa ./\n",
    "  aws s3 cp s3://crownproject/resources/hgr.gatk.fa.fai ./\n",
    "  aws s3 cp s3://crownproject/resources/hgr.gatk.dict ./\n",
    "  \n",
    "  java -Xmx12G -jar /home/ubuntu/software/GenomeAnalysisTK.jar \\\n",
    "  -R hgr.gatk.fa -T HaplotypeCaller \\\n",
    "  -ploidy 2 --max_alternate_alleles 6 \\\n",
    "  -I NA19240_lowcov.bam -o NA19240.rDNA_p2.vcf\n",
    "   # Memory issues, restrict to 45S region only\n",
    "       # -ploidy 100, 50, 20 failed... do 2 and analyze 45S further\n",
    "       \n",
    "  java -Xmx12G -jar /home/ubuntu/software/GenomeAnalysisTK.jar \\\n",
    "  -R hgr.gatk.fa -T HaplotypeCaller \\\n",
    "  -ploidy 50 --max_alternate_alleles 10 \\\n",
    "  -I NA19240_lowcov.45s.bam -o NA19240.45s_p50.vcf\n",
    "\n",
    "\n",
    "# Upload full file to amazon Glacier valut\n",
    "aws s3 cp s3://crownproject/scripts/treehash.py ./\n",
    " chmod 755 treehash.py\n",
    "aws s3 cp s3://crownproject/scripts/glacierupload.sh ./\n",
    " chmod 755 glacierupload.sh\n",
    " \n",
    " ./glacierupload.sh NA19240_lowcov.bam $AWSID\n",
    " \n",
    " aws s3 cp NA19240_lowcov.bam.glacer s3://crownproject/1kg_pilot/\n",
    "\n",
    "\n",
    "# Upload final output files to S3\n",
    " \n",
    "# 45S Only Alignent\n",
    " aws s3 cp NA19240_lowcov.45s.bam s3://crownproject/1kg_pilot/\n",
    " aws s3 cp NA19240_lowcov.45s.bam.bai s3://crownproject/1kg_pilot/\n",
    " \n",
    "# rDNA Only Alignent\n",
    " aws s3 cp NA19240_lowcov.rDNA.bam s3://crownproject/1kg_pilot/\n",
    " aws s3 cp NA19240_lowcov.rDNA.bam.bai s3://crownproject/1kg_pilot/\n",
    "\n",
    "#VCF over 45S\n",
    " aws s3 cp NA19240.45s_p50.vcf s3://crownproject/1kg_pilot/\n",
    " aws s3 cp NA19240.45s_p50.vcf.idx s3://crownproject/1kg_pilot/\n",
    " \n",
    "#VCF over rDNA\n",
    " aws s3 cp NA19240.rDNA_p2.vcf s3://crownproject/1kg_pilot/\n",
    " aws s3 cp NA19240.rDNA_p2.vcf.idx s3://crownproject/1kg_pilot/\n",
    "\n",
    "  \n",
    " # samtools view file.bam | grep queryname - > subset.sam\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Run took ~45 minutes\n",
    "\n",
    "##### Standard Output - bowtie2\n",
    "```\n",
    "35674078 reads; of these:\n",
    "  35674078 (100.00%) were paired; of these:\n",
    "    34209349 (95.89%) aligned concordantly 0 times\n",
    "    582559 (1.63%) aligned concordantly exactly 1 time\n",
    "    882170 (2.47%) aligned concordantly >1 times\n",
    "    ----\n",
    "    34209349 pairs aligned concordantly 0 times; of these:\n",
    "      48972 (0.14%) aligned discordantly 1 time\n",
    "    ----\n",
    "    34160377 pairs aligned 0 times concordantly or discordantly; of these:\n",
    "      68320754 mates make up the pairs; of these:\n",
    "        67131280 (98.26%) aligned 0 times\n",
    "        508149 (0.74%) aligned exactly 1 time\n",
    "        681325 (1.00%) aligned >1 times\n",
    "5.91% overall alignment rate\n",
    "```\n",
    "##### file-sizes post alignment\n",
    "```\n",
    "ls -alh\n",
    "total 29G\n",
    "drwxrwxr-x 2 ubuntu ubuntu 4.0K Jan  6 01:23 .\n",
    "drwxr-xr-x 9 ubuntu ubuntu 4.0K Jan  6 00:35 ..\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  18G Jan  6 01:13 aligned_reads.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 5.4G Jan  6 01:50 aligned_unsorted.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.1M Jan  6 00:01 hgr.1.bt2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  11K Jan  6 00:01 hgr.2.bt2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  125 Jan  6 00:01 hgr.3.bt2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  11K Jan  6 00:01 hgr.4.bt2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.1M Jan  6 00:00 hgr.fa\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  16K Jan  6 00:00 hgr.fa.fai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.1M Jan  6 00:01 hgr.rev.1.bt2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  11K Jan  6 00:01 hgr.rev.2.bt2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 2.9G Jan  6 00:18 SRR794330_1.fastq.gz\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 2.8G Jan  6 00:27 SRR794330_2.fastq.gz\n",
    "```\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "code_folding": [
     0
    ],
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "# 1kgpilot.sh script\n",
    "# AMI: crown-161229 - ami-22ca7b42\n",
    "# EC2: c4.2xlarge (8cpu / 15 gb)\n",
    "# Storage: 100 Gb\n",
    "# ALPHA\n",
    "# Start: 20:13\n",
    "# Alignment done: 20:34\n",
    "# Align.subset done: 21:07\n",
    "# End: 21:32\n",
    "# (note: this invludes revisions to fix bugs as it went along)\n",
    "\n",
    "# Control Panel -------------------------------\n",
    "# CPU\n",
    "\tTHREADS='6'\n",
    "\tAWSID='#########'\n",
    "\n",
    "# Sequencing Data\n",
    "\tLIBRARY='HG03118_lowcov' # Library/ File name\n",
    "\tFASTQ1='ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR187/ERR187387/ERR187387_1.fastq.gz'\n",
    "\tFASTQ2='ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR187/ERR187387/ERR187387_2.fastq.gz'\n",
    "    \n",
    "    # File-names\n",
    "    FQ1=$(basename $FASTQ1)\n",
    "    FQ2=$(basename $FASTQ2)\n",
    "\n",
    "# Read Group Data\n",
    "\tRGID='ERR187387' # Read Group ID. Accession Number\n",
    "\tRGSM='HG03118'   # Sample. Patient Identifer\n",
    "\tRGLB=$LIBRARY # Library Name. Accession Number\n",
    "\tRGPL='ILLUMINA'  # Sequencing Platform.\n",
    "\tRGPO='ESN' # Patient Population\n",
    "\t# Extract Sequencing Run Info\n",
    "\t#  RGPU=$(gzip -dc $FQ1 | head -n1 - | cut -f1 -d':' | cut -f2 -d' ')\n",
    "\n",
    "\n",
    "# Initialize wordir ---------------------------\n",
    "\n",
    "# Make working directory\n",
    "  mkdir -p align; cd align\n",
    "\n",
    "# Copy hgr genome and create bowtie2 index\n",
    "  cp ~/resources/hgr.fa ./\n",
    "  cp ~/resources/hgr.fa.fai ./\n",
    "  \n",
    "  bowtie2-build hgr.fa hgr\n",
    "  \n",
    "# Download Genome Sequencing Data\n",
    "  wget $FASTQ1\n",
    "  wget $FASTQ2\n",
    "\n",
    "    # Extract Sequencing Run Info\n",
    "    RGPU=$(gzip -dc $FQ1| head -n1 - | cut -f1 -d':' | cut -f2 -d' ')\n",
    "\n",
    "\n",
    "# Treehash calculators / Glacier Upload script\n",
    "aws s3 cp s3://crownproject/scripts/treehash.py ./\n",
    " chmod 755 treehash.py\n",
    "aws s3 cp s3://crownproject/scripts/glacierupload.sh ./\n",
    " chmod 755 glacierupload.sh\n",
    "\n",
    "# Primary Alignment -------------------------\n",
    "\n",
    "# Bowtie2: align to genome\n",
    "\n",
    "bowtie2 --very-sensitive-local -p $THREADS --rg-id $RGID --rg LB:$RGLB --rg SM:$RGSM --rg PL:$RGPL --rg PU:$RGPU -x hgr -1 $FQ1 -2 $FQ2 | samtools view -bS - > aligned_unsorted.bam\n",
    "\n",
    "# Sort alignment file\n",
    "  samtools sort aligned_unsorted.bam aligned\n",
    "  samtools index aligned.bam\n",
    "  rm aligned_unsorted.bam\n",
    "\n",
    "\n",
    "# Read Subset ------------------------------\n",
    "# Extract mapped reads, and their unmapped pairs\n",
    "\n",
    "  # Extract Header\n",
    "  samtools view -H aligned.bam > align.header.tmp\n",
    "\n",
    "  # Unmapped reads with mapped pairs\n",
    "  # Extract Mapped Reads\n",
    "  # and their unmapped pairs\n",
    "  samtools view -b -F 4 aligned.bam > align.F4.bam #mapped\n",
    "  samtools view -b -f 4 -F 8 aligned.bam > align.f4F8.bam #unmapped pairs\n",
    "  \n",
    "  # Extract just the 45S unit\n",
    "  echo -e \"chr13\\t999000\\t1013500\" > rDNA_45s.bed\n",
    "  samtools view -b -L rDNA_45s.bed align.F4.bam > align.F4.45s.bam\n",
    "  \n",
    "  # What are the 45S mapped readnames\n",
    "  samtools view align.F4.45s.bam | cut -f1 - > read.names.45s.tmp\n",
    "  \n",
    "  # Extract cases of read pairs mapped on edge of region of interest\n",
    "  # -------|======= R O I ======| ----------\n",
    "  # read:                  ====---====\n",
    "\n",
    "  samtools view align.F4.bam | grep -Ff read.names.45s.tmp - > align.F4.tmp.sam\n",
    "\n",
    "  # Complete mapped reads list\n",
    "  cut -f1 align.F4.tmp.sam > read.names.45s.long.tmp\n",
    "\n",
    "  # Extract unmapped reads with a mapped pair in 45s.long\n",
    "  samtools view align.f4F8.bam | grep -Ff read.names.45s.long.tmp - > align.f4F8.tmp.sam\n",
    "\n",
    "  # Re-compile bam file\n",
    "  cat align.header.tmp align.F4.tmp.sam align.f4F8.tmp.sam | samtools view -bS - > align.45s.tmp.bam\n",
    "    samtools sort align.45s.tmp.bam align.45s\n",
    "    samtools index align.45s.bam\n",
    "    rm align.45s.tmp.bam\n",
    "    \n",
    "  # Read Counts: align.45s\n",
    "  # 66736 Reads\n",
    "  # 60877 mapped (91.22%)\n",
    "  # 33368 read1; 33368 read2\n",
    "  \n",
    "  # Recompile the total rDNA mapped read pairs\n",
    "  samtools cat -h align.header.tmp -o align.rDNA.bam align.F4.bam align.f4F8.bam\n",
    "  samtools index align.rDNA.bam\n",
    "  \n",
    "# Calcualte library flagstats\n",
    "  samtools flagstat aligned.bam > aligned.flagstat\n",
    "  samtools flagstat align.45s.bam > align.45s.flagstat\n",
    "  samtools flagstat align.rDNA.bam > align.rDNA.flagstat\n",
    "  \n",
    "  rm *tmp* # Clean-up\n",
    "\n",
    "# Rename the final Bam Files\n",
    "  mv aligned.bam $LIBRARY.bam\n",
    "  mv aligned.bam.bai $LIBRARY.bam.bai\n",
    "  mv aligned.flagstat $LIBRARY.flagstat\n",
    "\n",
    "  mv align.rDNA.bam $LIBRARY.rDNA.bam\n",
    "  mv align.rDNA.bam.bai $LIBRARY.rDNA.bam.bai\n",
    "  mv align.rDNA.flagstat $LIBRARY.rDNA.flagstat\n",
    "\n",
    "  mv align.45s.bam $LIBRARY.45s.bam\n",
    "  mv align.45s.bam.bai $LIBRARY.45s.bam.bai\n",
    "  mv align.45s.flagstat $LIBRARY.45s.flagstat\n",
    "\n",
    "\n",
    "# GATK variant calling over 45S region\n",
    "  aws s3 cp s3://crownproject/resources/hgr.gatk.fa ./\n",
    "  aws s3 cp s3://crownproject/resources/hgr.gatk.fa.fai ./\n",
    "  aws s3 cp s3://crownproject/resources/hgr.gatk.dict ./\n",
    "  \n",
    "  java -Xmx12G -jar /home/ubuntu/software/GenomeAnalysisTK.jar \\\n",
    "  -R hgr.gatk.fa -T HaplotypeCaller \\\n",
    "  -ploidy 2 --max_alternate_alleles 6 \\\n",
    "  -I $LIBRARY.bam -o $LIBRARY.rDNA_p2.vcf\n",
    "   # Memory issues, restrict to 45S region only\n",
    "     # -ploidy 100, 50, 20 failed... do 2 and analyze 45S further\n",
    "     \n",
    "  java -Xmx12G -jar /home/ubuntu/software/GenomeAnalysisTK.jar \\\n",
    "  -R hgr.gatk.fa -T HaplotypeCaller \\\n",
    "  -ploidy 50 --max_alternate_alleles 10 \\\n",
    "  -I $LIBRARY.45s.bam -o $LIBRARY.45s_p50.vcf\n",
    "\n",
    "\n",
    "# Upload final output files to S3\n",
    " \n",
    "# 45S Only Alignent\n",
    " aws s3 cp $LIBRARY.45s.bam s3://crownproject/1kg_pilot/\n",
    " aws s3 cp $LIBRARY.45s.bam.bai s3://crownproject/1kg_pilot/\n",
    " aws s3 cp $LIBRARY.45s.flagstat s3://crownproject/1kg_pilot/\n",
    "\n",
    "# rDNA Only Alignent\n",
    " aws s3 cp $LIBRARY.rDNA.bam s3://crownproject/1kg_pilot/\n",
    " aws s3 cp $LIBRARY.rDNA.bam.bai s3://crownproject/1kg_pilot/\n",
    " aws s3 cp $LIBRARY.rDNA.flagstat s3://crownproject/1kg_pilot/\n",
    "\n",
    "#VCF over 45S\n",
    " aws s3 cp $LIBRARY.45s_p50.vcf s3://crownproject/1kg_pilot/\n",
    " aws s3 cp $LIBRARY.45s_p50.vcf.idx s3://crownproject/1kg_pilot/\n",
    " \n",
    "#VCF over rDNA\n",
    " aws s3 cp $LIBRARY.rDNA_p2.vcf s3://crownproject/1kg_pilot/\n",
    " aws s3 cp $LIBRARY.rDNA_p2.vcf.idx s3://crownproject/1kg_pilot/\n",
    "\n",
    "  \n",
    "# Upload full bam file to amazon Glacier vault\n",
    "  aws s3 cp $LIBRARY.flagstat s3://crownproject/1kg_pilot/\n",
    "  ./glacierupload.sh $LIBRARY.bam $AWSID\n",
    "  aws s3 cp $LIBRARY.bam.glacer s3://crownproject/1kg_pilot/\n",
    "\n",
    "\n",
    "# Script complete"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "code_folding": [],
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "# 1kgpilot.sh script\n",
    "# AMI: crown-161229 - ami-22ca7b42\n",
    "# EC2: c4.2xlarge (8cpu / 15 gb)\n",
    "# Storage: 100 Gb\n",
    "# BETA\n",
    "# Start: \n",
    "# Alignment done: \n",
    "# Align.subset done: \n",
    "# End:\n",
    "# (note: this invludes revisions to fix bugs as it went along)\n",
    "\n",
    "# Control Panel -------------------------------\n",
    "# CPU\n",
    "\tTHREADS='6'\n",
    "\tAWSID='#########' #Not needed if Glacier not used\n",
    "\n",
    "# Sequencing Data\n",
    "\tLIBRARY='HG02716_vlowcov' # Library/ File name\n",
    "\tFASTQ1='ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR583/SRR583743/SRR583743_1.fastq.gz'\n",
    "\tFASTQ2='ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR583/SRR583743/SRR583743_2.fastq.gz'\n",
    "\n",
    "    # File-names\n",
    "    FQ1=$(basename $FASTQ1)\n",
    "    FQ2=$(basename $FASTQ2)\n",
    "\n",
    "# Read Group Data\n",
    "\tRGID='SRR583743' # Read Group ID. Accession Number\n",
    "\tRGSM='HG02716'   # Sample. Patient Identifer\n",
    "\tRGLB=$LIBRARY # Library Name. Accession Number\n",
    "\tRGPL='ILLUMINA'  # Sequencing Platform.\n",
    "\tRGPO='GWD' # Patient Population\n",
    "\t# Extract Sequencing Run Info\n",
    "\t#  RGPU=$(gzip -dc $FQ1 | head -n1 - | cut -f1 -d':' | cut -f2 -d' ')\n",
    "\n",
    "\n",
    "# Initialize wordir ---------------------------\n",
    "\n",
    "# Make working directory\n",
    "  mkdir -p align; cd align\n",
    "\n",
    "# Copy hgr genome and create bowtie2 index\n",
    "  cp ~/resources/hgr.fa ./\n",
    "  cp ~/resources/hgr.fa.fai ./\n",
    "  \n",
    "  bowtie2-build hgr.fa hgr\n",
    "  \n",
    "# Download Genome Sequencing Data\n",
    "  wget $FASTQ1\n",
    "  wget $FASTQ2\n",
    "\n",
    "    # Extract Sequencing Run Info\n",
    "    RGPU=$(gzip -dc $FQ1| head -n1 - | cut -f1 -d':' | cut -f2 -d' ')\n",
    "\n",
    "\n",
    "# Treehash calculators / Glacier Upload script\n",
    "aws s3 cp s3://crownproject/scripts/treehash.py ./\n",
    " chmod 755 treehash.py\n",
    "aws s3 cp s3://crownproject/scripts/glacierupload.sh ./\n",
    " chmod 755 glacierupload.sh\n",
    "\n",
    "# Primary Alignment -------------------------\n",
    "\n",
    "# Bowtie2: align to genome\n",
    "\n",
    "bowtie2 --very-sensitive-local -p $THREADS --rg-id $RGID --rg LB:$RGLB --rg SM:$RGSM --rg PL:$RGPL --rg PU:$RGPU -x hgr -1 $FQ1 -2 $FQ2 | samtools view -bS - > aligned_unsorted.bam\n",
    "\n",
    "# Sort alignment file\n",
    "  samtools sort aligned_unsorted.bam aligned\n",
    "  samtools index aligned.bam\n",
    "  rm aligned_unsorted.bam\n",
    "\n",
    "\n",
    "# Read Subset ------------------------------\n",
    "# Extract mapped reads, and their unmapped pairs\n",
    "\n",
    "  # Extract Header\n",
    "  samtools view -H aligned.bam > align.header.tmp\n",
    "\n",
    "  # Unmapped reads with mapped pairs\n",
    "  # Extract Mapped Reads\n",
    "  # and their unmapped pairs\n",
    "  samtools view -b -F 4 aligned.bam > align.F4.bam #mapped\n",
    "  samtools view -b -f 4 -F 8 aligned.bam > align.f4F8.bam #unmapped pairs\n",
    "  \n",
    "  # Extract just the 45S unit\n",
    "  aws s3 cp s3://crownproject/resources/rDNA_45s.bed ./\n",
    "  samtools view -b -L rDNA_45s.bed align.F4.bam > align.F4.45s.bam\n",
    "  \n",
    "  # What are the 45S mapped readnames\n",
    "  samtools view align.F4.45s.bam | cut -f1 - > read.names.45s.tmp\n",
    "  \n",
    "  # Extract cases of read pairs mapped on edge of region of interest\n",
    "  # -------|======= R O I ======| ----------\n",
    "  # read:                  ====---====\n",
    "\n",
    "  samtools view align.F4.bam | grep -Ff read.names.45s.tmp - > align.F4.tmp.sam\n",
    "\n",
    "  # Complete mapped reads list\n",
    "  cut -f1 align.F4.tmp.sam > read.names.45s.long.tmp\n",
    "\n",
    "  # Extract unmapped reads with a mapped pair in 45s.long\n",
    "  samtools view align.f4F8.bam | grep -Ff read.names.45s.long.tmp - > align.f4F8.tmp.sam\n",
    "\n",
    "  # Re-compile bam file\n",
    "  cat align.header.tmp align.F4.tmp.sam align.f4F8.tmp.sam | samtools view -bS - > align.45s.tmp.bam\n",
    "    samtools sort align.45s.tmp.bam align.45s\n",
    "    samtools index align.45s.bam\n",
    "    rm align.45s.tmp.bam\n",
    "    \n",
    "  # Read Counts: align.45s\n",
    "  # 66736 Reads\n",
    "  # 60877 mapped (91.22%)\n",
    "  # 33368 read1; 33368 read2\n",
    "  \n",
    "  # Recompile the total rDNA mapped read pairs\n",
    "  samtools cat -h align.header.tmp -o align.rDNA.bam align.F4.bam align.f4F8.bam\n",
    "  samtools index align.rDNA.bam\n",
    "  \n",
    "# Calcualte library flagstats\n",
    "  samtools flagstat aligned.bam > aligned.flagstat\n",
    "  samtools flagstat align.45s.bam > align.45s.flagstat\n",
    "  samtools flagstat align.rDNA.bam > align.rDNA.flagstat\n",
    "  \n",
    "  rm *tmp* # Clean-up\n",
    "\n",
    "# Rename the final Bam Files\n",
    "  mv aligned.bam $LIBRARY.bam\n",
    "  mv aligned.bam.bai $LIBRARY.bam.bai\n",
    "  mv aligned.flagstat $LIBRARY.flagstat\n",
    "\n",
    "  mv align.rDNA.bam $LIBRARY.rDNA.bam\n",
    "  mv align.rDNA.bam.bai $LIBRARY.rDNA.bam.bai\n",
    "  mv align.rDNA.flagstat $LIBRARY.rDNA.flagstat\n",
    "\n",
    "  mv align.45s.bam $LIBRARY.45s.bam\n",
    "  mv align.45s.bam.bai $LIBRARY.45s.bam.bai\n",
    "  mv align.45s.flagstat $LIBRARY.45s.flagstat\n",
    "\n",
    "\n",
    "# GATK variant calling over 45S region\n",
    "  aws s3 cp s3://crownproject/resources/hgr.gatk.fa ./\n",
    "  aws s3 cp s3://crownproject/resources/hgr.gatk.fa.fai ./\n",
    "  aws s3 cp s3://crownproject/resources/hgr.gatk.dict ./\n",
    "  \n",
    "  java -Xmx12G -jar /home/ubuntu/software/GenomeAnalysisTK.jar \\\n",
    "  -R hgr.gatk.fa -T HaplotypeCaller \\\n",
    "  -ploidy 2 --max_alternate_alleles 6 \\\n",
    "  -I $LIBRARY.bam -o $LIBRARY.rDNA_p2.vcf\n",
    "   # Memory issues, restrict to 45S region only\n",
    "     # -ploidy 100, 50, 20 failed... do 2 and analyze 45S further\n",
    "     \n",
    "  java -Xmx12G -jar /home/ubuntu/software/GenomeAnalysisTK.jar \\\n",
    "  -R hgr.gatk.fa -T HaplotypeCaller \\\n",
    "  -ploidy 50 --max_alternate_alleles 10 \\\n",
    "  -I $LIBRARY.45s.bam -o $LIBRARY.45s_p50.vcf\n",
    "\n",
    "\n",
    "# Upload final output files to S3\n",
    " \n",
    "# 45S Only Alignent\n",
    " aws s3 cp $LIBRARY.45s.bam s3://crownproject/1kg_pilot/\n",
    " aws s3 cp $LIBRARY.45s.bam.bai s3://crownproject/1kg_pilot/\n",
    " aws s3 cp $LIBRARY.45s.flagstat s3://crownproject/1kg_pilot/\n",
    "\n",
    "# rDNA Only Alignent\n",
    " aws s3 cp $LIBRARY.rDNA.bam s3://crownproject/1kg_pilot/\n",
    " aws s3 cp $LIBRARY.rDNA.bam.bai s3://crownproject/1kg_pilot/\n",
    " aws s3 cp $LIBRARY.rDNA.flagstat s3://crownproject/1kg_pilot/\n",
    "\n",
    "#VCF over 45S\n",
    " aws s3 cp $LIBRARY.45s_p50.vcf s3://crownproject/1kg_pilot/\n",
    " aws s3 cp $LIBRARY.45s_p50.vcf.idx s3://crownproject/1kg_pilot/\n",
    " \n",
    "#VCF over rDNA\n",
    " aws s3 cp $LIBRARY.rDNA_p2.vcf s3://crownproject/1kg_pilot/\n",
    " aws s3 cp $LIBRARY.rDNA_p2.vcf.idx s3://crownproject/1kg_pilot/\n",
    "\n",
    "  \n",
    "# Upload full bam file to amazon Glacier vault (not done)\n",
    "  aws s3 cp $LIBRARY.flagstat s3://crownproject/1kg_pilot/\n",
    "\n",
    "# Alignments are so fast, it's nto worth storing on glacier, just-realign when needed\n",
    "  #./glacierupload.sh $LIBRARY.bam $AWSID\n",
    "  #aws s3 cp $LIBRARY.bam.glacer s3://crownproject/1kg_pilot/\n",
    "\n",
    "# Shutdown and Terminate instance\n",
    "EC2ID=$(ec2metadata --instance-id)\n",
    "aws ec2 terminate-instances --instance-ids $EC2ID\n",
    "\n",
    "# Script complete"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " Pilot Pilot alignments seems to be quite smooth. Runs within an hour.\n",
    " \n",
    " \n",
    " ## Pilot 1 Run\n",
    " \n",
    " Before running all 10 pilot genomes at once; I'll test one out to ensure it runs to completion and look at the output. Then I could script the run and do it en masse.\n",
    " \n",
    " Using the same `1kgpilot_beta.sh` script with modifications for it to accept input from the command line \n",
    "   "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "code_folding": [],
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "# 1kgpilot.sh script\n",
    "# AMI: crown-161229 - ami-22ca7b42\n",
    "# EC2: c4.2xlarge (8cpu / 15 gb)\n",
    "# Storage: 100-500 Gb\n",
    "\n",
    "# Control Panel -------------------------------\n",
    "# CPU\n",
    "\tTHREADS='7'\n",
    "\tAWSID='#########'\n",
    "\n",
    "# Sequencing Data\n",
    "\tLIBRARY=$1 # Library/ File name\n",
    "\tFASTQ1=$5\n",
    "\tFASTQ2=$6\n",
    "\n",
    "    # File-names\n",
    "    FQ1=$(basename $FASTQ1)\n",
    "    FQ2=$(basename $FASTQ2)\n",
    "\n",
    "# Read Group Data\n",
    "\tRGSM=$2   # Sample. Patient Identifer\n",
    "\tRGID=$3 # Read Group ID. Accession Number\n",
    "\tRGLB=$LIBRARY # Library Name. Accession Number\n",
    "\tRGPL='ILLUMINA'  # Sequencing Platform.\n",
    "\tRGPO=$4 # Patient Population\n",
    "\t# Extract Sequencing Run Info\n",
    "\t#  RGPU=$(gzip -dc $FQ1 | head -n1 - | cut -f1 -d':' | cut -f2 -d' ')\n",
    "\n",
    "\n",
    "# Initialize wordir ---------------------------\n",
    "\n",
    "# Make working directory\n",
    "  mkdir -p align; cd align\n",
    "\n",
    "# Copy hgr genome and create bowtie2 index\n",
    "  cp ~/resources/hgr.fa ./\n",
    "  cp ~/resources/hgr.fa.fai ./\n",
    "  \n",
    "  bowtie2-build hgr.fa hgr\n",
    "  \n",
    "# Download Genome Sequencing Data\n",
    "  wget $FASTQ1\n",
    "  wget $FASTQ2\n",
    "\n",
    "    # Extract Sequencing Run Info\n",
    "    RGPU=$(gzip -dc $FQ1| head -n1 - | cut -f1 -d':' | cut -f2 -d' ')\n",
    "\n",
    "\n",
    "# Treehash calculators / Glacier Upload script\n",
    "aws s3 cp s3://crownproject/scripts/treehash.py ./\n",
    " chmod 755 treehash.py\n",
    "aws s3 cp s3://crownproject/scripts/glacierupload.sh ./\n",
    " chmod 755 glacierupload.sh\n",
    "\n",
    "# Primary Alignment -------------------------\n",
    "\n",
    "# Bowtie2: align to genome\n",
    "\n",
    "bowtie2 --very-sensitive-local -p $THREADS --rg-id $RGID --rg LB:$RGLB --rg SM:$RGSM --rg PL:$RGPL --rg PU:$RGPU -x hgr -1 $FQ1 -2 $FQ2 | samtools view -bS - > aligned_unsorted.bam\n",
    "\n",
    "# Sort alignment file\n",
    "  samtools sort aligned_unsorted.bam aligned\n",
    "  samtools index aligned.bam\n",
    "  rm aligned_unsorted.bam\n",
    "\n",
    "\n",
    "# Read Subset ------------------------------\n",
    "# Extract mapped reads, and their unmapped pairs\n",
    "\n",
    "  # Extract Header\n",
    "  samtools view -H aligned.bam > align.header.tmp\n",
    "\n",
    "  # Unmapped reads with mapped pairs\n",
    "  # Extract Mapped Reads\n",
    "  # and their unmapped pairs\n",
    "  samtools view -b -F 4 aligned.bam > align.F4.bam #mapped\n",
    "  samtools view -b -f 4 -F 8 aligned.bam > align.f4F8.bam #unmapped pairs\n",
    "  \n",
    "  # Extract just the 45S unit\n",
    "  aws s3 cp s3://crownproject/resources/rDNA_45s.bed ./\n",
    "  samtools view -b -L rDNA_45s.bed align.F4.bam > align.F4.45s.bam\n",
    "  \n",
    "  # What are the 45S mapped readnames\n",
    "  samtools view align.F4.45s.bam | cut -f1 - > read.names.45s.tmp\n",
    "  \n",
    "  # Extract cases of read pairs mapped on edge of region of interest\n",
    "  # -------|======= R O I ======| ----------\n",
    "  # read:                  ====---====\n",
    "\n",
    "  samtools view align.F4.bam | grep -Ff read.names.45s.tmp - > align.F4.tmp.sam\n",
    "\n",
    "  # Complete mapped reads list\n",
    "  cut -f1 align.F4.tmp.sam > read.names.45s.long.tmp\n",
    "\n",
    "  # Extract unmapped reads with a mapped pair in 45s.long\n",
    "  samtools view align.f4F8.bam | grep -Ff read.names.45s.long.tmp - > align.f4F8.tmp.sam\n",
    "\n",
    "  # Re-compile bam file\n",
    "  cat align.header.tmp align.F4.tmp.sam align.f4F8.tmp.sam | samtools view -bS - > align.45s.tmp.bam\n",
    "    samtools sort align.45s.tmp.bam align.45s\n",
    "    samtools index align.45s.bam\n",
    "    rm align.45s.tmp.bam\n",
    "    \n",
    "  # Read Counts: align.45s\n",
    "  # 66736 Reads\n",
    "  # 60877 mapped (91.22%)\n",
    "  # 33368 read1; 33368 read2\n",
    "  \n",
    "  # Recompile the total rDNA mapped read pairs\n",
    "  samtools cat -h align.header.tmp -o align.rDNA.bam align.F4.bam align.f4F8.bam\n",
    "  samtools index align.rDNA.bam\n",
    "  \n",
    "# Calcualte library flagstats\n",
    "  samtools flagstat aligned.bam > aligned.flagstat\n",
    "  samtools flagstat align.45s.bam > align.45s.flagstat\n",
    "  samtools flagstat align.rDNA.bam > align.rDNA.flagstat\n",
    "  \n",
    "  rm *tmp* # Clean-up\n",
    "\n",
    "# Rename the final Bam Files\n",
    "  mv aligned.bam $LIBRARY.bam\n",
    "  mv aligned.bam.bai $LIBRARY.bam.bai\n",
    "  mv aligned.flagstat $LIBRARY.flagstat\n",
    "\n",
    "  mv align.rDNA.bam $LIBRARY.rDNA.bam\n",
    "  mv align.rDNA.bam.bai $LIBRARY.rDNA.bam.bai\n",
    "  mv align.rDNA.flagstat $LIBRARY.rDNA.flagstat\n",
    "\n",
    "  mv align.45s.bam $LIBRARY.45s.bam\n",
    "  mv align.45s.bam.bai $LIBRARY.45s.bam.bai\n",
    "  mv align.45s.flagstat $LIBRARY.45s.flagstat\n",
    "\n",
    "\n",
    "# GATK variant calling over 45S region\n",
    "  aws s3 cp s3://crownproject/resources/hgr.gatk.fa ./\n",
    "  aws s3 cp s3://crownproject/resources/hgr.gatk.fa.fai ./\n",
    "  aws s3 cp s3://crownproject/resources/hgr.gatk.dict ./\n",
    "  \n",
    "  java -Xmx12G -jar /home/ubuntu/software/GenomeAnalysisTK.jar \\\n",
    "  -R hgr.gatk.fa -T HaplotypeCaller \\\n",
    "  -ploidy 2 --max_alternate_alleles 6 \\\n",
    "  -I $LIBRARY.bam -o $LIBRARY.rDNA_p2.vcf\n",
    "   # Memory issues, restrict to 45S region only\n",
    "     # -ploidy 100, 50, 20 failed... do 2 and analyze 45S further\n",
    "     \n",
    "  java -Xmx12G -jar /home/ubuntu/software/GenomeAnalysisTK.jar \\\n",
    "  -R hgr.gatk.fa -T HaplotypeCaller \\\n",
    "  -ploidy 50 --max_alternate_alleles 10 \\\n",
    "  -I $LIBRARY.45s.bam -o $LIBRARY.45s_p50.vcf\n",
    "\n",
    "\n",
    "# Upload final output files to S3\n",
    " \n",
    "# 45S Only Alignent\n",
    " aws s3 cp $LIBRARY.45s.bam s3://crownproject/1kg_pilot/\n",
    " aws s3 cp $LIBRARY.45s.bam.bai s3://crownproject/1kg_pilot/\n",
    " aws s3 cp $LIBRARY.45s.flagstat s3://crownproject/1kg_pilot/\n",
    "\n",
    "# rDNA Only Alignent\n",
    " aws s3 cp $LIBRARY.rDNA.bam s3://crownproject/1kg_pilot/\n",
    " aws s3 cp $LIBRARY.rDNA.bam.bai s3://crownproject/1kg_pilot/\n",
    " aws s3 cp $LIBRARY.rDNA.flagstat s3://crownproject/1kg_pilot/\n",
    "\n",
    "#VCF over 45S\n",
    " aws s3 cp $LIBRARY.45s_p50.vcf s3://crownproject/1kg_pilot/\n",
    " aws s3 cp $LIBRARY.45s_p50.vcf.idx s3://crownproject/1kg_pilot/\n",
    " \n",
    "#VCF over rDNA\n",
    " aws s3 cp $LIBRARY.rDNA_p2.vcf s3://crownproject/1kg_pilot/\n",
    " aws s3 cp $LIBRARY.rDNA_p2.vcf.idx s3://crownproject/1kg_pilot/\n",
    "\n",
    "  \n",
    "# Upload full bam file to amazon Glacier vault\n",
    "  aws s3 cp $LIBRARY.flagstat s3://crownproject/1kg_pilot/\n",
    "\n",
    "# Alignments are so fast, it's nto worth storing on glacier, just-realign when needed\n",
    "  #./glacierupload.sh $LIBRARY.bam $AWSID\n",
    "  #aws s3 cp $LIBRARY.bam.glacer s3://crownproject/1kg_pilot/\n",
    "\n",
    "\n",
    "# Script complete"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Manually launch SPOT instance\n",
    "# Login to it\n",
    "# Try with 200 Gb of storage on C4.2xlarge\n",
    "# Start: 23:44\n",
    "# Alignment End: 05:17\n",
    "# Sorting expansion done: 6:50\n",
    "# Sort Done: 11:56\n",
    "# VCF1: 13:01\n",
    "# End: 13:12\n",
    "# Peak Memory:~190 Gb @ ~09:00\n",
    "# CPU Usage: Steady at 80% usage; could ramp up to 7/8 CPUs\n",
    "# (I thought the bam pipe would require more CPU usage)\n",
    "\n",
    "aws s3 cp s3://crownproject/scripts/1kgpilot.sh ./\n",
    "\n",
    "screen\n",
    "\n",
    "sh 1kgpilot.sh HG00731_pcr HG00731 ERR903028 PUR ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR903/ERR903028/ERR903028_1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR903/ERR903028/ERR903028_2.fastq.gz \\\n",
    "> HG00731.out.txt 2> HG00731.err.txt\n",
    "\n",
    "# Line 67 changed; optimized for multi-threading\n",
    "#  Delete fastq files before sort (reduce footprint to 130 Gb)\n",
    "#  samtools sort aligned_unsorted.bam aligned\n",
    "#  samtools sort -@ $THREADS -m 1537M aligned_unsorted.bam aligned\n",
    "\n",
    "# Remove 50 ploidy VCF calls, it failed on this run\n",
    "# make sure script runs even if a command fails (so\n",
    "# that the instance shutsdown)\n",
    "\n",
    "# Add shutdown call\n",
    "\n",
    "# Shutdown and Terminate instance\n",
    "EC2ID=$(ec2metadata --instance-id)\n",
    "aws ec2 terminate-instances --instance-ids $EC2ID\n",
    "\n",
    "# Updated 1kgpilot.sh script and uploaded to s3."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "total 54G\n",
    "drwxrwxr-x 2 ubuntu ubuntu  20K Jan 10 13:01 .\n",
    "drwxr-xr-x 9 ubuntu ubuntu 4.0K Jan 10 03:17 ..\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  32M Jan 10 11:52 align.F4.45s.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 2.5G Jan 10 11:23 align.F4.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.7G Jan 10 11:51 align.f4F8.bam\n",
    "-rwxr-xr-x 1 ubuntu ubuntu 2.5K Jan  7 02:23 glacierupload.sh\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  52M Jan 10 11:56 HG00731_pcr.45s.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  832 Jan 10 11:56 HG00731_pcr.45s.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  384 Jan 10 12:18 HG00731_pcr.45s.flagstat\n",
    "-rw-rw-r-- 1 ubuntu ubuntu    0 Jan 10 13:01 HG00731_pcr.45s_p50.vcf\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  45G Jan 10 10:30 HG00731_pcr.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  36K Jan 10 10:51 HG00731_pcr.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  399 Jan 10 12:18 HG00731_pcr.flagstat\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.1G Jan 10 11:56 HG00731_pcr.rDNA.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  397 Jan 10 12:20 HG00731_pcr.rDNA.flagstat\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 114K Jan 10 13:01 HG00731_pcr.rDNA_p2.vcf\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 2.4K Jan 10 13:01 HG00731_pcr.rDNA_p2.vcf.idx\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.1M Jan  9 23:44 hgr.1.bt2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  11K Jan  9 23:44 hgr.2.bt2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  125 Jan  9 23:44 hgr.3.bt2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  11K Jan  9 23:44 hgr.4.bt2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.1M Jan  9 23:44 hgr.fa\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  16K Jan  9 23:44 hgr.fa.fai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  135 Jan  6 02:41 hgr.gatk.dict\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.1M Jan  6 02:40 hgr.gatk.fa\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   22 Jan  6 02:41 hgr.gatk.fa.fai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.1M Jan  9 23:44 hgr.rev.1.bt2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  11K Jan  9 23:44 hgr.rev.2.bt2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   31 Nov 22 02:42 rDNA_45s.bed\n",
    "-rwxr-xr-x 1 ubuntu ubuntu 1.9K Jan  7 00:33 treehash.py\n",
    "```\n",
    "HG00731_pcr.45s.flagstat\n",
    "```\n",
    "571320 + 0 in total (QC-passed reads + QC-failed reads)\n",
    "0 + 0 duplicates\n",
    "464316 + 0 mapped (81.27%:-nan%)\n",
    "571320 + 0 paired in sequencing\n",
    "285660 + 0 read1\n",
    "285660 + 0 read2\n",
    "103350 + 0 properly paired (18.09%:-nan%)\n",
    "357312 + 0 with itself and mate mapped\n",
    "107004 + 0 singletons (18.73%:-nan%)\n",
    "0 + 0 with mate mapped to a different chr\n",
    "0 + 0 with mate mapped to a different chr (mapQ>=5)\n",
    "```\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Manually launch SPOT instance\n",
    "# Login to it\n",
    "# 200 Gb of storage on C4.2xlarge\n",
    "\n",
    "# Start: Tue Jan 10 19:47:44 UTC 2017...\n",
    "## Failed... There is some operational issue on EC2... posibly related.\n",
    "# will retry without the shutdown command for error checkign\n",
    "# also could be samtools sort command failure.\n",
    "\n",
    "# Start2: Wed Jan 11 03:36:17 UTC 2017\n",
    "## Failed... looks like samtools sort was 'killed' in the middle\n",
    "# of a sort. Reduce memory usage (no -m flag) and try again.\n",
    "\n",
    "# Start3: Wed Jan 11 17:39:31 UTC 2017\n",
    "# Finish: Wed Jan 11 21:00:00 UTC 2017\n",
    "# Commented out line 71 & 190 for test run\n",
    "\n",
    "aws s3 cp s3://crownproject/scripts/1kgpilot.sh ./\n",
    "screen\n",
    "\n",
    "sh 1kgpilot.sh NA19240_pcr NA19240 ERR894723 YRI ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR894/ERR894723/ERR894723_1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR894/ERR894723/ERR894723_2.fastq.gz \\\n",
    "> NA19240_pcr.out.txt 2> NA19240_pcr.err.txt\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 4 Pilot Genomes\n",
    "Thu Jan 12 11:06:45 PST 2017\n",
    "\n",
    "1kgpilot pipeline refined to run smoothly. I've completed five alignments; NA19240_lowcov, NA19240_pcr (high coverage, no PCR), HG00731_pcr, HG02716_vlowcov, HG03118_lowcov. Also included is the first high coverage NA19240 alignment I ran beforehand.\n",
    "\n",
    "![First 4 Alignments](../figure/20170112_1kg_4pilot.png)\n",
    "\n",
    "The 5' end of 18S is shown above; key observations\n",
    "* Each individual shares a conserved two-state variation upstream of 18S, ranging from 11% to 26% alternative allele. There are multiple such alelles.\n",
    "\n",
    "    \n",
    "    NA19240_pcr\n",
    "    chr13:1,003,640\n",
    "    <hr>Total count: 6795\n",
    "    A      : 778  (11%,     229+,   549- )\n",
    "    C      : 6003  (88%,     1813+,   4190- ) [Reference]\n",
    "    G      : 8  (0%,     1+,   7- )\n",
    "    T      : 6  (0%,     2+,   4- )\n",
    "    N      : 0\n",
    "    ---------------\n",
    "    DEL: 0\n",
    "    INS: 6\n",
    "\n",
    "\n",
    "* Some variations are shared between individuals; and absent in others.\n",
    "\n",
    "\n",
    "    NA19240_pcr (Yeruba) and HG00731 (Puerto Rico)\n",
    "    chr13:1,003,635\n",
    "    <hr>Total count: 6535\n",
    "    A      : 3  (0%,     2+,   1- )\n",
    "    C      : 6505  (100%,     1953+,   4552- )\n",
    "    G      : 16  (0%,     9+,   7- ) [Reference]\n",
    "    T      : 11  (0%,     4+,   7- )\n",
    "    N      : 0\n",
    "    ---------------\n",
    "    DEL: 0\n",
    "    INS: 1\n",
    "\n",
    "    HG03118_vlowcov[Esan] and HG002716 [Ghana]\n",
    "\n",
    "    chr13:1,003,635\n",
    "    <hr>Total count: 176\n",
    "    A      : 3  (2%,     3+,   0- )\n",
    "    C      : 146  (83%,     108+,   38- )\n",
    "    G      : 26  (15%,     26+,   0- ) [ Reference ]\n",
    "    T      : 1  (1%,     0+,   1- )\n",
    "    N      : 0\n",
    "    ---------------\n",
    "\n",
    "\n",
    "* The 'PCR-Free' genomes have good sequencing over the very-high GC content areas and don't have the same drop-out that PCR-libraries have.\n",
    "\n",
    "\n",
    "    NA19240_hgr\n",
    "    chr13:1,003,486\n",
    "    <hr>Total count: 2\n",
    "    A      : 0\n",
    "    C      : 0\n",
    "    G      : 2  (100%,     1+,   1- ) [Reference]\n",
    "    T      : 0\n",
    "    N      : 0\n",
    "    ---------------\n",
    "\n",
    "    NA19240_pcr\n",
    "    chr13:1,003,486\n",
    "    <hr>Total count: 5287\n",
    "    A      : 5  (0%,     3+,   2- )\n",
    "    C      : 28  (1%,     28+,   0- )\n",
    "    G      : 4927  (93%,     2440+,   2487- ) [Reference]\n",
    "    T      : 327  (6%,     164+,   163- )\n",
    "    N      : 0\n",
    "    ---------------\n",
    "    DEL: 10\n",
    "    INS: 3\n",
    "\n",
    "\n",
    "Even though the PCR-free libraries are high coverage, slow and expensive, it will be worth running the entire set as a first step. This will allow for expansion loop regions to be sequenced and help define the structural variation present in 45s for further ribo-population alignments. Otherwise there will be significant loss of structurally variant SNPs using normal alignment software.\n",
    "\n",
    "### g.C4751T (~4-5%)\n",
    "![g.C4751T variation](../figure/20170112_g.C4751T.png)\n",
    "\n",
    "This variant is interesting; it is at a more conserved position (Entropy: 0.396) and overlaps a contact site for S8. May be a functional variation, present in all libraries (but sub-0.05 allele frequency).\n",
    "\n",
    "Complete pilot study now...\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Pilot 10 Run\n",
    "\n",
    "Set-up seems to be working well. Run the next 8 genomes to complete 10 pilot genomes.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# 200 Gb of storage on C4.2xlarge\n",
    "# On each instance run the commands below:\n",
    "\n",
    "\n",
    "# Common Command\n",
    "aws s3 cp s3://crownproject/scripts/1kgpilot.sh ./\n",
    "screen\n",
    "\n",
    "\n",
    "# Date: Thu Jan 12 23:41:00 UTC 2017\n",
    "# Instance specific command\n",
    "# ec2-35-167-146-38.us-west-2.compute.amazonaws.com -- DONE\n",
    "sh 1kgpilot.sh  NA19238_pcr NA19238 ERR899706 YRI ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR899/ERR899706/ERR899706_1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR899/ERR899706/ERR899706_2.fastq.gz\n",
    "\n",
    "# ec2-35-167-146-34.us-west-2.compute.amazonaws.com -- DONE TRIO!\n",
    "sh 1kgpilot.sh NA19239_pcr NA19239 ERR899707 YRI ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR899/ERR899707/ERR899707_1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR899/ERR899707/ERR899707_2.fastq.gz\n",
    "\n",
    "# ec2-35-167-43-133.us-west-2.compute.amazonaws.com [Ran with 350 Gb space] -- FAILED: ~24 hours!\n",
    "# ec2-35-163-97-252.us-west-2.compute.amazonaws.com [Re-Ran with 500 Gb space] -- Failed too. Discard.\n",
    "sh 1kgpilot.sh NA12878_pcr NA12878 SRR622457 CEU ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR622/SRR622457/SRR622457_1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR622/SRR622457/SRR622457_2.fastq.gz\n",
    "\n",
    "# ec2-35-165-233-175.us-west-2.compute.amazonaws.com -- DONE \n",
    "sh 1kgpilot.sh NA20845 NA20845 SRR1295542 GIH ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR129/002/SRR1295542/SRR1295542_1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR129/002/SRR1295542/SRR1295542_2.fastq.gz\n",
    "\n",
    "# ec2-35-166-57-2.us-west-2.compute.amazonaws.com -- DONE\n",
    "sh 1kgpilot.sh NA18525 NA18525 SRR1295539 CHB ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR129/009/SRR1295539/SRR1295539_1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR129/009/SRR1295539/SRR1295539_2.fastq.gz\n",
    "\n",
    "# ec2-35-166-130-196.us-west-2.compute.amazonaws.com -- DONE\n",
    "sh 1kgpilot.sh HG00268 HG00268 SRR1291262 FIN ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR129/002/SRR1291262/SRR1291262_1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR129/002/SRR1291262/SRR1291262_2.fastq.gz\n",
    "\n",
    "# ec2-35-161-113-239.us-west-2.compute.amazonaws.com -- DONE\n",
    "sh 1kgpilot.sh HG00096 HG00096 SRR1291026 GBR ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR129/006/SRR1291026/SRR1291026_1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR129/006/SRR1291026/SRR1291026_2.fastq.gz\n",
    "\n",
    "# ec2-35-161-212-163.us-west-2.compute.amazonaws.com -- DONE\n",
    "sh 1kgpilot.sh NA19017 NA19017 SRR1295544 LWK ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR129/004/SRR1295544/SRR1295544_1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR129/004/SRR1295544/SRR1295544_2.fastq.gz\n",
    "\n",
    "\n",
    "# Called 'done' at Fri Jan 13 19:54:35 PST 2017\n",
    "# going to let NA12878 run in the background.\n",
    "\n",
    "# All *.45s.bam files downloaded locally for analysis in IGV\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Results & Discussion\n",
    "\n",
    "\n",
    "### Variation in the 45s rDNA components\n",
    "\n",
    "![28S Pilot Genomes Variation](../figure/20170114_28S_pilotVariation.png)\n",
    "\n",
    "Running the pilot genomes makes it immediatly apparant that the variations in each genome are based on some sort of biological/chemical truth. The variant allele frequencies are high enough and the sequencing depth deep enough that it's a high confidence call. An expansion/variable loop is shown above, which are the areas in which the variations cluster. \n",
    "\n",
    "Also above, you can note that in the NA19240 trio, the variations are heritable. And across the populations they look to branch, probably reflecting the underlying population structures. Sequencing more people in each population and looking for clustering may be an important finding, each population having it's own private pool of rDNA variation.\n",
    "\n",
    "I'll soon have to come up with a naming scheme for these variation/expansion loops (possibly adapt from previous publication) and come up with a way of aligning to these alternative loops better. I have some ideas in mind and may play around with it next before opening up the analysis to larger cohorts.\n",
    "\n",
    "#### 28S A 59 A/G\n",
    "\n",
    "One of the most interesting alleles is 28S.A59. In every genome examined, this nucleotide is maintained at near 50% allele frequency of A and G. Unlike most other variations, there is little variability in alternative allele frequency, and it's maintained. The evolutionary entropy at that position is also 0.565, meaning it's conserved.\n",
    "\n",
    "This is a very good indicator of a functional consequence to this variation; it's adjacent to L15 and L29 binding sites.\n",
    "\n",
    "    NA19240_pcr.bam\n",
    "    28S:A59 | A60 in ribovision\n",
    "    chr13:1,007,994\n",
    "    <hr>Total count: 5855\n",
    "    A      : 2092  (36%,     935+,   1157- )\n",
    "    C      : 6  (0%,     6+,   0- )\n",
    "    G      : 3751  (64%,     1716+,   2035- )\n",
    "    T      : 6  (0%,     3+,   3- )\n",
    "    N      : 0\n",
    "    ---------------\n",
    "    \n",
    "* 28S:A58 base pairs with 28S:U327\\338, which is bound by L13\n",
    "* 28S:G59 base-phosphate interaction with 28S:C29 (bound by L15, loop 11)\n",
    "\n",
    "![ 28S: G59-C29 interaction ](../figure/20170114_28s.G59.Loop11.png)\n",
    "\n",
    "\n",
    "![ 28S: G59-C29 structure](../figure/20170114_G59-C28.png)\n",
    "C28 on left, G59 on Right.\n",
    "\n",
    "Mutagenesis to C59 would destroy this interaction, the Adenine doesn't have a -NH2 group at C2.\n",
    "\n",
    "![ 28S: G59 - A336 Electrostatic? ](../figure/20170114_G59-A336_ectrostatic.png)\n",
    "G59 at bottom; A335\\336\n",
    "\n",
    "In this case the highlighted interaction is less favourable, two electron donors are in line; possibly the C6-NH2 on A335 could rotate into a favourable position in which case this would be a good electrostatic interaction. A59 on the other hand would form a favourable interaction between A59 C6-NH2 and A335 N1, especially since the ~4 angstrom C28-G59 interaction would be gone. \n",
    "\n",
    "Either way, there doesn't seem to be any direct contact with RPL15.\n",
    "\n",
    "This position is worth noting and opening a seperate notebook page for; good chance of selection for variation here.\n",
    "\n",
    "### YRI Trio - Heredity of variations\n",
    "\n",
    "![YRI Trio post-5.8S](../figure/20170112_YRI_Trio_1.png)\n",
    "\n",
    "![YRI Trio pre-18S](../figure/20170114_yri_trio.png)\n",
    "\n",
    "The two screenshots above show groupings of variations outside of final units but the same is true for ones within.\n",
    "\n",
    "There is clearly heredity of variants but it doesn't match as simple mixing hypothesis.\n",
    "\n",
    "\n",
    "For instance, the locus around chr13:1002659 is shown below\n",
    "![YRI Heredity](../figure/20170114_yri_heredity_2.png)\n",
    "\n",
    "Under a simple mixing/allele sampling hypothesis; the offspring allele frequency should be the average of the two parents but this doesn't hold true.\n",
    "\n",
    "There is some combination of homogenization, sub-sampling which is present. It will be informative to find positions where two alleles are close together and phased together, what probabity there is that they are inherited together. Roughly, it looks like this is true but there is also a very high rate of recombination past 10-20 nts.\n",
    "\n",
    "Needs more trios and a method to calculate this systematically.\n",
    "\n",
    "| Genome  | Position | A%| T%| G%| C%|\n",
    "|---------|----------|---|---|---|---|\n",
    "| Mat     | 2659     | 0 | 0 | 0 |100|\n",
    "| Pat     | 2659     | 0 | 14| 0 |86 |\n",
    "| Child   | 2659     | 0 | 0 | 0 |100|\n",
    "|||||||\n",
    "| Mat     | 2665     | 0 |10 | 0 |90 |\n",
    "| Pat     | 2665     | 0 | 2 | 0 |98 |\n",
    "| Child   | 2665     | 0 | 1 | 0 |99 |\n",
    "|||||||\n",
    "| Mat     | 2817     |98 | 0 | 1 | 1 |\n",
    "| Pat     | 2817     |92 | 0 | 7 | 1 |\n",
    "| Child   | 2817     |92 | 0 | 6 | 1 |\n",
    "|||||||\n",
    "| Mat     | 3233     |0 | 98| 1 | 1 |\n",
    "| Pat     | 2817     |0 | 90| 1 | 9 |\n",
    "| Child   | 2817     |0 | 89| 1 | 11|\n",
    "|||||||\n",
    "\n",
    "### PCR-free and Differences in Sequencing Technology\n",
    "\n",
    "The PCR-free libraries performed much cleaner over high GC areas but it also appears that newer sequencing runs in general were able to sequence across high GC content. The inclination here is to use as new as possible sequencing and figure out which parameters exactly were changed between the earlier and later 1kg sequencing runs.\n",
    "\n",
    "### Runtimes, Cost, Scalability\n",
    "\n",
    "This experiment cost about \\$15 including testing. Overall I can anticipate a sub-\\$1 cost per genome for analysis. With EC2/S3 I shouldn't have to worry too much about infrastructure but may have to downscale how much data is stored vs. how much can be recalculated when needed.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Addendum - 20170211\n",
    "\n",
    "### Getting NA12878 done\n",
    "I think I'd like to include NA12878; there's 10X genomics data for her for long-distance phasing. It'll be worth it to have the genome PCR-free sequencing too, it just will cost a slight bit more on computation time.\n",
    "\n",
    "EC2: m4.4xlarge (16 cpu 30 Gb memory). 1000 Gb hard-drive space.\n",
    "Run: ec2-52-32-189-193.us-west-2.compute.amazonaws.com\n",
    "Date: Sat Feb 11 23:53:06 UTC 2017\n",
    "\n",
    "Changed code to 14 cores and no automated exit.\n",
    "\n",
    "\n",
    "```\n",
    "aws s3 cp s3://crownproject/scripts/1kgpilot.sh ./\n",
    "screen\n",
    "\n",
    "sh 1kgpilot.sh NA12878_pcr NA12878 SRR622457 CEU ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR622/SRR622457/SRR622457_1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR622/SRR622457/SRR622457_2.fastq.gz\n",
    "```\n",
    "\n",
    "Failed again... The sort step seems to be failing. Re-run but stop pre-sort step\n",
    "\n",
    "Re-running on the same instance but with `samtools sort -@ 10 -m 4G ...` to hopefully fix teh sort issue\n",
    "No more pre-sort delete as well.\n",
    "\n",
    "### 10x Genomics to hgr\n",
    "To do long-distance phasing over hgr I'm going to take a look at this 10X genomics data and see if it may resolve which variants are found togther within an rDNA repeat.\n",
    "\n",
    "The data is available for download (in bam format) on S3. Opening a custom  EC2 instance to run the pipeline and get the fastq files manually.\n",
    "\n",
    "EC2: m4.4xlarge (16 cpu 30 Gb memory). 1000 Gb hard-drive space.\n",
    "Run: ec2-35-164-213-205.us-west-2.compute.amazonaws.com\n",
    "\n",
    "```\n",
    "aws s3 cp s3://crownproject/scripts/1kgpilot.sh ./\n",
    "\n",
    "aws s3 cp s3://giab/data/NA12878/10XGenomics/NA12878_phased_possorted_bam.bam ./\n",
    "\n",
    "# Extract unique barcodes from the data\n",
    "samtools view -h NA12878_phased_possorted_bam.bam | grep -oh 'BX:Z:[ATGC\\-1234]*' - | cut -f 3 -d':' - > barcodes.tmp\n",
    "\n",
    "sort -u barcodes.tmp | uniq - > barcodes.unq.tmp\n",
    "\n",
    "\n",
    "\n",
    "screen\n",
    "\n",
    "sh 1kgpilot.sh NA12878_pcr NA12878 SRR622457 CEU ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR622/SRR622457/SRR622457_1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR622/SRR622457/SRR622457_2.fastq.gz\n",
    "```\n",
    "\n",
    "Closed...\n",
    "\n",
    "I need a method to extract the reads and regain the BX barcode tag for each read. Emailed Jake to see what's up.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                           PRE 10XGenomics_sizeselected/\r\n",
      "2015-08-20 06:33:03 101415465865 NA12878_phased_possorted_bam.bam\r\n",
      "2015-08-20 06:01:21    9021936 NA12878_phased_possorted_bam.bam.bai\r\n",
      "2015-09-11 06:18:05       2790 README_10Xgenomes.txt\r\n"
     ]
    }
   ],
   "source": [
    "aws s3 ls s3://giab/data/NA12878/10XGenomics/ "
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
