{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Alternative Aligners\n",
    "```\n",
    "pi:ababaian\n",
    "start: 2016 12 1st ish\n",
    "complete : 2016 12 29\n",
    "```\n",
    "## Introduction\n",
    "\n",
    "Previously for alignment of reads to the 'hgr' genome I used the command (NA19240)\n",
    "```\n",
    "# Bowtie2 Alignment to hgr genome\n",
    "bowtie2 -x ~/Crown/resources/hgr/hgr -1 SRR794330_1.filt.fastq.gz -2 SRR794330_2.filt.fastq.gz --very-sensitive | samtools view -bS - > NA19240_hgr.bam\n",
    "```\n",
    "Yeilding flagstat\n",
    "```\n",
    "71087208 + 0 in total (QC-passed reads + QC-failed reads)\n",
    "0 + 0 secondary\n",
    "0 + 0 supplimentary\n",
    "0 + 0 duplicates\n",
    "596582 + 0 mapped (0.84%:-nan%)\n",
    "71087208 + 0 paired in sequencing\n",
    "35543604 + 0 read1\n",
    "35543604 + 0 read2\n",
    "322054 + 0 properly paired (0.45%:-nan%)\n",
    "347818 + 0 with itself and mate mapped\n",
    "248764 + 0 singletons (0.35%:-nan%)\n",
    "0 + 0 with mate mapped to a different chr\n",
    "0 + 0 with mate mapped to a different chr (mapQ>=5)\n",
    "```\n",
    "This is the stats to beat. Bowtie2 runs in a few hours and yieldis quite a good alignment. I've become a bit more skeptical if this is the best method for alignment especially if there are alternative haplotypes with multiple variations in a cluster.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "## Objective\n",
    "\n",
    "* Test alternative aligners and determine how well they perform relative to bowtie2.\n",
    "\n",
    "### Hypothesis\n",
    "    \n",
    "    * Using more sensitive (try-hard) short read aligners will place more divergent reads on the hgr genome\n",
    "    * More divergent reads being aligned will allow for more sensitive nucleotide variant calling over rDNA\n",
    "    * More divergent reads being aligned will allow for better structural variant calling over rDNA\n",
    "    \n",
    "    \n",
    "### Aligners to test\n",
    "\n",
    "* [Bowtie2](http://www.nature.com/nmeth/journal/v9/n4/full/nmeth.1923.html): My work-horse aligner; it's fast and does a good job. So far I've used this.\n",
    "* [deBGA](http://bioinformatics.oxfordjournals.org/content/32/21/3224): Uses deBruijn Graphs to read alignment. Theoretically a good dBG aligner with a genome containing known variants should out-perform seed-extend aligners.\n",
    "* [Stampy](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3106326/): Hybrid aligner which excels at indel detection. \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Methods\n",
    "### Bowtie2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/usr/bin/bowtie2-align-s version 2.2.6\r\n",
      "64-bit\r\n",
      "Built on lgw01-12\r\n",
      "Mon Dec 28 11:09:46 UTC 2015\r\n",
      "Compiler: gcc version 5.3.1 20151219 (Ubuntu 5.3.1-4ubuntu1) \r\n",
      "Options: -O3 -m64 -msse2  -funroll-loops -g3 -Wl,-Bsymbolic-functions -Wl,-z,relro -DPOPCNT_CAPABILITY -DWITH_TBB\r\n",
      "Sizeof {int, long, long long, void*, size_t, off_t}: {4, 8, 8, 8, 8, 8}\r\n"
     ]
    }
   ],
   "source": [
    "# I won't re-run bowtie2 alignment\n",
    "bowtie2 --version\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### deBGA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# These operations were run outside of the notebook in a seperate terminal\n",
    "\n",
    "git clone https://github.com/hitbc/deBGA.git\n",
    "\n",
    "cd deBGA\n",
    "make\n",
    "\n",
    "deBGA index -k 20 hgr.fa hgr\n",
    "# Genome index is 2 Gb\n",
    "\n",
    "# 1.57 seconds to allocate memory\n",
    "# 1238.4 seconds to map reads\n",
    "\n",
    "\n",
    "deBGA aln ~/Crown/resources/hgr/hgr SRR794330_1.filt.fastq.gz SRR794330_2.filt.fastq.gz NA19240.deBGA.sam\n",
    "\n",
    "# Pipe these together in the future\n",
    "samtools view -b NA19240.deBGA.sam > NA19240.deBGA.bam\n",
    "\n",
    "samtools sort NA19240.deBGA.bam -o NA19240.deBGA.sort.bam\n",
    "\n",
    "# rm NA19240.deBGA.bam NA19240.deBGA.sam\n",
    "\n",
    "samtools flagstat NA19240_hgr.bam > NA19240_bt2.flagstat\n",
    "# bowtie2: 596582 mapped reads. 0.84%\n",
    "\n",
    "samtools flagstat NA19240.deBGA.sort.bam > NA19240_bga.flagstat\n",
    "# deBGA: 3910 mapped reads. 0.01%\n",
    "\n",
    "mv NA19240.deBGA.sort.bam NA19240.deBGA.bam\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Stampy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Download stampy\n",
    "wget http://www.well.ox.ac.uk/bioinformatics/Software/Stampy-latest.tgz\n",
    "\n",
    "tar -xvf Stampy-latest.tgz\n",
    "\n",
    "cd stampy-1.0.30\n",
    "\n",
    "make\n",
    "# python scripts made here now\n",
    "\n",
    "# Made HGR genome\n",
    "./stampy.py -G hgr /home/artem/Crown/resources/hgr/hgr.fa\n",
    "\n",
    "# Index Genome\n",
    "./stampy.py -g hgr -H hgr\n",
    "\n",
    "mv hgr* ~/Crown/resources/hgr/hgr\n",
    "\n",
    "cd ~/Crown/data/1kgenomes/\n",
    "\n",
    "# First Run ----------------------------------\n",
    "### ~/Desktop/stampy-1.0.30/stampy.py -g ~/Crown/resources/hgr/hgr -h ~/Crown/resources/hgr/hgr -M SRR794330_1.filt.fastq.gz SRR794330_2.filt.fastq.gz | samtools view -b - > NA19240.stampy.bam\n",
    "\n",
    "# Note the command with piping above took 3 days to complete. Try again without\n",
    "# to see if it increases efficiency\n",
    "\n",
    "### samtools sort NA19240.stampy.bam -o NA19240.stampy.sort.bam\n",
    "### mv NA19240.stampy.sort.bam NA19240.stampy.bam\n",
    "\n",
    "# 7442219 + 0 mapped (13.00% : N/A) !! fuck\n",
    "# File corrupted after moving it... !! double fuck\n",
    "# Re-run\n",
    "\n",
    "# Second Run ---------------------------------\n",
    "\n",
    "~/Desktop/stampy-1.0.30/stampy.py -g ~/Crown/resources/hgr/hgr -h ~/Crown/resources/hgr/hgr -M SRR794330_1.filt.fastq.gz SRR794330_2.filt.fastq.gz > NA19240.stampy.sam\n",
    "\n",
    "# With a few suspend interruptions took ~ 2 days to align\n",
    "\n",
    "samtools view -bh NA19240.sam | samtools sort - -o NA19240.stampy.bam\n",
    "\n",
    "# Start 161202-080200\n",
    "# End  161204-192600\n",
    "#stampy: # Nucleotides (all/1/2):\t1376544600\t688272300\t688272300\n",
    "#stampy: # Variants:             \t117847472\t54546261\t63301211\n",
    "#stampy: # Fraction:             \t0.0856\t0.0793\t0.0920\n",
    "#stampy: # Paired-end insert size: 91.2 +/- 44.5  (5167630 pairs)y\n",
    "#stampy: Done\n",
    "\n",
    "samtools flagstat NA19240.stampy.bam > NA19240_stampy.flagstat"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Bowtie2 ---------------------------------------------\r\n",
      "71087208 + 0 in total (QC-passed reads + QC-failed reads)\r\n",
      "0 + 0 secondary\r\n",
      "0 + 0 supplementary\r\n",
      "0 + 0 duplicates\r\n",
      "596582 + 0 mapped (0.84% : N/A)\r\n",
      "71087208 + 0 paired in sequencing\r\n",
      "35543604 + 0 read1\r\n",
      "35543604 + 0 read2\r\n",
      "322054 + 0 properly paired (0.45% : N/A)\r\n",
      "347818 + 0 with itself and mate mapped\r\n",
      "248764 + 0 singletons (0.35% : N/A)\r\n",
      "0 + 0 with mate mapped to a different chr\r\n",
      "0 + 0 with mate mapped to a different chr (mapQ>=5)\r\n",
      "deBGA -----------------------------------------------\r\n",
      "71087208 + 0 in total (QC-passed reads + QC-failed reads)\r\n",
      "0 + 0 secondary\r\n",
      "0 + 0 supplementary\r\n",
      "0 + 0 duplicates\r\n",
      "3910 + 0 mapped (0.01% : N/A)\r\n",
      "71087208 + 0 paired in sequencing\r\n",
      "35543604 + 0 read1\r\n",
      "35543604 + 0 read2\r\n",
      "238 + 0 properly paired (0.00% : N/A)\r\n",
      "830 + 0 with itself and mate mapped\r\n",
      "3080 + 0 singletons (0.00% : N/A)\r\n",
      "0 + 0 with mate mapped to a different chr\r\n",
      "0 + 0 with mate mapped to a different chr (mapQ>=5)\r\n",
      "Stampy ----------------------------------------------\r\n",
      "71087208 + 0 in total (QC-passed reads + QC-failed reads)\r\n",
      "0 + 0 secondary\r\n",
      "0 + 0 supplementary\r\n",
      "0 + 0 duplicates\r\n",
      "7442219 + 0 mapped (10.47% : N/A)\r\n",
      "71087208 + 0 paired in sequencing\r\n",
      "35543604 + 0 read1\r\n",
      "35543604 + 0 read2\r\n",
      "3198988 + 0 properly paired (4.50% : N/A)\r\n",
      "5707378 + 0 with itself and mate mapped\r\n",
      "1734841 + 0 singletons (2.44% : N/A)\r\n",
      "0 + 0 with mate mapped to a different chr\r\n",
      "0 + 0 with mate mapped to a different chr (mapQ>=5)\r\n",
      "-----------------------------------------------------\r\n"
     ]
    }
   ],
   "source": [
    "cd ~/Crown/data/1kgenomes/\n",
    "\n",
    "# Flagstats from each aligner\n",
    "echo Bowtie2 ---------------------------------------------\n",
    "cat NA19240_bt2.flagstat\n",
    "echo deBGA -----------------------------------------------\n",
    "cat NA19240_bga.flagstat\n",
    "echo Stampy ----------------------------------------------\n",
    "cat NA19240_stampy.flagstat\n",
    "echo -----------------------------------------------------\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Notes:\n",
    "I was quite surprised by how much reads get aligned by Stampy, 7 million vs. 50k by bowtie2 is a huge discrepency. But this also is saying that 10% of the genome is in the rDNA which probably isn't a fair assessment. Since the aligner is good at placing 'divergent' reads a lot of genomic sequence is probably being forced onto what is available.\n",
    "\n",
    "The largest discrepency between Stampy and Bowtie2 is in how the algorithms treat extension from a strong seed sequence match. At chr13:1013900 is a poly-T tract for 30 bp followed by (TTC) repeats. This looks like it makes a great seed match for many sequences across the genome and Stampy puts LOTS of reads there\n",
    "\n",
    "![Poly-T Alignment](../figure/20161209_polyT_align.png)\n",
    "\n",
    "But looking at a broader picture of the entire 18S, stampy is better at aligning reads with sequence mis-matches. Allele cut-off set to 0.05 here:\n",
    "\n",
    "![Alignment comparison: 18S](../figure/20161209_align_compare_18S.png)\n",
    "\n",
    "deBGA obviously didn't work and needs to be optimized; kind of not stoked to do this. Also I'm not sure how much gains there will be from a dBG aligner before I have a 'population of genomes' to align to. Somethign to keep in the back-pocket for later.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Bowtie2: 18S -------------------------------------------\r\n",
      "13281 + 0 in total (QC-passed reads + QC-failed reads)\r\n",
      "0 + 0 secondary\r\n",
      "0 + 0 supplementary\r\n",
      "0 + 0 duplicates\r\n",
      "13145 + 0 mapped (98.98% : N/A)\r\n",
      "13281 + 0 paired in sequencing\r\n",
      "6640 + 0 read1\r\n",
      "6641 + 0 read2\r\n",
      "12383 + 0 properly paired (93.24% : N/A)\r\n",
      "13009 + 0 with itself and mate mapped\r\n",
      "136 + 0 singletons (1.02% : N/A)\r\n",
      "0 + 0 with mate mapped to a different chr\r\n",
      "0 + 0 with mate mapped to a different chr (mapQ>=5)\r\n",
      "\r\n",
      "Stampy: 18S -------------------------------------------\r\n",
      "13616 + 0 in total (QC-passed reads + QC-failed reads)\r\n",
      "0 + 0 secondary\r\n",
      "0 + 0 supplementary\r\n",
      "0 + 0 duplicates\r\n",
      "13494 + 0 mapped (99.10% : N/A)\r\n",
      "13616 + 0 paired in sequencing\r\n",
      "6808 + 0 read1\r\n",
      "6808 + 0 read2\r\n",
      "11161 + 0 properly paired (81.97% : N/A)\r\n",
      "13369 + 0 with itself and mate mapped\r\n",
      "125 + 0 singletons (0.92% : N/A)\r\n",
      "0 + 0 with mate mapped to a different chr\r\n",
      "0 + 0 with mate mapped to a different chr (mapQ>=5)\r\n",
      "\r\n",
      "deBGA: 18S -------------------------------------------\r\n",
      "5 + 0 in total (QC-passed reads + QC-failed reads)\r\n",
      "0 + 0 secondary\r\n",
      "0 + 0 supplementary\r\n",
      "0 + 0 duplicates\r\n",
      "5 + 0 mapped (100.00% : N/A)\r\n",
      "5 + 0 paired in sequencing\r\n",
      "2 + 0 read1\r\n",
      "3 + 0 read2\r\n",
      "0 + 0 properly paired (0.00% : N/A)\r\n",
      "5 + 0 with itself and mate mapped\r\n",
      "0 + 0 singletons (0.00% : N/A)\r\n",
      "0 + 0 with mate mapped to a different chr\r\n",
      "0 + 0 with mate mapped to a different chr (mapQ>=5)\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "# Let's compare the difference in alignment over exclusively the 18S DNA region\n",
    "# chr13:1,003,499-1,005,599  (excludes (AG) simple repeats)\n",
    "echo -e \"chr13\\t1003499\\t1005599\" > 18S.tmp\n",
    "echo Bowtie2: 18S -------------------------------------------\n",
    "samtools view -bh -L 18S.tmp NA19240_hgr.bam | samtools flagstat -\n",
    "echo ''\n",
    "echo Stampy: 18S -------------------------------------------\n",
    "samtools view -bh -L 18S.tmp NA19240.stampy.bam | samtools flagstat -\n",
    "echo ''\n",
    "echo deBGA: 18S -------------------------------------------\n",
    "samtools view -bh -L 18S.tmp NA19240.deBGA.bam| samtools flagstat -\n",
    "echo ''"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Over 18S there is a much smaller difference between Bowtie2 and Stampy. Looking at the alignments it seems that it stems from the difference in the two methods in aligning over indels. At chr13:1004360 there is a deletion in some sequences which exemplifies this. Position highlighted with black dotted line is chr13:1004382\n",
    "\n",
    "![Stampy alignment at 4360](../figure/20161209_del4360_stampy.png)\n",
    "```\n",
    "STAMPY ======================\n",
    "Deletion\n",
    "chr13:1,004,365\n",
    "<hr>Total count: 484\n",
    "A      : 1  (0%,     1+,   0- )\n",
    "C      : 3  (1%,     0+,   3- )\n",
    "G      : 480  (99%,     232+,   248- )\n",
    "T      : 0\n",
    "N      : 0\n",
    "---------------\n",
    "DEL: 18\n",
    "INS: 1\n",
    "\n",
    "Highlighted position\n",
    "chr13:1,004,382\n",
    "<hr>Total count: 521\n",
    "A      : 10  (2%,     5+,   5- )\n",
    "C      : 491  (94%,     245+,   246- )\n",
    "G      : 15  (3%,     9+,   6- )\n",
    "T      : 5  (1%,     3+,   2- )\n",
    "N      : 0\n",
    "---------------\n",
    "DEL: 1\n",
    "INS: 0\n",
    "```\n",
    "\n",
    "Compared to the bowtie2 alignment at the same position\n",
    "\n",
    "![Bowtie2 alignment at 4360](../figure/20161209_del4360_bt2.png)\n",
    "```\n",
    "BOWTIE2  =====================\n",
    "Deletion\n",
    "chr13:1,004,365\n",
    "<hr>Total count: 484\n",
    "A      : 1  (0%,     0+,   1- )\n",
    "C      : 3  (1%,     0+,   3- )\n",
    "G      : 480  (99%,     231+,   249- )\n",
    "T      : 0\n",
    "N      : 0\n",
    "---------------\n",
    "DEL: 2\n",
    "INS: 0\n",
    "\n",
    "Highlighted Position\n",
    "chr13:1,004,382\n",
    "<hr>Total count: 503\n",
    "A      : 1  (0%,     0+,   1- )\n",
    "C      : 496  (99%,     247+,   249- )\n",
    "G      : 2  (0%,     2+,   0- )\n",
    "T      : 4  (1%,     2+,   2- )\n",
    "N      : 0\n",
    "---------------\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### del4362-4372\n",
    "\n",
    "It's variants like this which are the 'trade-off' between bowtie2 and stampy. The deletion is present at 3.52% of the reads (18 / 511). It's highly unlikely that it's simply sequencing error, the rate is too high and the deletion is called over mostly the same 10bp window (+/- 1bp).\n",
    "\n",
    "Biologically what does this mean? It's tempting to write it off as a 'pseudo-retrogene' degraded elsewhere in the genome. At 3.5% with an estimate of 100 - 450 copies of rDNA that puts this at 3.5 - 15.75 copies in the genome. There also looks to be at least 5 distinct halpotypes of the variant which would mean each of them are between sub-1 - 3 copies. A simple solution here is that they are single-copy variants which are related to one another. But that still doesn't answer why there are multiple copies of this deletion containing copy.\n",
    "\n",
    "** Deletion 4362 Haplotype from NA19240 using Stampy **\n",
    "![del4362 Haplotype](../figure/20161209_del4362_hap.png)\n",
    "\n",
    "\n",
    "#### Possible Sources of Error\n",
    "\n",
    "* mt rDNA\n",
    "* Contaminating parasite / organism in sample\n",
    "* PCR Artifact (unlikely)\n",
    "* pseduo rRNA\n",
    "* an rRNA derived gene (not true 18S)\n",
    "\n",
    "\n",
    "\n",
    "## Conclusions\n",
    "\n",
    "* Stampy is a more sensitive aligner; I think there is little doubt about that especially over indels like it claims in the paper.\n",
    "* Stampy takes significantly longer to complete and alignment then bowtie2 (days instead of hours). May not be feasible for aligning dozens or hundreds of genomes.\n",
    "* A hybrid approach of using Stampy for a smaller number of genomes to detect indels and making 'variant' rDNA copies combined with the speed and accuracy of bowtie2 is probably the best approach.\n",
    "\n",
    "\n",
    "* Use stampy to align a pilot set of ~10 genomes and describe larger structural variants which are present (along with something like lumpy), using that information align a larger amount of genomes with bowtie2.\n",
    "\n",
    "qed"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
