{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 1kg hgr0 Alignment Pilot\n",
    "```\n",
    "pi:ababaian\n",
    "files: ~/Crown/data/1kg_hgr0\n",
    "start: 2017 02 20\n",
    "complete : 2017 02 22\n",
    "```\n",
    "## Introduction\n",
    "\n",
    "I've identified that there is bona-fide intra- and inter-personal variation in rDNA. All my hours of analysis have essentially focused on RNA45S and it's promoter, there is LOTS of analysis to be done with it and it's the low-hanging fruit. The remainder of the rDNA repeat is more difficult to study but I will get back to it (hopefully with help).\n",
    "\n",
    "To measure how much population-level variation exists, I want to quickly assay these 'interesting' regions across many samples so I've narrowed down a reference sequence of interest to something I'm calling [hgr0](./20170213_hgr0_reference_rDNA.ipynb), which contains 5S, in the context of a full 5S rDNA repeat (without simple-repeat sequences), and RNA45S and it's immediate upstream promoter (1.1kb).\n",
    "\n",
    "These alignments and variant calling should run very quickly so I'll be able to do my target of 100 human genomes quickly and analyze these.\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "## Objective\n",
    "\n",
    "* Run the `1kg_align_v0.sh` pipeline on NA19240 and her parents.\n",
    "* Informally compare sequence alignments in the Trio between `hgr` and `hgr0`."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Materials and Methods\n",
    "\n",
    "Alignments will be run on ec2 using a modified `1kgpilot.sh` script called `1kg_align_vo.sh`, as in this is the 'pipeline' version 0.\n",
    "\n",
    "This should be much faster then 1kgpilot.sh since alot of the simple repeat sequences have been taken out and the focus is on RNA45S and 5S itself as opposed to the rest of the rDNA repeat.\n",
    "\n",
    "### NA19240 Trio Files\n",
    "\n",
    "\n",
    "| Sample | Population | Sex | Notes       | Fastq | Size (KB) |\n",
    "|--------|:----------:|:---:|-------------|-------|-----------|\n",
    "|NA19240|YRI|F|WGS, PCR-free|  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR894/ERR894723/ERR894723_1.fastq.gz|24068394|\n",
    "|NA19240|YRI|F|WGS, PCR-free| ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR894/ERR894723/ERR894723_2.fastq.gz|24313932|\n",
    "|NA19238|YRI|F|WGS, PCR-free| ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR899/ERR899706/ERR899706_1.fastq.gz|5423970|\n",
    "|NA19238|YRI|F|WGS, PCR-free| ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR899/ERR899706/ERR899706_2.fastq.gz|5554683|\n",
    "|NA19239|YRI|M|WGS, PCR-free| ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR899/ERR899707/ERR899707_1.fastq.gz|8052464|\n",
    "|NA19239|YRI|M|WGS, PCR-free| ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR899/ERR899707/ERR899707_2.fastq.gz|8192156|\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 1kg_align_v0_a.sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "# 1kg_align_v0_a.sh script\n",
    "# AMI: crown-170220 - ami-66129306\n",
    "# EC2: c4.2xlarge (8cpu / 15 gb)\n",
    "# Storage: 100-500 Gb\n",
    "# Start: \n",
    "# Alignment done: \n",
    "# Align.subset done: \n",
    "# End:\n",
    "\n",
    "# Control Panel -------------------------------\n",
    "# CPU\n",
    "\tTHREADS='7'\n",
    "\tAWSID='#########'\n",
    "\n",
    "# Sequencing Data\n",
    "\tLIBRARY=$1 # Library/ File name\n",
    "\tFASTQ1=$5\n",
    "\tFASTQ2=$6\n",
    "\n",
    "    # File-names\n",
    "    FQ1=$(basename $FASTQ1)\n",
    "    FQ2=$(basename $FASTQ2)\n",
    "\n",
    "# Read Group Data\n",
    "\tRGSM=$2   # Sample. Patient Identifer\n",
    "\tRGID=$3 # Read Group ID. Accession Number\n",
    "\tRGLB=$LIBRARY # Library Name. Accession Number\n",
    "\tRGPL='ILLUMINA'  # Sequencing Platform.\n",
    "\tRGPO=$4 # Patient Population\n",
    "\t# Extract Sequencing Run Info\n",
    "\t#  RGPU=$(gzip -dc $FQ1 | head -n1 - | cut -f1 -d':' | cut -f2 -d' ')\n",
    "\n",
    "\n",
    "# Initialize wordir ---------------------------\n",
    "\n",
    "# Make working directory\n",
    "  mkdir -p align; cd align\n",
    "\n",
    "# Copy hgr0 genome and create bowtie2 index\n",
    "  cp ~/resources/hgr0.fa ./\n",
    "  cp ~/resources/hgr0.fa.fai ./\n",
    "  \n",
    "  bowtie2-build hgr0.fa hgr0\n",
    "  \n",
    "# Download Genome Sequencing Data\n",
    "  wget $FASTQ1\n",
    "  wget $FASTQ2\n",
    "\n",
    "    # Extract Sequencing Run Info\n",
    "    RGPU=$(gzip -dc $FQ1| head -n1 - | cut -f1 -d':' | cut -f2 -d' ')\n",
    "\n",
    "# Primary Alignment -------------------------\n",
    "\n",
    "# Bowtie2: align to genome\n",
    "\n",
    "bowtie2 --very-sensitive-local -p $THREADS --rg-id $RGID --rg LB:$RGLB --rg SM:$RGSM --rg PL:$RGPL --rg PU:$RGPU -x hgr0 -1 $FQ1 -2 $FQ2 | samtools view -bS - > aligned_unsorted.bam\n",
    "\n",
    "rm $FQ1 $FQ2 # Remove fastq files to save space\n",
    "\n",
    "# Sort alignment file\n",
    "  samtools sort -@ $THREADS aligned_unsorted.bam aligned\n",
    "  samtools index aligned.bam\n",
    "  rm aligned_unsorted.bam\n",
    "\n",
    "# Calcualte library flagstats\n",
    "  samtools flagstat aligned.bam > aligned.flagstat\n",
    "  \n",
    "# Rename the final Bam Files\n",
    "  mv aligned.bam $LIBRARY.bam\n",
    "  mv aligned.bam.bai $LIBRARY.bam.bai\n",
    "  mv aligned.flagstat $LIBRARY.flagstat\n",
    "\n",
    "# GATK variant calling over 45S region\n",
    "  aws s3 cp s3://crownproject/resources/hgr0.gatk.fa ./\n",
    "  aws s3 cp s3://crownproject/resources/hgr0.gatk.fa.fai ./\n",
    "  aws s3 cp s3://crownproject/resources/hgr0.gatk.dict ./\n",
    "  \n",
    "  java -Xmx12G -jar /home/ubuntu/software/GenomeAnalysisTK.jar \\\n",
    "  -R hgr0.gatk.fa -T HaplotypeCaller \\\n",
    "  -ploidy 2 --max_alternate_alleles 6 \\\n",
    "  -I $LIBRARY.bam -o $LIBRARY.rDNA_p2.vcf\n",
    "   # Memory issues, restrict to 45S region only\n",
    "     # -ploidy 100, 50, 20 failed... do 2 and analyze 45S further\n",
    "     \n",
    "# Upload final output files to S3\n",
    " \n",
    "# Alignments\n",
    " aws s3 cp $LIBRARY.bam s3://crownproject/1kg_hgr0/\n",
    " aws s3 cp $LIBRARY.bam.bai s3://crownproject/1kg_hgr0/\n",
    " aws s3 cp $LIBRARY.flagstat s3://crownproject/1kg_hgr0/\n",
    " \n",
    "# VCF\n",
    " aws s3 cp $LIBRARY.rDNA_p2.vcf s3://crownproject/1kg_hgr0/\n",
    " aws s3 cp $LIBRARY.rDNA_p2.vcf.idx s3://crownproject/1kg_hgr0/\n",
    " \n",
    " \n",
    "# Shutdown and Terminate instance\n",
    "EC2ID=$(ec2metadata --instance-id)\n",
    "aws ec2 terminate-instances --instance-ids $EC2ID\n",
    "\n",
    "# Script complete"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### EC2 Runs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# AMI crown-170220\n",
    "# 200 Gb of storage on C4.2xlarge\n",
    "# On each instance run the commands below:\n",
    "\n",
    "# Common Command\n",
    "screen\n",
    "aws s3 cp s3://crownproject/scripts/1kg_align_v0.sh ./\n",
    "\n",
    "# ec2-52-39-30-14.us-west-2.compute.amazonaws.com\n",
    "# DOES NOT AUTO-SHUTDOWN\n",
    "sh 1kg_align_v0.sh NA19240_pcr NA19240 ERR894723 YRI ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR894/ERR894723/ERR894723_1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR894/ERR894723/ERR894723_2.fastq.gz\n",
    "\n",
    "# ec2-52-36-62-251.us-west-2.compute.amazonaws.com\n",
    "sh 1kg_align_v0.sh NA19238_pcr NA19238 ERR899706 YRI ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR899/ERR899706/ERR899706_1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR899/ERR899706/ERR899706_2.fastq.gz\n",
    "\n",
    "# ec2-35-164-38-177.us-west-2.compute.amazonaws.com\n",
    "sh 1kg_align_v0.sh NA19239_pcr NA19239 ERR899707 YRI ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR899/ERR899707/ERR899707_1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR899/ERR899707/ERR899707_2.fastq.gz\n",
    "\n",
    "# Output moved to s3://crownproject/1kg_hgr0/na19240_trio/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 1kg_align_v0_b.sh\n",
    "\n",
    "* Alignment worked; files are large though (they include unmapped reads)\n",
    "* VCF didn't work; `hgr.gatk.fa.fai does not exist`. UPLOADED\n",
    "* It may be more efficient to re-work the pipeline to feed 'mapped reads' and 'unmapped reads' into two distinct files at the alignment step."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "# 1kg_align_v0_b.sh script\n",
    "# AMI: crown-170220 - ami-66129306\n",
    "# EC2: c4.2xlarge (8cpu / 15 gb)\n",
    "# Storage: 100-500 Gb\n",
    "# Start: \n",
    "# Alignment done: \n",
    "# Align.subset done: \n",
    "# End:\n",
    "\n",
    "# Control Panel -------------------------------\n",
    "# CPU\n",
    "\tTHREADS='7'\n",
    "\tAWSID='#########'\n",
    "\n",
    "# Sequencing Data\n",
    "\tLIBRARY=$1 # Library/ File name\n",
    "\tFASTQ1=$5\n",
    "\tFASTQ2=$6\n",
    "\n",
    "    # File-names\n",
    "    FQ1=$(basename $FASTQ1)\n",
    "    FQ2=$(basename $FASTQ2)\n",
    "\n",
    "# Read Group Data\n",
    "\tRGSM=$2   # Sample. Patient Identifer\n",
    "\tRGID=$3 # Read Group ID. Accession Number\n",
    "\tRGLB=$LIBRARY # Library Name. Accession Number\n",
    "\tRGPL='ILLUMINA'  # Sequencing Platform.\n",
    "\tRGPO=$4 # Patient Population\n",
    "\t# Extract Sequencing Run Info\n",
    "\t#  RGPU=$(gzip -dc $FQ1 | head -n1 - | cut -f1 -d':' | cut -f2 -d' ')\n",
    "\n",
    "\n",
    "# Initialize wordir ---------------------------\n",
    "\n",
    "# Make working directory\n",
    "  mkdir -p align; cd align\n",
    "\n",
    "# Copy hgr0 genome and create bowtie2 index\n",
    "  cp ~/resources/hgr0.fa ./\n",
    "  cp ~/resources/hgr0.fa.fai ./\n",
    "  \n",
    "  bowtie2-build hgr0.fa hgr0\n",
    "  \n",
    "# Download Genome Sequencing Data\n",
    "  wget $FASTQ1\n",
    "  wget $FASTQ2\n",
    "\n",
    "    # Extract Sequencing Run Info\n",
    "    RGPU=$(gzip -dc $FQ1| head -n1 - | cut -f1 -d':' | cut -f2 -d' ')\n",
    "\n",
    "# Primary Alignment -------------------------\n",
    "\n",
    "# Bowtie2: align to genome\n",
    "\n",
    "bowtie2 --very-sensitive-local -p $THREADS --rg-id $RGID --rg LB:$RGLB --rg SM:$RGSM \\\n",
    "--rg PL:$RGPL --rg PU:$RGPU -x hgr0 -1 $FQ1 -2 $FQ2 | samtools view -bS - > aligned_unsorted.bam\n",
    "\n",
    "rm $FQ1 $FQ2 # Remove fastq files to save space\n",
    "\n",
    "# Sort alignment file\n",
    "  samtools sort -@ $THREADS aligned_unsorted.bam aligned\n",
    "  samtools index aligned.bam\n",
    "  rm aligned_unsorted.bam\n",
    "\n",
    "# Calcualte library flagstats\n",
    "  samtools flagstat aligned.bam > aligned.flagstat\n",
    "  \n",
    "# Read Subset ------------------------------\n",
    "# Extract mapped reads, and their unmapped pairs\n",
    "\n",
    "  # Extract Header\n",
    "  samtools view -H aligned.bam > align.header.tmp\n",
    "\n",
    "  # Unmapped reads with mapped pairs\n",
    "  # Extract Mapped Reads\n",
    "  # and their unmapped pairs\n",
    "  samtools view -b -F 4 aligned.bam > align.F4.bam #mapped\n",
    "  samtools view -b -f 4 -F 8 aligned.bam > align.f4F8.bam #unmapped pairs\n",
    "  \n",
    "  # Extract just the 45S unit\n",
    "  #aws s3 cp s3://crownproject/resources/rDNA_45s.bed ./\n",
    "  #samtools view -b -L rDNA_45s.bed align.F4.bam > align.F4.45s.bam\n",
    "  \n",
    "  # What are the mapped readnames\n",
    "  samtools view align.F4.bam | cut -f1 - > read.names.tmp\n",
    "  \n",
    "  # Extract mapped reads\n",
    "  samtools view align.F4.bam | grep -Ff read.names.tmp - > align.F4.tmp.sam\n",
    "\n",
    "  \n",
    "  # Extract cases of read pairs mapped on edge of region of interest\n",
    "  # -------|======= R O I ======| ----------\n",
    "  # read:                  ====---====\n",
    "  samtools view align.F4.bam | grep -Ff read.names.tmp - > align.F4.tmp.sam\n",
    "\n",
    "  # Complete mapped reads list\n",
    "  #cut -f1 align.F4.tmp.sam > read.names.45s.long.tmp\n",
    "\n",
    "  # Extract unmapped reads with a mapped pair\n",
    "  samtools view align.f4F8.bam | grep -Ff read.names.tmp - > align.f4F8.tmp.sam\n",
    "\n",
    "  # Re-compile bam file\n",
    "  cat align.header.tmp align.F4.tmp.sam align.f4F8.tmp.sam | samtools view -bS - > align.hgr0.tmp.bam\n",
    "    samtools sort align.hgr0.tmp.bam align.hgr0\n",
    "    samtools index align.hgr0.bam\n",
    "    samtools flagstat align.hgr0.bam > align.hgr0.flagstat\n",
    "    \n",
    "  # Read Counts: align.hgr0.bam (NA19240_pcr)\n",
    "    # 651340 + 0 in total (QC-passed reads + QC-failed reads)\n",
    "    # 0 + 0 duplicates\n",
    "    # 614264 + 0 mapped (94.31%:-nan%)\n",
    "    # 651340 + 0 paired in sequencing\n",
    "    # 325670 + 0 read1\n",
    "    # 325670 + 0 read2\n",
    "    # 166576 + 0 properly paired (25.57%:-nan%)\n",
    "    # 577188 + 0 with itself and mate mapped\n",
    "    # 37076 + 0 singletons (5.69%:-nan%)\n",
    "    # 0 + 0 with mate mapped to a different chr\n",
    "    # 0 + 0 with mate mapped to a different chr (mapQ>=5)\n",
    "  \n",
    "  rm *tmp* align.F4.bam align.f4F8.bam # Clean-up\n",
    "\n",
    "# Rename the total Bam Files\n",
    "  mv aligned.bam $LIBRARY.bam\n",
    "  mv aligned.bam.bai $LIBRARY.bam.bai\n",
    "  mv aligned.flagstat $LIBRARY.flagstat\n",
    "\n",
    "# Rename the hgr0 Bam files\n",
    "  mv align.hgr0.bam $LIBRARY.hgr0.bam\n",
    "  mv align.hgr0.bam.bai $LIBRARY.hgr0.bam.bai\n",
    "  mv align.hgr0.flagstat $LIBRARY.hgr0.flagstat\n",
    "  \n",
    "# Primary VCF ----------------------------\n",
    "\n",
    "# GATK variant calling over 45S region\n",
    "  aws s3 cp s3://crownproject/resources/hgr0.gatk.fa ./\n",
    "  aws s3 cp s3://crownproject/resources/hgr0.gatk.fa.fai ./\n",
    "  aws s3 cp s3://crownproject/resources/hgr0.gatk.dict ./\n",
    "  \n",
    "  java -Xmx12G -jar /home/ubuntu/software/GenomeAnalysisTK.jar \\\n",
    "  -R hgr0.gatk.fa -T HaplotypeCaller \\\n",
    "  -ploidy 2 --max_alternate_alleles 6 \\\n",
    "  -I $LIBRARY.bam -o $LIBRARY.hgr0.vcf\n",
    "   # Memory issues, restrict to 45S region only\n",
    "     # -ploidy 100, 50, 20 failed... do 2 and analyze 45S further\n",
    "     \n",
    "# Upload final output files to S3\n",
    " \n",
    "# Alignments (Full)\n",
    " #aws s3 cp $LIBRARY.bam s3://crownproject/1kg_hgr0/\n",
    " #aws s3 cp $LIBRARY.bam.bai s3://crownproject/1kg_hgr0/\n",
    " aws s3 cp $LIBRARY.flagstat s3://crownproject/1kg_hgr0/\n",
    "\n",
    "# Alignments (Aligned)\n",
    "  aws s3 cp $LIBRARY.hgr0.bam s3://crownproject/1kg_hgr0/\n",
    "  aws s3 cp $LIBRARY.hgr0.bam.bai s3://crownproject/1kg_hgr0/\n",
    "  aws s3 cp $LIBRARY.hgr0.flagstat s3://crownproject/1kg_hgr0/\n",
    "\n",
    "# VCF\n",
    " aws s3 cp $LIBRARY.hgr0.vcf s3://crownproject/1kg_hgr0/\n",
    " aws s3 cp $LIBRARY.hgr0.vcf.idx s3://crownproject/1kg_hgr0/\n",
    " \n",
    "# Shutdown and Terminate instance\n",
    "EC2ID=$(ec2metadata --instance-id)\n",
    "aws ec2 terminate-instances --instance-ids $EC2ID\n",
    "\n",
    "# Script complete"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Results - Discussion\n",
    "\n",
    "After a bit of hand-holding and refining the script to `1kg_align_v0_b.sh` (note: still called `1kg_align_v0.sh` on disk) the alignments worked.\n",
    "\n",
    "This is a much faster pipeline and I can 'open the hose'. I talked with S.Jackman last night about ways I can auto-launch EC2 instances with a script and he suggested coding the ssh command to login to the EC2 machine and run the commands automatically. The hope is that the EC2-launch / run / close is all automated and I can do 1000 genomes alignments from a single csv file.\n",
    "\n",
    "![NA19240 Trio hgr0 Variation](../figure/20170222_na19240_trio_hgr0.png)\n",
    "\n",
    "Time to open up the throttle on these alignments; the components are in place."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Addendum - 170228 NA12878\n",
    "\n",
    "From the 1kgpilot experiment, I left the NA1278 instance 'online' but paused. I didn't want to re-download the data (which was stupid and expensive in the end).\n",
    "\n",
    "Anyways, I resumed the m4.4xlarge instance and ran the 1kg_hgr0 pipeline with the command:\n",
    "\n",
    "```\n",
    "sh 1kg_align_v0.sh NA12878_pcr NA12878 SRR622457 CEU ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR622/SRR622457/SRR622457_1.fastq.gz ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR622/SRR622457/SRR622457_2.fastq.gz\n",
    "```"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
