{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# hgr1 bcftools analysis / variant analysis\n",
    "```\n",
    "pi:ababaian\n",
    "files: ~/Crown/data/hgr1_vcf/\n",
    "start: 2017 03 07\n",
    "complete : 2017 03 13\n",
    "```\n",
    "## Introduction\n",
    "\n",
    "I have the hgr1_v1 pipeline up and running; part of the output includes variant calling from GATK.\n",
    "\n",
    "```\n",
    "  java -Xmx12G -jar /home/ubuntu/software/GenomeAnalysisTK.jar \\\n",
    "  -R hgr1.gatk.fa -T HaplotypeCaller \\\n",
    "  -ploidy 2 --max_alternate_alleles 6 \\\n",
    "  -I $LIBRARY.hgr1.bam -o $LIBRARY.hgr1.vcf\n",
    "```\n",
    "\n",
    "This is the downstream analysis of those files to broadly describe variation in rDNA across many individuals."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "## Objective\n",
    "\n",
    "- Automate VCF analysis\n",
    "- Calculate which intra-individual variants are found in the population and which are maintained in humans\n",
    "- Calculate the frequency of variation in different regions of the rDNA loci\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Materials and Methods\n",
    "\n",
    "### Key Files\n",
    "\n",
    "- `100g_hgr1_individual.tar.gz` : Each individual hgr1_v1 output vcf file from 107 genomes (3 platinum + 104 population)\n",
    "- \n",
    "\n",
    "### Current VCF files - 30 genomes\n",
    "\n",
    "#### Manipulation of VCF files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": []
    }
   ],
   "source": [
    "cd ~/Crown/data/\n",
    "mkdir -p hgr1_vcf/; cd hgr1_vcf\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Tue Mar  7 16:55:32 PST 2017\r\n",
      "2017-03-07 14:24:07      61563 HG00478.hgr1.vcf\r\n",
      "2017-03-07 14:24:07       4324 HG00478.hgr1.vcf.idx\r\n",
      "2017-03-07 16:08:21      58909 HG00536.hgr1.vcf\r\n",
      "2017-03-07 16:08:22       2298 HG00536.hgr1.vcf.idx\r\n",
      "2017-03-07 16:09:22      37016 HG00557.hgr1.vcf\r\n",
      "2017-03-07 16:09:22        782 HG00557.hgr1.vcf.idx\r\n",
      "2017-03-07 16:11:04      17098 HG00717.hgr1.vcf\r\n",
      "2017-03-07 16:11:05        284 HG00717.hgr1.vcf.idx\r\n",
      "2017-03-06 23:18:53      63888 HG00851.hgr1.vcf\r\n",
      "2017-03-06 23:18:53       4325 HG00851.hgr1.vcf.idx\r\n",
      "2017-03-06 22:54:06      43098 HG00978.hgr1.vcf\r\n",
      "2017-03-06 22:54:07        781 HG00978.hgr1.vcf.idx\r\n",
      "2017-03-07 11:28:46      53048 HG00982.hgr1.vcf\r\n",
      "2017-03-07 11:28:46       2301 HG00982.hgr1.vcf.idx\r\n",
      "2017-03-06 22:18:41      28059 HG02190.hgr1.vcf\r\n",
      "2017-03-06 22:18:42        284 HG02190.hgr1.vcf.idx\r\n",
      "2017-03-06 16:16:48      26310 HG02283.hgr1.vcf\r\n",
      "2017-03-06 16:16:49        283 HG02283.hgr1.vcf.idx\r\n",
      "2017-03-06 16:03:32      23316 HG02343.hgr1.vcf\r\n",
      "2017-03-06 16:03:33        284 HG02343.hgr1.vcf.idx\r\n",
      "2017-03-06 17:22:04      42688 HG02479.hgr1.vcf\r\n",
      "2017-03-06 17:22:04       1284 HG02479.hgr1.vcf.idx\r\n",
      "2017-03-06 17:18:00      44321 HG02508.hgr1.vcf\r\n",
      "2017-03-06 17:18:00        782 HG02508.hgr1.vcf.idx\r\n",
      "2017-03-06 20:39:59      27550 HG03604.hgr1.vcf\r\n",
      "2017-03-06 20:40:00        283 HG03604.hgr1.vcf.idx\r\n",
      "2017-03-06 18:45:52      29420 HG03607.hgr1.vcf\r\n",
      "2017-03-06 18:45:53        284 HG03607.hgr1.vcf.idx\r\n",
      "2017-03-06 23:35:54      28702 HG03616.hgr1.vcf\r\n",
      "2017-03-06 23:35:54        283 HG03616.hgr1.vcf.idx\r\n",
      "2017-03-06 20:56:57      26647 HG04162.hgr1.vcf\r\n",
      "2017-03-06 20:56:57        284 HG04162.hgr1.vcf.idx\r\n",
      "2017-03-07 11:09:14      27959 NA06985.hgr1.vcf\r\n",
      "2017-03-07 11:09:14        282 NA06985.hgr1.vcf.idx\r\n",
      "2017-03-07 11:40:47      31776 NA11881.hgr1.vcf\r\n",
      "2017-03-07 11:40:47        284 NA11881.hgr1.vcf.idx\r\n",
      "2017-03-07 12:47:51      36404 NA12234.hgr1.vcf\r\n",
      "2017-03-07 12:47:52        780 NA12234.hgr1.vcf.idx\r\n",
      "2017-03-07 12:41:01      23532 NA12283.hgr1.vcf\r\n",
      "2017-03-07 12:41:02        283 NA12283.hgr1.vcf.idx\r\n",
      "2017-03-05 11:16:48      33001 NA12878_pp.hgr1.vcf\r\n",
      "2017-03-05 11:31:26        288 NA12878_pp.hgr1.vcf.idx\r\n",
      "2017-03-05 11:15:19      30814 NA12891_pp.hgr1.vcf\r\n",
      "2017-03-05 11:15:19        287 NA12891_pp.hgr1.vcf.idx\r\n",
      "2017-03-06 00:14:00      29637 NA12892_pp.hgr1.vcf\r\n",
      "2017-03-06 00:14:00        286 NA12892_pp.hgr1.vcf.idx\r\n",
      "2017-03-07 14:36:12      19482 NA18533.hgr1.vcf\r\n",
      "2017-03-07 14:36:13        282 NA18533.hgr1.vcf.idx\r\n",
      "2017-03-07 13:23:47      48767 NA18622.hgr1.vcf\r\n",
      "2017-03-07 13:23:47       2301 NA18622.hgr1.vcf.idx\r\n",
      "2017-03-07 13:24:24      32073 NA18632.hgr1.vcf\r\n",
      "2017-03-07 13:24:24        272 NA18632.hgr1.vcf.idx\r\n",
      "2017-03-07 14:36:35      54286 NA18647.hgr1.vcf\r\n",
      "2017-03-07 14:36:36        782 NA18647.hgr1.vcf.idx\r\n",
      "2017-03-06 19:27:14      37320 NA19984.hgr1.vcf\r\n",
      "2017-03-06 19:27:14        780 NA19984.hgr1.vcf.idx\r\n",
      "2017-03-06 18:11:22      24555 NA20357.hgr1.vcf\r\n",
      "2017-03-06 18:11:23        285 NA20357.hgr1.vcf.idx\r\n",
      "2017-03-06 19:29:44      29104 NA20359.hgr1.vcf\r\n",
      "2017-03-06 19:29:45        281 NA20359.hgr1.vcf.idx\r\n",
      "2017-03-05 23:18:07      28970 testGenome.hgr1.vcf\r\n",
      "2017-03-05 23:18:07        288 testGenome.hgr1.vcf.idx\r\n"
     ]
    }
   ],
   "source": [
    "date # this is in flux, will be updated\n",
    "aws s3 ls s3://crownproject/1kg_hgr1/ | grep \".vcf\" -"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Completed 16.7 KiB/1.0 MiB with 31 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG00717.hgr1.vcf to ./HG00717.hgr1.vcf\r\n",
      "Completed 16.7 KiB/1.0 MiB with 30 file(s) remaining\r",
      "Completed 42.4 KiB/1.0 MiB with 30 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG02283.hgr1.vcf to ./HG02283.hgr1.vcf\r\n",
      "Completed 42.4 KiB/1.0 MiB with 29 file(s) remaining\r",
      "Completed 78.5 KiB/1.0 MiB with 29 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG00557.hgr1.vcf to ./HG00557.hgr1.vcf\r\n",
      "Completed 78.5 KiB/1.0 MiB with 28 file(s) remaining\r",
      "Completed 105.9 KiB/1.0 MiB with 28 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG02190.hgr1.vcf to ./HG02190.hgr1.vcf\r\n",
      "Completed 105.9 KiB/1.0 MiB with 27 file(s) remaining\r",
      "Completed 168.3 KiB/1.0 MiB with 27 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG00851.hgr1.vcf to ./HG00851.hgr1.vcf\r\n",
      "Completed 168.3 KiB/1.0 MiB with 26 file(s) remaining\r",
      "Completed 210.0 KiB/1.0 MiB with 26 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG02479.hgr1.vcf to ./HG02479.hgr1.vcf\r\n",
      "Completed 210.0 KiB/1.0 MiB with 25 file(s) remaining\r",
      "Completed 267.5 KiB/1.0 MiB with 25 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG00536.hgr1.vcf to ./HG00536.hgr1.vcf\r\n",
      "Completed 267.5 KiB/1.0 MiB with 24 file(s) remaining\r",
      "Completed 290.3 KiB/1.0 MiB with 24 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG02343.hgr1.vcf to ./HG02343.hgr1.vcf\r\n",
      "Completed 290.3 KiB/1.0 MiB with 23 file(s) remaining\r",
      "Completed 332.4 KiB/1.0 MiB with 23 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG00978.hgr1.vcf to ./HG00978.hgr1.vcf\r\n",
      "Completed 332.4 KiB/1.0 MiB with 22 file(s) remaining\r",
      "Completed 392.5 KiB/1.0 MiB with 22 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG00478.hgr1.vcf to ./HG00478.hgr1.vcf\r\n",
      "Completed 392.5 KiB/1.0 MiB with 21 file(s) remaining\r",
      "Completed 420.6 KiB/1.0 MiB with 21 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG03616.hgr1.vcf to ./HG03616.hgr1.vcf\r\n",
      "Completed 420.6 KiB/1.0 MiB with 20 file(s) remaining\r",
      "Completed 447.5 KiB/1.0 MiB with 20 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG03604.hgr1.vcf to ./HG03604.hgr1.vcf\r\n",
      "Completed 447.5 KiB/1.0 MiB with 19 file(s) remaining\r",
      "Completed 476.2 KiB/1.0 MiB with 19 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG03607.hgr1.vcf to ./HG03607.hgr1.vcf\r\n",
      "Completed 476.2 KiB/1.0 MiB with 18 file(s) remaining\r",
      "Completed 519.5 KiB/1.0 MiB with 18 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG02508.hgr1.vcf to ./HG02508.hgr1.vcf\r\n",
      "Completed 519.5 KiB/1.0 MiB with 17 file(s) remaining\r",
      "Completed 542.5 KiB/1.0 MiB with 17 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA12283.hgr1.vcf to ./NA12283.hgr1.vcf\r\n",
      "Completed 542.5 KiB/1.0 MiB with 16 file(s) remaining\r",
      "Completed 572.5 KiB/1.0 MiB with 16 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA12891_pp.hgr1.vcf to ./NA12891_pp.hgr1.vcf\r\n",
      "Completed 572.5 KiB/1.0 MiB with 15 file(s) remaining\r",
      "Completed 598.6 KiB/1.0 MiB with 15 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG04162.hgr1.vcf to ./HG04162.hgr1.vcf\r\n",
      "Completed 598.6 KiB/1.0 MiB with 14 file(s) remaining\r",
      "Completed 629.6 KiB/1.0 MiB with 14 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA11881.hgr1.vcf to ./NA11881.hgr1.vcf\r\n",
      "Completed 629.6 KiB/1.0 MiB with 13 file(s) remaining\r",
      "Completed 656.9 KiB/1.0 MiB with 13 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA06985.hgr1.vcf to ./NA06985.hgr1.vcf\r\n",
      "Completed 656.9 KiB/1.0 MiB with 12 file(s) remaining\r",
      "Completed 692.5 KiB/1.0 MiB with 12 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA12234.hgr1.vcf to ./NA12234.hgr1.vcf\r\n",
      "Completed 692.5 KiB/1.0 MiB with 11 file(s) remaining\r",
      "Completed 711.5 KiB/1.0 MiB with 11 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA18533.hgr1.vcf to ./NA18533.hgr1.vcf\r\n",
      "Completed 711.5 KiB/1.0 MiB with 10 file(s) remaining\r",
      "Completed 743.7 KiB/1.0 MiB with 10 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA12878_pp.hgr1.vcf to ./NA12878_pp.hgr1.vcf\r\n",
      "Completed 743.7 KiB/1.0 MiB with 9 file(s) remaining\r",
      "Completed 772.6 KiB/1.0 MiB with 9 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA12892_pp.hgr1.vcf to ./NA12892_pp.hgr1.vcf\r\n",
      "Completed 772.6 KiB/1.0 MiB with 8 file(s) remaining\r",
      "Completed 801.1 KiB/1.0 MiB with 8 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA20359.hgr1.vcf to ./NA20359.hgr1.vcf\r\n",
      "Completed 801.1 KiB/1.0 MiB with 7 file(s) remaining\r",
      "Completed 829.4 KiB/1.0 MiB with 7 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/testGenome.hgr1.vcf to ./testGenome.hgr1.vcf\r\n",
      "Completed 829.4 KiB/1.0 MiB with 6 file(s) remaining\r",
      "Completed 877.0 KiB/1.0 MiB with 6 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA18622.hgr1.vcf to ./NA18622.hgr1.vcf\r\n",
      "Completed 877.0 KiB/1.0 MiB with 5 file(s) remaining\r",
      "Completed 913.4 KiB/1.0 MiB with 5 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA19984.hgr1.vcf to ./NA19984.hgr1.vcf\r\n",
      "Completed 913.4 KiB/1.0 MiB with 4 file(s) remaining\r",
      "Completed 966.4 KiB/1.0 MiB with 4 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA18647.hgr1.vcf to ./NA18647.hgr1.vcf\r\n",
      "Completed 966.4 KiB/1.0 MiB with 3 file(s) remaining\r",
      "Completed 1018.2 KiB/1.0 MiB with 3 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG00982.hgr1.vcf to ./HG00982.hgr1.vcf\r\n",
      "Completed 1018.2 KiB/1.0 MiB with 2 file(s) remaining\r",
      "Completed 1.0 MiB/1.0 MiB with 2 file(s) remaining   \r",
      "download: s3://crownproject/1kg_hgr1/NA18632.hgr1.vcf to ./NA18632.hgr1.vcf\r\n",
      "Completed 1.0 MiB/1.0 MiB with 1 file(s) remaining\r",
      "Completed 1.0 MiB/1.0 MiB with 1 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA20357.hgr1.vcf to ./NA20357.hgr1.vcf\r\n"
     ]
    }
   ],
   "source": [
    "aws s3 cp s3://crownproject/1kg_hgr1/ ./ --recursive --exclude \"*\" --include \"*.vcf\"\n",
    "\n",
    "rm testGenome.hgr1.vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": []
    }
   ],
   "source": [
    "# convert files to BCF\n",
    "VCF=$(ls *.vcf)\n",
    "\n",
    "for FILE in $(ls *.vcf)\n",
    "do\n",
    "    # Convert VCF to BCF File\n",
    "    bcftools view -O b $FILE -o $FILE.bcf\n",
    "    bcftools index $FILE.bcf\n",
    "done\n",
    "\n",
    "bcftools merge $(ls *.bcf) -o 100genomes_merged.vcf\n",
    "bcftools view -O b 100genomes_merged.vcf -o 100genomes_merged.bcf\n",
    "bcftools index 100genomes_merged.bcf\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[E::main_vcfindex] unknown filetype; expected bgzip compressed VCF or BCF\r\n",
      "[E::main_vcfindex] was the VCF/BCF compressed with bgzip?\r\n"
     ]
    }
   ],
   "source": [
    "# Filter for variants with 30x coverage\n",
    "\n",
    "bcftools view -i \"DP>=30\" 100genomes_merged.vcf -o 100genomes_DP30.vcf\n",
    "bcftools index 100genomes_DP30.vcf\n",
    "#bcftools filter -i \"DP>=30\" 100genomes_merged.bcf | bcftools view -O b 100genomes.DP30.bcf\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Summary Plots for VCF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Parsing bcftools stats output: 100g.dp30.vcfplot\r\n",
      "Plotting graphs: python plot_dp30/100g-plot.py\r\n",
      "Creating PDF: pdflatex 100g-summary.tex >100g-plot-vcfstats.log 2>&1\r\n",
      "Finished: plot_dp30/100g-summary.pdf\r\n",
      "Parsing bcftools stats output: 100g.vcfplot\r\n",
      "Plotting graphs: python plot/100g-plot.py\r\n",
      "Creating PDF: pdflatex 100g-summary.tex >100g-plot-vcfstats.log 2>&1\r\n",
      "Finished: plot/100g-summary.pdf\r\n"
     ]
    }
   ],
   "source": [
    "## vcfplot (summary stats)\n",
    "\n",
    "# DP30 filtered\n",
    "bcftools stats 100genomes_DP30.vcf > 100g.dp30.vcfplot\n",
    "plot-vcfstats -p plot_dp30/100g 100g.dp30.vcfplot\n",
    "\n",
    "# nonDP30 filtered\n",
    "bcftools stats 100genomes_merged.vcf > 100g.vcfplot\n",
    "plot-vcfstats -p plot/100g 100g.vcfplot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "## Measure the density of variants in each region\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### vcfR Analysis - statistical analysis in R\n",
    "\n",
    "To do the heavy lifting number-wise I went to an R based package called `vcfR` it's actually really good!\n",
    "\n",
    "Still I have to do some custom code stuff for rRNA variation analysis since it's outside the norm but I can extract information from the VCF file into a data.frame fairly quiickly with vcfR. Script copy below, subject to change on disk.\n",
    "\n",
    "` cd ~/Crown/data/vcfR_analysis/`\n",
    "\n",
    "vcf_100g.r - script for analysis\n",
    "\n",
    "```\n",
    "# vcfR analysis\n",
    "# 100 genomes hgr1\n",
    "# 170308\n",
    "#\n",
    "# hgr1_v1 vcf (variant input only) analysis\n",
    "\n",
    "# Install vcfR\n",
    "#install.packages(\"vcfR\")\n",
    "library(\"vcfR\")\n",
    "library(\"ggplot2\")\n",
    "library(\"reshape2\")\n",
    "\n",
    "# File pointers\n",
    "\n",
    "  vcf_file = '30genomes_merged.vcf.gz'\n",
    "  dna_file = 'rDNA.fa'\n",
    "  gff_file = 'rDNA.gff'\n",
    "\n",
    "# output Prefix (for plots)\n",
    "  \n",
    "  outPrefix='30genomes'\n",
    "\n",
    "# SCRIPT ============================================================\n",
    "#====================================================================\n",
    "# Import VCF / DNA / GFF\n",
    "  VCF = read.vcfR(vcf_file)\n",
    "  DNA = ape::read.dna(dna_file, format = 'fasta')\n",
    "  GFF = read.table(gff_file, sep=\"\\t\", quote = \"\")\n",
    "  \n",
    "  \n",
    "# Create chromR Object\n",
    "  chrom = create.chromR(name=\"rDNA\", vcf=VCF, seq=DNA, ann=GFF)\n",
    "\n",
    "  #plot(chrom)\n",
    " #chromoqc(chrom, dp.alpha = 22)\n",
    "\n",
    "\n",
    "# Variant Statistics ================================================\n",
    "\n",
    "# Intra-individual variation (_i)\n",
    "  \n",
    "  # Depth of coverage at each variant\n",
    "  DP = extract.gt(chrom, element=\"DP\", as.numeric=TRUE)\n",
    "\n",
    "  # Reference allele-only depth\n",
    "  RAD = extract.gt(chrom, element=\"AD\", as.numeric=TRUE)\n",
    "\n",
    "  # Variant allele frequency (intra-individual)\n",
    "  iVAF = (DP-RAD)/DP\n",
    "  \n",
    "  \n",
    "# Population variation (called variants)\n",
    "  \n",
    "  # Number of individual genomes in total population\n",
    "  N_pop = length(DP[1,])\n",
    "  \n",
    "  # Called Variant allele count (population)\n",
    "  pV = rowSums(!is.na(iVAF))\n",
    "  pVAF = pV / N_pop\n",
    "  \n",
    "  # Average intra-genomic variant allele frequency\n",
    "  mean_iVAF = apply(iVAF,1,mean, na.rm = T)\n",
    "  sd_iVAF = apply(iVAF,1,sd, na.rm = T)\n",
    "\n",
    "# Population Variant Statistics\n",
    "  \n",
    "POPVAR = data.frame(pVAF, mean_iVAF, sd_iVAF)\n",
    "\n",
    "# Plot--------------------------------------------------------------\n",
    "# Mean Intra-individual Variant Allele Frequency vs.\n",
    "# Population-level Variant Allele Frequency (called vs not called)\n",
    "\n",
    "# open PDF device\n",
    "pdf(file = paste0(outPrefix,\".MeaniVAF_pVAF.pdf\"), width = 5, height = 5)\n",
    "\n",
    "PLOT1 = ggplot(POPVAR, aes(mean_iVAF, pVAF))\n",
    "PLOT1 = PLOT1 + geom_point(alpha = 2/10, stroke = 0, aes(size = 0.5))\n",
    "PLOT1 = PLOT1 + theme_bw()\n",
    "PLOT1 = PLOT1 + theme(legend.position=\"none\")\n",
    "PLOT1 = PLOT1 + xlab('Mean Intra-individual Variant Allele Frequency')\n",
    "PLOT1 = PLOT1 + ylab('Population Variant Allele Frequency')\n",
    "PLOT1\n",
    "\n",
    "dev.off()\n",
    "# ------------------------------------------------------------------ \n",
    "  \n",
    "# Stratify variants by their mean Intra-individual allele frequency\n",
    "# the hypothesis was that there are two classes of variants;\n",
    "# Directional Variants, which can range from 0 - 1\n",
    "# Stabilized Variants, which are maintained in a narrower range\n",
    "\n",
    "# strat0_2 =  which(mean_iVAF <= 0.2)\n",
    "# strat2_4 =  which(mean_iVAF > 0.2 & mean_iVAF <= 0.4)\n",
    "# strat4_6 =  which(mean_iVAF > 0.4 & mean_iVAF <= 0.6)\n",
    "# strat6_8 =  which(mean_iVAF > 0.6 & mean_iVAF <= 0.8)\n",
    "# strat8_1 =  which(mean_iVAF > 0.8)\n",
    "\n",
    "\n",
    "Stratify = function(iVAF, minVAF, maxVAF){\n",
    "  \n",
    "  # calcualte the average intra-individual VAF for each variant\n",
    "  mean_iVAF = apply(iVAF,1, mean, na.rm = T)\n",
    "  \n",
    "  # Subselect (index) mean iVAF between a range of values\n",
    "  strat_lim = which(mean_iVAF > minVAF & mean_iVAF <= maxVAF)\n",
    "  \n",
    "  # Actually subselect by stratification\n",
    "  iVAF_str = apply(iVAF[strat_lim,], 1, sort, decreasing = TRUE)\n",
    "  \n",
    "  N_strats = length(iVAF_str) # number of variants in this stratification\n",
    "  N_samples = length(iVAF[1,]) # number of samples in the analysis (people)\n",
    "  \n",
    "  iVAF_str_matrix = matrix( rep(0, N_strats * N_samples), nrow = N_strats)\n",
    "  \n",
    "  for (X in 1:N_strats){\n",
    "    LINE_VALUES = unlist(iVAF_str[X])\n",
    "    length(LINE_VALUES) = N_samples\n",
    "    LINE_VALUES[is.na(LINE_VALUES)] <- 0\n",
    "    \n",
    "    iVAF_str_matrix[X,] = iVAF_str_matrix[X,] + LINE_VALUES\n",
    "  }\n",
    "  \n",
    "  return(iVAF_str_matrix)\n",
    "  \n",
    "}\n",
    "\n",
    "Stratify_rownames = function(iVAF, minVAF, maxVAF){\n",
    "  # calcualte the average intra-individual VAF for each variant\n",
    "  mean_iVAF = apply(iVAF,1, mean, na.rm = T)\n",
    "  \n",
    "  # Subselect (index) mean iVAF between a range of values\n",
    "  strat_lim = which(mean_iVAF > minVAF & mean_iVAF <= maxVAF)\n",
    "  \n",
    "  return(names(strat_lim))\n",
    "}\n",
    "\n",
    "StratifyPlot = function(iVAF, minVAF, maxVAF){\n",
    "  \n",
    "  ## Stratify variants by intra-individual variant allele frequency\n",
    "  iVAF_SUB_matrix = data.frame(Stratify(iVAF, minVAF, maxVAF), row.names = Stratify_rownames(iVAF, minVAF, maxVAF))\n",
    "  \n",
    "  # Shape data to be plotted\n",
    "  iVAF_SUB_matrix = melt(iVAF_SUB_matrix)\n",
    "  iVAF_SUB_matrix$rowid = Stratify_rownames(iVAF, minVAF, maxVAF)\n",
    "\n",
    "    # Plot--------------------------------------------------------------\n",
    "    # Mean Intra-individual Variant Allele Frequency vs.\n",
    "    # Population-level Variant Allele Frequency (called vs not called)\n",
    "    plotTitle = paste0(\"Mean VAF: \",minVAF,\" - \",maxVAF)\n",
    "  \n",
    "    PLOT2 = ggplot(iVAF_SUB_matrix, aes(variable, value, group = factor(rowid)))\n",
    "    PLOT2 = PLOT2 + geom_line(aes(color = factor(rowid)))\n",
    "    PLOT2 = PLOT2 + theme_bw()\n",
    "    PLOT2 = PLOT2 + theme(legend.position=\"none\")\n",
    "    PLOT2 = PLOT2 + ylim(0,1)\n",
    "    PLOT2 = PLOT2 + scale_x_discrete(breaks = c(1,N_pop))\n",
    "    PLOT2 = PLOT2 + ggtitle(plotTitle)\n",
    "    #PLOT2 = PLOT2 + ylab('Intra-individual Variant Allele Frequency')\n",
    "    #PLOT2 = PLOT2 + xlab('Genomes')\n",
    "    ## -----------------------------------------------------------------\n",
    "    #PLOT2\n",
    "    \n",
    "    return(PLOT2)\n",
    "}\n",
    "\n",
    "\n",
    "# Plot Stratify for multiple ranges\n",
    "minVAF=0 # initialize with min VAF at 0\n",
    "nStratification=5 # How many equal parts to divide the VAF from 0-1\n",
    "\n",
    "# range width of each stratification\n",
    "widthStratification=1/nStratification\n",
    "\n",
    "for (nS in 1:nStratification){\n",
    "\n",
    "  maxVAF = minVAF + widthStratification\n",
    "  \n",
    "  # Generate Plot\n",
    "  PLOT_VAF = StratifyPlot(iVAF, minVAF, maxVAF)\n",
    "  \n",
    "  # Open PDF file to write to\n",
    "  pdf(file = paste0(outPrefix,\".\",nS,\".strat.pdf\"),\n",
    "     width = 5, height = 5)\n",
    "  \n",
    "    print(PLOT_VAF)\n",
    "  \n",
    "  dev.off()\n",
    "  \n",
    "  minVAF = minVAF + widthStratification\n",
    "}\n",
    "\n",
    "\n",
    "# Heatmap ================================================\n",
    "# Log transform depth-data at each variant\n",
    "logDP = log(DP)\n",
    "\n",
    "# # chrom from tutotrial\n",
    "\n",
    "heatmap.bp(iVAF, cbarplot = F, rbarplot = F, rlabels = F)\n",
    "\n",
    "heatmap.bp(logDP, cbarplot = F, rbarplot = F, rlabels = F)\n",
    "\n",
    "```\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Output from vcfR analysis + plotVCF\n",
    "\n",
    "Summary Statistics (read-depth 30 filtered) for 30 genomes\n",
    "![Summary Statistics (read-depth 30) for 30 genomes](../figure/20170308_summary_30g.png)\n",
    "\n",
    "Intra-genomic Variant Allele Frequency (VAF) vs. Population VAF\n",
    "![iVAF vs. pVAF](../figure/20170308_30g_iVAF_v_pVAF.png)\n",
    "\n",
    "- Bottom right panel is iVAF vs. pVAF for every variant in the dataset.\n",
    "- Note the variants which are variable within a high fraction of genomes, namely at (0.48,1.0) which is the 28S.A49G variation. It's present in 100% of genomes. Evidence for stabalizing selection of this allele in all humans\n",
    "- The line graphs are little involved. Variants were stratified by their average intra-genomic VAF. In the 0.4 - 0.6 graph this is best exemplified. The vast majority of individual variants (lines) are present at a high intragenomic frequency (y-axis) in a few individuals (x-axis) and absent in others. The exception is again 28S.A49G which is intra-genomically variant in all genomes.\n",
    "- Keep in mind this doesn't account for 'allelic dropout' when sequencing depth is poor over high GC regions so other 'stabalized' intragenomic variants may have individual samples where that variant looks 'fixed' or 'absent'. This needs to be refined a little bit but it's incredibly strong evidence for paralogous rRNA in humans\n",
    "\n",
    "30 Genomes - Intra-genomic variant allele frequency for each called variant\n",
    "![30g iVAF heatmap](../figure/20170308_30g_iVAF_hm.png)\n",
    "\n",
    "30 Genomes - Read depth of each called variant.\n",
    "![30g DP heatmap](../figure/20170308_30g_log_DP_hm.png)\n",
    "\n",
    "- Column = individual genome\n",
    "- Row = individual variants\n",
    "- Scale = logDP 1 - 3200; iVAF 0.0 - 1.0"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Full dataset - 107 genomes\n",
    "\n",
    "I moved the joined 30 genomes VCF file (and renamed it) to the 30g folder in ~/Crown/data/hgr1_vcf\n",
    "\n",
    "Download the complete set of .vcf files from 104 genomes and re-analyze as above.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "30g.dp30.vcfplot    30genomes_DP30.bcf.csi  30genomes_merged.vcf  plot\r\n",
      "30genomes_DP30.bcf  30genomes_DP30.vcf\t    30g.vcfplot\t\t  plot_dp30\r\n",
      "Fri Mar 10 10:19:47 PST 2017\r\n",
      "2017-03-08 12:31:14      39989 HG00128.hgr1.vcf\r\n",
      "2017-03-08 12:31:15        781 HG00128.hgr1.vcf.idx\r\n",
      "2017-03-08 12:41:08      26601 HG00139.hgr1.vcf\r\n",
      "2017-03-08 12:41:09        282 HG00139.hgr1.vcf.idx\r\n",
      "2017-03-08 13:53:28      60138 HG00234.hgr1.vcf\r\n",
      "2017-03-08 13:53:29       2300 HG00234.hgr1.vcf.idx\r\n",
      "2017-03-08 12:17:24      22863 HG00253.hgr1.vcf\r\n",
      "2017-03-08 12:17:24        282 HG00253.hgr1.vcf.idx\r\n",
      "2017-03-08 00:48:57      41492 HG00337.hgr1.vcf\r\n",
      "2017-03-08 00:48:57        782 HG00337.hgr1.vcf.idx\r\n",
      "2017-03-08 00:15:02      67236 HG00353.hgr1.vcf\r\n",
      "2017-03-08 00:15:02       4325 HG00353.hgr1.vcf.idx\r\n",
      "2017-03-08 00:13:24      49655 HG00358.hgr1.vcf\r\n",
      "2017-03-08 00:13:24        780 HG00358.hgr1.vcf.idx\r\n",
      "2017-03-08 01:07:43      27207 HG00362.hgr1.vcf\r\n",
      "2017-03-08 01:07:43        281 HG00362.hgr1.vcf.idx\r\n",
      "2017-03-07 14:24:07      61563 HG00478.hgr1.vcf\r\n",
      "2017-03-07 14:24:07       4324 HG00478.hgr1.vcf.idx\r\n",
      "2017-03-07 16:08:21      58909 HG00536.hgr1.vcf\r\n",
      "2017-03-07 16:08:22       2298 HG00536.hgr1.vcf.idx\r\n",
      "2017-03-07 16:09:22      37016 HG00557.hgr1.vcf\r\n",
      "2017-03-07 16:09:22        782 HG00557.hgr1.vcf.idx\r\n",
      "2017-03-07 16:11:04      17098 HG00717.hgr1.vcf\r\n",
      "2017-03-07 16:11:05        284 HG00717.hgr1.vcf.idx\r\n",
      "2017-03-09 19:43:26      23630 HG00734.hgr1.vcf\r\n",
      "2017-03-09 19:43:26        281 HG00734.hgr1.vcf.idx\r\n",
      "2017-03-09 19:11:00      22875 HG00740.hgr1.vcf\r\n",
      "2017-03-09 19:11:00        283 HG00740.hgr1.vcf.idx\r\n",
      "2017-03-06 23:18:53      63888 HG00851.hgr1.vcf\r\n",
      "2017-03-06 23:18:53       4325 HG00851.hgr1.vcf.idx\r\n",
      "2017-03-06 22:54:06      43098 HG00978.hgr1.vcf\r\n",
      "2017-03-06 22:54:07        781 HG00978.hgr1.vcf.idx\r\n",
      "2017-03-07 11:28:46      53048 HG00982.hgr1.vcf\r\n",
      "2017-03-07 11:28:46       2301 HG00982.hgr1.vcf.idx\r\n",
      "2017-03-09 19:48:39      24915 HG01066.hgr1.vcf\r\n",
      "2017-03-09 19:48:40        284 HG01066.hgr1.vcf.idx\r\n",
      "2017-03-07 18:17:56      24663 HG01125.hgr1.vcf\r\n",
      "2017-03-07 18:17:56        283 HG01125.hgr1.vcf.idx\r\n",
      "2017-03-09 21:26:17      28607 HG01176.hgr1.vcf\r\n",
      "2017-03-09 21:26:17        285 HG01176.hgr1.vcf.idx\r\n",
      "2017-03-07 21:55:31      63904 HG01277.hgr1.vcf\r\n",
      "2017-03-07 21:55:32       1285 HG01277.hgr1.vcf.idx\r\n",
      "2017-03-07 21:52:28      22165 HG01348.hgr1.vcf\r\n",
      "2017-03-07 21:52:28        270 HG01348.hgr1.vcf.idx\r\n",
      "2017-03-07 20:20:28      27827 HG01464.hgr1.vcf\r\n",
      "2017-03-07 20:20:28        283 HG01464.hgr1.vcf.idx\r\n",
      "2017-03-09 15:22:45      18594 HG01589.hgr1.vcf\r\n",
      "2017-03-09 15:22:45        285 HG01589.hgr1.vcf.idx\r\n",
      "2017-03-08 21:05:54      39140 HG01620.hgr1.vcf\r\n",
      "2017-03-08 21:05:54        780 HG01620.hgr1.vcf.idx\r\n",
      "2017-03-08 18:02:22      18830 HG01626.hgr1.vcf\r\n",
      "2017-03-08 18:02:22        283 HG01626.hgr1.vcf.idx\r\n",
      "2017-03-08 20:50:21      49106 HG01630.hgr1.vcf\r\n",
      "2017-03-08 20:50:21        779 HG01630.hgr1.vcf.idx\r\n",
      "2017-03-08 23:48:17      28264 HG02035.hgr1.vcf\r\n",
      "2017-03-08 23:48:17        284 HG02035.hgr1.vcf.idx\r\n",
      "2017-03-08 23:43:32      27762 HG02040.hgr1.vcf\r\n",
      "2017-03-08 23:43:33        283 HG02040.hgr1.vcf.idx\r\n",
      "2017-03-08 23:49:06      30633 HG02046.hgr1.vcf\r\n",
      "2017-03-08 23:49:06        283 HG02046.hgr1.vcf.idx\r\n",
      "2017-03-09 04:20:28      24893 HG02078.hgr1.vcf\r\n",
      "2017-03-09 04:20:29        284 HG02078.hgr1.vcf.idx\r\n",
      "2017-03-09 15:33:28      56000 HG02090.hgr1.vcf\r\n",
      "2017-03-09 15:33:28       4326 HG02090.hgr1.vcf.idx\r\n",
      "2017-03-06 22:18:41      28059 HG02190.hgr1.vcf\r\n",
      "2017-03-06 22:18:42        284 HG02190.hgr1.vcf.idx\r\n",
      "2017-03-08 18:01:42      18158 HG02219.hgr1.vcf\r\n",
      "2017-03-08 18:01:42        283 HG02219.hgr1.vcf.idx\r\n",
      "2017-03-09 13:45:14      45077 HG02272.hgr1.vcf\r\n",
      "2017-03-09 13:45:15        779 HG02272.hgr1.vcf.idx\r\n",
      "2017-03-09 15:56:54      45271 HG02277.hgr1.vcf\r\n",
      "2017-03-09 15:56:55       1285 HG02277.hgr1.vcf.idx\r\n",
      "2017-03-06 16:16:48      26310 HG02283.hgr1.vcf\r\n",
      "2017-03-06 16:16:49        283 HG02283.hgr1.vcf.idx\r\n",
      "2017-03-09 15:56:34      60943 HG02286.hgr1.vcf\r\n",
      "2017-03-09 15:56:34       1285 HG02286.hgr1.vcf.idx\r\n",
      "2017-03-06 16:03:32      23316 HG02343.hgr1.vcf\r\n",
      "2017-03-06 16:03:33        284 HG02343.hgr1.vcf.idx\r\n",
      "2017-03-06 17:22:04      42688 HG02479.hgr1.vcf\r\n",
      "2017-03-06 17:22:04       1284 HG02479.hgr1.vcf.idx\r\n",
      "2017-03-06 17:18:00      44321 HG02508.hgr1.vcf\r\n",
      "2017-03-06 17:18:00        782 HG02508.hgr1.vcf.idx\r\n",
      "2017-03-09 17:33:48      17293 HG02780.hgr1.vcf\r\n",
      "2017-03-09 17:33:48        283 HG02780.hgr1.vcf.idx\r\n",
      "2017-03-08 17:24:07      28047 HG02839.hgr1.vcf\r\n",
      "2017-03-08 17:24:08        283 HG02839.hgr1.vcf.idx\r\n",
      "2017-03-08 17:16:29      29529 HG02888.hgr1.vcf\r\n",
      "2017-03-08 17:16:29        283 HG02888.hgr1.vcf.idx\r\n",
      "2017-03-08 15:11:59      26174 HG02891.hgr1.vcf\r\n",
      "2017-03-08 15:11:59        284 HG02891.hgr1.vcf.idx\r\n",
      "2017-03-07 21:26:50      26285 HG02946.hgr1.vcf\r\n",
      "2017-03-07 21:26:50        283 HG02946.hgr1.vcf.idx\r\n",
      "2017-03-09 17:37:23      23782 HG03019.hgr1.vcf\r\n",
      "2017-03-09 17:37:23        283 HG03019.hgr1.vcf.idx\r\n",
      "2017-03-07 21:14:57      25991 HG03195.hgr1.vcf\r\n",
      "2017-03-07 21:14:57        283 HG03195.hgr1.vcf.idx\r\n",
      "2017-03-08 15:16:28      29320 HG03240.hgr1.vcf\r\n",
      "2017-03-08 15:16:28        283 HG03240.hgr1.vcf.idx\r\n",
      "2017-03-07 23:16:32      27708 HG03268.hgr1.vcf\r\n",
      "2017-03-07 23:16:33        282 HG03268.hgr1.vcf.idx\r\n",
      "2017-03-07 21:21:04      33232 HG03295.hgr1.vcf\r\n",
      "2017-03-07 21:21:05        781 HG03295.hgr1.vcf.idx\r\n",
      "2017-03-09 10:53:21      21186 HG03432.hgr1.vcf\r\n",
      "2017-03-09 10:53:22        283 HG03432.hgr1.vcf.idx\r\n",
      "2017-03-09 11:01:12      32407 HG03470.hgr1.vcf\r\n",
      "2017-03-09 11:01:13        781 HG03470.hgr1.vcf.idx\r\n",
      "2017-03-09 10:44:51      23790 HG03548.hgr1.vcf\r\n",
      "2017-03-09 10:44:51        283 HG03548.hgr1.vcf.idx\r\n",
      "2017-03-09 10:44:09      20503 HG03558.hgr1.vcf\r\n",
      "2017-03-09 10:44:09        283 HG03558.hgr1.vcf.idx\r\n",
      "2017-03-06 20:39:59      27550 HG03604.hgr1.vcf\r\n",
      "2017-03-06 20:40:00        283 HG03604.hgr1.vcf.idx\r\n",
      "2017-03-06 18:45:52      29420 HG03607.hgr1.vcf\r\n",
      "2017-03-06 18:45:53        284 HG03607.hgr1.vcf.idx\r\n",
      "2017-03-06 23:35:54      28702 HG03616.hgr1.vcf\r\n",
      "2017-03-06 23:35:54        283 HG03616.hgr1.vcf.idx\r\n",
      "2017-03-09 17:32:22      24340 HG03624.hgr1.vcf\r\n",
      "2017-03-09 17:32:22        283 HG03624.hgr1.vcf.idx\r\n",
      "2017-03-08 20:23:17      25961 HG03720.hgr1.vcf\r\n",
      "2017-03-08 20:23:17        283 HG03720.hgr1.vcf.idx\r\n",
      "2017-03-09 18:29:49      27921 HG03753.hgr1.vcf\r\n",
      "2017-03-09 18:29:50        284 HG03753.hgr1.vcf.idx\r\n",
      "2017-03-08 21:59:38      26051 HG03790.hgr1.vcf\r\n",
      "2017-03-08 21:59:39        283 HG03790.hgr1.vcf.idx\r\n",
      "2017-03-09 20:28:40      29064 HG03836.hgr1.vcf\r\n",
      "2017-03-09 20:28:41        284 HG03836.hgr1.vcf.idx\r\n",
      "2017-03-09 20:32:27      25195 HG03856.hgr1.vcf\r\n",
      "2017-03-09 20:32:27        281 HG03856.hgr1.vcf.idx\r\n",
      "2017-03-08 20:33:40      28377 HG03861.hgr1.vcf\r\n",
      "2017-03-08 20:33:40        284 HG03861.hgr1.vcf.idx\r\n",
      "2017-03-08 20:31:06      33622 HG03867.hgr1.vcf\r\n",
      "2017-03-08 20:31:07        780 HG03867.hgr1.vcf.idx\r\n",
      "2017-03-09 20:35:00      24319 HG03890.hgr1.vcf\r\n",
      "2017-03-09 20:35:00        283 HG03890.hgr1.vcf.idx\r\n",
      "2017-03-06 20:56:57      26647 HG04162.hgr1.vcf\r\n",
      "2017-03-06 20:56:57        284 HG04162.hgr1.vcf.idx\r\n",
      "2017-03-07 11:09:14      27959 NA06985.hgr1.vcf\r\n",
      "2017-03-07 11:09:14        282 NA06985.hgr1.vcf.idx\r\n",
      "2017-03-07 11:40:47      31776 NA11881.hgr1.vcf\r\n",
      "2017-03-07 11:40:47        284 NA11881.hgr1.vcf.idx\r\n",
      "2017-03-07 12:47:51      36404 NA12234.hgr1.vcf\r\n",
      "2017-03-07 12:47:52        780 NA12234.hgr1.vcf.idx\r\n",
      "2017-03-07 12:41:01      23532 NA12283.hgr1.vcf\r\n",
      "2017-03-07 12:41:02        283 NA12283.hgr1.vcf.idx\r\n",
      "2017-03-05 11:16:48      33001 NA12878_pp.hgr1.vcf\r\n",
      "2017-03-05 11:31:26        288 NA12878_pp.hgr1.vcf.idx\r\n",
      "2017-03-05 11:15:19      30814 NA12891_pp.hgr1.vcf\r\n",
      "2017-03-05 11:15:19        287 NA12891_pp.hgr1.vcf.idx\r\n",
      "2017-03-06 00:14:00      29637 NA12892_pp.hgr1.vcf\r\n",
      "2017-03-06 00:14:00        286 NA12892_pp.hgr1.vcf.idx\r\n",
      "2017-03-07 14:36:12      19482 NA18533.hgr1.vcf\r\n",
      "2017-03-07 14:36:13        282 NA18533.hgr1.vcf.idx\r\n",
      "2017-03-07 13:23:47      48767 NA18622.hgr1.vcf\r\n",
      "2017-03-07 13:23:47       2301 NA18622.hgr1.vcf.idx\r\n",
      "2017-03-07 13:24:24      32073 NA18632.hgr1.vcf\r\n",
      "2017-03-07 13:24:24        272 NA18632.hgr1.vcf.idx\r\n",
      "2017-03-07 14:36:35      54286 NA18647.hgr1.vcf\r\n",
      "2017-03-07 14:36:36        782 NA18647.hgr1.vcf.idx\r\n",
      "2017-03-08 23:50:10      26879 NA18981.hgr1.vcf\r\n",
      "2017-03-08 23:50:11        283 NA18981.hgr1.vcf.idx\r\n",
      "2017-03-08 23:04:20      46039 NA19001.hgr1.vcf\r\n",
      "2017-03-08 23:04:21        780 NA19001.hgr1.vcf.idx\r\n",
      "2017-03-08 22:48:56      28963 NA19006.hgr1.vcf\r\n",
      "2017-03-08 22:48:57        283 NA19006.hgr1.vcf.idx\r\n",
      "2017-03-09 02:07:23      25184 NA19042.hgr1.vcf\r\n",
      "2017-03-09 02:07:23        283 NA19042.hgr1.vcf.idx\r\n",
      "2017-03-08 22:41:18      45156 NA19091.hgr1.vcf\r\n",
      "2017-03-08 22:41:18        781 NA19091.hgr1.vcf.idx\r\n",
      "2017-03-10 01:35:16      21253 NA19098.hgr1.vcf\r\n",
      "2017-03-10 01:35:16        283 NA19098.hgr1.vcf.idx\r\n",
      "2017-03-10 01:28:35      20513 NA19144.hgr1.vcf\r\n",
      "2017-03-10 01:28:36        283 NA19144.hgr1.vcf.idx\r\n",
      "2017-03-10 01:14:41      23136 NA19153.hgr1.vcf\r\n",
      "2017-03-10 01:14:41        283 NA19153.hgr1.vcf.idx\r\n",
      "2017-03-09 22:53:41      22115 NA19159.hgr1.vcf\r\n",
      "2017-03-09 22:53:41        285 NA19159.hgr1.vcf.idx\r\n",
      "2017-03-10 01:25:01      27899 NA19240.hgr1.vcf\r\n",
      "2017-03-10 01:25:02        283 NA19240.hgr1.vcf.idx\r\n",
      "2017-03-09 01:26:39      25162 NA19320.hgr1.vcf\r\n",
      "2017-03-09 01:26:39        285 NA19320.hgr1.vcf.idx\r\n",
      "2017-03-09 02:39:57      25154 NA19351.hgr1.vcf\r\n",
      "2017-03-09 02:39:58        281 NA19351.hgr1.vcf.idx\r\n",
      "2017-03-09 01:56:52      38055 NA19475.hgr1.vcf\r\n",
      "2017-03-09 01:56:53        782 NA19475.hgr1.vcf.idx\r\n",
      "2017-03-09 14:09:41      22693 NA19658.hgr1.vcf\r\n",
      "2017-03-09 14:09:41        284 NA19658.hgr1.vcf.idx\r\n",
      "2017-03-09 13:37:50      20161 NA19670.hgr1.vcf\r\n",
      "2017-03-09 13:37:50        282 NA19670.hgr1.vcf.idx\r\n",
      "2017-03-09 12:49:22      31541 NA19734.hgr1.vcf\r\n",
      "2017-03-09 12:49:22        283 NA19734.hgr1.vcf.idx\r\n",
      "2017-03-09 12:47:11      54940 NA19795.hgr1.vcf\r\n",
      "2017-03-09 12:47:11       2300 NA19795.hgr1.vcf.idx\r\n",
      "2017-03-06 19:27:14      37320 NA19984.hgr1.vcf\r\n",
      "2017-03-06 19:27:14        780 NA19984.hgr1.vcf.idx\r\n",
      "2017-03-06 18:11:22      24555 NA20357.hgr1.vcf\r\n",
      "2017-03-06 18:11:23        285 NA20357.hgr1.vcf.idx\r\n",
      "2017-03-06 19:29:44      29104 NA20359.hgr1.vcf\r\n",
      "2017-03-06 19:29:45        281 NA20359.hgr1.vcf.idx\r\n",
      "2017-03-09 23:52:39      21930 NA20503.hgr1.vcf\r\n",
      "2017-03-09 23:52:39        283 NA20503.hgr1.vcf.idx\r\n",
      "2017-03-09 23:55:39      24763 NA20514.hgr1.vcf\r\n",
      "2017-03-09 23:55:40        283 NA20514.hgr1.vcf.idx\r\n",
      "2017-03-09 20:55:16      20564 NA20827.hgr1.vcf\r\n",
      "2017-03-09 20:55:16        284 NA20827.hgr1.vcf.idx\r\n",
      "2017-03-09 22:36:19      20385 NA20832.hgr1.vcf\r\n",
      "2017-03-09 22:36:19        284 NA20832.hgr1.vcf.idx\r\n",
      "2017-03-08 12:07:40      31276 NA21109.hgr1.vcf\r\n",
      "2017-03-08 12:07:40        283 NA21109.hgr1.vcf.idx\r\n",
      "2017-03-08 16:15:43      34920 NA21112.hgr1.vcf\r\n",
      "2017-03-08 16:15:43        781 NA21112.hgr1.vcf.idx\r\n",
      "2017-03-08 15:45:11      35582 NA21113.hgr1.vcf\r\n",
      "2017-03-08 15:45:11        781 NA21113.hgr1.vcf.idx\r\n",
      "2017-03-08 18:45:53      26910 NA21125.hgr1.vcf\r\n",
      "2017-03-08 18:45:53        282 NA21125.hgr1.vcf.idx\r\n",
      "2017-03-05 23:18:07      28970 testGenome.hgr1.vcf\r\n",
      "2017-03-05 23:18:07        288 testGenome.hgr1.vcf.idx\r\n",
      "Completed 26.0 KiB/~1.7 MiB with ~51 file(s) remaining (calculating...)\r",
      "download: s3://crownproject/1kg_hgr1/HG00139.hgr1.vcf to ./HG00139.hgr1.vcf\r\n",
      "Completed 26.0 KiB/~1.8 MiB with ~53 file(s) remaining (calculating...)\r",
      "Completed 52.5 KiB/~1.8 MiB with ~55 file(s) remaining (calculating...)\r",
      "download: s3://crownproject/1kg_hgr1/HG00362.hgr1.vcf to ./HG00362.hgr1.vcf\r\n",
      "Completed 52.5 KiB/~1.8 MiB with ~54 file(s) remaining (calculating...)\r",
      "Completed 111.3 KiB/~2.1 MiB with ~64 file(s) remaining (calculating...)\r",
      "download: s3://crownproject/1kg_hgr1/HG00234.hgr1.vcf to ./HG00234.hgr1.vcf\r\n",
      "Completed 111.3 KiB/~2.2 MiB with ~66 file(s) remaining (calculating...)\r",
      "Completed 176.9 KiB/~2.3 MiB with ~71 file(s) remaining (calculating...)\r",
      "Completed 199.3 KiB/~2.4 MiB with ~74 file(s) remaining (calculating...)\r",
      "download: s3://crownproject/1kg_hgr1/HG00353.hgr1.vcf to ./HG00353.hgr1.vcf\r\n",
      "Completed 199.3 KiB/~2.4 MiB with ~73 file(s) remaining (calculating...)\r",
      "Completed 239.8 KiB/~2.5 MiB with ~75 file(s) remaining (calculating...)\r",
      "download: s3://crownproject/1kg_hgr1/HG00253.hgr1.vcf to ./HG00253.hgr1.vcf\r\n",
      "Completed 239.8 KiB/~2.5 MiB with ~74 file(s) remaining (calculating...)\r",
      "download: s3://crownproject/1kg_hgr1/HG00337.hgr1.vcf to ./HG00337.hgr1.vcf\r\n",
      "Completed 239.8 KiB/~2.6 MiB with ~76 file(s) remaining (calculating...)\r",
      "Completed 262.9 KiB/~2.8 MiB with ~83 file(s) remaining (calculating...)\r",
      "download: s3://crownproject/1kg_hgr1/HG00734.hgr1.vcf to ./HG00734.hgr1.vcf\r\n",
      "Completed 262.9 KiB/~2.9 MiB with ~87 file(s) remaining (calculating...)\r",
      "Completed 279.6 KiB/~3.2 MiB with ~97 file(s) remaining (calculating...)\r",
      "download: s3://crownproject/1kg_hgr1/HG00717.hgr1.vcf to ./HG00717.hgr1.vcf\r\n",
      "Completed 279.6 KiB/3.3 MiB with 100 file(s) remaining\r",
      "Completed 315.7 KiB/3.3 MiB with 100 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG00557.hgr1.vcf to ./HG00557.hgr1.vcf\r\n",
      "Completed 315.7 KiB/3.3 MiB with 99 file(s) remaining\r",
      "Completed 373.2 KiB/3.3 MiB with 99 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG00536.hgr1.vcf to ./HG00536.hgr1.vcf\r\n",
      "Completed 373.2 KiB/3.3 MiB with 98 file(s) remaining\r",
      "Completed 421.7 KiB/3.3 MiB with 98 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG00358.hgr1.vcf to ./HG00358.hgr1.vcf\r\n",
      "Completed 421.7 KiB/3.3 MiB with 97 file(s) remaining\r",
      "Completed 446.1 KiB/3.3 MiB with 97 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG01066.hgr1.vcf to ./HG01066.hgr1.vcf\r\n",
      "Completed 446.1 KiB/3.3 MiB with 96 file(s) remaining\r",
      "Completed 470.1 KiB/3.3 MiB with 96 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG01125.hgr1.vcf to ./HG01125.hgr1.vcf\r\n",
      "Completed 470.1 KiB/3.3 MiB with 95 file(s) remaining\r",
      "Completed 512.2 KiB/3.3 MiB with 95 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG00978.hgr1.vcf to ./HG00978.hgr1.vcf\r\n",
      "Completed 512.2 KiB/3.3 MiB with 94 file(s) remaining\r",
      "Completed 574.6 KiB/3.3 MiB with 94 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG00851.hgr1.vcf to ./HG00851.hgr1.vcf\r\n",
      "Completed 574.6 KiB/3.3 MiB with 93 file(s) remaining\r",
      "Completed 596.3 KiB/3.3 MiB with 93 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG01348.hgr1.vcf to ./HG01348.hgr1.vcf\r\n",
      "Completed 596.3 KiB/3.3 MiB with 92 file(s) remaining\r",
      "Completed 623.4 KiB/3.3 MiB with 92 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG01464.hgr1.vcf to ./HG01464.hgr1.vcf\r\n",
      "Completed 623.4 KiB/3.3 MiB with 91 file(s) remaining\r",
      "Completed 641.6 KiB/3.3 MiB with 91 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG01589.hgr1.vcf to ./HG01589.hgr1.vcf\r\n",
      "Completed 641.6 KiB/3.3 MiB with 90 file(s) remaining\r",
      "Completed 663.9 KiB/3.3 MiB with 90 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG00740.hgr1.vcf to ./HG00740.hgr1.vcf\r\n",
      "Completed 663.9 KiB/3.3 MiB with 89 file(s) remaining\r",
      "Completed 702.2 KiB/3.3 MiB with 89 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG01620.hgr1.vcf to ./HG01620.hgr1.vcf\r\n",
      "Completed 702.2 KiB/3.3 MiB with 88 file(s) remaining\r",
      "Completed 764.6 KiB/3.3 MiB with 88 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG01277.hgr1.vcf to ./HG01277.hgr1.vcf\r\n",
      "Completed 764.6 KiB/3.3 MiB with 87 file(s) remaining\r",
      "Completed 816.4 KiB/3.3 MiB with 87 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG00982.hgr1.vcf to ./HG00982.hgr1.vcf\r\n",
      "Completed 816.4 KiB/3.3 MiB with 86 file(s) remaining\r",
      "Completed 844.0 KiB/3.3 MiB with 86 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG02035.hgr1.vcf to ./HG02035.hgr1.vcf\r\n",
      "Completed 844.0 KiB/3.3 MiB with 85 file(s) remaining\r",
      "Completed 904.1 KiB/3.3 MiB with 85 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG00478.hgr1.vcf to ./HG00478.hgr1.vcf\r\n",
      "Completed 904.1 KiB/3.3 MiB with 84 file(s) remaining\r",
      "Completed 922.5 KiB/3.3 MiB with 84 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG01626.hgr1.vcf to ./HG01626.hgr1.vcf\r\n",
      "Completed 922.5 KiB/3.3 MiB with 83 file(s) remaining\r",
      "Completed 952.4 KiB/3.3 MiB with 83 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG02046.hgr1.vcf to ./HG02046.hgr1.vcf\r\n",
      "Completed 952.4 KiB/3.3 MiB with 82 file(s) remaining\r",
      "Completed 976.7 KiB/3.3 MiB with 82 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG02078.hgr1.vcf to ./HG02078.hgr1.vcf\r\n",
      "Completed 976.7 KiB/3.3 MiB with 81 file(s) remaining\r",
      "Completed 994.4 KiB/3.3 MiB with 81 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG02219.hgr1.vcf to ./HG02219.hgr1.vcf\r\n",
      "Completed 994.4 KiB/3.3 MiB with 80 file(s) remaining\r",
      "Completed 1.0 MiB/3.3 MiB with 80 file(s) remaining  \r",
      "download: s3://crownproject/1kg_hgr1/HG01630.hgr1.vcf to ./HG01630.hgr1.vcf\r\n",
      "Completed 1.0 MiB/3.3 MiB with 79 file(s) remaining\r",
      "Completed 1.0 MiB/3.3 MiB with 79 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG01176.hgr1.vcf to ./HG01176.hgr1.vcf\r\n",
      "Completed 1.0 MiB/3.3 MiB with 78 file(s) remaining\r",
      "Completed 1.1 MiB/3.3 MiB with 78 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG02283.hgr1.vcf to ./HG02283.hgr1.vcf\r\n",
      "Completed 1.1 MiB/3.3 MiB with 77 file(s) remaining\r",
      "Completed 1.1 MiB/3.3 MiB with 77 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG02040.hgr1.vcf to ./HG02040.hgr1.vcf\r\n",
      "Completed 1.1 MiB/3.3 MiB with 76 file(s) remaining\r",
      "Completed 1.1 MiB/3.3 MiB with 76 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG02272.hgr1.vcf to ./HG02272.hgr1.vcf\r\n",
      "Completed 1.1 MiB/3.3 MiB with 75 file(s) remaining\r",
      "Completed 1.2 MiB/3.3 MiB with 75 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG02343.hgr1.vcf to ./HG02343.hgr1.vcf\r\n",
      "Completed 1.2 MiB/3.3 MiB with 74 file(s) remaining\r",
      "Completed 1.2 MiB/3.3 MiB with 74 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG02286.hgr1.vcf to ./HG02286.hgr1.vcf\r\n",
      "Completed 1.2 MiB/3.3 MiB with 73 file(s) remaining\r",
      "Completed 1.2 MiB/3.3 MiB with 73 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG02780.hgr1.vcf to ./HG02780.hgr1.vcf\r\n",
      "Completed 1.2 MiB/3.3 MiB with 72 file(s) remaining\r",
      "Completed 1.3 MiB/3.3 MiB with 72 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG02479.hgr1.vcf to ./HG02479.hgr1.vcf\r\n",
      "Completed 1.3 MiB/3.3 MiB with 71 file(s) remaining\r",
      "Completed 1.3 MiB/3.3 MiB with 71 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG02839.hgr1.vcf to ./HG02839.hgr1.vcf\r\n",
      "Completed 1.3 MiB/3.3 MiB with 70 file(s) remaining\r",
      "Completed 1.4 MiB/3.3 MiB with 70 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG02090.hgr1.vcf to ./HG02090.hgr1.vcf\r\n",
      "Completed 1.4 MiB/3.3 MiB with 69 file(s) remaining\r",
      "Completed 1.4 MiB/3.3 MiB with 69 file(s) remaining\r",
      "Completed 1.4 MiB/3.3 MiB with 69 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG02891.hgr1.vcf to ./HG02891.hgr1.vcf\r\n",
      "Completed 1.4 MiB/3.3 MiB with 68 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG02277.hgr1.vcf to ./HG02277.hgr1.vcf\r\n",
      "Completed 1.4 MiB/3.3 MiB with 67 file(s) remaining\r",
      "Completed 1.5 MiB/3.3 MiB with 67 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG02946.hgr1.vcf to ./HG02946.hgr1.vcf\r\n",
      "Completed 1.5 MiB/3.3 MiB with 66 file(s) remaining\r",
      "Completed 1.5 MiB/3.3 MiB with 66 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG02888.hgr1.vcf to ./HG02888.hgr1.vcf\r\n",
      "Completed 1.5 MiB/3.3 MiB with 65 file(s) remaining\r",
      "Completed 1.5 MiB/3.3 MiB with 65 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG03195.hgr1.vcf to ./HG03195.hgr1.vcf\r\n",
      "Completed 1.5 MiB/3.3 MiB with 64 file(s) remaining\r",
      "Completed 1.5 MiB/3.3 MiB with 64 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG03019.hgr1.vcf to ./HG03019.hgr1.vcf\r\n",
      "Completed 1.5 MiB/3.3 MiB with 63 file(s) remaining\r",
      "Completed 1.6 MiB/3.3 MiB with 63 file(s) remaining\r",
      "Completed 1.6 MiB/3.3 MiB with 63 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG03295.hgr1.vcf to ./HG03295.hgr1.vcf\r\n",
      "Completed 1.6 MiB/3.3 MiB with 62 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG03268.hgr1.vcf to ./HG03268.hgr1.vcf\r\n",
      "Completed 1.6 MiB/3.3 MiB with 61 file(s) remaining\r",
      "Completed 1.6 MiB/3.3 MiB with 61 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG03470.hgr1.vcf to ./HG03470.hgr1.vcf\r\n",
      "Completed 1.6 MiB/3.3 MiB with 60 file(s) remaining\r",
      "Completed 1.6 MiB/3.3 MiB with 60 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG02190.hgr1.vcf to ./HG02190.hgr1.vcf\r\n",
      "Completed 1.6 MiB/3.3 MiB with 59 file(s) remaining\r",
      "Completed 1.7 MiB/3.3 MiB with 59 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG03240.hgr1.vcf to ./HG03240.hgr1.vcf\r\n",
      "Completed 1.7 MiB/3.3 MiB with 58 file(s) remaining\r",
      "Completed 1.7 MiB/3.3 MiB with 58 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG03432.hgr1.vcf to ./HG03432.hgr1.vcf\r\n",
      "Completed 1.7 MiB/3.3 MiB with 57 file(s) remaining\r",
      "Completed 1.7 MiB/3.3 MiB with 57 file(s) remaining\r",
      "Completed 1.8 MiB/3.3 MiB with 57 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG02508.hgr1.vcf to ./HG02508.hgr1.vcf\r\n",
      "Completed 1.8 MiB/3.3 MiB with 56 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG03604.hgr1.vcf to ./HG03604.hgr1.vcf\r\n",
      "Completed 1.8 MiB/3.3 MiB with 55 file(s) remaining\r",
      "Completed 1.8 MiB/3.3 MiB with 55 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG03607.hgr1.vcf to ./HG03607.hgr1.vcf\r\n",
      "Completed 1.8 MiB/3.3 MiB with 54 file(s) remaining\r",
      "Completed 1.8 MiB/3.3 MiB with 54 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG03548.hgr1.vcf to ./HG03548.hgr1.vcf\r\n",
      "Completed 1.8 MiB/3.3 MiB with 53 file(s) remaining\r",
      "Completed 1.8 MiB/3.3 MiB with 53 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG03616.hgr1.vcf to ./HG03616.hgr1.vcf\r\n",
      "Completed 1.8 MiB/3.3 MiB with 52 file(s) remaining\r",
      "Completed 1.9 MiB/3.3 MiB with 52 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG03558.hgr1.vcf to ./HG03558.hgr1.vcf\r\n",
      "Completed 1.9 MiB/3.3 MiB with 51 file(s) remaining\r",
      "Completed 1.9 MiB/3.3 MiB with 51 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG03790.hgr1.vcf to ./HG03790.hgr1.vcf\r\n",
      "Completed 1.9 MiB/3.3 MiB with 50 file(s) remaining\r",
      "Completed 1.9 MiB/3.3 MiB with 50 file(s) remaining\r",
      "Completed 1.9 MiB/3.3 MiB with 50 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG03624.hgr1.vcf to ./HG03624.hgr1.vcf\r\n",
      "Completed 1.9 MiB/3.3 MiB with 49 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG03720.hgr1.vcf to ./HG03720.hgr1.vcf\r\n",
      "Completed 1.9 MiB/3.3 MiB with 48 file(s) remaining\r",
      "Completed 2.0 MiB/3.3 MiB with 48 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG03861.hgr1.vcf to ./HG03861.hgr1.vcf\r\n",
      "Completed 2.0 MiB/3.3 MiB with 47 file(s) remaining\r",
      "Completed 2.0 MiB/3.3 MiB with 47 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG03753.hgr1.vcf to ./HG03753.hgr1.vcf\r\n",
      "Completed 2.0 MiB/3.3 MiB with 46 file(s) remaining\r",
      "Completed 2.0 MiB/3.3 MiB with 46 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG03867.hgr1.vcf to ./HG03867.hgr1.vcf\r\n",
      "Completed 2.0 MiB/3.3 MiB with 45 file(s) remaining\r",
      "Completed 2.0 MiB/3.3 MiB with 45 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG03890.hgr1.vcf to ./HG03890.hgr1.vcf\r\n",
      "Completed 2.0 MiB/3.3 MiB with 44 file(s) remaining\r",
      "Completed 2.1 MiB/3.3 MiB with 44 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG03856.hgr1.vcf to ./HG03856.hgr1.vcf\r\n",
      "Completed 2.1 MiB/3.3 MiB with 43 file(s) remaining\r",
      "Completed 2.1 MiB/3.3 MiB with 43 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG04162.hgr1.vcf to ./HG04162.hgr1.vcf\r\n",
      "Completed 2.1 MiB/3.3 MiB with 42 file(s) remaining\r",
      "Completed 2.1 MiB/3.3 MiB with 42 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG03836.hgr1.vcf to ./HG03836.hgr1.vcf\r\n",
      "Completed 2.1 MiB/3.3 MiB with 41 file(s) remaining\r",
      "Completed 2.1 MiB/3.3 MiB with 41 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA12283.hgr1.vcf to ./NA12283.hgr1.vcf\r\n",
      "Completed 2.1 MiB/3.3 MiB with 40 file(s) remaining\r",
      "Completed 2.2 MiB/3.3 MiB with 40 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA06985.hgr1.vcf to ./NA06985.hgr1.vcf\r\n",
      "Completed 2.2 MiB/3.3 MiB with 39 file(s) remaining\r",
      "Completed 2.2 MiB/3.3 MiB with 39 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA12878_pp.hgr1.vcf to ./NA12878_pp.hgr1.vcf\r\n",
      "Completed 2.2 MiB/3.3 MiB with 38 file(s) remaining\r",
      "Completed 2.2 MiB/3.3 MiB with 38 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA12891_pp.hgr1.vcf to ./NA12891_pp.hgr1.vcf\r\n",
      "Completed 2.2 MiB/3.3 MiB with 37 file(s) remaining\r",
      "Completed 2.3 MiB/3.3 MiB with 37 file(s) remaining\r",
      "Completed 2.3 MiB/3.3 MiB with 37 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA11881.hgr1.vcf to ./NA11881.hgr1.vcf\r\n",
      "Completed 2.3 MiB/3.3 MiB with 36 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA12234.hgr1.vcf to ./NA12234.hgr1.vcf\r\n",
      "Completed 2.3 MiB/3.3 MiB with 35 file(s) remaining\r",
      "Completed 2.3 MiB/3.3 MiB with 35 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA18981.hgr1.vcf to ./NA18981.hgr1.vcf\r\n",
      "Completed 2.3 MiB/3.3 MiB with 34 file(s) remaining\r",
      "Completed 2.3 MiB/3.3 MiB with 34 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA18632.hgr1.vcf to ./NA18632.hgr1.vcf\r\n",
      "Completed 2.3 MiB/3.3 MiB with 33 file(s) remaining\r",
      "Completed 2.4 MiB/3.3 MiB with 33 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA18533.hgr1.vcf to ./NA18533.hgr1.vcf\r\n",
      "Completed 2.4 MiB/3.3 MiB with 32 file(s) remaining\r",
      "Completed 2.4 MiB/3.3 MiB with 32 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA12892_pp.hgr1.vcf to ./NA12892_pp.hgr1.vcf\r\n",
      "Completed 2.4 MiB/3.3 MiB with 31 file(s) remaining\r",
      "Completed 2.4 MiB/3.3 MiB with 31 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA18647.hgr1.vcf to ./NA18647.hgr1.vcf\r\n",
      "Completed 2.4 MiB/3.3 MiB with 30 file(s) remaining\r",
      "Completed 2.5 MiB/3.3 MiB with 30 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA19001.hgr1.vcf to ./NA19001.hgr1.vcf\r\n",
      "Completed 2.5 MiB/3.3 MiB with 29 file(s) remaining\r",
      "Completed 2.5 MiB/3.3 MiB with 29 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA19098.hgr1.vcf to ./NA19098.hgr1.vcf\r\n",
      "Completed 2.5 MiB/3.3 MiB with 28 file(s) remaining\r",
      "Completed 2.6 MiB/3.3 MiB with 28 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA18622.hgr1.vcf to ./NA18622.hgr1.vcf\r\n",
      "Completed 2.6 MiB/3.3 MiB with 27 file(s) remaining\r",
      "Completed 2.6 MiB/3.3 MiB with 27 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA19042.hgr1.vcf to ./NA19042.hgr1.vcf\r\n",
      "Completed 2.6 MiB/3.3 MiB with 26 file(s) remaining\r",
      "Completed 2.6 MiB/3.3 MiB with 26 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA19144.hgr1.vcf to ./NA19144.hgr1.vcf\r\n",
      "Completed 2.6 MiB/3.3 MiB with 25 file(s) remaining\r",
      "Completed 2.6 MiB/3.3 MiB with 25 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA19159.hgr1.vcf to ./NA19159.hgr1.vcf\r\n",
      "Completed 2.6 MiB/3.3 MiB with 24 file(s) remaining\r",
      "Completed 2.6 MiB/3.3 MiB with 24 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA19240.hgr1.vcf to ./NA19240.hgr1.vcf\r\n",
      "Completed 2.6 MiB/3.3 MiB with 23 file(s) remaining\r",
      "Completed 2.7 MiB/3.3 MiB with 23 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA19091.hgr1.vcf to ./NA19091.hgr1.vcf\r\n",
      "Completed 2.7 MiB/3.3 MiB with 22 file(s) remaining\r",
      "Completed 2.7 MiB/3.3 MiB with 22 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA19320.hgr1.vcf to ./NA19320.hgr1.vcf\r\n",
      "Completed 2.7 MiB/3.3 MiB with 21 file(s) remaining\r",
      "Completed 2.7 MiB/3.3 MiB with 21 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA19153.hgr1.vcf to ./NA19153.hgr1.vcf\r\n",
      "Completed 2.7 MiB/3.3 MiB with 20 file(s) remaining\r",
      "Completed 2.8 MiB/3.3 MiB with 20 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA19670.hgr1.vcf to ./NA19670.hgr1.vcf\r\n",
      "Completed 2.8 MiB/3.3 MiB with 19 file(s) remaining\r",
      "Completed 2.8 MiB/3.3 MiB with 19 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA19351.hgr1.vcf to ./NA19351.hgr1.vcf\r\n",
      "Completed 2.8 MiB/3.3 MiB with 18 file(s) remaining\r",
      "Completed 2.8 MiB/3.3 MiB with 18 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA19475.hgr1.vcf to ./NA19475.hgr1.vcf\r\n",
      "Completed 2.8 MiB/3.3 MiB with 17 file(s) remaining\r",
      "Completed 2.8 MiB/3.3 MiB with 17 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA19658.hgr1.vcf to ./NA19658.hgr1.vcf\r\n",
      "Completed 2.8 MiB/3.3 MiB with 16 file(s) remaining\r",
      "Completed 2.9 MiB/3.3 MiB with 16 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA19734.hgr1.vcf to ./NA19734.hgr1.vcf\r\n",
      "Completed 2.9 MiB/3.3 MiB with 15 file(s) remaining\r",
      "Completed 2.9 MiB/3.3 MiB with 15 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA20359.hgr1.vcf to ./NA20359.hgr1.vcf\r\n",
      "Completed 2.9 MiB/3.3 MiB with 14 file(s) remaining\r",
      "Completed 2.9 MiB/3.3 MiB with 14 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA19984.hgr1.vcf to ./NA19984.hgr1.vcf\r\n",
      "Completed 2.9 MiB/3.3 MiB with 13 file(s) remaining\r",
      "Completed 3.0 MiB/3.3 MiB with 13 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA19006.hgr1.vcf to ./NA19006.hgr1.vcf\r\n",
      "Completed 3.0 MiB/3.3 MiB with 12 file(s) remaining\r",
      "Completed 3.0 MiB/3.3 MiB with 12 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA20514.hgr1.vcf to ./NA20514.hgr1.vcf\r\n",
      "Completed 3.0 MiB/3.3 MiB with 11 file(s) remaining\r",
      "Completed 3.0 MiB/3.3 MiB with 11 file(s) remaining\r",
      "Completed 3.0 MiB/3.3 MiB with 11 file(s) remaining\r",
      "Completed 3.0 MiB/3.3 MiB with 11 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA20832.hgr1.vcf to ./NA20832.hgr1.vcf\r\n",
      "Completed 3.0 MiB/3.3 MiB with 10 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA20357.hgr1.vcf to ./NA20357.hgr1.vcf\r\n",
      "Completed 3.0 MiB/3.3 MiB with 9 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA20827.hgr1.vcf to ./NA20827.hgr1.vcf\r\n",
      "Completed 3.0 MiB/3.3 MiB with 8 file(s) remaining\r",
      "Completed 3.1 MiB/3.3 MiB with 8 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA19795.hgr1.vcf to ./NA19795.hgr1.vcf\r\n",
      "Completed 3.1 MiB/3.3 MiB with 7 file(s) remaining\r",
      "Completed 3.1 MiB/3.3 MiB with 7 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA21109.hgr1.vcf to ./NA21109.hgr1.vcf\r\n",
      "Completed 3.1 MiB/3.3 MiB with 6 file(s) remaining\r",
      "Completed 3.1 MiB/3.3 MiB with 6 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA20503.hgr1.vcf to ./NA20503.hgr1.vcf\r\n",
      "Completed 3.1 MiB/3.3 MiB with 5 file(s) remaining\r",
      "Completed 3.2 MiB/3.3 MiB with 5 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/testGenome.hgr1.vcf to ./testGenome.hgr1.vcf\r\n",
      "Completed 3.2 MiB/3.3 MiB with 4 file(s) remaining\r",
      "Completed 3.2 MiB/3.3 MiB with 4 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA21113.hgr1.vcf to ./NA21113.hgr1.vcf\r\n",
      "Completed 3.2 MiB/3.3 MiB with 3 file(s) remaining\r",
      "Completed 3.2 MiB/3.3 MiB with 3 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA21125.hgr1.vcf to ./NA21125.hgr1.vcf\r\n",
      "Completed 3.2 MiB/3.3 MiB with 2 file(s) remaining\r",
      "Completed 3.3 MiB/3.3 MiB with 2 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA21112.hgr1.vcf to ./NA21112.hgr1.vcf\r\n",
      "Completed 3.3 MiB/3.3 MiB with 1 file(s) remaining\r",
      "Completed 3.3 MiB/3.3 MiB with 1 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG00128.hgr1.vcf to ./HG00128.hgr1.vcf\r\n",
      "Completed 282 Bytes/~29.6 KiB with ~27 file(s) remaining (calculating...)\r",
      "Completed 1.0 KiB/~29.9 KiB with ~28 file(s) remaining (calculating...)  \r",
      "download: s3://crownproject/1kg_hgr1/HG00128.hgr1.vcf.idx to ./HG00128.hgr1.vcf.idx\r\n",
      "Completed 1.0 KiB/~30.4 KiB with ~29 file(s) remaining (calculating...)\r",
      "download: s3://crownproject/1kg_hgr1/HG00139.hgr1.vcf.idx to ./HG00139.hgr1.vcf.idx\r\n",
      "Completed 1.0 KiB/~30.4 KiB with ~28 file(s) remaining (calculating...)\r",
      "Completed 3.3 KiB/~40.8 KiB with ~39 file(s) remaining (calculating...)\r",
      "download: s3://crownproject/1kg_hgr1/HG00234.hgr1.vcf.idx to ./HG00234.hgr1.vcf.idx\r\n",
      "Completed 3.3 KiB/~42.4 KiB with ~42 file(s) remaining (calculating...)\r",
      "Completed 4.0 KiB/~42.4 KiB with ~42 file(s) remaining (calculating...)\r",
      "Completed 4.3 KiB/~42.4 KiB with ~42 file(s) remaining (calculating...)\r",
      "Completed 4.6 KiB/~42.7 KiB with ~43 file(s) remaining (calculating...)\r",
      "Completed 4.9 KiB/~43.5 KiB with ~46 file(s) remaining (calculating...)\r",
      "download: s3://crownproject/1kg_hgr1/HG00557.hgr1.vcf.idx to ./HG00557.hgr1.vcf.idx\r\n",
      "Completed 4.9 KiB/~43.8 KiB with ~46 file(s) remaining (calculating...)\r",
      "Completed 5.6 KiB/~43.8 KiB with ~46 file(s) remaining (calculating...)\r",
      "Completed 6.4 KiB/~43.8 KiB with ~46 file(s) remaining (calculating...)\r",
      "download: s3://crownproject/1kg_hgr1/HG00734.hgr1.vcf.idx to ./HG00734.hgr1.vcf.idx\r\n",
      "Completed 6.4 KiB/~43.8 KiB with ~45 file(s) remaining (calculating...)\r",
      "Completed 8.6 KiB/~43.8 KiB with ~45 file(s) remaining (calculating...)\r",
      "Completed 12.9 KiB/~44.6 KiB with ~46 file(s) remaining (calculating...)\r",
      "download: s3://crownproject/1kg_hgr1/HG00253.hgr1.vcf.idx to ./HG00253.hgr1.vcf.idx\r\n",
      "Completed 12.9 KiB/~44.6 KiB with ~45 file(s) remaining (calculating...)\r",
      "download: s3://crownproject/1kg_hgr1/HG00362.hgr1.vcf.idx to ./HG00362.hgr1.vcf.idx\r\n",
      "Completed 12.9 KiB/~44.8 KiB with ~45 file(s) remaining (calculating...)\r",
      "download: s3://crownproject/1kg_hgr1/HG00358.hgr1.vcf.idx to ./HG00358.hgr1.vcf.idx\r\n",
      "Completed 12.9 KiB/~45.4 KiB with ~46 file(s) remaining (calculating...)\r",
      "download: s3://crownproject/1kg_hgr1/HG00536.hgr1.vcf.idx to ./HG00536.hgr1.vcf.idx\r\n",
      "Completed 12.9 KiB/~45.4 KiB with ~45 file(s) remaining (calculating...)\r",
      "download: s3://crownproject/1kg_hgr1/HG00337.hgr1.vcf.idx to ./HG00337.hgr1.vcf.idx\r\n",
      "Completed 12.9 KiB/~46.1 KiB with ~45 file(s) remaining (calculating...)\r",
      "download: s3://crownproject/1kg_hgr1/HG00478.hgr1.vcf.idx to ./HG00478.hgr1.vcf.idx\r\n",
      "Completed 12.9 KiB/~46.4 KiB with ~45 file(s) remaining (calculating...)\r",
      "Completed 13.1 KiB/~46.7 KiB with ~46 file(s) remaining (calculating...)\r",
      "Completed 17.4 KiB/~46.7 KiB with ~46 file(s) remaining (calculating...)\r",
      "download: s3://crownproject/1kg_hgr1/HG00717.hgr1.vcf.idx to ./HG00717.hgr1.vcf.idx\r\n",
      "Completed 17.4 KiB/~47.5 KiB with ~48 file(s) remaining (calculating...)\r",
      "Completed 17.6 KiB/~47.8 KiB with ~49 file(s) remaining (calculating...)\r",
      "Completed 21.9 KiB/~47.8 KiB with ~49 file(s) remaining (calculating...)\r",
      "download: s3://crownproject/1kg_hgr1/HG00353.hgr1.vcf.idx to ./HG00353.hgr1.vcf.idx\r\n",
      "Completed 21.9 KiB/~48.1 KiB with ~49 file(s) remaining (calculating...)\r",
      "Completed 22.6 KiB/~48.1 KiB with ~49 file(s) remaining (calculating...)\r",
      "download: s3://crownproject/1kg_hgr1/HG00851.hgr1.vcf.idx to ./HG00851.hgr1.vcf.idx\r\n",
      "Completed 22.6 KiB/~48.9 KiB with ~51 file(s) remaining (calculating...)\r",
      "download: s3://crownproject/1kg_hgr1/HG00740.hgr1.vcf.idx to ./HG00740.hgr1.vcf.idx\r\n",
      "Completed 22.6 KiB/~49.7 KiB with ~51 file(s) remaining (calculating...)\r",
      "Completed 24.9 KiB/~50.0 KiB with ~52 file(s) remaining (calculating...)\r",
      "download: s3://crownproject/1kg_hgr1/HG00978.hgr1.vcf.idx to ./HG00978.hgr1.vcf.idx\r\n",
      "Completed 24.9 KiB/~50.0 KiB with ~51 file(s) remaining (calculating...)\r",
      "Completed 25.2 KiB/~50.0 KiB with ~51 file(s) remaining (calculating...)\r",
      "Completed 25.4 KiB/~50.0 KiB with ~51 file(s) remaining (calculating...)\r",
      "download: s3://crownproject/1kg_hgr1/HG00982.hgr1.vcf.idx to ./HG00982.hgr1.vcf.idx\r\n",
      "Completed 25.4 KiB/~51.5 KiB with ~54 file(s) remaining (calculating...)\r",
      "download: s3://crownproject/1kg_hgr1/HG01125.hgr1.vcf.idx to ./HG01125.hgr1.vcf.idx\r\n",
      "Completed 25.4 KiB/~51.8 KiB with ~54 file(s) remaining (calculating...)\r",
      "download: s3://crownproject/1kg_hgr1/HG01066.hgr1.vcf.idx to ./HG01066.hgr1.vcf.idx\r\n",
      "Completed 25.4 KiB/~52.1 KiB with ~54 file(s) remaining (calculating...)\r",
      "Completed 26.7 KiB/~52.1 KiB with ~54 file(s) remaining (calculating...)\r",
      "Completed 27.0 KiB/~52.1 KiB with ~54 file(s) remaining (calculating...)\r",
      "Completed 27.2 KiB/~55.2 KiB with ~58 file(s) remaining (calculating...)\r",
      "download: s3://crownproject/1kg_hgr1/HG01277.hgr1.vcf.idx to ./HG01277.hgr1.vcf.idx\r\n",
      "Completed 27.2 KiB/~55.2 KiB with ~57 file(s) remaining (calculating...)\r",
      "Completed 27.5 KiB/~55.2 KiB with ~57 file(s) remaining (calculating...)\r",
      "download: s3://crownproject/1kg_hgr1/HG01176.hgr1.vcf.idx to ./HG01176.hgr1.vcf.idx\r\n",
      "Completed 27.5 KiB/~55.2 KiB with ~56 file(s) remaining (calculating...)\r",
      "Completed 27.8 KiB/~55.2 KiB with ~56 file(s) remaining (calculating...)\r",
      "Completed 28.1 KiB/~56.2 KiB with ~58 file(s) remaining (calculating...)\r",
      "download: s3://crownproject/1kg_hgr1/HG01348.hgr1.vcf.idx to ./HG01348.hgr1.vcf.idx\r\n",
      "Completed 28.1 KiB/~56.2 KiB with ~57 file(s) remaining (calculating...)\r",
      "download: s3://crownproject/1kg_hgr1/HG01589.hgr1.vcf.idx to ./HG01589.hgr1.vcf.idx\r\n",
      "Completed 28.1 KiB/~57.0 KiB with ~57 file(s) remaining (calculating...)\r",
      "download: s3://crownproject/1kg_hgr1/HG01464.hgr1.vcf.idx to ./HG01464.hgr1.vcf.idx\r\n",
      "Completed 28.1 KiB/~57.3 KiB with ~57 file(s) remaining (calculating...)\r",
      "Completed 28.8 KiB/~57.5 KiB with ~58 file(s) remaining (calculating...)\r",
      "download: s3://crownproject/1kg_hgr1/HG01626.hgr1.vcf.idx to ./HG01626.hgr1.vcf.idx\r\n",
      "Completed 28.8 KiB/~57.5 KiB with ~57 file(s) remaining (calculating...)\r",
      "download: s3://crownproject/1kg_hgr1/HG01620.hgr1.vcf.idx to ./HG01620.hgr1.vcf.idx\r\n",
      "Completed 28.8 KiB/~59.4 KiB with ~61 file(s) remaining (calculating...)\r",
      "Completed 29.1 KiB/~64.3 KiB with ~70 file(s) remaining (calculating...)\r",
      "download: s3://crownproject/1kg_hgr1/HG02040.hgr1.vcf.idx to ./HG02040.hgr1.vcf.idx\r\n",
      "Completed 29.1 KiB/~65.9 KiB with ~73 file(s) remaining (calculating...)\r",
      "Completed 29.9 KiB/~65.9 KiB with ~73 file(s) remaining (calculating...)\r",
      "download: s3://crownproject/1kg_hgr1/HG01630.hgr1.vcf.idx to ./HG01630.hgr1.vcf.idx\r\n",
      "Completed 29.9 KiB/~67.0 KiB with ~76 file(s) remaining (calculating...)\r",
      "Completed 30.1 KiB/69.1 KiB with 80 file(s) remaining                   \r",
      "download: s3://crownproject/1kg_hgr1/HG02046.hgr1.vcf.idx to ./HG02046.hgr1.vcf.idx\r\n",
      "Completed 30.1 KiB/69.1 KiB with 79 file(s) remaining\r",
      "Completed 30.4 KiB/69.1 KiB with 79 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG02035.hgr1.vcf.idx to ./HG02035.hgr1.vcf.idx\r\n",
      "Completed 30.4 KiB/69.1 KiB with 78 file(s) remaining\r",
      "Completed 30.7 KiB/69.1 KiB with 78 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG02190.hgr1.vcf.idx to ./HG02190.hgr1.vcf.idx\r\n",
      "Completed 30.7 KiB/69.1 KiB with 77 file(s) remaining\r",
      "Completed 31.9 KiB/69.1 KiB with 77 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG02277.hgr1.vcf.idx to ./HG02277.hgr1.vcf.idx\r\n",
      "Completed 31.9 KiB/69.1 KiB with 76 file(s) remaining\r",
      "Completed 36.2 KiB/69.1 KiB with 76 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG02090.hgr1.vcf.idx to ./HG02090.hgr1.vcf.idx\r\n",
      "Completed 36.2 KiB/69.1 KiB with 75 file(s) remaining\r",
      "Completed 36.4 KiB/69.1 KiB with 75 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG02219.hgr1.vcf.idx to ./HG02219.hgr1.vcf.idx\r\n",
      "Completed 36.4 KiB/69.1 KiB with 74 file(s) remaining\r",
      "Completed 36.7 KiB/69.1 KiB with 74 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG02078.hgr1.vcf.idx to ./HG02078.hgr1.vcf.idx\r\n",
      "Completed 36.7 KiB/69.1 KiB with 73 file(s) remaining\r",
      "Completed 38.0 KiB/69.1 KiB with 73 file(s) remaining\r",
      "Completed 38.3 KiB/69.1 KiB with 73 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG02286.hgr1.vcf.idx to ./HG02286.hgr1.vcf.idx\r\n",
      "Completed 38.3 KiB/69.1 KiB with 72 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG02343.hgr1.vcf.idx to ./HG02343.hgr1.vcf.idx\r\n",
      "Completed 38.3 KiB/69.1 KiB with 71 file(s) remaining\r",
      "Completed 39.0 KiB/69.1 KiB with 71 file(s) remaining\r",
      "Completed 39.8 KiB/69.1 KiB with 71 file(s) remaining\r",
      "Completed 40.1 KiB/69.1 KiB with 71 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG02272.hgr1.vcf.idx to ./HG02272.hgr1.vcf.idx\r\n",
      "Completed 40.1 KiB/69.1 KiB with 70 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG02508.hgr1.vcf.idx to ./HG02508.hgr1.vcf.idx\r\n",
      "Completed 40.1 KiB/69.1 KiB with 69 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG02283.hgr1.vcf.idx to ./HG02283.hgr1.vcf.idx\r\n",
      "Completed 40.1 KiB/69.1 KiB with 68 file(s) remaining\r",
      "Completed 40.3 KiB/69.1 KiB with 68 file(s) remaining\r",
      "Completed 41.6 KiB/69.1 KiB with 68 file(s) remaining\r",
      "Completed 41.9 KiB/69.1 KiB with 68 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG02839.hgr1.vcf.idx to ./HG02839.hgr1.vcf.idx\r\n",
      "Completed 41.9 KiB/69.1 KiB with 67 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG02780.hgr1.vcf.idx to ./HG02780.hgr1.vcf.idx\r\n",
      "Completed 41.9 KiB/69.1 KiB with 66 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG02479.hgr1.vcf.idx to ./HG02479.hgr1.vcf.idx\r\n",
      "Completed 41.9 KiB/69.1 KiB with 65 file(s) remaining\r",
      "Completed 42.1 KiB/69.1 KiB with 65 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG02891.hgr1.vcf.idx to ./HG02891.hgr1.vcf.idx\r\n",
      "Completed 42.1 KiB/69.1 KiB with 64 file(s) remaining\r",
      "Completed 42.4 KiB/69.1 KiB with 64 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG03240.hgr1.vcf.idx to ./HG03240.hgr1.vcf.idx\r\n",
      "Completed 42.4 KiB/69.1 KiB with 63 file(s) remaining\r",
      "Completed 43.2 KiB/69.1 KiB with 63 file(s) remaining\r",
      "Completed 43.5 KiB/69.1 KiB with 63 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG03019.hgr1.vcf.idx to ./HG03019.hgr1.vcf.idx\r\n",
      "Completed 43.5 KiB/69.1 KiB with 62 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG03295.hgr1.vcf.idx to ./HG03295.hgr1.vcf.idx\r\n",
      "Completed 43.5 KiB/69.1 KiB with 61 file(s) remaining\r",
      "Completed 43.7 KiB/69.1 KiB with 61 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG03558.hgr1.vcf.idx to ./HG03558.hgr1.vcf.idx\r\n",
      "Completed 43.7 KiB/69.1 KiB with 60 file(s) remaining\r",
      "Completed 44.0 KiB/69.1 KiB with 60 file(s) remaining\r",
      "Completed 44.3 KiB/69.1 KiB with 60 file(s) remaining\r",
      "Completed 44.6 KiB/69.1 KiB with 60 file(s) remaining\r",
      "Completed 44.8 KiB/69.1 KiB with 60 file(s) remaining\r",
      "Completed 45.1 KiB/69.1 KiB with 60 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG02946.hgr1.vcf.idx to ./HG02946.hgr1.vcf.idx\r\n",
      "Completed 45.1 KiB/69.1 KiB with 59 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG03432.hgr1.vcf.idx to ./HG03432.hgr1.vcf.idx\r\n",
      "Completed 45.1 KiB/69.1 KiB with 58 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG03548.hgr1.vcf.idx to ./HG03548.hgr1.vcf.idx\r\n",
      "Completed 45.1 KiB/69.1 KiB with 57 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG02888.hgr1.vcf.idx to ./HG02888.hgr1.vcf.idx\r\n",
      "Completed 45.1 KiB/69.1 KiB with 56 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG03268.hgr1.vcf.idx to ./HG03268.hgr1.vcf.idx\r\n",
      "Completed 45.1 KiB/69.1 KiB with 55 file(s) remaining\r",
      "Completed 45.9 KiB/69.1 KiB with 55 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG03470.hgr1.vcf.idx to ./HG03470.hgr1.vcf.idx\r\n",
      "Completed 45.9 KiB/69.1 KiB with 54 file(s) remaining\r",
      "Completed 46.1 KiB/69.1 KiB with 54 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG03604.hgr1.vcf.idx to ./HG03604.hgr1.vcf.idx\r\n",
      "Completed 46.1 KiB/69.1 KiB with 53 file(s) remaining\r",
      "Completed 46.4 KiB/69.1 KiB with 53 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG03607.hgr1.vcf.idx to ./HG03607.hgr1.vcf.idx\r\n",
      "Completed 46.4 KiB/69.1 KiB with 52 file(s) remaining\r",
      "Completed 46.7 KiB/69.1 KiB with 52 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG03195.hgr1.vcf.idx to ./HG03195.hgr1.vcf.idx\r\n",
      "Completed 46.7 KiB/69.1 KiB with 51 file(s) remaining\r",
      "Completed 47.0 KiB/69.1 KiB with 51 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG03720.hgr1.vcf.idx to ./HG03720.hgr1.vcf.idx\r\n",
      "Completed 47.0 KiB/69.1 KiB with 50 file(s) remaining\r",
      "Completed 47.3 KiB/69.1 KiB with 50 file(s) remaining\r",
      "Completed 47.5 KiB/69.1 KiB with 50 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG03836.hgr1.vcf.idx to ./HG03836.hgr1.vcf.idx\r\n",
      "Completed 47.5 KiB/69.1 KiB with 49 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG03753.hgr1.vcf.idx to ./HG03753.hgr1.vcf.idx\r\n",
      "Completed 47.5 KiB/69.1 KiB with 48 file(s) remaining\r",
      "Completed 47.8 KiB/69.1 KiB with 48 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG03790.hgr1.vcf.idx to ./HG03790.hgr1.vcf.idx\r\n",
      "Completed 47.8 KiB/69.1 KiB with 47 file(s) remaining\r",
      "Completed 48.1 KiB/69.1 KiB with 47 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG03624.hgr1.vcf.idx to ./HG03624.hgr1.vcf.idx\r\n",
      "Completed 48.1 KiB/69.1 KiB with 46 file(s) remaining\r",
      "Completed 48.4 KiB/69.1 KiB with 46 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG03616.hgr1.vcf.idx to ./HG03616.hgr1.vcf.idx\r\n",
      "Completed 48.4 KiB/69.1 KiB with 45 file(s) remaining\r",
      "Completed 48.6 KiB/69.1 KiB with 45 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG03861.hgr1.vcf.idx to ./HG03861.hgr1.vcf.idx\r\n",
      "Completed 48.6 KiB/69.1 KiB with 44 file(s) remaining\r",
      "Completed 48.9 KiB/69.1 KiB with 44 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG03856.hgr1.vcf.idx to ./HG03856.hgr1.vcf.idx\r\n",
      "Completed 48.9 KiB/69.1 KiB with 43 file(s) remaining\r",
      "Completed 49.2 KiB/69.1 KiB with 43 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA06985.hgr1.vcf.idx to ./NA06985.hgr1.vcf.idx\r\n",
      "Completed 49.2 KiB/69.1 KiB with 42 file(s) remaining\r",
      "Completed 50.0 KiB/69.1 KiB with 42 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG03867.hgr1.vcf.idx to ./HG03867.hgr1.vcf.idx\r\n",
      "Completed 50.0 KiB/69.1 KiB with 41 file(s) remaining\r",
      "Completed 50.7 KiB/69.1 KiB with 41 file(s) remaining\r",
      "Completed 51.0 KiB/69.1 KiB with 41 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG04162.hgr1.vcf.idx to ./HG04162.hgr1.vcf.idx\r\n",
      "Completed 51.0 KiB/69.1 KiB with 40 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA12234.hgr1.vcf.idx to ./NA12234.hgr1.vcf.idx\r\n",
      "Completed 51.0 KiB/69.1 KiB with 39 file(s) remaining\r",
      "Completed 51.3 KiB/69.1 KiB with 39 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA12891_pp.hgr1.vcf.idx to ./NA12891_pp.hgr1.vcf.idx\r\n",
      "Completed 51.3 KiB/69.1 KiB with 38 file(s) remaining\r",
      "Completed 51.6 KiB/69.1 KiB with 38 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA12878_pp.hgr1.vcf.idx to ./NA12878_pp.hgr1.vcf.idx\r\n",
      "Completed 51.6 KiB/69.1 KiB with 37 file(s) remaining\r",
      "Completed 51.8 KiB/69.1 KiB with 37 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA12283.hgr1.vcf.idx to ./NA12283.hgr1.vcf.idx\r\n",
      "Completed 51.8 KiB/69.1 KiB with 36 file(s) remaining\r",
      "Completed 52.1 KiB/69.1 KiB with 36 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA12892_pp.hgr1.vcf.idx to ./NA12892_pp.hgr1.vcf.idx\r\n",
      "Completed 52.1 KiB/69.1 KiB with 35 file(s) remaining\r",
      "Completed 54.4 KiB/69.1 KiB with 35 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA18622.hgr1.vcf.idx to ./NA18622.hgr1.vcf.idx\r\n",
      "Completed 54.4 KiB/69.1 KiB with 34 file(s) remaining\r",
      "Completed 54.6 KiB/69.1 KiB with 34 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA18533.hgr1.vcf.idx to ./NA18533.hgr1.vcf.idx\r\n",
      "Completed 54.6 KiB/69.1 KiB with 33 file(s) remaining\r",
      "Completed 54.9 KiB/69.1 KiB with 33 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/HG03890.hgr1.vcf.idx to ./HG03890.hgr1.vcf.idx\r\n",
      "Completed 54.9 KiB/69.1 KiB with 32 file(s) remaining\r",
      "Completed 55.7 KiB/69.1 KiB with 32 file(s) remaining\r",
      "Completed 55.9 KiB/69.1 KiB with 32 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA18647.hgr1.vcf.idx to ./NA18647.hgr1.vcf.idx\r\n",
      "Completed 55.9 KiB/69.1 KiB with 31 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA11881.hgr1.vcf.idx to ./NA11881.hgr1.vcf.idx\r\n",
      "Completed 55.9 KiB/69.1 KiB with 30 file(s) remaining\r",
      "Completed 56.2 KiB/69.1 KiB with 30 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA18632.hgr1.vcf.idx to ./NA18632.hgr1.vcf.idx\r\n",
      "Completed 56.2 KiB/69.1 KiB with 29 file(s) remaining\r",
      "Completed 56.5 KiB/69.1 KiB with 29 file(s) remaining\r",
      "Completed 56.8 KiB/69.1 KiB with 29 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA18981.hgr1.vcf.idx to ./NA18981.hgr1.vcf.idx\r\n",
      "Completed 56.8 KiB/69.1 KiB with 28 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA19042.hgr1.vcf.idx to ./NA19042.hgr1.vcf.idx\r\n",
      "Completed 56.8 KiB/69.1 KiB with 27 file(s) remaining\r",
      "Completed 57.0 KiB/69.1 KiB with 27 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA19006.hgr1.vcf.idx to ./NA19006.hgr1.vcf.idx\r\n",
      "Completed 57.0 KiB/69.1 KiB with 26 file(s) remaining\r",
      "Completed 57.3 KiB/69.1 KiB with 26 file(s) remaining\r",
      "Completed 58.1 KiB/69.1 KiB with 26 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA19098.hgr1.vcf.idx to ./NA19098.hgr1.vcf.idx\r\n",
      "Completed 58.1 KiB/69.1 KiB with 25 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA19091.hgr1.vcf.idx to ./NA19091.hgr1.vcf.idx\r\n",
      "Completed 58.1 KiB/69.1 KiB with 24 file(s) remaining\r",
      "Completed 58.4 KiB/69.1 KiB with 24 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA19144.hgr1.vcf.idx to ./NA19144.hgr1.vcf.idx\r\n",
      "Completed 58.4 KiB/69.1 KiB with 23 file(s) remaining\r",
      "Completed 58.6 KiB/69.1 KiB with 23 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA19240.hgr1.vcf.idx to ./NA19240.hgr1.vcf.idx\r\n",
      "Completed 58.6 KiB/69.1 KiB with 22 file(s) remaining\r",
      "Completed 59.4 KiB/69.1 KiB with 22 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA19475.hgr1.vcf.idx to ./NA19475.hgr1.vcf.idx\r\n",
      "Completed 59.4 KiB/69.1 KiB with 21 file(s) remaining\r",
      "Completed 59.7 KiB/69.1 KiB with 21 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA19320.hgr1.vcf.idx to ./NA19320.hgr1.vcf.idx\r\n",
      "Completed 59.7 KiB/69.1 KiB with 20 file(s) remaining\r",
      "Completed 60.0 KiB/69.1 KiB with 20 file(s) remaining\r",
      "Completed 60.2 KiB/69.1 KiB with 20 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA19159.hgr1.vcf.idx to ./NA19159.hgr1.vcf.idx\r\n",
      "Completed 60.2 KiB/69.1 KiB with 19 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA19153.hgr1.vcf.idx to ./NA19153.hgr1.vcf.idx\r\n",
      "Completed 60.2 KiB/69.1 KiB with 18 file(s) remaining\r",
      "Completed 60.5 KiB/69.1 KiB with 18 file(s) remaining\r",
      "Completed 61.3 KiB/69.1 KiB with 18 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA19658.hgr1.vcf.idx to ./NA19658.hgr1.vcf.idx\r\n",
      "Completed 61.3 KiB/69.1 KiB with 17 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA19001.hgr1.vcf.idx to ./NA19001.hgr1.vcf.idx\r\n",
      "Completed 61.3 KiB/69.1 KiB with 16 file(s) remaining\r",
      "Completed 63.5 KiB/69.1 KiB with 16 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA19795.hgr1.vcf.idx to ./NA19795.hgr1.vcf.idx\r\n",
      "Completed 63.5 KiB/69.1 KiB with 15 file(s) remaining\r",
      "Completed 63.8 KiB/69.1 KiB with 15 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA19351.hgr1.vcf.idx to ./NA19351.hgr1.vcf.idx\r\n",
      "Completed 63.8 KiB/69.1 KiB with 14 file(s) remaining\r",
      "Completed 64.1 KiB/69.1 KiB with 14 file(s) remaining\r",
      "Completed 64.8 KiB/69.1 KiB with 14 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA19670.hgr1.vcf.idx to ./NA19670.hgr1.vcf.idx\r\n",
      "Completed 64.8 KiB/69.1 KiB with 13 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA19984.hgr1.vcf.idx to ./NA19984.hgr1.vcf.idx\r\n",
      "Completed 64.8 KiB/69.1 KiB with 12 file(s) remaining\r",
      "Completed 65.1 KiB/69.1 KiB with 12 file(s) remaining\r",
      "Completed 65.4 KiB/69.1 KiB with 12 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA20357.hgr1.vcf.idx to ./NA20357.hgr1.vcf.idx\r\n",
      "Completed 65.4 KiB/69.1 KiB with 11 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA19734.hgr1.vcf.idx to ./NA19734.hgr1.vcf.idx\r\n",
      "Completed 65.4 KiB/69.1 KiB with 10 file(s) remaining\r",
      "Completed 65.7 KiB/69.1 KiB with 10 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA20359.hgr1.vcf.idx to ./NA20359.hgr1.vcf.idx\r\n",
      "Completed 65.7 KiB/69.1 KiB with 9 file(s) remaining\r",
      "Completed 65.9 KiB/69.1 KiB with 9 file(s) remaining\r",
      "Completed 66.2 KiB/69.1 KiB with 9 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA20503.hgr1.vcf.idx to ./NA20503.hgr1.vcf.idx\r\n",
      "Completed 66.2 KiB/69.1 KiB with 8 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA20514.hgr1.vcf.idx to ./NA20514.hgr1.vcf.idx\r\n",
      "Completed 66.2 KiB/69.1 KiB with 7 file(s) remaining\r",
      "Completed 66.5 KiB/69.1 KiB with 7 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA20832.hgr1.vcf.idx to ./NA20832.hgr1.vcf.idx\r\n",
      "Completed 66.5 KiB/69.1 KiB with 6 file(s) remaining\r",
      "Completed 67.2 KiB/69.1 KiB with 6 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA21113.hgr1.vcf.idx to ./NA21113.hgr1.vcf.idx\r\n",
      "Completed 67.2 KiB/69.1 KiB with 5 file(s) remaining\r",
      "Completed 67.5 KiB/69.1 KiB with 5 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA21109.hgr1.vcf.idx to ./NA21109.hgr1.vcf.idx\r\n",
      "Completed 67.5 KiB/69.1 KiB with 4 file(s) remaining\r",
      "Completed 68.3 KiB/69.1 KiB with 4 file(s) remaining\r",
      "Completed 68.6 KiB/69.1 KiB with 4 file(s) remaining\r",
      "Completed 68.8 KiB/69.1 KiB with 4 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA21112.hgr1.vcf.idx to ./NA21112.hgr1.vcf.idx\r\n",
      "Completed 68.8 KiB/69.1 KiB with 3 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA21125.hgr1.vcf.idx to ./NA21125.hgr1.vcf.idx\r\n",
      "Completed 68.8 KiB/69.1 KiB with 2 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/testGenome.hgr1.vcf.idx to ./testGenome.hgr1.vcf.idx\r\n",
      "Completed 68.8 KiB/69.1 KiB with 1 file(s) remaining\r",
      "Completed 69.1 KiB/69.1 KiB with 1 file(s) remaining\r",
      "download: s3://crownproject/1kg_hgr1/NA20827.hgr1.vcf.idx to ./NA20827.hgr1.vcf.idx\r\n"
     ]
    }
   ],
   "source": [
    "# Move to hgr1_vcf folder\n",
    "cd ~/Crown/data/hgr1_vcf/; ls\n",
    "mkdir -p 100g; cd 100g\n",
    "mkdir -p vcf; cd vcf\n",
    "\n",
    "date\n",
    "\n",
    "aws s3 ls s3://crownproject/1kg_hgr1/ | grep \".vcf\" -\n",
    "\n",
    "aws s3 cp s3://crownproject/1kg_hgr1/ ./ --recursive --exclude \"*\" --include \"*.vcf\"\n",
    "aws s3 cp s3://crownproject/1kg_hgr1/ ./ --recursive --exclude \"*\" --include \"*.vcf.idx\"\n",
    "\n",
    "\n",
    "rm testGenome.hgr1.vcf testGenome.hgr1.vcf.idx"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed to open 100g_hgr1_v1.bcf: could not load index\r\n",
      "[E::main_vcfindex] unknown filetype; expected bgzip compressed VCF or BCF\r\n",
      "[E::main_vcfindex] was the VCF/BCF compressed with bgzip?\r\n"
     ]
    }
   ],
   "source": [
    "# convert files to BCF\n",
    "cd ~/Crown/data/hgr1_vcf/100g_raw/vcf\n",
    "mkdir -p ../bcf #bcf storage folder\n",
    "\n",
    "VCF=$(ls *.vcf)\n",
    "\n",
    "for FILE in $(ls *.vcf)\n",
    "do\n",
    "    # Convert VCF to BCF File\n",
    "    bcftools view -O b $FILE -o ../bcf/$FILE.bcf\n",
    "    bcftools index ../bcf/$FILE.bcf\n",
    "done\n",
    "\n",
    "cd ../bcf\n",
    "\n",
    "bcftools merge $(ls *.bcf) -o ../100g_hgr1_v1.vcf\n",
    "bcftools view -O b ../100g_hgr1_v1.vcf -o 100g_hgr1_v1.bcf\n",
    "bcftools index 100g_hgr1_v1.bcf\n",
    "\n",
    "# Filter for variants with 30x coverage\n",
    "cd ..\n",
    "\n",
    "bcftools view -i \"DP>=30\" 100g_hgr1_v1.vcf -o 100g_hgr1_v1.dp30.vcf\n",
    "bcftools index 100g_hgr1_v1.dp30.vcf\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Parsing bcftools stats output: 100g_hgr1_v1.vcfplot\r\n",
      "Plotting graphs: python plot/100g-plot.py\r\n",
      "Creating PDF: pdflatex 100g-summary.tex >100g-plot-vcfstats.log 2>&1\r\n",
      "Finished: plot/100g-summary.pdf\r\n",
      "Parsing bcftools stats output: 100g_hgr1_v1.dp30.vcfplot\r\n",
      "Plotting graphs: python plot_dp30/100g-plot.py\r\n",
      "Creating PDF: pdflatex 100g-summary.tex >100g-plot-vcfstats.log 2>&1\r\n",
      "Finished: plot_dp30/100g-summary.pdf\r\n"
     ]
    }
   ],
   "source": [
    "## vcfplot (summary stats)\n",
    "cd ~/Crown/data/hgr1_vcf/100g_raw/\n",
    "\n",
    "# nonDP30 filtered\n",
    "bcftools stats 100g_hgr1_v1.vcf > 100g_hgr1_v1.vcfplot\n",
    "plot-vcfstats -p plot/100g 100g_hgr1_v1.vcfplot\n",
    "\n",
    "# DP30 filtered\n",
    "bcftools stats 100g_hgr1_v1.dp30.vcf > 100g_hgr1_v1.dp30.vcfplot\n",
    "plot-vcfstats -p plot_dp30/100g 100g_hgr1_v1.dp30.vcfplot\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": []
    }
   ],
   "source": [
    "# I'd like to re-name the files based on their Super-population_Population_ID\n",
    "# so they alpha-sort to world populations\n",
    "# I'll likely set-this up in excel (a conversion table)\n",
    "\n",
    "cd ~/Crown/data/hgr1_vcf/\n",
    "mkdir -p renamed_100g\n",
    "\n",
    "#cp -r 100g_raw/vcf/*.vcf renamed_100g/vcf/\n",
    "ln -s ln -s $PWD/100g_raw/vcf/ $PWD/renamed_100g/vcf \n",
    "\n",
    "cd renamed_100g"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NA12878_pp\tCEU\tEUR\t4\t4.EUR.CEU_NA12878_pp\r\n",
      "NA12891_pp\tCEU\tEUR\t4\t4.EUR.CEU_NA12891_pp\r\n",
      "NA12892_pp\tCEU\tEUR\t4\t4.EUR.CEU_NA12892_pp\r\n",
      "HG02283\tACB\tAFR\t1\t1.AFR.ACB_HG02283\r\n",
      "HG02343\tACB\tAFR\t1\t1.AFR.ACB_HG02343\r\n",
      "HG02508\tACB\tAFR\t1\t1.AFR.ACB_HG02508\r\n",
      "HG02479\tACB\tAFR\t1\t1.AFR.ACB_HG02479\r\n",
      "NA20357\tASW\tAFR\t1\t1.AFR.ASW_NA20357\r\n",
      "NA20359\tASW\tAFR\t1\t1.AFR.ASW_NA20359\r\n",
      "NA20362\tASW\tAFR\t1\t1.AFR.ASW_NA20362\r\n",
      "NA19984\tASW\tAFR\t1\t1.AFR.ASW_NA19984\r\n",
      "HG03607\tBEB\tSAS\t5\t5.SAS.BEB_HG03607\r\n",
      "HG03604\tBEB\tSAS\t5\t5.SAS.BEB_HG03604\r\n",
      "HG04162\tBEB\tSAS\t5\t5.SAS.BEB_HG04162\r\n",
      "HG03616\tBEB\tSAS\t5\t5.SAS.BEB_HG03616\r\n",
      "HG02190\tCDX\tEAS\t3\t3.EAS.CDX_HG02190\r\n",
      "HG00851\tCDX\tEAS\t3\t3.EAS.CDX_HG00851\r\n",
      "HG00978\tCDX\tEAS\t3\t3.EAS.CDX_HG00978\r\n",
      "HG00982\tCDX\tEAS\t3\t3.EAS.CDX_HG00982\r\n",
      "NA06985\tCEU\tEUR\t4\t4.EUR.CEU_NA06985\r\n",
      "NA11881\tCEU\tEUR\t4\t4.EUR.CEU_NA11881\r\n",
      "NA12234\tCEU\tEUR\t4\t4.EUR.CEU_NA12234\r\n",
      "NA12283\tCEU\tEUR\t4\t4.EUR.CEU_NA12283\r\n",
      "NA18622\tCHB\tEAS\t3\t3.EAS.CHB_NA18622\r\n",
      "NA18632\tCHB\tEAS\t3\t3.EAS.CHB_NA18632\r\n",
      "NA18533\tCHB\tEAS\t3\t3.EAS.CHB_NA18533\r\n",
      "NA18647\tCHB\tEAS\t3\t3.EAS.CHB_NA18647\r\n",
      "HG00478\tCHS\tEAS\t3\t3.EAS.CHS_HG00478\r\n",
      "HG00536\tCHS\tEAS\t3\t3.EAS.CHS_HG00536\r\n",
      "HG00557\tCHS\tEAS\t3\t3.EAS.CHS_HG00557\r\n",
      "HG00717\tCHS\tEAS\t3\t3.EAS.CHS_HG00717\r\n",
      "HG01125\tCLM\tAMR\t2\t2.AMR.CLM_HG01125\r\n",
      "HG01464\tCLM\tAMR\t2\t2.AMR.CLM_HG01464\r\n",
      "HG01348\tCLM\tAMR\t2\t2.AMR.CLM_HG01348\r\n",
      "HG01277\tCLM\tAMR\t2\t2.AMR.CLM_HG01277\r\n",
      "HG03295\tESN\tAFR\t1\t1.AFR.ESN_HG03295\r\n",
      "HG03195\tESN\tAFR\t1\t1.AFR.ESN_HG03195\r\n",
      "HG02946\tESN\tAFR\t1\t1.AFR.ESN_HG02946\r\n",
      "HG03268\tESN\tAFR\t1\t1.AFR.ESN_HG03268\r\n",
      "HG00358\tFIN\tEUR\t4\t4.EUR.FIN_HG00358\r\n",
      "HG00353\tFIN\tEUR\t4\t4.EUR.FIN_HG00353\r\n",
      "HG00337\tFIN\tEUR\t4\t4.EUR.FIN_HG00337\r\n",
      "HG00362\tFIN\tEUR\t4\t4.EUR.FIN_HG00362\r\n",
      "HG00234\tGBR\tEUR\t4\t4.EUR.GBR_HG00234\r\n",
      "HG00128\tGBR\tEUR\t4\t4.EUR.GBR_HG00128\r\n",
      "HG00253\tGBR\tEUR\t4\t4.EUR.GBR_HG00253\r\n",
      "HG00139\tGBR\tEUR\t4\t4.EUR.GBR_HG00139\r\n",
      "NA21109\tGIH\tSAS\t5\t5.SAS.GIH_NA21109\r\n",
      "NA21113\tGIH\tSAS\t5\t5.SAS.GIH_NA21113\r\n",
      "NA21112\tGIH\tSAS\t5\t5.SAS.GIH_NA21112\r\n",
      "NA21125\tGIH\tSAS\t5\t5.SAS.GIH_NA21125\r\n",
      "HG03240\tGWD\tAFR\t1\t1.AFR.GWD_HG03240\r\n",
      "HG02891\tGWD\tAFR\t1\t1.AFR.GWD_HG02891\r\n",
      "HG02888\tGWD\tAFR\t1\t1.AFR.GWD_HG02888\r\n",
      "HG02839\tGWD\tAFR\t1\t1.AFR.GWD_HG02839\r\n",
      "HG02219\tIBS\tEUR\t4\t4.EUR.IBS_HG02219\r\n",
      "HG01626\tIBS\tEUR\t4\t4.EUR.IBS_HG01626\r\n",
      "HG01630\tIBS\tEUR\t4\t4.EUR.IBS_HG01630\r\n",
      "HG01620\tIBS\tEUR\t4\t4.EUR.IBS_HG01620\r\n",
      "HG03720\tITU\tSAS\t5\t5.SAS.ITU_HG03720\r\n",
      "HG03861\tITU\tSAS\t5\t5.SAS.ITU_HG03861\r\n",
      "HG03867\tITU\tSAS\t5\t5.SAS.ITU_HG03867\r\n",
      "HG03790\tITU\tSAS\t5\t5.SAS.ITU_HG03790\r\n",
      "NA19001\tJPT\tEAS\t3\t3.EAS.JPT_NA19001\r\n",
      "NA18981\tJPT\tEAS\t3\t3.EAS.JPT_NA18981\r\n",
      "NA19091\tJPT\tEAS\t3\t3.EAS.JPT_NA19091\r\n",
      "NA19006\tJPT\tEAS\t3\t3.EAS.JPT_NA19006\r\n",
      "HG02046\tKHV\tEAS\t3\t3.EAS.KHV_HG02046\r\n",
      "HG02035\tKHV\tEAS\t3\t3.EAS.KHV_HG02035\r\n",
      "HG02040\tKHV\tEAS\t3\t3.EAS.KHV_HG02040\r\n",
      "HG02078\tKHV\tEAS\t3\t3.EAS.KHV_HG02078\r\n",
      "NA19475\tLWK\tAFR\t1\t1.AFR.LWK_NA19475\r\n",
      "NA19320\tLWK\tAFR\t1\t1.AFR.LWK_NA19320\r\n",
      "NA19351\tLWK\tAFR\t1\t1.AFR.LWK_NA19351\r\n",
      "NA19042\tLWK\tAFR\t1\t1.AFR.LWK_NA19042\r\n",
      "HG03432\tMSL\tAFR\t1\t1.AFR.MSL_HG03432\r\n",
      "HG03558\tMSL\tAFR\t1\t1.AFR.MSL_HG03558\r\n",
      "HG03470\tMSL\tAFR\t1\t1.AFR.MSL_HG03470\r\n",
      "HG03548\tMSL\tAFR\t1\t1.AFR.MSL_HG03548\r\n",
      "NA19795\tMXL\tAMR\t2\t2.AMR.MXL_NA19795\r\n",
      "NA19734\tMXL\tAMR\t2\t2.AMR.MXL_NA19734\r\n",
      "NA19658\tMXL\tAMR\t2\t2.AMR.MXL_NA19658\r\n",
      "NA19670\tMXL\tAMR\t2\t2.AMR.MXL_NA19670\r\n",
      "HG02272\tPEL\tAMR\t2\t2.AMR.PEL_HG02272\r\n",
      "HG02286\tPEL\tAMR\t2\t2.AMR.PEL_HG02286\r\n",
      "HG02090\tPEL\tAMR\t2\t2.AMR.PEL_HG02090\r\n",
      "HG02277\tPEL\tAMR\t2\t2.AMR.PEL_HG02277\r\n",
      "HG01589\tPJL\tSAS\t5\t5.SAS.PJL_HG01589\r\n",
      "HG03624\tPJL\tSAS\t5\t5.SAS.PJL_HG03624\r\n",
      "HG03019\tPJL\tSAS\t5\t5.SAS.PJL_HG03019\r\n",
      "HG02780\tPJL\tSAS\t5\t5.SAS.PJL_HG02780\r\n",
      "HG01176\tPUR\tAMR\t2\t2.AMR.PUR_HG01176\r\n",
      "HG00740\tPUR\tAMR\t2\t2.AMR.PUR_HG00740\r\n",
      "HG01066\tPUR\tAMR\t2\t2.AMR.PUR_HG01066\r\n",
      "HG00734\tPUR\tAMR\t2\t2.AMR.PUR_HG00734\r\n",
      "HG03753\tSTU\tSAS\t5\t5.SAS.STU_HG03753\r\n",
      "HG03836\tSTU\tSAS\t5\t5.SAS.STU_HG03836\r\n",
      "HG03856\tSTU\tSAS\t5\t5.SAS.STU_HG03856\r\n",
      "HG03890\tSTU\tSAS\t5\t5.SAS.STU_HG03890\r\n",
      "NA20827\tTSI\tEUR\t4\t4.EUR.TSI_NA20827\r\n",
      "NA20514\tTSI\tEUR\t4\t4.EUR.TSI_NA20514\r\n",
      "NA20503\tTSI\tEUR\t4\t4.EUR.TSI_NA20503\r\n",
      "NA20832\tTSI\tEUR\t4\t4.EUR.TSI_NA20832\r\n",
      "NA19159\tYRI\tAFR\t1\t1.AFR.YRI_NA19159\r\n",
      "NA19153\tYRI\tAFR\t1\t1.AFR.YRI_NA19153\r\n",
      "NA19098\tYRI\tAFR\t1\t1.AFR.YRI_NA19098\r\n",
      "NA19144\tYRI\tAFR\t1\t1.AFR.YRI_NA19144\r\n",
      "NA19240\tYRI\tAFR\t1\t1.AFR.YRI_NA19240\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "# I made a name conversion table in the main spreadsheet\n",
    "cat nameConversion.txt\n",
    "cp nameConversion.txt ../\n",
    "\n",
    "# I'll be renaming files/ID to\n",
    "# <SuperPopulation_Number>.<SuperPopulation>.<Population>.<Individual>\n",
    "#\n",
    "# This means the vcf files will alpha-sort into populations easily\n",
    "# as opposed to the chronological order in which theby were entered into\n",
    "# the project\n",
    "\n",
    "## For figure make the Utah Trio super-group 0 (they will be 3 genomes at the top)\n",
    "## in ~/Crown/data/hgr1_vcf/renamed_100g\n",
    "##\n",
    "## > nameConversion_figure.txt\n",
    "##"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[E::bcf_hdr_add_sample] Empty sample name: trailing spaces/tabs in the header line?\r\n",
      "Aborted (core dumped)\r\n",
      "[E::hts_open_format] fail to open file '.bcf'\r\n",
      "Failed to read .bcf\r\n",
      "[E::bcf_hdr_add_sample] Empty sample name: trailing spaces/tabs in the header line?\r\n",
      "Aborted (core dumped)\r\n",
      "[E::hts_open_format] fail to open file '.bcf'\r\n",
      "Failed to read .bcf\r\n",
      "[E::bcf_hdr_add_sample] Empty sample name: trailing spaces/tabs in the header line?\r\n",
      "Aborted (core dumped)\r\n",
      "[E::hts_open_format] fail to open file '.bcf'\r\n",
      "Failed to read .bcf\r\n"
     ]
    }
   ],
   "source": [
    "# Iterate through every VCF file in the VCF folder;\n",
    "# rename the file; rename the ID tag in the file\n",
    "# create VCF index for it\n",
    "cd ~/Crown/data/hgr1_vcf/renamed_100g/\n",
    "mkdir -p renaming\n",
    "\n",
    "cp -r vcf/* renaming/\n",
    "\n",
    "cd renaming\n",
    "\n",
    "VCF=$(ls *.vcf)\n",
    "\n",
    "for FILE in $(ls *.vcf)\n",
    "do\n",
    "    originalID=$(echo $FILE | cut -f1 -d'.' -)\n",
    "    \n",
    "    convertID=$(grep $originalID ../nameConversion.txt | cut -f 5 - )\n",
    "    \n",
    "\n",
    "    # Underscores are removed from contigID so search without it\n",
    "    dropUnderScore=$(echo $originalID | cut -f1 -d'_' -)\n",
    "    convertID_ds=$(echo $convertID | cut -f1 -d'_' -)\n",
    "    \n",
    "    # Within VCF file, convert ID\n",
    "    sed -i \"s/\\t$dropUnderScore/\\t$convertID_ds/g\" $FILE\n",
    "    \n",
    "    # Rename the file itself (retain underscores)\n",
    "    mv $FILE $convertID.vcf\n",
    "    \n",
    "    # Convert to bcf file\n",
    "    bcftools view -O b $convertID.vcf -o $convertID.bcf\n",
    "    bcftools index $convertID.bcf\n",
    "done\n",
    "\n",
    "\n",
    "bcftools merge $(ls *.bcf) -o ../100g_hgr1_pop.vcf\n",
    "cd ..\n",
    "\n",
    "gzip 100g_hgr1_pop.vcf\n",
    "\n",
    "rm -r renaming\n",
    "#bcftools view -O b 100g_hgr1_pop.vcf -o 100g_hgr1_pop.bcf\n",
    "#bcftools index 100g_hgr1_pop.bcf"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Discussion\n",
    "\n",
    "Data loaded into IGV and used to make figure. Roughly...\n",
    "\n",
    "![Raw 100 genome variants viewed in IGV and made into figure](../figure/20170313_hgr1_100g_raw.png)\n",
    "\n",
    "The main problem with this data is that it doesn't distinguish 'No Data' with reference allele only. Needs to be re-run using g.vcf file format."
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
