{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# VCF Density Analysis\n",
    "```\n",
    "pi:ababaian\n",
    "files: ~/Crown/data/vcf_correlation\n",
    "start: 2017 03 14\n",
    "complete : 2017 03 20\n",
    "```\n",
    "## Introduction\n",
    "\n",
    "Lots of data made and I have my 'definitive' g.vcf calls from 107 genomes. One key question is what is the number and density of variants in different regions of `hgr1`. What is the number of intra-genomic variants per genome (measure in trio) and what's the relationship between evolutionary information, protein binding contacts and variation.\n",
    "\n",
    "Such 'correlative' analysis is well suited for BED intersection analysis. The methods outlined here should also be broadly applicable when doing this with different data-sets.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "## Objective\n",
    "\n",
    "- Determine the density of variants in the Utah Trio\n",
    "- Count number of variants per individual\n",
    "- Classify variants by region, shannon entropy (evolutionary), protein contact points and possibly GC-content."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Materials and Methods\n",
    "\n",
    "### Initialize folder\n",
    "\n",
    "Initialized `~/Crown/data/vcf_correlation/`. From throughout the rest of the project I copied over the annotation tracks, bam files for Utah Trio, GC-content for hgr1 wigs, sequences of each region in hgr1 and the `100g.hgr1.g.vcf.gz` vcf file from 1kg_hgr1_v1 pipeline.\n",
    "\n",
    "### Extract  Utah Trio g.vcf Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "cd ~/Crown/data/vcf_correlation/vcf\n",
    "\n",
    "# From the unified genotype calls extract each sample indepndently\n",
    "bcftools view -s NA12878 100g.hgr1.g.vcf > NA12878.hgr1.g.vcf\n",
    "bcftools view -s NA12891 100g.hgr1.g.vcf > NA12891.hgr1.g.vcf\n",
    "bcftools view -s NA12892 100g.hgr1.g.vcf > NA12892.hgr1.g.vcf"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Bed File Defintions\n",
    "\n",
    "There are a few bed Files which I'm using as 'region defintions' for downstream analysis\n",
    "\n",
    "**rDNA.bed**\n",
    "```\n",
    "    chr13\t10219\t10340\t5S\n",
    "    chr13\t1000000\t1013408\t45S\n",
    "    chr13\t1003660\t1005529\t18S\n",
    "    chr13\t1006622\t1006779\t5.8S\n",
    "    chr13\t1007947\t1013018\t28S\n",
    "```\n",
    "\n",
    "**hgr1_regions.bed**: manually defined\n",
    "```\n",
    "    chr13\t10000\t10219\tup_5S\n",
    "    chr13\t10219\t10340\t5S\n",
    "    chr13\t10340\t10376\tdown_5S\n",
    "    chr13\t10376\t10650\tNmask_Alu\n",
    "    chr13\t10650\t11590\tspacer1\n",
    "    chr13\t11590\t11897\tNmask_SR\n",
    "    chr13\t11897\t12241\tspacer2\n",
    "    chr13\t999000\t1000000\tRNA45S_promoter\n",
    "    chr13\t1000000\t1013408\tRNA45S\n",
    "    chr13\t1000000\t1003660\t5ETS\n",
    "    chr13\t1003660\t1005529\t18S\n",
    "    chr13\t1005529\t1006622\tITS1\n",
    "    chr13\t1006622\t1006779\t5.8S\n",
    "    chr13\t1006779\t1007947\tITS2\n",
    "    chr13\t1007947\t1013018\t28S\n",
    "    chr13\t1013018\t1013408\t3ETS\n",
    "    chr13\t1013408\t1013558\tIGS_start\n",
    "```\n",
    "\n",
    "**hgr1_wigRegions.bed**: manually defined\n",
    "```\n",
    "    chr13\t10000\t10376\tA5S\n",
    "    chr13\t10650\t11590\tB5S\n",
    "    chr13\t11897\t12241\tC5S\n",
    "    chr13\t998900\t1013558\tD45S\n",
    "```\n",
    "\n",
    "### Windowed Bed Files\n",
    "\n",
    "Using bedtools to make sliding windows of `hgr1` with different parameters."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": []
    }
   ],
   "source": [
    "cd ~/Crown/data/vcf_correlation/annot\n",
    "# bed windows\n",
    "## w = window size\n",
    "## s = slide length\n",
    "\n",
    "# Window 1\n",
    "bedtools makewindows -b hgr1_wigRegions.bed -w 1 -s 1 > covRegions.w1s1.bed\n",
    "\n",
    "# Window 10\n",
    "bedtools makewindows -b hgr1_wigRegions.bed -w 10 -s 10 > covRegions.w10s1.bed\n",
    "bedtools makewindows -b hgr1_wigRegions.bed -w 10 -s 10 > covRegions.w10s10.bed\n",
    "\n",
    "# Window 50\n",
    "bedtools makewindows -b hgr1_wigRegions.bed -w 100 -s 50 > covRegions.w50s25.bed\n",
    "bedtools makewindows -b hgr1_wigRegions.bed -w 50 -s 50 > covRegions.w50s50.bed\n",
    "\n",
    "# Window 100\n",
    "bedtools makewindows -b hgr1_wigRegions.bed -w 100 -s 50 > covRegions.w100s50.bed\n",
    "bedtools makewindows -b hgr1_wigRegions.bed -w 100 -s 100 > covRegions.w100s100.bed\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Measure Coverage Over Bed Files\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "cd ~/Crown/data/vcf_correlation/\n",
    "\n",
    "## Bed Coverage over the 'regions' of hgr1\n",
    "samtools bedcov annot/hgr1_regions.bed \\\n",
    "  bam/NA12878_pp.hgr1.bam bam/NA12891_pp.hgr1.bam bam/NA12892_pp.hgr1.bam\n",
    "  \n",
    "## Data saved in vcfC_data.xlsx : Region_Coverage"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": []
    }
   ],
   "source": [
    "cd ~/Crown/data/vcf_correlation/\n",
    "\n",
    "## Coverage per base of `hgr1`\n",
    "samtools bedcov annot/covRegions.w1s1.bed bam/NA12878_pp.hgr1.bam > data/na12878_cov.w1s1.bed\n",
    "\n",
    "samtools bedcov annot/covRegions.w1s1.bed bam/NA12891_pp.hgr1.bam > data/na12891_cov.w1s1.bed\n",
    "\n",
    "samtools bedcov annot/covRegions.w1s1.bed bam/NA12892_pp.hgr1.bam > data/na12892_cov.w1s1.bed"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "### R-based vcfR Analysis\n",
    "\n",
    "I wrote a master 'analysis' script which generates the central figures of this analysis. This makes the figures which went into the manuscript. I'll be re-writing the script to be more descriptive and 'cleaned up' which I can put online share more easily but for now this will do.\n",
    "\n",
    "`gvcf_100g.r` - 2017 03 23 Version\n",
    "\n",
    "* Calculates population-level statistics such as pVDF and iVAF\n",
    "* Quality control of gvcf files plots\n",
    "* Measures density of variants per region of RNA45S\n",
    "* Measures variants per CEPH-1436 genome\n",
    "* CDF Plot\n",
    "* Stratify iVAF variatns and plot distribution\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "# vcfR analysis\n",
    "# 100 genomes hgr1\n",
    "# 170311\n",
    "#\n",
    "# 100g_gvcf  (g variant input only) analysis\n",
    "\n",
    "# Install vcfR\n",
    "#install.packages(\"vcfR\")\n",
    "library(\"vcfR\")\n",
    "library(\"ggplot2\")\n",
    "library(\"reshape2\")\n",
    "\n",
    "# File pointers\n",
    "\n",
    "  vcf_file = '100g.hgr1.g.vcf'\n",
    "  dna_file = 'rDNA.fa'\n",
    "  gff_file = 'rDNA.gff'\n",
    "\n",
    "# output Prefix (for plots)\n",
    "  outPrefix='100g_gvcf'\n",
    "  \n",
    "# FUNCTIONS =========================================================\n",
    "  \n",
    "## Count number of NODATA per variant position\n",
    "N_NODATA = function(inputROW){\n",
    "  # For an input ROW of DP (or any data)\n",
    "  # returns how many in the row are NA\n",
    "  \n",
    "  return(  length(which(is.na(inputROW))) )\n",
    "  \n",
    "}\n",
    "\n",
    "## Count Variant positions (iVAF > 0)\n",
    "# N_VAR = function(inputROW){\n",
    "#   # For an input ROW of DP (or any data)\n",
    "#   # returns how many in the row are NA\n",
    "#   \n",
    "#   return( length(which(inputROW > 0 )) )\n",
    "#   \n",
    "# }\n",
    "\n",
    "## Count Variant positions (iVAF > 0)\n",
    "N_VAR = function(inputROW){\n",
    "  # For an input ROW of DP (or any data)\n",
    "  # returns how many in the row are NA\n",
    "  \n",
    "  return( length(which(inputROW > 0.02 )) )\n",
    "  \n",
    "}\n",
    "\n",
    "# SCRIPT ============================================================\n",
    "#====================================================================\n",
    "# Import VCF / DNA / GFF\n",
    "  VCF = read.vcfR(vcf_file)\n",
    "  DNA = ape::read.dna(dna_file, format = 'fasta')\n",
    "  GFF = read.table(gff_file, sep=\"\\t\", quote = \"\")\n",
    "  \n",
    "# Create chromR Object\n",
    "  chrom = create.chromR(name=\"rDNA\", vcf=VCF, seq=DNA, ann=GFF)\n",
    "\n",
    "#plot(chrom)\n",
    "\n",
    "## Quality Metrics ==================================================\n",
    " plot(chrom) # Standard chromR plot for VCF\n",
    " chromoqc(chrom, dp.alpha = 22)\n",
    "\n",
    "\n",
    " # Quality of each variant Vector\n",
    " #system(paste0(\"cut -f 6 \",vcf_file,\" > tmp.qual\"))\n",
    " #vcf.quality = read.csv('tmp.qual', sep = \"\\t\", quote =\"\", col.names = T)\n",
    "\n",
    "# # Describe Quality stats (log scale)\n",
    "# ggplot(as.data.frame(vcf.quality), aes(vcf.quality)) + geom_histogram() +\n",
    "#    theme_minimal() + scale_x_log10()\n",
    "#    xlab('Alternative Genotype Quality, PHRED') +\n",
    "#    ylab('Count')\n",
    "\n",
    "  \n",
    "# Variant Statistics ================================================\n",
    "\n",
    "# Intra-individual variation (i*)\n",
    "  \n",
    "  # Depth of coverage at each variant (used reads)\n",
    "  DP = extract.gt(chrom, element=\"DP\", as.numeric=TRUE)\n",
    "\n",
    "  # Extract Genotype Quality per sample\n",
    "  GQ = extract.gt(chrom, element=\"GQ\", as.numeric=TRUE)\n",
    "  \n",
    "  # Reference allele-only depth\n",
    "  RAD = extract.gt(chrom, element=\"AD\", as.numeric=TRUE)\n",
    "\n",
    "  # Variant allele frequency (intra-individual)\n",
    "  # Reference Allele Frequency (RAD / DP)\n",
    "  iVAF = (DP-RAD)/DP\n",
    "  \n",
    "# Population variation (called variants)\n",
    "  \n",
    "  # Number of individual genomes in total population\n",
    "  N_pop = length(DP[1,])\n",
    "  \n",
    "  # Column-length vector of how many NODATA there are in each\n",
    "    # column of DP\n",
    "  N_nd = apply(DP, 1, N_NODATA)\n",
    "  \n",
    "  # Number of individuals with measured data at each variant\n",
    "  N_measured = N_pop - N_nd\n",
    "  \n",
    "  # Variant allele count (population)\n",
    "  # number of people carrying measured variant\n",
    "    # pV = rowSums(!is.na(iVAF))\n",
    "  pV = apply(iVAF, 1, N_VAR)\n",
    "  \n",
    "  # Population Variant Allele Frequency\n",
    "  pVAF = pV / N_measured\n",
    "  \n",
    "  # Remove 'zero' from the data\n",
    "  # when measuring intra-genomic VAF, only consider\n",
    "  # when the variant is present\n",
    "  noZero_iVAF = iVAF\n",
    "  noZero_iVAF[which(iVAF == 0)] = NA\n",
    "  \n",
    "  # Average intra-genomic variant allele frequency\n",
    "  mean_iVAF = apply(iVAF,1,mean, na.rm = T)\n",
    "  sd_iVAF = apply(iVAF,1,sd, na.rm = T)\n",
    "  \n",
    "  # Average intra-genomic variant allele frequency in only variant called\n",
    "  mean_noZero_iVAF = apply(noZero_iVAF,1,mean, na.rm = T)\n",
    "  sd_noZero_iVAF = apply(noZero_iVAF, 1, sd, na.rm = T)\n",
    "  \n",
    "# Population Variant Statistics\n",
    "  POPVAR = data.frame(pV, N_measured, pVAF, mean_iVAF, sd_iVAF, mean_noZero_iVAF, sd_noZero_iVAF)\n",
    "  \n",
    "# Highlight variants at a mean_iVAF between two values ( 0.33 < iVAF < 0.66)\n",
    "# Common, abundant variants\n",
    "  CAV = which(0.33 < POPVAR$mean_noZero_iVAF & POPVAR$mean_noZero_iVAF < 0.66 & POPVAR$pVAF > 0.5)\n",
    "  #highlight2 = which(POPVAR$mean_noZero_iVAF >= 0.66)\n",
    "  \n",
    "  POPVAR$colr = 'black'\n",
    "  POPVAR$colr[CAV] = 'red'\n",
    "  #POPVAR$colr[highlight2] = 'purple'\n",
    "\n",
    "# Plot--------------------------------------------------------------\n",
    "\n",
    "# Describe population stats\n",
    "ggplot(as.data.frame(pV), aes(pV)) + geom_histogram() +\n",
    "  theme_minimal() + scale_x_log10() +\n",
    "  xlab('# Genomes variant was measured in') + \n",
    "  ylab('Count')\n",
    "\n",
    "ggplot(as.data.frame(mean_noZero_iVAF), aes(mean_noZero_iVAF)) + geom_histogram() +\n",
    "  theme_minimal() + \n",
    "  xlab('Intra-genomic VAF') + \n",
    "  ylab('Count')\n",
    "\n",
    "ggplot(as.data.frame(pVAF), aes(pVAF)) + geom_histogram() +\n",
    "  theme_minimal() +\n",
    "  xlab('Population VAF') + \n",
    "  ylab('Count')\n",
    "\n",
    "# Mean Intra-individual Variant Allele Frequency vs.\n",
    "# Population-level Variant Allele Frequency (called vs not called)\n",
    "# open PDF device\n",
    "pdf(file = paste0(outPrefix,\".MeaniVAF_pVAF_Nm.pdf\"), width = 6.5, height = 5)\n",
    "\n",
    "PLOT1 = ggplot(POPVAR, aes(pVAF, mean_noZero_iVAF ))\n",
    "#PLOT1 = PLOT1 + geom_point(alpha = 0.5, stroke = 0, aes(size = pV/2, color = colr))\n",
    "PLOT1 = PLOT1 + geom_point(alpha = 0.25, stroke = 0, aes(size = N_measured/2, color = colr))\n",
    "PLOT1 = PLOT1 + scale_color_manual(values =  c('gray80','red') )\n",
    "PLOT1 = PLOT1 + theme_bw()\n",
    "#PLOT1 = PLOT1 + theme(legend.position=\"none\")\n",
    "#PLOT1 = PLOT1 + theme(legend.scale)\n",
    "PLOT1 = PLOT1 + ylab('Mean Intra-individual Variant Allele Frequency')\n",
    "PLOT1 = PLOT1 + xlab('Population Variant Allele Frequency')\n",
    "PLOT1\n",
    "\n",
    "dev.off()\n",
    "\n",
    "# ------------------------------------------------------------------ \n",
    "# ------------------------------------------------------------------ \n",
    "# ------------------------------------------------------------------ \n",
    "\n",
    "# Stratify variants by their mean Intra-individual allele frequency\n",
    "# the hypothesis was that there are two classes of variants;\n",
    "# Directional Variants, which can range from 0 - 1\n",
    "# Stabilized Variants, which are maintained in a narrower range\n",
    "\n",
    "Stratify = function(iVAF, minVAF, maxVAF, noZero = TRUE){\n",
    "  \n",
    "  if (noZero){\n",
    "    iVAF[which(iVAF == 0)] = NA #noZero\n",
    "  }\n",
    "  \n",
    "  # calcualte the average intra-individual VAF for each variant\n",
    "  mean_iVAF = apply(iVAF,1, mean, na.rm = T)\n",
    "\n",
    "  # Subselect (index) mean iVAF between a range of values\n",
    "  strat_lim = which(mean_iVAF > minVAF & mean_iVAF <= maxVAF)\n",
    "  \n",
    "  # Actually subselect by stratification\n",
    "  iVAF_str = apply(iVAF[strat_lim,], 1, sort, decreasing = TRUE)\n",
    "  \n",
    "  N_strats = length(iVAF_str) # number of variants in this stratification\n",
    "  N_samples = length(iVAF[1,]) # number of samples in the analysis (people)\n",
    "  \n",
    "  iVAF_str_matrix = matrix( rep(0, N_strats * N_samples), nrow = N_strats)\n",
    "  \n",
    "  for (X in 1:N_strats){\n",
    "    LINE_VALUES = unlist(iVAF_str[X])\n",
    "    length(LINE_VALUES) = N_samples\n",
    "    LINE_VALUES[is.na(LINE_VALUES)] <- 0\n",
    "    \n",
    "    iVAF_str_matrix[X,] = iVAF_str_matrix[X,] + LINE_VALUES\n",
    "  }\n",
    "  \n",
    "  return(iVAF_str_matrix)\n",
    "  \n",
    "}\n",
    "\n",
    "Stratify_rownames = function(iVAF, minVAF, maxVAF, noZero = TRUE){\n",
    "  \n",
    "  if (noZero){\n",
    "    iVAF[which(iVAF == 0)] = NA #noZero\n",
    "  }\n",
    "  \n",
    "  # calcualte the average intra-individual VAF for each variant\n",
    "  mean_iVAF = apply(iVAF,1, mean, na.rm = T)\n",
    "  \n",
    "  # Subselect (index) mean iVAF between a range of values\n",
    "  strat_lim = which(mean_iVAF > minVAF & mean_iVAF <= maxVAF)\n",
    "  \n",
    "  return(names(strat_lim))\n",
    "}\n",
    "\n",
    "StratifyPlot = function(iVAF, minVAF, maxVAF){\n",
    "  \n",
    "  ## Stratify variants by intra-individual variant allele frequency\n",
    "  iVAF_SUB_matrix = data.frame(Stratify(iVAF, minVAF, maxVAF), row.names = Stratify_rownames(iVAF, minVAF, maxVAF))\n",
    "  \n",
    "  # Shape data to be plotted\n",
    "  iVAF_SUB_matrix = melt(iVAF_SUB_matrix)\n",
    "  iVAF_SUB_matrix$rowid = Stratify_rownames(iVAF, minVAF, maxVAF)\n",
    "\n",
    "    # Plot--------------------------------------------------------------\n",
    "    # Mean Intra-individual Variant Allele Frequency vs.\n",
    "    # Population-level Variant Allele Frequency (called vs not called)\n",
    "    plotTitle = paste0(\"Mean VAF: \", minVAF,\" - \",maxVAF)\n",
    "  \n",
    "    PLOT2 = ggplot(iVAF_SUB_matrix, aes(variable, value, group = factor(rowid)))\n",
    "    PLOT2 = PLOT2 + geom_line(aes(color = factor(rowid)))\n",
    "    #PLOT2 = PLOT2 + geom_line(aes(color = )) COLOR BY STANDRD DEVIATION OF VARIANT ACROSS ALL MEAUSRED SAMPLES\n",
    "    PLOT2 = PLOT2 + theme_bw()\n",
    "    PLOT2 = PLOT2 + theme(legend.position=\"none\")\n",
    "    PLOT2 = PLOT2 + ylim(0,1)\n",
    "    PLOT2 = PLOT2 + scale_x_discrete(breaks = c(1,N_pop))\n",
    "    PLOT2 = PLOT2 + ggtitle(plotTitle)\n",
    "    #PLOT2 = PLOT2 + ylab('Intra-individual Variant Allele Frequency')\n",
    "    #PLOT2 = PLOT2 + xlab('Genomes')\n",
    "    ## -----------------------------------------------------------------\n",
    "    #PLOT2\n",
    "    \n",
    "    return(PLOT2)\n",
    "}\n",
    "\n",
    "StratifyBoxPlot = function(iVAF, minVAF, maxVAF){\n",
    "  \n",
    "  ## Stratify variants by intra-individual variant allele frequency\n",
    "  iVAF_SUB_matrix = data.frame(Stratify(iVAF, minVAF, maxVAF), row.names = Stratify_rownames(iVAF, minVAF, maxVAF))\n",
    "  \n",
    "  # Shape data to be plotted\n",
    "  iVAF_SUB_matrix = melt(iVAF_SUB_matrix)\n",
    "  iVAF_SUB_matrix$rowid = Stratify_rownames(iVAF, minVAF, maxVAF)\n",
    "  \n",
    "  # Plot--------------------------------------------------------------\n",
    "  # Mean Intra-individual Variant Allele Frequency vs.\n",
    "  # Population-level Variant Allele Frequency (called vs not called)\n",
    "  plotTitle = paste0(\"Mean VAF: \", minVAF,\" - \",maxVAF)\n",
    "  \n",
    "  PLOT2 = ggplot(iVAF_SUB_matrix, aes(variable, value, group = factor(rowid)))\n",
    "  PLOT2 = PLOT2 + geom_boxplot(aes(color = factor(rowid)))\n",
    "  #PLOT2 = PLOT2 + geom_line(aes(color = )) COLOR BY STANDRD DEVIATION OF VARIANT ACROSS ALL MEAUSRED SAMPLES\n",
    "  PLOT2 = PLOT2 + theme_bw()\n",
    "  PLOT2 = PLOT2 + theme(legend.position=\"none\")\n",
    "  PLOT2 = PLOT2 + ylim(0,1)\n",
    "  PLOT2 = PLOT2 + scale_x_discrete(breaks = c(1,N_pop))\n",
    "  PLOT2 = PLOT2 + ggtitle(plotTitle)\n",
    "  #PLOT2 = PLOT2 + ylab('Intra-individual Variant Allele Frequency')\n",
    "  #PLOT2 = PLOT2 + xlab('Genomes')\n",
    "  ## -----------------------------------------------------------------\n",
    "  #PLOT2\n",
    "  \n",
    "  return(PLOT2)\n",
    "}\n",
    "\n",
    "\n",
    "# Plot Stratify for multiple ranges\n",
    "minVAF=0 # initialize with min VAF at 0\n",
    "nStratification=3 # How many equal parts to divide the VAF from 0-1\n",
    "\n",
    "# range width of each stratification\n",
    "widthStratification=1/nStratification\n",
    "\n",
    "for (nS in 1:nStratification){\n",
    "\n",
    "  maxVAF = minVAF + widthStratification\n",
    "  \n",
    "  # Generate Plot\n",
    "  PLOT_VAF = StratifyPlot(iVAF, minVAF, maxVAF)\n",
    "  \n",
    "  # Open PDF file to write to\n",
    "  pdf(file = paste0(outPrefix,\".\",nS,\".strat.pdf\"),\n",
    "     width = 5, height = 5)\n",
    "  \n",
    "    print(PLOT_VAF)\n",
    "  \n",
    "  dev.off()\n",
    "  \n",
    "  minVAF = minVAF + widthStratification\n",
    "}\n",
    "\n",
    "# Stratify plot for only CAV\n",
    "\n",
    "StratifyPlot(iVAF[CAV,], 0.3, 0.9)\n",
    "StratifyBoxPlot(iVAF[CAV,], 0.3, 0.9)\n",
    "\n",
    "\n",
    "\n",
    "# Heatmap ================================================\n",
    "# Log transform depth-data at each variant\n",
    "logDP = log(DP)\n",
    "\n",
    "# # chrom from tutotrial\n",
    "heatmap.bp(iVAF)\n",
    "heatmap.bp(iVAF, cbarplot = F, rbarplot = F, rlabels = F)\n",
    "\n",
    "heatmap.bp(logDP, cbarplot = F, rbarplot = F, rlabels = F)\n",
    "\n",
    "## Table of Variants ====================================\n",
    "\n",
    "# Function to report rRNA variant stats from imported VCF\n",
    "\n",
    "REGIONNAME= c(\"up_5S\",\"5S\",\"down_5S\",\"Nmask_Alu\",\"spacer1\",\n",
    "              \"Nmask_SR\",\"spacer2\",\"RNA45S_promoter\",\"RNA45S\",\n",
    "              \"5ETS\",\"18S\",\"ITS1\",\"5.8S\",\"ITS2\",\"28S\",\"3ETS\",\n",
    "              \"IGS_start\")\n",
    "\n",
    "REGIONSTART = c(10000,10219,10340,10376,10650,11590,11897,999000,\n",
    "                1000000,1000000,1003660,1005529,1006622,1006779,\n",
    "                1007947,1013018,1013408)\n",
    "\n",
    "REGIONEND   = c(10219,10340,10376,10650,11590,11897,12241,1000000,\n",
    "                1013408,1003660,1005529,1006622,1006779,1007947,\n",
    "                1013018,1013408,1013558)\n",
    "           \n",
    "\n",
    "RDNA_LOCUS = data.frame(REGIONNAME, REGIONSTART, REGIONEND)\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "# Count per Genome ==================================================\n",
    "\n",
    "# How much data is missing?------------------------------------------\n",
    "\n",
    "NA_DP = apply(DP,1,is.na) # bin vector of present/missing\n",
    "\n",
    "# How data is missing\n",
    "NA_freq = length(which(NA_DP)) / length(NA_DP)\n",
    "\n",
    "NA_perSite =  data.frame( colSums(NA_DP) / length(NA_DP[,1]) )\n",
    "colnames(NA_perSite) = c('percentNA')\n",
    "\n",
    "\n",
    "ggplot(as.data.frame(NA_perSite), aes(percentNA)) + geom_histogram() +\n",
    "  theme_minimal() + \n",
    "  xlab('Intra-genomic VAF') + \n",
    "  ylab('Count')\n",
    "\n",
    "# Number of variantions per individual ------------------------------\n",
    "df_iVAF = data.frame(iVAF) # iVAF as data.frame\n",
    "\n",
    "#length( which( ( colSums(NA_DP) / length(NA_DP[,1])) > .5 ) ) / 926\n",
    "\n",
    "\n",
    "# Go through rDNA_LOCUS (apply) and return the count of how many variants are in a range\n",
    "\n",
    "countVCF = function( RDNA_ROW, inVCF){\n",
    "  # for RDNA_ROW from RDNA_LOCUS\n",
    "  # how many variants of inVCF are within the coordinates\n",
    "  # of this region\n",
    "  variantID = rownames(inVCF)\n",
    "  variantPOS = read.table(text = variantID, header = T, sep = \"_\")[,2] # position of variant\n",
    "  \n",
    "  CNT = which(variantPOS >= as.integer(RDNA_ROW[2]) &\n",
    "              variantPOS <  as.integer(RDNA_ROW[3]) )\n",
    "  \n",
    "  return(length(CNT))\n",
    "}\n",
    "\n",
    "# Which variants are in each of the CEPH-1436 libraries\n",
    "  var_78 = which(df_iVAF$NA12878 > 0.02 & df_iVAF$NA12878 < 0.998)\n",
    "  var_91 = which(df_iVAF$NA12891 > 0.02 & df_iVAF$NA12891 < 0.998)\n",
    "  var_92 = which(df_iVAF$NA12892 > 0.02 & df_iVAF$NA12892 < 0.998)\n",
    "  \n",
    "  var_trio = sort(unique(c(var_78, var_91, var_92)))\n",
    "\n",
    "# Count per region in the Utah Trio\n",
    "VariantsPerRegion = \n",
    "data.frame(as.character(RDNA_LOCUS$REGIONNAME),\n",
    "  apply(RDNA_LOCUS, 1, countVCF, inVCF = df_iVAF[var_78,]),\n",
    "  apply(RDNA_LOCUS, 1, countVCF, inVCF = df_iVAF[var_91,]),\n",
    "  apply(RDNA_LOCUS, 1, countVCF, inVCF = df_iVAF[var_92,]))\n",
    "  colnames(VariantsPerRegion) = c('Region','78','91','92')\n",
    "  \n",
    "\n",
    "VariantsPerRegion[1,2:4] = \n",
    "  VariantsPerRegion[1,2:4] +\n",
    "  colSums(VariantsPerRegion[3:7, 2:4])\n",
    "\n",
    "VariantsPerRegion = \n",
    "  VariantsPerRegion[-c(3,4,5,6,7,9,17),]\n",
    "\n",
    "VariantsPerRegion$mean = apply(VariantsPerRegion[,2:4],1,mean)\n",
    "VariantsPerRegion$sd = apply(VariantsPerRegion[,2:4],1,sd)\n",
    "\n",
    "VariantsPerRegion$forceSORT = c('A','B','C','D','E','F','G','H','I','J')\n",
    "VariantsPerRegion$Region = as.character(VariantsPerRegion$Region)\n",
    "\n",
    "pdf('VariantsPerRegion.pdf',width = 5, height = 5)\n",
    "\n",
    "\n",
    "PLOT3 = ggplot(VariantsPerRegion, aes(x = forceSORT, mean))\n",
    "PLOT3 = PLOT3 + geom_bar(stat = 'identity')\n",
    "PLOT3 = PLOT3 + geom_errorbar(aes(ymax=mean+sd, ymin=mean-sd), width = 0)\n",
    "PLOT3 = PLOT3 + scale_x_discrete(labels=VariantsPerRegion$Region)\n",
    "PLOT3 = PLOT3 + theme_bw()\n",
    "PLOT3\n",
    "\n",
    "dev.off()\n",
    "\n",
    "PLOT1 = PLOT1 + geom_point(alpha = 0.5, stroke = 0, aes(size = pV/2, color = colr))\n",
    "PLOT1 = PLOT1 + scale_color_manual(values =  c('gray80','gray80','red') )\n",
    "PLOT1 = PLOT1 + theme_bw()\n",
    "#PLOT1 = PLOT1 + theme(legend.position=\"none\")\n",
    "PLOT1 = PLOT1 + theme(legend.scale)\n",
    "PLOT1 = PLOT1 + ylab('Mean Intra-individual Variant Allele Frequency')\n",
    "PLOT1 = PLOT1 + xlab('Population Variant Allele Frequency')\n",
    "PLOT1\n",
    "\n",
    "\n",
    "# # Which variants are less than 2% minor allele frequency within the genome\n",
    "# A = length(df_iVAF$NA12878[-which(df_iVAF$NA12878 < 0.02 | df_iVAF$NA12878 > 0.98)])\n",
    "# B = length(df_iVAF$NA12891[-which(df_iVAF$NA12891 < 0.02 | df_iVAF$NA12878 > 0.98)])\n",
    "# C = length(df_iVAF$NA12892[-which(df_iVAF$NA12892 < 0.02 | df_iVAF$NA12878 > 0.98)])\n",
    "# \n",
    "# # Which variants are less than 10% minor allele frequency\n",
    "# A1 = length(df_iVAF$NA12878[-which(df_iVAF$NA12878 < 0.02 | df_iVAF$NA12878 > 0.98)])\n",
    "# B1 = length(df_iVAF$NA12891[-which(df_iVAF$NA12891 < 0.02 | df_iVAF$NA12878 > 0.98)])\n",
    "# C1 = length(df_iVAF$NA12892[-which(df_iVAF$NA12892 < 0.02 | df_iVAF$NA12878 > 0.98)])\n",
    "# \n",
    "# # Fraction of variants in sample which are between 2-10%\n",
    "# mA = (A-A1)/A\n",
    "# mB = (B-B1)/B\n",
    "# mC = (C-C1)/C\n",
    "\n",
    "\n",
    "## CDF Plots per Individual ----------------------------------------\n",
    "\n",
    "# Extract iVAF from TRIO, correct for consensus (i.e. Minor Allele Frequency)\n",
    "NA12878_iVAF = df_iVAF$NA12878\n",
    "  #NA12878_iVAF[which(NA12878_iVAF > 0.5)] = 1 - NA12878_iVAF[which(NA12878_iVAF > 0.5)]\n",
    "\n",
    "NA12891_iVAF = df_iVAF$NA12891\n",
    "  #NA12891_iVAF[which(NA12891_iVAF > 0.5)] = 1 - NA12891_iVAF[which(NA12891_iVAF > 0.5)]\n",
    "  \n",
    "NA12892_iVAF = df_iVAF$NA12892\n",
    "  #NA12892_iVAF[which(NA12892_iVAF > 0.5)] = 1 - NA12892_iVAF[which(NA12892_iVAF > 0.5)]\n",
    "  \n",
    "TRIO_iVAF = data.frame(NA12878_iVAF, NA12891_iVAF, NA12892_iVAF)\n",
    "TRIO_iVAF = melt(TRIO_iVAF)\n",
    "\n",
    "# PLOT4 = ggplot(TRIO_iVAF, aes(x = value))\n",
    "# PLOT4 = PLOT4 + geom_histogram(data=subset(TRIO_iVAF,variable == 'NA12878_iVAF'),fill = \"blue\", alpha = 0.2, bins = 10)\n",
    "# PLOT4 = PLOT4 + geom_histogram(data=subset(TRIO_iVAF,variable == 'NA12891_iVAF'),fill = \"green\", alpha = 0.2, bins = 10)\n",
    "# PLOT4 = PLOT4 + geom_histogram(data=subset(TRIO_iVAF,variable == 'NA12892_iVAF'),fill = \"red\", alpha = 0.2, bins = 10)\n",
    "# PLOT4 = PLOT4 + theme_bw() + theme(legend.position=\"none\")\n",
    "# #PLOT4 = PLOT4 + ylim(c(0,150))\n",
    "# #PLOT4 = PLOT4 + geom_vline(xintercept = 0.02)\n",
    "# PLOT4\n",
    "\n",
    "pdf('CEPH_TRIO_iVAF_CDF_2.pdf', height = 5, width = 5)\n",
    "PLOT4 = ggplot(TRIO_iVAF, aes(value, color = variable )) + stat_ecdf(geom = \"step\")\n",
    "PLOT4 = PLOT4 + theme_bw() + theme(legend.position=\"none\")\n",
    "PLOT4 = PLOT4 + geom_vline(xintercept = 0.02)\n",
    "PLOT4\n",
    "dev.off()\n",
    "\n",
    "NA12891_VCF = df_iVAF[which(df_iVAF$NA12891 > 0.02 & df_iVAF$NA12891 < 0.998),]\n",
    "NA12892_VCF = df_iVAF[which(df_iVAF$NA12892 > 0.02 & df_iVAF$NA12892 < 0.998),]\n",
    "\n",
    "\n",
    "## Evolutionary Conservation fo Variants ---------------------------\n",
    "ENTROPY = read.table('hgr1_entropy.bed', header = F, sep = '\\t')\n",
    "  colnames(ENTROPY) = c('chr','start','end','S')\n",
    "  ENTROPY$info = 2 - ENTROPY$S\n",
    "\n",
    "df_trio = df_iVAF[var_trio, c('NA12878','NA12891','NA12892')]  \n",
    "  \n",
    "TrioVariants = rownames(df_trio)\n",
    "TrioVariants = read.table(text = TrioVariants, header = F, sep = \"_\")[,2] # position of variant\n",
    "\n",
    "TrioENTROPY = ENTROPY[which(ENTROPY$end %in% TrioVariants),]\n",
    "trio_mean_iVAF = apply(df_trio[which(TrioVariants %in% ENTROPY$end),],1 , mean)\n",
    "TrioENTROPY$mean_iVAF = trio_mean_iVAF\n",
    "\n",
    "TrioENTROPY = TrioENTROPY[-which(TrioENTROPY$start == 1013018),]\n",
    "# Removing the last position variants. These are insertions at the last position\n",
    "# of rRNA, I can't define them as being inside 28S since I don't know if the mature\n",
    "# 28S molecule retains this 'long tail' or not.\n",
    "# worth looking into in the RNAseq datas.\n",
    "\n",
    "pdf('CEPH_TRIO_Evolution.pdf', height = 5, width = 5)\n",
    "  PLOT5 = ggplot(TrioENTROPY, aes(info, mean_iVAF))\n",
    "  PLOT5 = PLOT5 + geom_point(alpha = 0.25, stroke = 0, size = 4)\n",
    "  PLOT5 = PLOT5 + scale_color_manual(values =  c('gray80') )\n",
    "  PLOT5 = PLOT5 + theme_bw() + theme(legend.position=\"none\")\n",
    "  PLOT5\n",
    "dev.off()\n",
    "\n",
    "\n",
    "pdf(file = paste0(outPrefix,\".MeaniVAF_pVAF_Nm.pdf\"), width = 6.5, height = 5)\n",
    "\n",
    "PLOT1 = ggplot(POPVAR, aes(pVAF, mean_noZero_iVAF ))\n",
    "#PLOT1 = PLOT1 + geom_point(alpha = 0.5, stroke = 0, aes(size = pV/2, color = colr))\n",
    "PLOT1 = PLOT1 + geom_point(alpha = 0.25, stroke = 0, aes(size = N_measured/2, color = colr))\n",
    "PLOT1 = PLOT1 + scale_color_manual(values =  c('gray80','red') )\n",
    "PLOT1 = PLOT1 + theme_bw()\n",
    "#PLOT1 = PLOT1 + theme(legend.position=\"none\")\n",
    "PLOT1 = PLOT1 + ylab('Mean Intra-individual Variant Allele Frequency')\n",
    "PLOT1 = PLOT1 + xlab('Population Variant Allele Frequency')\n",
    "\n",
    "\n",
    "# Table 1 Variants --------------------------------------------------\n",
    "\n",
    "varSelection = which(0.33 < POPVAR$mean_noZero_iVAF & POPVAR$mean_noZero_iVAF < 0.66 & pVAF > 0.5)\n",
    "\n",
    "TABLE1 = data.frame(names(varSelection))\n",
    "\n",
    "  TABLE1 = cbind( TABLE1, POPVAR$mean_noZero_iVAF[varSelection],\n",
    "                  POPVAR$sd_noZero_iVAF[varSelection],\n",
    "                  POPVAR$pVAF[varSelection],\n",
    "                  POPVAR$pV[varSelection])\n",
    "  colnames(TABLE1) = c('Coordinate', 'iVAF','sd_iVAF','pVAF','pV')\n",
    "\n",
    "# Supplementary Table 1 ---------------------------------------------\n",
    "  \n",
    "  write.table(POPVAR,file = 'supplementary_Table_1.csv', quote = F, sep = \"\\t\")\n",
    "\n",
    "  \n",
    " ```\n",
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Results + Discussion\n",
    "\n",
    "This made up the main chunk of analysis of the data made up to this point. R+D explored in [the manuscript](http://biorxiv.org/content/early/2017/03/21/118760).\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "QED"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
