{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 5S Subunit Sequence\n",
    "```\n",
    "pi:ababaian\n",
    "file: ~/Crown/data/5s_seq/\n",
    "start: 2017 02 02\n",
    "complete : 2017 02 13\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": []
    }
   ],
   "source": [
    "cd ~/Crown/data/5s_seq/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Introduction\n",
    "\n",
    "Analysis to this point has focused on 45S sequence which matures to the 18S, 5.8S and 28S rRNA\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "The final RNA species which is part of the ribosome is the 5S rRNA. 5S is transcribed from a distinct locus on chromosome 1. The 5S locus is also in tandem array, with a copy number ~equal to 45S array copy number.\n",
    "\n",
    "In the human genome the 5S locus is at 1q42.13.\n",
    "\n",
    "![1q42.13](../figure/20170206_1q42.13.png)\n",
    "\n",
    "Zoomed into a single repeat\n",
    "\n",
    "![5S repeat](../figure/20170206_5s_repeat_unit.png)\n",
    "\n",
    "\n",
    "There is already some dbSNP data for 5S repeats. In the edge-repeat 18 there is a SNP within 5S [rs102046644](http://www.ncbi.nlm.nih.gov/SNP/snp_ref.cgi?type=rs&rs=rs12046644).\n",
    "\n",
    "![dbSNP 147](../figure/20170206_5SP18_snp.png)\n",
    "\n",
    "It's a logical extension to include this locus in rRNA variation analysis (namely cancer-specific variations) and to this end I'll be looking at how to best include a reference sequence."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true,
    "heading_collapsed": true
   },
   "source": [
    "## Objective\n",
    "\n",
    "- Develop a 5S rRNA repeat reference sequence for alignment/variation analysis in 1kg project data."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Materials and Methods\n",
    "- Input DNAseq\n",
    "- Input 5s Sequences\n",
    "- Alignment"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### NA19240 Sequence for alignment\n",
    "\n",
    "Re-downloaded NA19240 PCR-free data:\n",
    "\n",
    "`ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR894/ERR894723/ERR894723_1.fastq.gz` and `ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR894/ERR894723/ERR894723_2.fastq.gz`\n",
    "\n",
    "Extracted first 10-million reads (40 million lines) from each respective file and saved locally as:\n",
    "\n",
    "`na19240_pcr_10m_1.fq` and `na19240_pcr_10m_1.fq`\n",
    "\n",
    "### 5S rRNA Sequences\n",
    "\n",
    "5s rRNA (RNA5S) sequence from 4UG0 structure / hg38\n",
    "```\n",
    ">5s_rRNA_Structure\n",
    "GTCTACGGCCATACCACCCTGAACGCGCCCGATCTCGTCTGATCTCGGAAGCTAAGCAGGGTCGGGCCTGGTTAGTACTTGGATGGGAGACCGCCTGGGAATACCGGGTGCTGTAGGCTTT\n",
    "```\n",
    "\n",
    "Added flanking N's and hg38 bases to each side. (call this RNA5S goign forward)\n",
    "\n",
    "- RNA5S.fa\n",
    "\n",
    "```\n",
    ">chrR_5s\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "GTCTACGGCCATACCACCCTGAACGCGCCCGATCTCGTCTGATCTCGGAA\n",
    "GCTAAGCAGGGTCGGGCCTGGTTAGTACTTGGATGGGAGACCGCCTGGGA\n",
    "ATACCGGGTGCTGTAGGCTTTNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "```\n",
    "\n",
    "Full 5s Repeat from hg38:\n",
    "\n",
    "- r5S.fa\n",
    "```\n",
    ">r5s\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "gtgggccctgggccctgacgcctcggagcactccctgctccgagcgggcc\n",
    "cgatgtggtggaagctcgggagcgcgggagccgggggaaggccgcgggcc\n",
    "agcggctcgggggtccccgatccgagccccgcggccccgggctggcggtg\n",
    "tcggctgcaatccggcgggcacggccgggccgggctgggctcttggggca\n",
    "gccaggcgcctccttcagcGTCTACGGCCATACCACCCTGAACGCGCCCG\n",
    "ATCTCGTCTGATCTCGGAAGCTAAGCAGGGTCGGGCCTGGTTAGTACTTG\n",
    "GATGGGAGACCGCCTGGGAATACCGGGTGCTGTAGGCTTTttctttggct\n",
    "ttttgctgtttctttccttttcttccagacggagtctcgccctgtcgccc\n",
    "aggctggagtgcagtggcgccatctcggctcactgcaagctccgcctccc\n",
    "gggttcacgccattccccggcctcagcctcccgagtagctgggcctacag\n",
    "gcgcccgccaccacgcccggctactttgttctatttttcctagagacggg\n",
    "ctttcaccctgttagccgggatggtctggagctcctgacctcgtgatcca\n",
    "cccgcctcggcctcccagagtgctgggattacaggcgcgagccaccgcgc\n",
    "ccgcccggcctgctgtaggcttttgtggcttccccgctgcctcccttccc\n",
    "cccacagtcgccatgcttcccaacctcccctgactctgctccccctttac\n",
    "cgcccacctacacccccgccgcagccgcagccggggtcctcctgctgggg\n",
    "gtccgcccctactgcacgccggccgggcagcagcatcccaccgcttccgc\n",
    "ctcgccgccgccccgccaggagcccggctccagcctgggagggcaggggg\n",
    "ccggaccccaaaggcgcagccgctgggttccctgccgttcgcggtgcctt\n",
    "cccgctcccggaacgcccaggcgattcaattcacccatcgggcgccgtcg\n",
    "tcgcacccttccaaaccgggggaaggggcgggcaggggcagcgggtgcca\n",
    "cagacgccagccaagacgtcggctccggaacgcatgggctgctttacccg\n",
    "ggggaaggacattgcttcgccagccaccgggaaaacaggccctgtgcacc\n",
    "cgggattcccaatgccccccgcttcgtgtcgacgactccagtcccgagga\n",
    "ctcgccagagacccaggcctccgggcccgcccggtgccacggctcccgcc\n",
    "aaacgggcgggcgcgctctgcaaatctcggggcccgccgcaaggcaccca\n",
    "gagcacagggaggtgccaagaaaggcaggagcctacgaaacccacctcca\n",
    "aagcaagcaattcatccaagaaaacgcccgcctcagcgctccgttggtcc\n",
    "tctcgcacggaccgcctggcccccgtgttctgggcgcagcccaagccccc\n",
    "tccaccctatcccggcctgctcaagagggcgctgcctaacggagccgggc\n",
    "gcttcctctctaaggctctatcgctctcgctctctagctccctccgcctc\n",
    "tctcttctgggtttccccctggacctcgcgctacttttgtcgttttccct\n",
    "ctgtctctctgtctctctctctctctttctctgtgcctctctctttctca\n",
    "gcctctctgtctgtctccttccctctcgccctcccctgtctctctcgatc\n",
    "gctgtctctctccctccctcggtttctatctctccatccatctcgtcctt\n",
    "gctctccttcaagccgcgtgtgtgtgtgcgtgcgtgtgcgcgtgcgtgtg\n",
    "tgcgtgtgtgtctgtgtgtctgtgtgtgtgtgtgtgagtccgcgcgcgcg\n",
    "cgtgcgagcgcgcccgggtgtgtctgtgtgtgggggagtggatttgctcc\n",
    "tggtggcggtggggtgtgtctgggtttctctcaggccctctcacccgaga\n",
    "tcaggccgccgcctctagtgccagcccggggcaaaacagggccacccccc\n",
    "gacccgctacaccccacgccctcttgcccccccggccgggtcttggtcgg\n",
    "gacaagcgaccgtggtgggggcgttgtgagagaaaggccccgcgcggctg\n",
    "ggccggctgttcgccttcggccagccctgacggctctgggtgggtggggc\n",
    "aagagggggcctcgcaggagcccctgtgcggcgagggatccaaaacgctg\n",
    "cctccgcgacagggcggaggaccggagggcgtcccaggatcNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "```\n",
    "\n",
    "Full 5s Repeat with the Alu element sequence N-masked\n",
    "- r5S_nAlu.fa\n",
    "```\n",
    ">r5s_nAlu\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "gtgggccctgggccctgacgcctcggagcactccctgctccgagcgggcc\n",
    "cgatgtggtggaagctcgggagcgcgggagccgggggaaggccgcgggcc\n",
    "agcggctcgggggtccccgatccgagccccgcggccccgggctggcggtg\n",
    "tcggctgcaatccggcgggcacggccgggccgggctgggctcttggggca\n",
    "gccaggcgcctccttcagcGTCTACGGCCATACCACCCTGAACGCGCCCG\n",
    "ATCTCGTCTGATCTCGGAAGCTAAGCAGGGTCGGGCCTGGTTAGTACTTG\n",
    "GATGGGAGACCGCCTGGGAATACCGGGTGCTGTAGGCTTTttctttggct\n",
    "ttttgctgtttctttccttttcttccNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "ccgcccggcctgctgtaggcttttgtggcttccccgctgcctcccttccc\n",
    "cccacagtcgccatgcttcccaacctcccctgactctgctccccctttac\n",
    "cgcccacctacacccccgccgcagccgcagccggggtcctcctgctgggg\n",
    "gtccgcccctactgcacgccggccgggcagcagcatcccaccgcttccgc\n",
    "ctcgccgccgccccgccaggagcccggctccagcctgggagggcaggggg\n",
    "ccggaccccaaaggcgcagccgctgggttccctgccgttcgcggtgcctt\n",
    "cccgctcccggaacgcccaggcgattcaattcacccatcgggcgccgtcg\n",
    "tcgcacccttccaaaccgggggaaggggcgggcaggggcagcgggtgcca\n",
    "cagacgccagccaagacgtcggctccggaacgcatgggctgctttacccg\n",
    "ggggaaggacattgcttcgccagccaccgggaaaacaggccctgtgcacc\n",
    "cgggattcccaatgccccccgcttcgtgtcgacgactccagtcccgagga\n",
    "ctcgccagagacccaggcctccgggcccgcccggtgccacggctcccgcc\n",
    "aaacgggcgggcgcgctctgcaaatctcggggcccgccgcaaggcaccca\n",
    "gagcacagggaggtgccaagaaaggcaggagcctacgaaacccacctcca\n",
    "aagcaagcaattcatccaagaaaacgcccgcctcagcgctccgttggtcc\n",
    "tctcgcacggaccgcctggcccccgtgttctgggcgcagcccaagccccc\n",
    "tccaccctatcccggcctgctcaagagggcgctgcctaacggagccgggc\n",
    "gcttcctctctaaggctctatcgctctcgctctctagctccctccgcctc\n",
    "tctcttctgggtttccccctggacctcgcgctacttttgtcgttttccct\n",
    "ctgtctctctgtctctctctctctctttctctgtgcctctctctttctca\n",
    "gcctctctgtctgtctccttccctctcgccctcccctgtctctctcgatc\n",
    "gctgtctctctccctccctcggtttctatctctccatccatctcgtcctt\n",
    "gctctccttcaagccgcgtgtgtgtgtgcgtgcgtgtgcgcgtgcgtgtg\n",
    "tgcgtgtgtgtctgtgtgtctgtgtgtgtgtgtgtgagtccgcgcgcgcg\n",
    "cgtgcgagcgcgcccgggtgtgtctgtgtgtgggggagtggatttgctcc\n",
    "tggtggcggtggggtgtgtctgggtttctctcaggccctctcacccgaga\n",
    "tcaggccgccgcctctagtgccagcccggggcaaaacagggccacccccc\n",
    "gacccgctacaccccacgccctcttgcccccccggccgggtcttggtcgg\n",
    "gacaagcgaccgtggtgggggcgttgtgagagaaaggccccgcgcggctg\n",
    "ggccggctgttcgccttcggccagccctgacggctctgggtgggtggggc\n",
    "aagagggggcctcgcaggagcccctgtgcggcgagggatccaaaacgctg\n",
    "cctccgcgacagggcggaggaccggagggcgtcccaggatcNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "```\n",
    "\n",
    "### Alignment Parameters\n",
    "For each of the above files an index was generated\n",
    "\n",
    "```\n",
    "samtools faidx 5s.fa\n",
    "bowtie2-build 5s.fa 5s\n",
    "```\n",
    "and local bowtie2 alignment was ran\n",
    "```\n",
    "bowtie2 --very-sensitive-local -x 5s -1 ../na19240_pcr_10m_1.fq -2 ../na19240_pcr_10m_2.fq | samtools view -bS - > aligned_unsorted.bam\n",
    "\n",
    "samtools sort aligned_unsorted.bam > na19240_5s.bam\n",
    "\n",
    "rm aligned_unsorted.bam\n",
    "```\n",
    "\n",
    "### Adaconis Alignment\n",
    "Aligning the 'Total RNA' K562 Adaconis sequencing to RNA5S to see if RNA of this size is align-able in that data-set. Edge-cases of size-selection for 200 nt should crop up but it might be neccesary to find a small-RNA sequencing library to use instead.\n",
    "\n",
    "```\n",
    "cd ~/Crown/data/adaconis/totalRNA1/\n",
    "mkdir rna5s; cd rna5s;\n",
    "cp ../../5s_seq/RNA5S.fa ./\n",
    "\n",
    "samtools faidx RNA5S.fa\n",
    "bowtie2-build RNA5S.fa RNA5S\n",
    "\n",
    "bowtie2 --very-sensitive-local -x RNA5S -1 ../SRR567450_1.fastq.gz -2 ../SRR567450_2.fastq.gz | samtools view -bS - > aligned_unsorted.bam \n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Results\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Alignment to RNA5S\n",
    "\n",
    "![RNA5S Alignment](../figure/20170206_5s_alignment.png)\n",
    "\n",
    "- There are two distinct groups of sequences; (lower) a more homogenized 5S sequence with the same flanking sequences and (upper) a more variable 5S sequence with more heterogenous flanking sequences.\n",
    "- Likely represents true-RNA5S vs. pseudo-RNA5S\n",
    "- Near position 280 there is another region where right-clipping is starting. Read coverage drops from ~160 to ~50 over this which implies it's a common site for loss in pseudogenization. Also implies there are at least 2x as many pseudo-5S sequences are in the genome relative to RNA5S.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20000000 + 0 in total (QC-passed reads + QC-failed reads)\r\n",
      "0 + 0 secondary\r\n",
      "0 + 0 supplementary\r\n",
      "0 + 0 duplicates\r\n",
      "228 + 0 mapped (0.00% : N/A)\r\n",
      "20000000 + 0 paired in sequencing\r\n",
      "10000000 + 0 read1\r\n",
      "10000000 + 0 read2\r\n",
      "4 + 0 properly paired (0.00% : N/A)\r\n",
      "4 + 0 with itself and mate mapped\r\n",
      "224 + 0 singletons (0.00% : N/A)\r\n",
      "0 + 0 with mate mapped to a different chr\r\n",
      "0 + 0 with mate mapped to a different chr (mapQ>=5)\r\n"
     ]
    }
   ],
   "source": [
    "samtools flagstat ~/Crown/data/5s_seq/RNA5S/na19240_5s.bam"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Alignment to r5s\n",
    "\n",
    "![r5S Alignment](../figure/20170206_r5S_alignment.png)\n",
    "\n",
    "- There is a consistent 40-fold coverage over most of r5S.\n",
    "- The first bump in coverage is RNA5S which goes up to ~160 coverage\n",
    "- The large increase in coverage is an Alu element; ~90,000 coverage\n",
    "- The distal increase in coverage is a simple-CT rich region; ~1000 coverage"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### TotalRNA Alignment to RNA5S\n",
    "Aligned adaconis_1 to RNA5S\n",
    "\n",
    "```\n",
    "23813162 + 0 in total (QC-passed reads + QC-failed reads)\n",
    "0 + 0 secondary\n",
    "0 + 0 supplementary\n",
    "0 + 0 duplicates\n",
    "541 + 0 mapped (0.00% : N/A)\n",
    "23813162 + 0 paired in sequencing\n",
    "11906581 + 0 read1\n",
    "11906581 + 0 read2\n",
    "20 + 0 properly paired (0.00% : N/A)\n",
    "20 + 0 with itself and mate mapped\n",
    "521 + 0 singletons (0.00% : N/A)\n",
    "0 + 0 with mate mapped to a different chr\n",
    "0 + 0 with mate mapped to a different chr (mapQ>=5)\n",
    "```\n",
    "\n",
    "Total RNA alignment is very surprising. Keep in mind this is from K562 cells though.\n",
    "\n",
    "#### Contig 1\n",
    "\n",
    "![total RNA to RNA5S](../figure/20170206_totalRNA_RNA5S.png)\n",
    "\n",
    "\n",
    "- The major species of reads (76 bp) has 44 bases aligned to RNA5S and then soft-clipping. 20 read-pairs from this region were extracted and aligned/assembled by hand. See `~/Crown/data/5s_seq/totalRNA1/rna5s_totalRNA_assembly.fa`\n",
    "\n",
    "```\n",
    ">RNA5S_totalRNA_Contig1\n",
    "CAGAGTCTCGTTCGTTATCGGAATTAACCAGACAAATCGCTCCACCAACT\n",
    "AAGAACGGCCATACCACCCTGAACGCGCCCGATCTCGTCTGATCTCGG\n",
    "```\n",
    "\n",
    "The paired reads from those reads which aligned to the above contig were assembled by hand into a second contig\n",
    "```\n",
    ">RNA5S_totalRNA_Contig1_mates\n",
    "CACTAAACCATCCAATCGGTAGTAGCGACGGGCGGTGTGTACAAAGGGCA\n",
    "GGGACTTAATCAACGCAAGCTTATGACCCGCACTTACTGGGAATTCCTCG\n",
    "TTCATGGGGAATAATTGCAATCCCCGATCCCCATCACGAATGGGGTTCAA\n",
    "CGGGTTACCCGCGCCTGCCGGCGTAGGGTAGGCACACGCTGAGCCAGTCA\n",
    "GTGTAGCGCGCGTGCAGCCCCGGACATCTAAGGGCATCACAGACCTGTTA\n",
    "TTGCTCAATCTCGGGTGGCTGAACGCCACTTGTCCCTCTAAGAAGTTG\n",
    "```\n",
    "\n",
    "![Blat to hg38](../figure/20170206_totalRNA_Contigs.png)\n",
    "\n",
    "\n",
    "The first contig can be broken down into\n",
    "\n",
    "A) `CAGAGTCTCGTTCGTTATCGGAATTAACCAGACAAATCGCTCCACCAACTAAGAACGGCCATACC`\n",
    "\n",
    "+\n",
    "\n",
    "B) `ACCCTGAACGCGCCCGATCTCGTCTGATCTCGG`\n",
    "\n",
    "The first sequence blats to the p-arm of chr21 in hg38. It's part of the SSU (18S) rRNA, negative strand, and the second sequence blats to RNA5S in hg38, positive strand. There is no known DNA-fusion between 18S and 5S and in NA19240_pcr aligned to 45S there wasn't a sequence which soft-clips to make such a sequence.\n",
    "\n",
    "There is a 6-base overlap `ACCACCC / GGGGTGGT` at this region between 5S and 18S which means this could be a cDNA-synthesis / PCR-artifact. That doesn't explain the sequence fully since it doesn't continue until the edge of the mature 5S sequence but ends prematurely after ~40 bases.\n",
    "\n",
    "```\n",
    "...-------- contigs ----------|\n",
    "\n",
    "                           ~~~~~~~~~~~~~~~~~~~~~~~\n",
    "    <---------- 18S- ----==\n",
    "                        ~==-- 5S+ --->\n",
    "```\n",
    "\n",
    "#### Contig 2\n",
    "The second most abundant species of 5S soft-clipping \n",
    "```\n",
    ">rna5S_totalRNA_contig2\n",
    "TGGTTAGTTTCTTTTCCTCCGCTGACTAATATGCTTAAATTCAGCGGGTC\n",
    "GCCACGTCTGATCTCGGAAGCTAAGCAGGGTCGGGCCTGGTTAGTACTTG\n",
    "GATGGGAGACCGCCTGGGAATACCGGGTGCTGTAGGCTTT\n",
    "```\n",
    "\n",
    "A) `TGGTTAGTTTCTTTTCCTCCGCTGACTAATATGCTTAAATTCAGCGGGTCGCCA`\n",
    "\n",
    "+\n",
    "\n",
    "B) `CGTCTGATCTCGGAAGCTAAGCAGGGTCGGGCCTGGTTAGTACTTGGATGGGAGACCGCCTGGGAATACCGGGTGCTGTAGGCTTT`\n",
    "\n",
    "Where A) maps to the negative strand of 28S rRNA and B) is the positive strand of 5S rRNA\n",
    "\n",
    "#### Re-alignment with Adaconis 2\n",
    "\n",
    "If these are technical artificats then they may be unlikely to arise again in the technical replicate. I'm creating a three-part genome to align adaconis_2 against\n",
    "- RNA5S_fs.fa (fusions)\n",
    "```\n",
    ">RNA5S\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNagccaggcgcctccttcagc\n",
    "GTCTACGGCCATACCACCCTGAACGCGCCCGATCTCGTCTGATCTCGGAA\n",
    "GCTAAGCAGGGTCGGGCCTGGTTAGTACTTGGATGGGAGACCGCCTGGGA\n",
    "ATACCGGGTGCTGTAGGCTTTttctttggctttttgctgtttctttcctt\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    ">contig1\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "CAGAGTCTCGTTCGTTATCGGAATTAACCAGACAAATCGCTCCACCAACT\n",
    "AAGAACGGCCATACCACCCTGAACGCGCCCGATCTCGTCTGATCTCGGAA\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    ">contig2\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNTGGTTAGTTTCTTTTCCTC\n",
    "CGCTGACTAATATGCTTAAATTCAGCGGGTCGCCACGTCTGATCTCGGAA\n",
    "GCTAAGCAGGGTCGGGCCTGGTTAGTACTTGGATGGGAGACCGCCTGGGA\n",
    "ATACCGGGTGCTGTAGGCTTTNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "```\n",
    "\n",
    "In ~/Crown/data/5s_seq/totalRNA2 (a sym link to ~/Crown/data/adaconis/totalRNA2/rna5s_fs)\n",
    "\n",
    "`bowtie2 --very-sensitive-local -x RNA5Sfs -1 ../SRR567451_1.fastq.gz -2 ../SRR567451_2.fastq.gz | samtools view -bS - > aligned_unsorted.bam`\n",
    "\n",
    "#### Re-alignment; Adaconis 1 to hgr\n",
    "\n",
    "I re-ran adaconis1 aligned to hgr.fa with the bowtie2 command (instead of tophat). The alignment looks much better but there are many more examples of these 'rRNA fusions' such as \n",
    "\n",
    "```\n",
    ">SRR567450.11668700\n",
    "TCCGCACTTACTGGGAATTCCTCGTTCATGGGGAATAATTGCAATCCCCGATCCCCATTCGTGATGGGGATCGGGG\n",
    "```\n",
    "\n",
    "Where the first 52 bases map to the negative strand of 18S and the last 24 to the positive strand at 1,005,246. This appears to happen quite a bit throughout the rRNA in the adaconis samples. So far all examples have been positive-negative strand fusions which leads me to believe it's a hybridization/pcr artifact. Unless there is DNA-seq based evidence for such events they probably should be ignored.\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Alu Insertion in 5S Array\n",
    "\n",
    "I've been thinking about how to deal with this Alu element. For the first iteration, of the analysis it's probably best to mask the Alu to N's and focus on sequence variation in the 5S itself.\n",
    "\n",
    "![From Talk](../figure/20170213_5s_Alu.png)\n",
    "\n",
    "- There is a high proportion of pseudo-5S rDNA which maps to this locus when no genomic sequence is avaialble.\n",
    "- The better matching 'true 5s' sequence which is in the context of the 5S-repeat (flanking sequence matches) can also be non-RT pseudo 5S arrays but it's worth noting that all of those sequences in this low coverage experiment flank into the same Alu sequence, labeled 'true 5S'.\n",
    "- It appears that all 5S repeats contain the same Alu insertion."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Reference 5S array\n",
    "\n",
    "For next round of alignments, I will include a single 5S repeat with AluY element masked. Reasoning here that sequence variation in the 5S itself will be the 'interesting bit'. If the above alignment is any indication this will be much messier, but if 'concordant' read pairs are looked at then the sequence analysis should clean up quite significantly.\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
