{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# rRNA expression in Senescent Cells\n",
    "```\n",
    "pi:ababaian\n",
    "start: 2017 01 10\n",
    "complete : 2017 01 25\n",
    "addendum: 2017 03 25\n",
    "```\n",
    "## Introduction\n",
    "\n",
    "Variation (at the rDNA level) is all well and good, what's the consequence? Is there differential expression of these RNAs under different conditions?\n",
    "\n",
    "From LIONS I had RNA-seq from Senescent cells aligned to hg19r, this seemed like a good place to start 'functional' analysis of the variation.\n",
    "\n",
    "## Hypothesis\n",
    "- Senescent / Quiscent cells will have differential expression of rRNA relative to replicating cells, assocaited with more restrictive translation reflecting low activity.\n",
    "- 'Cell Replicating' variation should be present in different models of senescence; replicative and induced.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "## Materials and Methods\n",
    "\n",
    "### MDAH041 RNAseq\n",
    "\n",
    "- [Purcell Senescence Dataset](https://www.ncbi.nlm.nih.gov/pubmed/25483067)\n",
    "- RNAseq downloaded from SRA\n",
    "\n",
    "\n",
    "#### MDAH041 RNAseq - Replicative Senescence Model\n",
    "p53[+\\-] cells\n",
    "\n",
    "- Passage 11\n",
    "- Passage 18\n",
    "- Replicative Senescence\n",
    "\n",
    "#### MDAH041 RNAseq - Replicative Senescence Model\n",
    "p53[-\\-] immortalized cells\n",
    "\n",
    "- Immortalized Replicating Control\n",
    "- 5-aza treated Senescent\n",
    "- H2O2 treated Senescent\n",
    "- Adrimyocin treated Senescent\n",
    "- Quiescent control\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Files downloaded from SRA\n",
    "# Accession numbers given below\n",
    "\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/000/SRR1544480/SRR1544480_1.fastq.gz;\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/000/SRR1544480/SRR1544480_2.fastq.gz;\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/001/SRR1544481/SRR1544481_2.fastq.gz;\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/001/SRR1544481/SRR1544481_1.fastq.gz;\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/002/SRR1544482/SRR1544482_2.fastq.gz;\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/002/SRR1544482/SRR1544482_1.fastq.gz;\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/003/SRR1544483/SRR1544483_2.fastq.gz;\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/003/SRR1544483/SRR1544483_1.fastq.gz;\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/004/SRR1544484/SRR1544484_1.fastq.gz;\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/004/SRR1544484/SRR1544484_2.fastq.gz;\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/005/SRR1544485/SRR1544485_1.fastq.gz;\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/005/SRR1544485/SRR1544485_2.fastq.gz;\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/006/SRR1544486/SRR1544486_2.fastq.gz;\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/006/SRR1544486/SRR1544486_1.fastq.gz;\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/007/SRR1544487/SRR1544487_2.fastq.gz;\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/007/SRR1544487/SRR1544487_1.fastq.gz;\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/008/SRR1544488/SRR1544488_1.fastq.gz;\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/008/SRR1544488/SRR1544488_2.fastq.gz;\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/009/SRR1544489/SRR1544489_2.fastq.gz;\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/009/SRR1544489/SRR1544489_1.fastq.gz;\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/000/SRR1544490/SRR1544490_2.fastq.gz;\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/000/SRR1544490/SRR1544490_1.fastq.gz;\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/001/SRR1544491/SRR1544491_2.fastq.gz;\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/001/SRR1544491/SRR1544491_1.fastq.gz;\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/002/SRR1544492/SRR1544492_1.fastq.gz;\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/002/SRR1544492/SRR1544492_2.fastq.gz;\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/003/SRR1544493/SRR1544493_1.fastq.gz;\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/003/SRR1544493/SRR1544493_2.fastq.gz;\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/004/SRR1544494/SRR1544494_2.fastq.gz;\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/004/SRR1544494/SRR1544494_1.fastq.gz;\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/005/SRR1544495/SRR1544495_2.fastq.gz;\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/005/SRR1544495/SRR1544495_1.fastq.gz;\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/006/SRR1544496/SRR1544496_2.fastq.gz;\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/006/SRR1544496/SRR1544496_1.fastq.gz;\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/007/SRR1544497/SRR1544497_1.fastq.gz;\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/007/SRR1544497/SRR1544497_2.fastq.gz;\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/008/SRR1544498/SRR1544498_1.fastq.gz;\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/008/SRR1544498/SRR1544498_2.fastq.gz;\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/009/SRR1544499/SRR1544499_1.fastq.gz;\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/009/SRR1544499/SRR1544499_2.fastq.gz;\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/000/SRR1544500/SRR1544500_1.fastq.gz;\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/000/SRR1544500/SRR1544500_2.fastq.gz;\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/001/SRR1544501/SRR1544501_2.fastq.gz;\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/001/SRR1544501/SRR1544501_1.fastq.gz;\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/002/SRR1544502/SRR1544502_1.fastq.gz;\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/002/SRR1544502/SRR1544502_2.fastq.gz;\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/003/SRR1544503/SRR1544503_1.fastq.gz;\n",
    "wget -N ftp.sra.ebi.ac.uk/vol1/fastq/SRR154/003/SRR1544503/SRR1544503_2.fastq.gz;"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#!/bin/sh\n",
    "\n",
    "BAM=$1\n",
    "\n",
    "# Read Subset ------------------------------\n",
    "# Extract mapped reads, and their unmapped pairs\n",
    "\n",
    "  # Extract Header\n",
    "  samtools view -H $BAM > align.header.tmp\n",
    "\n",
    "  # Unmapped reads with mapped pairs\n",
    "  # Extract Mapped Reads\n",
    "  # and their unmapped pairs\n",
    "  samtools view -b -F 4 $BAM > align.F4.bam #mapped\n",
    "  samtools view -b -f 4 -F 8 $BAM > align.f4F8.bam #unmapped pairs\n",
    "  \n",
    "  # Extract just the 45S unit\n",
    "  aws s3 cp s3://crownproject/resources/rDNA_45s.bed ./\n",
    "  samtools view -b -L rDNA_45s.bed align.F4.bam > align.F4.45s.bam\n",
    "  \n",
    "  # What are the 45S mapped readnames\n",
    "  samtools view align.F4.45s.bam | cut -f1 - > read.names.45s.tmp\n",
    "  \n",
    "  # Extract cases of read pairs mapped on edge of region of interest\n",
    "  # -------|======= R O I ======| ----------\n",
    "  # read:                  ====---====\n",
    "\n",
    "  samtools view align.F4.bam | grep -Ff read.names.45s.tmp - > align.F4.tmp.sam\n",
    "\n",
    "  # Complete mapped reads list\n",
    "  cut -f1 align.F4.tmp.sam > read.names.45s.long.tmp\n",
    "\n",
    "  # Extract unmapped reads with a mapped pair in 45s.long\n",
    "  samtools view align.f4F8.bam | grep -Ff read.names.45s.long.tmp - > align.f4F8.tmp.sam\n",
    "\n",
    "  # Re-compile bam file\n",
    "  cat align.header.tmp align.F4.tmp.sam align.f4F8.tmp.sam | samtools view -bS - > align.45s.tmp.bam\n",
    "    samtools sort align.45s.tmp.bam FINAL.45s\n",
    "    samtools index FINAL.45s.bam\n",
    "    rm *.tmp*\n",
    "    rm align.*.bam\n",
    "    \n",
    "#### End of Script"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Commands run on GSC - Xhost10 to extract data from other alignments\n",
    "\n",
    "#  sh extract45s.sh ~/Gprojects/LIONS/projects/azaSen/p11_1/alignment/p11_1.bam \n",
    "#  sh extract45s.sh ~/Gprojects/LIONS/projects/azaSen/p18_1/alignment/p18_1.bam \n",
    "#  sh extract45s.sh ~/Gprojects/LIONS/projects/azaSen/natSen_1/alignment/natSen_1.bam \n",
    "\n",
    "\n",
    "sh extract45s.sh ~/Gprojects/LIONS/projects/azaSen/p11_2/alignment/p11_2.bam \n",
    "mv FINAL.45s.bam p11_2.45s.bam\n",
    "mv FINAL.45s.bam.bai p11_2.45s.bam.bai\n",
    "\n",
    "sh extract45s.sh ~/Gprojects/LIONS/projects/azaSen/p11_3/alignment/p11_3.bam \n",
    "mv FINAL.45s.bam p11_3.45s.bam\n",
    "mv FINAL.45s.bam.bai p11_3.45s.bam.bai\n",
    "\n",
    "\n",
    "sh extract45s.sh ~/Gprojects/LIONS/projects/azaSen/p18_2/alignment/p18_2.bam \n",
    "mv FINAL.45s.bam p18_2.45s.bam\n",
    "mv FINAL.45s.bam.bai p18_2.45s.bam.bai\n",
    "\n",
    "sh extract45s.sh ~/Gprojects/LIONS/projects/azaSen/p18_3/alignment/p18_3.bam \n",
    "mv FINAL.45s.bam p18_3.45s.bam\n",
    "mv FINAL.45s.bam.bai p18_3.45s.bam.bai\n",
    "\n",
    "\n",
    "sh extract45s.sh ~/Gprojects/LIONS/projects/azaSen/natSen_2/alignment/natSen_2.bam \n",
    "mv FINAL.45s.bam natSen_2.45s.bam\n",
    "mv FINAL.45s.bam.bai natSen_2.45s.bam.bai\n",
    "\n",
    "sh extract45s.sh ~/Gprojects/LIONS/projects/azaSen/natSen_3/alignment/natSen_3.bam \n",
    "mv FINAL.45s.bam natSen_3.45s.bam\n",
    "mv FINAL.45s.bam.bai natSen_3.45s.bam.bai\n",
    "\n",
    "# done: Fri Jan 13 15:42:24 PST 2017\n",
    "\n",
    "# Look at p53 -/- immortalized cells for variation too\n",
    "# Mon Jan 16 11:44:28 PST 2017\n",
    "\n",
    "sh extract45s.sh ~/Gprojects/LIONS/projects/azaSen/imm_1/alignment/imm_1.bam \n",
    "mv FINAL.45s.bam imm_1.45s.bam\n",
    "mv FINAL.45s.bam.bai imm_1.45s.bam.bai\n",
    "\n",
    "sh extract45s.sh ~/Gprojects/LIONS/projects/azaSen/imm_2/alignment/imm_2.bam \n",
    "mv FINAL.45s.bam imm_2.45s.bam\n",
    "mv FINAL.45s.bam.bai imm_2.45s.bam.bai\n",
    "\n",
    "sh extract45s.sh ~/Gprojects/LIONS/projects/azaSen/imm_3/alignment/imm_3.bam \n",
    "mv FINAL.45s.bam imm_3.45s.bam\n",
    "mv FINAL.45s.bam.bai imm_3.45s.bam.bai\n",
    "\n",
    "\n",
    "# H2O2 Treatment\n",
    "sh extract45s.sh ~/Gprojects/LIONS/projects/azaSen/h202_1/alignment/h202_1.bam \n",
    "mv FINAL.45s.bam h202_1.45s.bam\n",
    "mv FINAL.45s.bam.bai h202_1.45s.bam.bai\n",
    "\n",
    "sh extract45s.sh ~/Gprojects/LIONS/projects/azaSen/h202_2/alignment/h202_2.bam \n",
    "mv FINAL.45s.bam h202_2.45s.bam\n",
    "mv FINAL.45s.bam.bai h202_2.45s.bam.bai\n",
    "\n",
    "sh extract45s.sh ~/Gprojects/LIONS/projects/azaSen/h202_3/alignment/h202_3.bam \n",
    "mv FINAL.45s.bam h202_3.45s.bam\n",
    "mv FINAL.45s.bam.bai h202_3.45s.bam.bai\n",
    "\n",
    "# aza Treatment\n",
    "\n",
    "sh extract45s.sh ~/Gprojects/LIONS/projects/azaSen/5aza_1/alignment/5aza_1.bam \n",
    "mv FINAL.45s.bam 5aza_1.45s.bam\n",
    "mv FINAL.45s.bam.bai 5aza_1.45s.bam.bai\n",
    "\n",
    "sh extract45s.sh ~/Gprojects/LIONS/projects/azaSen/5aza_2/alignment/5aza_2.bam \n",
    "mv FINAL.45s.bam 5aza_2.45s.bam\n",
    "mv FINAL.45s.bam.bai 5aza_2.45s.bam.bai\n",
    "\n",
    "sh extract45s.sh ~/Gprojects/LIONS/projects/azaSen/5aza_3/alignment/5aza_3.bam \n",
    "mv FINAL.45s.bam 5aza_3.45s.bam\n",
    "mv FINAL.45s.bam.bai 5aza_3.45s.bam.bai\n",
    "\n",
    "# quiescent cells\n",
    "\n",
    "sh extract45s.sh ~/Gprojects/LIONS/projects/azaSen/qui_1/alignment/qui_1.bam \n",
    "mv FINAL.45s.bam qui_1.45s.bam\n",
    "mv FINAL.45s.bam.bai qui_1.45s.bam.bai\n",
    "\n",
    "sh extract45s.sh ~/Gprojects/LIONS/projects/azaSen/qui_2/alignment/qui_2.bam \n",
    "mv FINAL.45s.bam qui_2.45s.bam\n",
    "mv FINAL.45s.bam.bai qui_2.45s.bam.bai\n",
    "\n",
    "sh extract45s.sh ~/Gprojects/LIONS/projects/azaSen/qui_3/alignment/qui_3.bam \n",
    "mv FINAL.45s.bam qui_3.45s.bam\n",
    "mv FINAL.45s.bam.bai qui_3.45s.bam.bai\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# March 25th 2017\n",
    "# herp derp. I forgot to do adrimycin.\n",
    "\n",
    "sh extract45s.sh ~/Gprojects/LIONS/projects/azaSen/adri_1/alignment/adri_1.bam \n",
    "mv FINAL.45s.bam adri_1.45s.bam\n",
    "mv FINAL.45s.bam.bai adri_1.45s.bam.bai\n",
    "\n",
    "sh extract45s.sh ~/Gprojects/LIONS/projects/azaSen/adri_2/alignment/adri_2.bam \n",
    "mv FINAL.45s.bam adri_2.45s.bam\n",
    "mv FINAL.45s.bam.bai adri_2.45s.bam.bai\n",
    "\n",
    "sh extract45s.sh ~/Gprojects/LIONS/projects/azaSen/adri_3/alignment/adri_3.bam \n",
    "mv FINAL.45s.bam adri_3.45s.bam\n",
    "mv FINAL.45s.bam.bai adri_3.45s.bam.bai"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# All files were downloaded from GSC with scp and stored locally"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Results\n",
    "\n",
    "Without dragging it out; there were no positions in the sequencing which showed SNV in a Senescent-specific (replicative or induced), or Immortalized-specific (p53 heterozygote vs. homozygote disruption, or any treatment (5aza, H2O2, Adrimyocin) specific manner.\n",
    "\n",
    "This disproves the hypothesis originally stated; there are no SNV positions which provide evidence for senescent-specific rRNA molecules being expressed.\n",
    "\n",
    "Some positions showed variation in dosage between groups; but not a clear all or none effect.\n",
    "\n",
    "End of primary analysis"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
